{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "from util import sanitize_filename\n",
    "import os\n",
    "from collections import defaultdict\n",
    "from llm import chat_openai\n",
    "from datetime import datetime"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "chunk_size = 4000 # How many characters to read per chunk in website text\n",
    "overlap = 25 # How much overlap between chunks\n",
    "MAX_TOKENS = 10000 # Roughly 3K tokens, $0.10 per MAX_TOKENs, max tokens to read per key question\n",
    "search_query = \"How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Can be moved to util.py\n",
    "# Get the list of key questions\n",
    "def get_key_questions_list_string(search_query_file_safe):\n",
    "    file_path = f'autoscious_logs/{search_query_file_safe}/decompositions/improved_decomposition.json'\n",
    "\n",
    "    with open(file_path, 'r') as f:\n",
    "        question_decomposition = json.load(f)\n",
    "    key_questions = question_decomposition['key_drivers']['1']['hypotheses']['1']['key_questions']\n",
    "    print(\"\\nkey_questions\\n\", key_questions)\n",
    "\n",
    "    numbered_key_questions_string = \"\"\n",
    "    for key, value in key_questions.items():\n",
    "        first_two_words = ' '.join(value.split())\n",
    "        numbered_key_questions_string += f'{int(key) - 1}. {first_two_words}\\n'\n",
    "\n",
    "    print(\"\\nnumbered_key_questions_string\\n\", numbered_key_questions_string)\n",
    "    return key_questions, numbered_key_questions_string"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "key_questions\n",
      " {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity compare to other bacteria?', '3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?', '4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}\n",
      "\n",
      "numbered_key_questions_string\n",
      " 0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "key_questions_dict {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity compare to other bacteria?', '3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?', '4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}\n"
     ]
    }
   ],
   "source": [
    "# 1) Get key questions\n",
    "search_query_file_safe = sanitize_filename(search_query)\n",
    "key_questions_dict, key_questions_list_string = get_key_questions_list_string(search_query_file_safe)\n",
    "print(\"key_questions_dict\", key_questions_dict)\n",
    "\n",
    "# 2) create facts folder and subfolders\n",
    "facts_folder_path = f'autoscious_logs/{sanitize_filename(search_query)}/facts'\n",
    "if not os.path.exists(facts_folder_path):\n",
    "    os.makedirs(facts_folder_path)\n",
    "for kq_idx in key_questions_dict:\n",
    "    kq_facts_folder_path = f'{facts_folder_path}/kq{int(kq_idx)-1}'\n",
    "    if not os.path.exists(kq_facts_folder_path):\n",
    "        os.makedirs(kq_facts_folder_path)\n",
    "    with open(f'{kq_facts_folder_path}/facts.txt', 'w') as f:\n",
    "        pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Can be broken to prompts.py and util.py\n",
    "import os\n",
    "\n",
    "def extract_facts_from_website_text(search_query_file_safe, key_questions_list_string, website_title, website_text, website_url):\n",
    "    seed_initial_question_decomposition_prompt = f'''\n",
    "Key questions (index : question): \n",
    "{key_questions_list_string}\n",
    "\n",
    "Task: \n",
    "Extract accurate and useful direct quotes from the source text that are relevant to answering the key questions and also output the quote's most relevant key question index. Format as a JSON.\n",
    "```json\n",
    "{{\n",
    "  \"<insert unique quote index>\": {{\n",
    "    \"reasoning\": <one sentence explaining how this direct quote helps answer the key question>\n",
    "    \"quote\": \"<insert direct quote>\",\n",
    "    \"index\": <insert most relevant key question index>\n",
    "  }},\n",
    "  etc.\n",
    "}}\n",
    "```\n",
    "\n",
    "Source text: {website_text}\n",
    "\n",
    "Respond only with the output, with no explanation or conversation.\n",
    "'''\n",
    "    # Ask GPT the prompt\n",
    "    print(\"seed_initial_question_decomposition_prompt\", seed_initial_question_decomposition_prompt)\n",
    "    res = chat_openai(seed_initial_question_decomposition_prompt, model=\"gpt-3.5-turbo\")\n",
    "    print(\"Extracted quotes: \", res[0])\n",
    "\n",
    "    # Save the quote to the corresponding key question index file\n",
    "    res_json = json.loads(res[0])\n",
    "    for key, value in res_json.items():\n",
    "        index = value['index']\n",
    "        quote = value['quote']\n",
    "\n",
    "        # Only log if there is a quote\n",
    "        if quote:\n",
    "          file_name = f'autoscious_logs/{search_query_file_safe}/facts/kq{index}/facts.txt'\n",
    "\n",
    "          with open(file_name, 'a', encoding='utf-8') as f:\n",
    "              f.write(quote.replace('/\"', '\"') + f\"[{website_url}]\" + os.linesep)\n",
    "\n",
    "    return"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Can be moved to util\n",
    "def chunk_text(text: str, chunk_size: int, overlap: int) -> list[str]:\n",
    "    chunks = []\n",
    "    for i in range(0, len(text), chunk_size - overlap):\n",
    "        chunks.append(text[i : i + chunk_size])\n",
    "        print(\"len(chunk): \", text[i : i + chunk_size])\n",
    "    return chunks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "len(chunk):  ContentsSignificanceAbstractResultsDiscussionMaterials and MethodsData AvailabilityAcknowledgmentsSupporting InformationReferencesInformation & AuthorsMetrics & CitationsView OptionsReferencesMediaShareSignificanceCarboxylases capture and convert CO2, which makes them key enzymes in photosynthesis and the global carbon cycle. However, the question how enzymes bind atmospheric CO2 is still unsolved. We studied enoyl-CoA carboxylases/reductases (Ecrs), the fastest CO2-fixing enzymes in nature, using structural biology, biochemistry, and advanced computational methods. Ecrs create a highly specific CO2-binding pocket with 4 amino acids at the active site. The pocket controls the fate of the gaseous molecule during catalysis and shields the catalytic center from oxygen and water. This exquisite control makes Ecrs highly efficient carboxylases outcompeting RuBisCO, the key enzyme of photosynthesis, by an order of magnitude. Our findings define the atomic framework for the future development of CO2-converting catalysts in biology and chemistry.AbstractCarboxylases are biocatalysts that capture and convert carbon dioxide (CO2) under mild conditions and atmospheric concentrations at a scale of more than 400 Gt annually. However, how these enzymes bind and control the gaseous CO2 molecule during catalysis is only poorly understood. One of the most efficient classes of carboxylating enzymes are enoyl-CoA carboxylases/reductases (Ecrs), which outcompete the plant enzyme RuBisCO in catalytic efficiency and fidelity by more than an order of magnitude. Here we investigated the interactions of CO2 within the active site of Ecr from Kitasatospora setae. Combining experimental biochemistry, protein crystallography, and advanced computer simulations we show that 4 amino acids, N81, F170, E171, and H365, are required to create a highly efficient CO2-fixing enzyme. Together, these 4 residues anchor and position the CO2 molecule for the attack by a reactive enolate created during the catalytic cycle. Notably, a highly ordered water molecule plays an important role in an active site that is otherwise carefully shielded from water, which is detrimental to CO2 fixation. Altogether, our study reveals unprecedented molecular details of selective CO2 binding and C–C-bond formation during the catalytic cycle of nature’s most efficient CO2-fixing enzyme. This knowledge provides the basis for the future development of catalytic frameworks for the capture and conversion of CO2 in biology and chemistry.\n",
      "Sign up for PNAS alerts.\n",
      "Get alerts for new articles, or get an alert when an article is cited.\n",
      "Manage alerts\n",
      "The efficient capture and conversion of atmospheric carbon dioxide (CO2) is a prerequisite to develop a carbon-neutral, circular future economy. In biology, carbon fixation is performed under mild conditions and at atmospheric concentrations of CO2 (0.04 vol.%) by enzymes called carboxylases (1). It is estimated that 400 Gt of CO2 are fixed annually by a single biocatalyst, ribulose-1,5-bisphophate-carboxylase/oxygenase (RuBisCO), the key enzyme of photosynthesis (2, 3). In comparison, the chemical conversion of CO2 in industry accounts for only 0.1 Gt of carbon annually and uses pressurized CO2 (4), which emphasizes our need to understand the molecular mechanism that allows (bio)catalysts to selectively interact with a low-concentrated CO2 molecule during catalysis.Carboxylases catalyze the formation of a C–C bond between an acceptor substrate and a CO2 molecule (1) where the latter represents the electrophile (5). To facilitate C–C bond formation, most carboxylases activate their respective nucleophilic substrate (usually a thioester, α-ketoacid, or ketone) by converting it into an enol(ate) (6). Enol(ate)s are strong nucleophiles and highly reactive. A key requirement of CO2-fixation catalysis is the tight control of the reaction between the activated acceptor substrate and CO2. Any loss of catalytic control over the enol(ate) or the CO2 molecule b\n",
      "len(chunk):  te) or the CO2 molecule bears the danger of side reactions and reduces the efficiency of carbon fixation (7). The most prominent example is RuBisCO, which is known to feature several side reactions, most notably an oxygenation reaction (8). One in every 5 turnovers RuBisCO will incorporate an oxygen (O2) molecule instead of CO2, which leads to the formation of 2-phosphoglycolate, a side product that is toxic to the cell and has to be recycled in an energy-demanding process, highlighting the need of carboxylases to control the reaction of the activated acceptor with CO2.Another challenge in this respect is the accessibility of water (or protic amino acids) to the active site of carboxylases. Protons are better electrophiles than the CO2 molecule, which can directly quench the enolate. As a consequence, it is not sufficient that carboxylases enrich a low abundant gaseous CO2 molecule; they also need to efficiently suppress any competing (re)protonation reactions. Altogether, these examples show that controlling the fate of CO2 at the molecular level is a crucial feature of carboxylases. However, to date only very limited biochemical, let alone structural, information on CO2 binding in carboxylases (and other proteins) is available, besides some theoretical considerations (9–14).Here, we focused on a class of carboxylases, enoyl-CoA carboxylases/reductases (Ecrs) (15, 16) that show the fastest turnover frequencies among all carboxylases to date and exclusively react with CO2 in the presence of O2. These features make Ecrs excellent model systems to understand the details of selective CO2 binding and C–C-bond formation in proteins.The best-studied Ecr is crotonyl-CoA carboxylase/reductase (Ccr) that catalyzes the NADPH-dependent reductive carboxylation of crotonyl-CoA into (2S)-ethylmalonyl-CoA. While Ccr does not show side reactivity with O2, the enzyme catalyzes the reduction of crotonyl-CoA to butyryl-CoA as a side reaction, but only in the absence of CO2 and at low catalytic efficiency (Scheme 1) (17, 18). It has been suggested that this side reaction is an evolutionary remnant of Ecrs, which are evolutionary related to enoyl thioester reductases that catalyze the ordinary reduction of enoyl-CoA esters (6, 19). Apparently, Ecrs evolved from simple reductases into reductive carboxylases by acquiring a CO2-fixation function along their evolutionary trajectory.Reaction catalyzed by Ccr. Carboxylation to (2S)-ethylmalonyl-CoA in the presence of CO2 and reduction to butyryl-CoA in the absence of CO2.Open in viewerIn a previous structural study, CinF, an Ecr from Streptomyces sp. JS360, was crystallized with NADP+ and octenoyl-CoA [Protein Data Bank (PDB); PDB ID code: 4A0S (20)]. A putative CO2 binding pocket was proposed to be composed of Asn77, Phe166, and Glu167, which are all highly conserved in Ecrs. It was suggested that CO2 is held in position by hydrogen bonding to Asn77 and Glu167 while Phe166 would undergo hydrophobic interactions with CO2. Mutation of these residues suppressed the carboxylation of octenoyl-CoA (20). However, the exact role of the individual residues directing and controlling the carboxylation reaction remains enigmatic: in particular, how the gaseous CO2 molecule is aligned at the active site and how the reduction side reaction is efficiently suppressed.Here we combine experimental biochemistry, protein crystallography, and computer simulations to define the molecular interactions of CO2 during C–C-bond formation at the active site of Ecrs. Our results suggest that 4 amino acids are sufficient to convert an ordinary reductase into a highly efficient carboxylase. Together, these 4 residues anchor and lock the CO2 molecule in a favorable position for the attack by the reactive enolate created during catalysis. Notably, a highly ordered water molecule plays an essential role in coordinating the CO2 molecule, while the active site is otherwise effectively shielded from water to suppress the reduction side re\n",
      "len(chunk):  ess the reduction side reaction. Altogether our computational and experimental studies reveal the details of selective CO2 binding and C–C-bond formation in the catalytic cycle of nature’s most efficient CO2-fixing enzyme.ResultsCrystal Structure of KsCcr with Ethylmalonyl-CoA and NADPH.We solved the structure of Ccr from Kitasatospora setae (KsCcr) cocrystallized with NADPH and soaked with ethylmalonyl-CoA, the product of the carboxylation reaction, at 1.7-Å resolution (Fig. 1 and SI Appendix, Table S1, PDB ID code: 6OWE). The active site of KsCcr shares similar features with the CinF homolog previously reported (20) [PDB ID codes: 4A0S (20), 4Y0K (21), and 4GI2]. In contrast to these structures that did not capture the interaction of the protein with CO2, neither as free gas or covalently bound to the acyl-CoA moiety, our structure shows densities that can be interpreted as carboxylated product (SI Appendix, Fig. S1). This enabled us to identify 4 residues that potentially interact with CO2, namely Asn81, Phe170, Glu171, and His365. His365 also interacts with NADPH via hydrogen bonding (3.0 Å) to the carboxamide group of the nicotinamide, indicating a second function of His365 in coordinating the NADPH cofactor during catalysis. Similar to previously published structures we also observed an ordered water molecule between His365 and Glu171 at a distance of 2.9 and 2.7 Å, respectively. This feature is absent in structures lacking substrate and cofactor (PDB ID codes: 3HZZ and 3KRT) suggesting a role of the water molecule in the active enzyme complex. To test the function of these residues during catalysis, we characterized different active site variants and addressed the reaction mechanism with molecular-dynamics simulations along the minimum free-energy path within the quantum mechanics/molecular mechanics methodology.Structure of KsCcr complexed with NADPH and ethylmalonyl-CoA. (Left) KsCcr forms a dimer of dimers of open- and closed-form subunits. The subunits are highlighted in gray and green (closed form with NADPH and ethylmalonyl-CoA, both represented in spheres) and blue and orange (open form with NADPH only, represented in spheres). The rectangle represents close-up of the active site shown in the middle and right. (Middle) The active site with the CO2-binding residues His365, Glu171, Phe170, and Asn81 in green, ethylmalonyl-CoA (salmon), and NADPH (cyan); oxygen and nitrogen atoms are colored in red and blue, respectively. (Right) Same as Middle but rotated by 180° about the viewing direction.Open in viewerKinetic Characterization of KsCcr Wild Type.KsCcr wild type (WT) showed an apparent turnover frequency (kcat) of 103 ± 3 s−1 which is well in line with previously reported value of 104 s−1 for the Ccr of Rhodobacter Sphaeroides (16). Apparent KM values were 21 ± 2 µM (crotonyl-CoA), 37 ± 4 µM (NADPH), and 90 ± 10 µM (CO2), respectively, and substrate inhibition for crotonyl-CoA was observed at a Ki of 3,650 ± 810 µM. Under saturating amounts of CO2, the enzyme showed 100% carboxylation activity and exclusively formed (2S)-ethylmalonyl-CoA. In the absence of CO2, the enzyme catalyzed the reduction of crotonyl-CoA to butyryl-CoA (Table 1). Stereochemical analysis of the butyryl-CoA in D2O showed that 94 ± 2% of the deuterium label was retained (Table 1), demonstrating that the reduction side reaction took place in a stereospecific manner.Apparent steady-state parameters for KsCcr and its mutants expressed as mean value ± SEEnzymeSubstrateKM, µMKi, µMkcat, s−1% EMC*% Label retention†WTCrotonyl-CoA21 ± 23,650 ± 810103 ± 310094.3 ± 1.8 NADPH37 ± 4—86 ± 2 CO2‡90 ± 10—78 ± 2N81LCrotonyl-CoAND§NDND1958.9 ± 1.1 NADPHNDNDNDF170YCrotonyl-CoA10 ± 1558 ± 8083 ± 4100ND NADPH36 ± 3—56 ± 1 CO2‡150 ± 20—56 ± 2F170ACrotonyl-CoA31 ± 6—8.3 ± 0.41787 ± 0.6 NADPH11 ± 0.6—11 ± 0.1H365NCrotonyl-CoA29.8 ± 4.2—5.0 ± 0.29363.3 ± 0.5 NADPH22 ± 2—8.1 ± 0.3 CO2‡1310 ± 220—7.4 ± 0.7E171ACrotonyl-CoA500 ± 62—5.1 ± 0.29791.4 ± 0.2 NADPH112 ± 8—6.0\n",
      "len(chunk):  .4 ± 0.2 NADPH112 ± 8—6.0 ± 0.2 CO2155 ± 30—5.1 ± 0.3EXPAND FOR MORESI Appendix, Fig. S2 shows the Michaelis–Menten graphs of the original data.*Percentage of (2S)-ethylmalonyl-CoA (EMC) over total amount of products.†Deuterium label retention at the α-position of crotonyl-CoA expressed as mean value ± SD.‡Apparent KM values for CO2 were calculated from the HCO3− concentration in solution at pH = 8.§Not determined due to accumulation of the covalent C2-ene adduct.Open in viewerAsn81 Anchors the CO2 Molecule.How is the CO2 molecule bound in the active site? A key residue is Asn81, which defines one end of the putative CO2 binding pocket. Simulations of the WT enzyme exhibited a hydrogen-bond interaction between the carboxamide NH2 group of Asn81 and the CO2 molecule (Fig. 2A and Movie S1). When we experimentally characterized the reaction of the N81L variant in more detail, we observed a strongly decreased carboxylation reaction, as well as accumulation of a covalent reaction intermediate, a so-called C2-ene adduct (SI Appendix, Fig. S3). C2-ene adducts are also observed in WT Ecrs when the catalytic cycle is stalled, for example when CO2 is omitted from the reaction mixture. The fact that a C2-ene adduct is observed in the reaction of the N81L variant even under saturating CO2 conditions suggests that the interaction of the enzyme with CO2 is severely disturbed by the N81L mutation. Simulations of the N81L variant revealed that most of the catalytic residues and water molecules remain in the same position, while the CO2 molecule appears increasingly disordered (Fig. 2D and Movie S2). As a consequence, the minimum energy profile from the C2-adduct to the product (2S)-ethylmalonyl-CoA of N81L becomes endothermic and shows significantly higher barriers for the C–C-bond formation step compared with the WT. This explains the reduced carboxylation activity, as well as the accumulation of the C2-ene adduct that we experimentally observed in the N81L variant (Fig. 2 C and F).\n",
      "KsCcr WT\n",
      "KsCcr N81LLeft, Middle, and Right columns represent the WT, N81L, and F170A variants respectively. (A, D, and G) Active site of KsCcr variants before the nucleophilic attack of the enolate onto CO2. An overlay of different conformations of the CO2 molecule visualizes the tumbling motion in the different enzyme variants. CoA-ester is shown in salmon and NADP+ in cyan, and the red sphere represents the conserved water molecule coordinated by His365 and Glu171. (B, E, and H) Hydration sites located within 5 Å of the Cα obtained with SSTMap. For each enzyme, representative structures of the reactant state were subjected to 1 ns of constrained quantum mechanics/molecular mechanics simulations. The different positions sampled by the water molecules allowed the determination of clusters showing the preferential location of the solvent within the active site. (C, F, and I) Minimum free-energy path from the C2-ene adduct through the enolate to the products (2S)-ethylmalonyl-CoA and NADP+. Parameter D represents the distance between Cα and C2 minus the distance between Cα and CO2.Open in viewerOur simulations show that the active site of N81L does not become more solvent accessible, so that the reduction side reaction is still suppressed at the enzyme’s active site. But, why does N81L show 89% butyryl-CoA formation? Note that C2-ene adducts are unstable and prone to spontaneous decay into butyryl-CoA and NADP+ in free solution. Accordingly, the “apparent” side reaction in N81L is nonenzymatic and caused by release of the C2-ene adduct from the active site followed by its spontaneous decay in the solvent, as described before (18, 22, 23). This hypothesis is supported by our observation that stereospecificity of butyryl-CoA formation is almost completely lost in N81L compared with the WT (Table 1). Altogether, our experimental findings are well in line with the higher calculated free-energy barriers for the carboxylation step and endothermic product formation in the\n",
      "len(chunk):   product formation in the N81L variant (Fig. 2F), highlighting how crucial Asn81 is for correct positioning of CO2.How is the interaction of the carboxamide group of Asn81 with CO2 controlled? Analysis of the interaction network of the amino acid shows hydrogen bonding of Asn81 to 2 residues in the second shell of the active site, Thr82 and Ser119. We hypothesized that these interactions are essential in pointing the carboxamide NH2 group Asn81 toward the active site to enforce its interaction with the CO2 molecule. Indeed, the variants T82D and S119A showed almost full carboxylation, but at more than 50-fold reduced turnover frequency (SI Appendix, Table S3), demonstrating the importance of these secondary shell residues in increasing catalytic activity of Asn81. In summary, both simulation and experimental data suggest that Asn81 is crucial to position CO2 and to establish favorable interactions of the gas molecule with the enzyme during catalysis. Absence of this residue leads to lowered carboxylation efficiency and increased formation of the labile C2-ene adduct, which is not further processed by the enzyme and leaves the active site upon which it spontaneously decays in solution.Phe170 Shields the Active Site from Water.The reactive nature of the enolate that is formed during the catalytic cycle of Ccr mandates that the enolate does not get into contact with water at the active site, which would inevitably lead to its protonation and formation of the butyryl-CoA side product. A role in shielding the active site from water had been previously suggested for Phe170 (20). Simulations of an F170A variant reveal conformational changes that result in a disorganization of the active site and an increase in the number of hydration sites compared with the WT (Fig. 2; compare Fig. 2 B and H, Movie S3, and SI Appendix, Supplementary Pymol and Chimera Files 1a and b and 3a and b). These changes also perturb the interaction of the CO2 molecule with Asn81 (Fig. 2G). Accordingly, the CO2 molecule loses its favorable position for the reaction with the enolate so that both activation barriers are increased (Fig. 2I), which is in agreement with our experimental data (Table 1). In the F170A variant the carboxylation activity is decreased to 17% at the expense of increased reduction side reactivity. Unlike in the N81L variant, however, protonation takes place in F170A with almost WT stereospecificity (Table 1), confirming that in the F170A enzyme water is able to reach the active site and directly protonate the enolate.\n",
      "KsCcr F170ASome Ecrs feature a tyrosine at position 170 instead of the phenylalanine. When we tested an F170Y variant, the enzyme showed a slightly increased substrate inhibition, but otherwise very similar kinetic parameters as the WT. Most importantly, the F170Y variant displayed full conversion of crotonyl-CoA to (2S)-ethylmalonyl-CoA in the presence of saturating amounts of CO2, indicating that the presence of the hydroxyl group does not affect carboxylation activity. In summary, these experiments together with the simulations showed that the phenyl rings of phenylalanine (and tyrosine) play an important role in suppressing the reduction reaction of Ecrs by water shielding.His365 and Glu171 Coordinate an Ordered Water Molecule Interacting with CO2.Opposite of Asn81 and at the other end of the putative CO2-binding pocket the residues His365 and Glu171 are located. Together, these 2 residues coordinate a water molecule. In our simulations the ordered water molecule participates in hydrogen-bonded network of 3 water molecules, which interact directly with the CO2 molecule during the carboxylation step (Fig. 3). What is the exact contribution of these 2 residues to catalysis, in particular in respect to CO2 binding and water accessibility?Left and Right columns represent the E171A and H365 N variants, respectively. (A and D). Active site of KsCcr variants before the nucleophilic attack of the enolate onto CO2. The CoA-ester is\n",
      "len(chunk):  nto CO2. The CoA-ester is shown in salmon and NADP+ in cyan. Different orientations of carbon dioxide in the simulations represent the tumbling motion of the molecule before product formation. (B and E) Hydration sites located within 5 Å of the Cα obtained with SSTMap. (C and F) Minimum free-energy paths from the C2-ene adduct through the enolate to the products (2S)-ethylmalonyl-CoA and NADP+. Parameter D represents the distance between Cα and C2 minus the distance between Cα and CO2.Open in viewerHis365 serves a dual role by also coordinating the nicotinamide ring of NADPH. To preserve interaction of residue 365 with the NADPH cofactor, but interrupt its coordination of the ordered water molecule, we generated KsCcr H365N. The H365N variant showed a 20-fold decreased activity compared with the WT enzyme but still displayed 93% of carboxylated product, even though the KM for CO2 was raised by more than 1 order of magnitude. These data suggest that a defect in water coordination negatively affects C–C-bond formation activity in the H365N variant. However, this does not lead to a complete hydration of the active site. Simulations of the H365N mutant revealed broken interactions of the CO2 molecule with Asn81 (Fig. 3D and Movie S4). The coordination to the ordered water molecule that bridges to Glu171 is lost, and the latter residue is rotated out of the active site disfavoring CO2 binding, which explains the experimentally observed increased KM for CO2. In our simulations, the CO2 molecule shows an increased rotational tumbling at the active site of the H365N variant compared with the WT, which is reflected by an increased rmsd for CO2 (2.4 Å versus 0.46 Å). Together, experiment and simulations indicated reduced carboxylation efficiency because of decreased control of CO2 (but not water) at the active site in the H365N variant, lowering the chances of productive Michaelis complex formation.\n",
      "KsCcr E171ASimilar to the H365N variant, replacement of Glu171 with alanine also resulted in a kcat decrease (17-fold), and carboxylation of crotonyl-CoA was also maintained (97%). Additionally, however, the KM for crotonyl-CoA increased 25-fold, indicating an additional role of Glu171 in positioning the CoA substrate. In our simulations of the E171A mutant, interaction of CO2 with Asn81 is partially maintained, while Phe170 adopts the position of the mutated glutamate in the WT (Fig. 3A and Movie S5). As for H365N, the E171A variant showed increased CO2 tumbling at the active site, which is reflected by an rmsd of 2.0 Å (compared with 0.46 Å of the WT). Altogether, these results suggest that in the E171A variant, similar to the H365N variant, control of the CO2 molecule (but not water) is affected.\n",
      "KsCcr H365NWhile H365N and E171A show a similar carboxylation behavior, the free-energy paths of the 2 variants show distinct differences (Fig. 3 C and F). In both variants, the 2 main barriers along the minimum free-energy path of the reaction––the one that leads to the enolate from the C2-ene adduct and the one that adds CO2 to the enolate––appear increased compared with the WT (for WT see Fig. 2C). This is experimentally reflected by the decreased catalytic activity of the 2 variants and can be related to a distorted water network and increased tumbling of CO2 at the active site of these enzymes. In H365N, however, the first energy barrier is higher compared with E171A. This indicates that enolate formation is disfavored and suggests that the C2-ene adduct accumulates in the H365N variant. This was experimentally confirmed by measuring the stereospecificity of the reduction reaction in the absence of CO2. In the H365N variant, stereospecificity was lost (Table 1), indicating that the H365N is additionally affected in enolate formation compared with the E171A mutant and the WT, respectively.In summary, His365 and Glu171 can partially compensate each other so that carboxylation function is maintained. However, only the combined action of the 2 r\n",
      "len(chunk):  ombined action of the 2 residues allows full control over the CO2 molecule and thus a fast carboxylation rate.DiscussionCcr from K. setae carboxylates crotonyl-CoA at a turnover frequency of more than 100 s−1. This is almost 1 order of magnitude faster than an average RuBisCO homolog and one of the fastest CO2-conversion rates described to date. Combining X-ray crystallography, experimental biochemistry, and molecular dynamics simulations we characterized the role of individual amino acids at the active site of KsCcr in CO2 binding and C–C-bond formation. In KsCcr the active site is optimized to accommodate CO2 and at the same time exclude water to suppress the competing reduction reaction. All this is apparently achieved by only 4 amino acids: Asn81, His365, Glu171, and Phe170. The amide group of Asn81 is responsible for anchoring the CO2 from one side, while a water network organized from an ordered water molecule coordinated between His365 and Glu171 serves as an additional anchor point for CO2 from the opposite side. The aromatic ring of Phe170 side chain actively prevents the diffusion of water into the active site.Why is the CO2-fixation reaction of Ecrs so much faster but still more specific compared with the reaction catalyzed by RuBisCO, although both enzymes react through an enolate? Note that there are fundamental differences in the catalytic mechanisms of RuBisCO and Ecrs. In RuBisCO, enolization of the substrate ribulose 1,5-bisphosphate is achieved by abstraction of the H3 proton through an active-site carbamylated lysine, which is a reversible process and very close to equilibrium (24). (Re)protonation of the enolate yields the substrate again, which can undergo another round of activation until it reacts with a CO2 (or O2) molecule, which pulls the reaction further (25). In contrast, in Ecrs the enolate is formed by hydride transfer from NADPH to the β-position of crotonyl-CoA (17–19). This provides a more “unidirectional” reaction path and leaves the enolate committed for a nucleophilic attack. Accordingly, Ecrs are required to carefully control the further fate of the enolate and especially to prevent its protonation, which would irreversibly quench the reaction (in stark contrast to the case of RuBisCO). When CO2 is absent from the active site of WT Ecr (18), or its positioning is disturbed (e.g., by H365N or N81L mutation, this study), the enolate is not simply transformed back into the starting substrates as in RuBisCO, but tends to collapse into the C2-ene adduct, which is in line with the idea of unidirectionality in Ecr catalysis. It might be tempting to speculate that the C2-ene adduct serves as a way to “store” the reactive enolate until a resolving CO2 electrophile becomes available, thereby increasing the overall reactivity of Ecrs compared with RuBisCOs.RuBisCO and Ecrs probably both evolved from non–CO2-fixing ancestors. While it has been speculated that RuBisCO emerged from an ancestral sugar phosphate isomerase it is thought that Ecrs evolved from a primordial enoyl-CoA reductase (6, 26). Clearly, this put more constraints onto the active-site topology of Ecrs, because unlike the CO2 fixation reaction in RuBisCO that could be simply built on top of a reversible isomerization reaction, the reduction reaction needed to be suppressed and replaced by the carboxylation reaction in the Ecr scaffold because of the unidirectionality of Ecr’s catalytic mechanism. The situation is reminiscent of 2-ketopropyl coenzyme M oxidoreductase/carboxylase that evolved within the superfamily of NAD(P)H disulfide oxidoreductases for which the enzyme also had to replace an active-site topology prone to reduction reactions by a CO2-fixing active site (11). In both cases, the active-site transformation was achieved with only little changes and notably without introduction of a competing oxygenation reaction, posing the question whether the evolutionary circumstances or the nature of the catalytic cycle of both enzymes w\n",
      "len(chunk):  c cycle of both enzymes were responsible to achieve this superior selectivity against oxygen compared with RuBisCO.In summary, our findings provide detailed insights into the molecular control of CO2 at one of nature’s most efficient carbon-fixing enzymes. These insights will be helpful in the future design of catalytic frameworks for the capture and conversion of CO2 in chemistry and biology (27), but also for efforts that aim at using Ecrs as key enzymes in the development of synthetic cycles for the sustainable and efficient fixation of CO2 (28, 29).Materials and MethodsChemicals.Crotonic Anhydride and Carbonic anhydrase from bovine erythrocytes were purchased from Sigma-Aldrich AG, CoA trilithium salt and DNase I from Roche Diagnostics, and NADPH Na4 (98%) from Carl Roth GmbH. Solvents and salts were all analytical grade or better. Crotonyl-CoA was synthesized as previously reported (30).Cloning and Mutagenesis.The KsCcr gene was provided by the Joint Genome Institute. Enzyme variants were generated with the QuikChange Site‐Directed Mutagenesis Kit (Stratagene) using primer pairs listed in SI Appendix, Table S2.Protein Expression and Purification.Details about expression and purification are described in SI Appendix.Quantification of Reaction Products.Details about quantification of reaction products are described in SI Appendix.Spectrophotometric Enzyme Assays.Assays were performed on a Cary-60 UV/Vis spectrophotometer (Agilent) at 30 °C using quartz cuvettes (1- or 10-mm path length; Hellma). Reactions contained 20 µg/mL carbonic anhydrase and were performed in 100 mM K2HPO4 pH = 8.0. Kinetic parameters for 1 substrate were determined by varying its concentration while the others were kept constant at 10× their KM value. Reaction procedure was monitored by following the oxidation of NADPH at 365 nm (εNADPH,365nm = 3.33 mM−1⋅cm−1). Each concentration was measured in triplicates and the obtained curves were fit using GraphPad Prism 8. Hyperbolic curves were fit to the Michaelis–Menten equation to obtain apparent kcat and KM values. For mutants revealing substrate inhibition, the data were fit to v0 = (VMax [S])/(KM + [S] ((1 + [S])/Ki)).Determining the Stereochemistry of Protonation.Isotopic label incorporation experiments were performed analogous to a previously described method (SI Appendix, Fig. S4) (22).Molecular Dynamics Simulations.Description of the workflow of the molecular dynamics simulations is described in SI Appendix.Data AvailabilityData deposition: Crystallography, atomic coordinates, and structure factors have been deposited in the Protein Data Bank (PDB), http://www.rcsb.org/; PDB ID code: 6OWE.AcknowledgmentsD.A.S. and E.V.-M. acknowledge K. Zinovjev for the implementation of the adaptive string method and fruitful discussions. Diffraction data were collected at Advanced Photon Source Beamline 23ID-B, Argonne National Laboratory. This work was supported by the European Research Council Grant 637675 (Combining synthetic biology and chemistry to create novel CO2-fixing enzymes, organelles and organisms; SYBORG) granted to T.J.E., and the Max-Planck-Society Partnergroup Program, granted to E.V.-M. D.A.S. was supported by a SYNMIKRO fellowship. H.D. acknowledges support from NSF Science and Technology Centers Grant NSF-1231306 (Biology with X-ray Lasers, BioXFEL). S.W. acknowledges support from U.S. Department of Energy (DOE) Office of Science, Biological and Environmental Research; Stanford Precourt Institute; and SLAC Laboratory Directed Research and Development. The work conducted by the DOE Joint Genome Institute, a DOE Office of Science User Facility, is supported under Contract DE-AC02-05CH11231.Supporting InformationAppendix (PDF)Download1.36 MB\n",
      "Movie S1.KsCcr WTDownload5.00 MB\n",
      "Movie S2.KsCcr N81LDownload4.92 MB\n",
      "Movie S3.KsCcr F170ADownload8.17 MB\n",
      "Movie S4.KsCcr E171ADownload8.71 MB\n",
      "Movie S5.KsCcr H365NDownload7.20 MBReferences1T. J. Erb, Carboxylases in natural and synthetic microbial pathways. Appl. \n",
      "len(chunk):  icrobial pathways. Appl. Environ. Microbiol. 77, 8466–8477 (2011).CrossrefPubMedGoogle Scholara [...] (0.04 vol.%) by enzymes called carboxylases b [...] molecule 2R. J. Ellis, Most abundant protein in the world. Trends Biochem. Sci. 4, 241–244 (1979).Go to referenceCrossrefGoogle Scholar3R. Phillips, R. Milo, A feeling for the numbers in biology. Proc. Natl. Acad. Sci. U.S.A. 106, 21465–21471 (2009).Go to referenceCrossrefPubMedGoogle Scholar4E. Alper, O. Y. Orhan, CO2 utilization: Developments in conversion processes. Petroleum 3, 109–126 (2017).Go to referenceCrossrefGoogle Scholar5M. Aresta, A. Dibenedetto, Utilisation of CO2 as a chemical feedstock: Opportunities and challenges. Dalton Trans., 2975–2992 (2007).Go to referenceCrossrefPubMedGoogle Scholar6L. Schada von Borzyskowski, R. G. Rosenthal, T. J. Erb, Evolutionary history and biotechnological future of carboxylases. J. Biotechnol. 168, 243–251 (2013).CrossrefPubMedGoogle Scholara [...] or ketone) by converting it into an enol(ate) b [...] catalyze the ordinary reduction of enoyl-CoA esters c [...] Ecrs evolved from a primordial enoyl-CoA reductase 7D. L. Edmondson, H. J. Kane, T. J. Andrews, Substrate isomerization inhibits ribulosebisphospate carboxylase-oxygenase during catalysis. FEBS Lett. 260, 62–66 (1990).Go to referenceCrossrefGoogle Scholar8G. Bowes, W. L. Ogren, R. H. Hageman, Phosphoglycolate production catalyzed by ribulose diphosphate carboxylase. Biochem. Biophys. Res. Commun. 45, 716–722 (1971).Go to referenceCrossrefPubMedGoogle Scholar9G. A. Prussia et al., Substitution of a conserved catalytic dyad into 2-KPCC causes loss of carboxylation activity. FEBS Lett. 590, 2991–2996 (2016).Go to referenceCrossrefPubMedGoogle Scholar10M. A. Kofoed, D. A. Wampler, A. S. Pandey, J. W. Peters, S. A. Ensign, Roles of the redox-active disulfide and histidine residues forming a catalytic dyad in reactions catalyzed by 2-ketopropyl coenzyme M oxidoreductase/carboxylase. J. Bacteriol. 193, 4904–4913 (2011).CrossrefPubMedGoogle Scholar11A. S. Pandey, D. W. Mulder, S. A. Ensign, J. W. Peters, Structural basis for carbon dioxide binding by 2-ketopropyl coenzyme M oxidoreductase/carboxylase. FEBS Lett. 585, 459–464 (2011).Go to referenceCrossrefPubMedGoogle Scholar12B. Stec, Structural mechanism of RuBisCO activation by carbamylation of the active site lysine. Proc. Natl. Acad. Sci. U.S.A. 109, 18785–18790 (2012).CrossrefPubMedGoogle Scholar13M. van Lun, J. S. Hub, D. van der Spoel, I. Andersson, CO2 and O2 distribution in Rubisco suggests the small subunit functions as a CO2 reservoir. J. Am. Chem. Soc. 136, 3165–3171 (2014).CrossrefPubMedGoogle Scholar14Y. Ito, H. Kondo, Y. Shiota, K. Yoshizawa, Theoretical analysis of the reaction mechanism of biotin carboxylase. J. Chem. Theory Comput. 4, 366–374 (2008).Go to referenceCrossrefPubMedGoogle Scholar15D. M. Peter et al., Screening and engineering the synthetic potential of carboxylating reductases from central metabolism and polyketide biosynthesis. Angew. Chem. Int. Ed. Engl. 54, 13457–13461 (2015).Go to referenceCrossrefPubMedGoogle Scholar16T. J. Erb et al., Synthesis of C5-dicarboxylic acids from C2-units involving crotonyl-CoA carboxylase/reductase: The ethylmalonyl-CoA pathway. Proc. Natl. Acad. Sci. U.S.A. 104, 10631–10636 (2007).CrossrefPubMedGoogle Scholara [...] carboxylases, enoyl-CoA carboxylases/reductases (Ecrs) b [...] Rhodobacter Sphaeroides 17T. J. Erb, V. Brecht, G. Fuchs, M. Müller, B. E. Alber, Carboxylation mechanism and stereochemistry of crotonyl-CoA carboxylase/reductase, a carboxylating enoyl-thioester reductase. Proc. Natl. Acad. Sci. U.S.A. 106, 8871–8876 (2009).CrossrefPubMedGoogle Scholara [...] and at low catalytic efficiency b [...] transfer from NADPH to the β-position of crotonyl-CoA 18R. G. Rosenthal et al., Direct evidence for a covalent ene adduct intermediate in NAD(P)H-dependent enzymes. Nat. Chem. Biol. 10, 50–55 (2014).CrossrefPubMedGoogle Scholara [...] and at low catalytic effic\n",
      "len(chunk):  nd at low catalytic efficiency b [...] spontaneous decay in the solvent, as described before c [...] is absent from the active site of WT Ecr 19R. G. Rosenthal et al., The use of ene adducts to study and engineer enoyl-thioester reductases. Nat. Chem. Biol. 11, 398–400 (2015).CrossrefPubMedGoogle Scholara [...] catalyze the ordinary reduction of enoyl-CoA esters b [...] transfer from NADPH to the β-position of crotonyl-CoA 20N. Quade, L. Huo, S. Rachid, D. W. Heinz, R. Müller, Unusual carbon fixation gives rise to diverse polyketide extender units. Nat. Chem. Biol. 8, 117–124 (2011).CrossrefPubMedGoogle Scholara [...] [Protein Data Bank (PDB); PDB ID code: 4A0S b [...] residues suppressed the carboxylation of octenoyl-CoA c [...] features with the CinF homolog previously reported d [...] ) [PDB ID codes: 4A0S e [...] from water had been previously suggested for Phe170 21L. Zhang et al., Rational control of polyketide extender units by structure-based engineering of a crotonyl-CoA carboxylase/reductase in antimycin biosynthesis. Angew. Chem. Int. Ed. Engl. 54, 13462–13465 (2015).Go to referenceCrossrefPubMedGoogle Scholar22B. Vögeli et al., InhA, the enoyl-thioester reductase from Mycobacterium tuberculosis forms a covalent adduct during catalysis. J. Biol. Chem. 293, 17200–17207 (2018).CrossrefPubMedGoogle Scholara [...] spontaneous decay in the solvent, as described before b [...] performed analogous to a previously described method 23R. D. Libby, R. A. Mehl, Characterization of covalent Ene adduct intermediates in “hydride equivalent” transfers in a dihydropyridine model for NADH reduction reactions. Bioorg. Chem. 40, 57–66 (2012).Go to referenceCrossrefPubMedGoogle Scholar24G. Tcherkez, Modelling the reaction mechanism of ribulose-1,5-bisphosphate carboxylase/oxygenase and consequences for kinetic parameters. Plant Cell Environ. 36, 1586–1596 (2013).Go to referenceCrossrefPubMedGoogle Scholar25G. G. Tcherkez, G. D. Farquhar, T. J. Andrews, Despite slow catalysis and confused substrate specificity, all ribulose bisphosphate carboxylases may be nearly perfectly optimized. Proc. Natl. Acad. Sci. U.S.A. 103, 7246–7251 (2006).Go to referenceCrossrefPubMedGoogle Scholar26T. J. Erb, J. Zarzycki, A short history of RubisCO: The rise and fall (?) of Nature’s predominant CO2 fixing enzyme. Curr. Opin. Biotechnol. 49, 100–107 (2018).Go to referenceCrossrefPubMedGoogle Scholar27A. M. Appel et al., Frontiers, opportunities, and challenges in biochemical and chemical catalysis of CO2 fixation. Chem. Rev. 113, 6621–6658 (2013).Go to referenceCrossrefPubMedGoogle Scholar28T. Schwander, L. Schada von Borzyskowski, S. Burgener, N. S. Cortina, T. J. Erb, A synthetic pathway for the fixation of carbon dioxide in vitro. Science 354, 900–904 (2016).Go to referenceCrossrefPubMedGoogle Scholar29A. Bar-Even, E. Noor, N. E. Lewis, R. Milo, Design and analysis of synthetic carbon fixation pathways. Proc. Natl. Acad. Sci. U.S.A. 107, 8889–8894 (2010).Go to referenceCrossrefPubMedGoogle Scholar30D. M. Peter, B. Vögeli, N. S. Cortina, T. J. Erb, A chemo-enzymatic road map to the synthesis of CoA esters. Molecules 21, 517 (2016).Go to referenceCrossrefPubMedGoogle ScholarShow all referencesInformationAuthorsInformationPublished in\n",
      "Proceedings of the National Academy of SciencesVol. 116 | No. 28July 9, 2019PubMed: 31243147ClassificationsBiological SciencesBiochemistryCopyrightCopyright © 2019 the Author(s). Published by PNAS. This open access article is distributed under Creative Commons Attribution-NonCommercial-NoDerivatives License 4.0 (CC BY-NC-ND).Data AvailabilityData deposition: Crystallography, atomic coordinates, and structure factors have been deposited in the Protein Data Bank (PDB), http://www.rcsb.org/; PDB ID code: 6OWE.Submission historyPublished online: June 26, 2019Published in issue: July 9, 2019Keywordscarboxylasesenzyme mechanismscarbon dioxideCO2 fixationRuBisCOAcknowledgmentsD.A.S. and E.V.-M. acknowledge K. Zinovjev for the implemen\n",
      "len(chunk):  Zinovjev for the implementation of the adaptive string method and fruitful discussions. Diffraction data were collected at Advanced Photon Source Beamline 23ID-B, Argonne National Laboratory. This work was supported by the European Research Council Grant 637675 (Combining synthetic biology and chemistry to create novel CO2-fixing enzymes, organelles and organisms; SYBORG) granted to T.J.E., and the Max-Planck-Society Partnergroup Program, granted to E.V.-M. D.A.S. was supported by a SYNMIKRO fellowship. H.D. acknowledges support from NSF Science and Technology Centers Grant NSF-1231306 (Biology with X-ray Lasers, BioXFEL). S.W. acknowledges support from U.S. Department of Energy (DOE) Office of Science, Biological and Environmental Research; Stanford Precourt Institute; and SLAC Laboratory Directed Research and Development. The work conducted by the DOE Joint Genome Institute, a DOE Office of Science User Facility, is supported under Contract DE-AC02-05CH11231.NotesThis article is a PNAS Direct Submission.AuthorsAffiliationsExpand AllGabriele M. M. StoffelDepartment of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, D-35043 Marburg, Germany;Center for Synthetic Microbiology, D-35043 Marburg, Germany;View all articles by this authorDavid Adrian Saez https://orcid.org/0000-0001-5555-8720Departamento de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, 4070371 Concepción, Chile;View all articles by this authorHasan DeMirciBiosciences Division, SLAC National Accelerator Laboratory, Menlo Park, CA 94025;Stanford PULSE Institute, SLAC National Accelerator Laboratory, Menlo Park, CA 94025;View all articles by this authorBastian VögeliDepartment of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, D-35043 Marburg, Germany;View all articles by this authorYashas RaoBiosciences Division, SLAC National Accelerator Laboratory, Menlo Park, CA 94025;View all articles by this authorJan Zarzycki https://orcid.org/0000-0002-6307-6497Department of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, D-35043 Marburg, Germany;Center for Synthetic Microbiology, D-35043 Marburg, Germany;View all articles by this authorYasuo Yoshikuni https://orcid.org/0000-0002-8372-640XDepartment of Energy Joint Genome Institute, Walnut Creek, CA 94598;View all articles by this authorSoichi Wakatsuki1 https://orcid.org/0000-0001-5896-7968 toerb@mpi-marburg.mpg.deBiosciences Division, SLAC National Accelerator Laboratory, Menlo Park, CA 94025;Structural Biology Department, Stanford University, Stanford, CA 94305View all articles by this authorEsteban Vöhringer-Martinez1 toerb@mpi-marburg.mpg.deDepartamento de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, 4070371 Concepción, Chile;View all articles by this authorTobias J. Erb1 toerb@mpi-marburg.mpg.deDepartment of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, D-35043 Marburg, Germany;Center for Synthetic Microbiology, D-35043 Marburg, Germany;View all articles by this authorNotes1To whom correspondence may be addressed. Email: toerb@mpi-marburg.mpg.de, soichi.wakatsuki@stanford.edu (for X-ray crystallography), or evohringer@udec.cl (for computational correspondence).Author contributions: G.M.M.S., D.A.S., H.D., B.V., Y.R., S.W., E.V.-M., and T.J.E. designed research; G.M.M.S., D.A.S., H.D., Y.R., and E.V.-M. performed research; D.A.S. and Y.Y. contributed new reagents/analytic tools; G.M.M.S., D.A.S., H.D., Y.R., and J.Z. analyzed data; and G.M.M.S., D.A.S., H.D., Y.R., S.W., E.V.-M., and T.J.E. wrote the paper.Competing InterestsThe authors declare no conflict of interest.MetricsCitationsMetrics\n",
      "Article usageViewsCitationsNo data available.020406080MarAprMayJunJulAug6,33227Total6 Months12 MonthsTotal number of\n",
      "views and citations\n",
      "for the most recent 6 whole calendar months.\n",
      "Note: The article usage is pres\n",
      "len(chunk):  The article usage is presented with a three- to four-day delay and will update daily once available. Due to ths delay, usage data will not appear immediately following publication. Citation information is sourced from Crossref Cited-by service.\n",
      "Citation statements\n",
      "411531Smart Citations411531Citing PublicationsSupportingMentioningContrastingView CitationsSee how this article has been cited at scite.aiscite shows how a scientific paper has been cited by providing the context of the citation, a classification describing whether it supports, mentions, or contrasts the cited claim, and a label indicating in which section the citation was made.\n",
      "Altmetrics\n",
      "See more details\n",
      "Picked up by 8 news outlets\n",
      "Blogged by 1\n",
      "Tweeted by 10\n",
      "73 readers on Mendeley\n",
      "Citations\n",
      "If you have the appropriate software installed, you can download article citation data to the citation manager of your choice. Simply select your manager software from the list below and click Download.\n",
      "Format\n",
      "Please select one from the list\n",
      "RIS (ProCite, Reference Manager)\n",
      "EndNote\n",
      "BibTex\n",
      "Medlars\n",
      "RefWorks\n",
      "Direct Import\n",
      "Cited byLoading...\n",
      "View options PDF formatDownload this article as a PDF fileDOWNLOAD PDFFiguresTablesOtherFiguresReaction catalyzed by Ccr. Carboxylation to (2S)-ethylmalonyl-CoA in the presence of CO2 and reduction to butyryl-CoA in the absence of CO2.Go to FigureOpen in ViewerStructure of KsCcr complexed with NADPH and ethylmalonyl-CoA. (Left) KsCcr forms a dimer of dimers of open- and closed-form subunits. The subunits are highlighted in gray and green (closed form with NADPH and ethylmalonyl-CoA, both represented in spheres) and blue and orange (open form with NADPH only, represented in spheres). The rectangle represents close-up of the active site shown in the middle and right. (Middle) The active site with the CO2-binding residues His365, Glu171, Phe170, and Asn81 in green, ethylmalonyl-CoA (salmon), and NADPH (cyan); oxygen and nitrogen atoms are colored in red and blue, respectively. (Right) Same as Middle but rotated by 180° about the viewing direction.Go to FigureOpen in ViewerLeft, Middle, and Right columns represent the WT, N81L, and F170A variants respectively. (A, D, and G) Active site of KsCcr variants before the nucleophilic attack of the enolate onto CO2. An overlay of different conformations of the CO2 molecule visualizes the tumbling motion in the different enzyme variants. CoA-ester is shown in salmon and NADP+ in cyan, and the red sphere represents the conserved water molecule coordinated by His365 and Glu171. (B, E, and H) Hydration sites located within 5 Å of the Cα obtained with SSTMap. For each enzyme, representative structures of the reactant state were subjected to 1 ns of constrained quantum mechanics/molecular mechanics simulations. The different positions sampled by the water molecules allowed the determination of clusters showing the preferential location of the solvent within the active site. (C, F, and I) Minimum free-energy path from the C2-ene adduct through the enolate to the products (2S)-ethylmalonyl-CoA and NADP+. Parameter D represents the distance between Cα and C2 minus the distance between Cα and CO2.Go to FigureOpen in ViewerLeft and Right columns represent the E171A and H365 N variants, respectively. (A and D). Active site of KsCcr variants before the nucleophilic attack of the enolate onto CO2. The CoA-ester is shown in salmon and NADP+ in cyan. Different orientations of carbon dioxide in the simulations represent the tumbling motion of the molecule before product formation. (B and E) Hydration sites located within 5 Å of the Cα obtained with SSTMap. (C and F) Minimum free-energy paths from the C2-ene adduct through the enolate to the products (2S)-ethylmalonyl-CoA and NADP+. Parameter D represents the distance between Cα and C2 minus the distance between Cα and CO2.Go to FigureOpen in ViewerTablesApparent steady-state parameters for KsCcr and its mutants expressed as mean value ± SEGo to TableOpen in ViewerOtherShare\n",
      "len(chunk):  eOpen in ViewerOtherShareShare article linkhttps://www.pnas.org/doi/10.1073/pnas.1901471116COPY LINKCopied!Copying failed.Share on social mediaFacebookTwitterLinkedinGmailemailReferences1T. J. Erb, Carboxylases in natural and synthetic microbial pathways. Appl. Environ. Microbiol. 77, 8466–8477 (2011).CrossrefPubMedGoogle Scholara [...] (0.04 vol.%) by enzymes called carboxylases b [...] molecule 2R. J. Ellis, Most abundant protein in the world. Trends Biochem. Sci. 4, 241–244 (1979).Go to referenceCrossrefGoogle Scholar3R. Phillips, R. Milo, A feeling for the numbers in biology. Proc. Natl. Acad. Sci. U.S.A. 106, 21465–21471 (2009).Go to referenceCrossrefPubMedGoogle Scholar4E. Alper, O. Y. Orhan, CO2 utilization: Developments in conversion processes. Petroleum 3, 109–126 (2017).Go to referenceCrossrefGoogle Scholar5M. Aresta, A. Dibenedetto, Utilisation of CO2 as a chemical feedstock: Opportunities and challenges. Dalton Trans., 2975–2992 (2007).Go to referenceCrossrefPubMedGoogle Scholar6L. Schada von Borzyskowski, R. G. Rosenthal, T. J. Erb, Evolutionary history and biotechnological future of carboxylases. J. Biotechnol. 168, 243–251 (2013).CrossrefPubMedGoogle Scholara [...] or ketone) by converting it into an enol(ate) b [...] catalyze the ordinary reduction of enoyl-CoA esters c [...] Ecrs evolved from a primordial enoyl-CoA reductase 7D. L. Edmondson, H. J. Kane, T. J. Andrews, Substrate isomerization inhibits ribulosebisphospate carboxylase-oxygenase during catalysis. FEBS Lett. 260, 62–66 (1990).Go to referenceCrossrefGoogle Scholar8G. Bowes, W. L. Ogren, R. H. Hageman, Phosphoglycolate production catalyzed by ribulose diphosphate carboxylase. Biochem. Biophys. Res. Commun. 45, 716–722 (1971).Go to referenceCrossrefPubMedGoogle Scholar9G. A. Prussia et al., Substitution of a conserved catalytic dyad into 2-KPCC causes loss of carboxylation activity. FEBS Lett. 590, 2991–2996 (2016).Go to referenceCrossrefPubMedGoogle Scholar10M. A. Kofoed, D. A. Wampler, A. S. Pandey, J. W. Peters, S. A. Ensign, Roles of the redox-active disulfide and histidine residues forming a catalytic dyad in reactions catalyzed by 2-ketopropyl coenzyme M oxidoreductase/carboxylase. J. Bacteriol. 193, 4904–4913 (2011).CrossrefPubMedGoogle Scholar11A. S. Pandey, D. W. Mulder, S. A. Ensign, J. W. Peters, Structural basis for carbon dioxide binding by 2-ketopropyl coenzyme M oxidoreductase/carboxylase. FEBS Lett. 585, 459–464 (2011).Go to referenceCrossrefPubMedGoogle Scholar12B. Stec, Structural mechanism of RuBisCO activation by carbamylation of the active site lysine. Proc. Natl. Acad. Sci. U.S.A. 109, 18785–18790 (2012).CrossrefPubMedGoogle Scholar13M. van Lun, J. S. Hub, D. van der Spoel, I. Andersson, CO2 and O2 distribution in Rubisco suggests the small subunit functions as a CO2 reservoir. J. Am. Chem. Soc. 136, 3165–3171 (2014).CrossrefPubMedGoogle Scholar14Y. Ito, H. Kondo, Y. Shiota, K. Yoshizawa, Theoretical analysis of the reaction mechanism of biotin carboxylase. J. Chem. Theory Comput. 4, 366–374 (2008).Go to referenceCrossrefPubMedGoogle Scholar15D. M. Peter et al., Screening and engineering the synthetic potential of carboxylating reductases from central metabolism and polyketide biosynthesis. Angew. Chem. Int. Ed. Engl. 54, 13457–13461 (2015).Go to referenceCrossrefPubMedGoogle Scholar16T. J. Erb et al., Synthesis of C5-dicarboxylic acids from C2-units involving crotonyl-CoA carboxylase/reductase: The ethylmalonyl-CoA pathway. Proc. Natl. Acad. Sci. U.S.A. 104, 10631–10636 (2007).CrossrefPubMedGoogle Scholara [...] carboxylases, enoyl-CoA carboxylases/reductases (Ecrs) b [...] Rhodobacter Sphaeroides 17T. J. Erb, V. Brecht, G. Fuchs, M. Müller, B. E. Alber, Carboxylation mechanism and stereochemistry of crotonyl-CoA carboxylase/reductase, a carboxylating enoyl-thioester reductase. Proc. Natl. Acad. Sci. U.S.A. 106, 8871–8876 (2009).CrossrefPubMedGoogle Scholara [...] and at low catalytic efficiency b [...] transfer from NADPH to t\n",
      "len(chunk):   transfer from NADPH to the β-position of crotonyl-CoA 18R. G. Rosenthal et al., Direct evidence for a covalent ene adduct intermediate in NAD(P)H-dependent enzymes. Nat. Chem. Biol. 10, 50–55 (2014).CrossrefPubMedGoogle Scholara [...] and at low catalytic efficiency b [...] spontaneous decay in the solvent, as described before c [...] is absent from the active site of WT Ecr 19R. G. Rosenthal et al., The use of ene adducts to study and engineer enoyl-thioester reductases. Nat. Chem. Biol. 11, 398–400 (2015).CrossrefPubMedGoogle Scholara [...] catalyze the ordinary reduction of enoyl-CoA esters b [...] transfer from NADPH to the β-position of crotonyl-CoA 20N. Quade, L. Huo, S. Rachid, D. W. Heinz, R. Müller, Unusual carbon fixation gives rise to diverse polyketide extender units. Nat. Chem. Biol. 8, 117–124 (2011).CrossrefPubMedGoogle Scholara [...] [Protein Data Bank (PDB); PDB ID code: 4A0S b [...] residues suppressed the carboxylation of octenoyl-CoA c [...] features with the CinF homolog previously reported d [...] ) [PDB ID codes: 4A0S e [...] from water had been previously suggested for Phe170 21L. Zhang et al., Rational control of polyketide extender units by structure-based engineering of a crotonyl-CoA carboxylase/reductase in antimycin biosynthesis. Angew. Chem. Int. Ed. Engl. 54, 13462–13465 (2015).Go to referenceCrossrefPubMedGoogle Scholar22B. Vögeli et al., InhA, the enoyl-thioester reductase from Mycobacterium tuberculosis forms a covalent adduct during catalysis. J. Biol. Chem. 293, 17200–17207 (2018).CrossrefPubMedGoogle Scholara [...] spontaneous decay in the solvent, as described before b [...] performed analogous to a previously described method 23R. D. Libby, R. A. Mehl, Characterization of covalent Ene adduct intermediates in “hydride equivalent” transfers in a dihydropyridine model for NADH reduction reactions. Bioorg. Chem. 40, 57–66 (2012).Go to referenceCrossrefPubMedGoogle Scholar24G. Tcherkez, Modelling the reaction mechanism of ribulose-1,5-bisphosphate carboxylase/oxygenase and consequences for kinetic parameters. Plant Cell Environ. 36, 1586–1596 (2013).Go to referenceCrossrefPubMedGoogle Scholar25G. G. Tcherkez, G. D. Farquhar, T. J. Andrews, Despite slow catalysis and confused substrate specificity, all ribulose bisphosphate carboxylases may be nearly perfectly optimized. Proc. Natl. Acad. Sci. U.S.A. 103, 7246–7251 (2006).Go to referenceCrossrefPubMedGoogle Scholar26T. J. Erb, J. Zarzycki, A short history of RubisCO: The rise and fall (?) of Nature’s predominant CO2 fixing enzyme. Curr. Opin. Biotechnol. 49, 100–107 (2018).Go to referenceCrossrefPubMedGoogle Scholar27A. M. Appel et al., Frontiers, opportunities, and challenges in biochemical and chemical catalysis of CO2 fixation. Chem. Rev. 113, 6621–6658 (2013).Go to referenceCrossrefPubMedGoogle Scholar28T. Schwander, L. Schada von Borzyskowski, S. Burgener, N. S. Cortina, T. J. Erb, A synthetic pathway for the fixation of carbon dioxide in vitro. Science 354, 900–904 (2016).Go to referenceCrossrefPubMedGoogle Scholar29A. Bar-Even, E. Noor, N. E. Lewis, R. Milo, Design and analysis of synthetic carbon fixation pathways. Proc. Natl. Acad. Sci. U.S.A. 107, 8889–8894 (2010).Go to referenceCrossrefPubMedGoogle Scholar30D. M. Peter, B. Vögeli, N. S. Cortina, T. J. Erb, A chemo-enzymatic road map to the synthesis of CoA esters. Molecules 21, 517 (2016).Go to referenceCrossrefPubMedGoogle ScholarView full textDownload PDF\n",
      "Further reading in this issue\n",
      "Research ArticleJune 21, 2019Multiplicative fitness, rapid haplotype discovery, and fitness decay explain evolution of human MHC\n",
      "Alexander E. Lobkovsky,\n",
      "Lee Levi,\n",
      "Yuri I. Wolf,\n",
      "Martin Maiers,\n",
      "Loren Gragert,\n",
      "Idan Alter,\n",
      "Yoram Louzoun,\n",
      "[...]Eugene V. Koonin,\n",
      "+5 authors\n",
      "This is an addendum toComplex viscosity of helical and doubly helical polymeric liquids from general rigid bead-rod theoryResearch ArticleJune 24, 2019Histone H2B monoubiquitination regulates heart development via epigenetic control of \n",
      "len(chunk):  ia epigenetic control of cilia motility\n",
      "Andrew Robson,\n",
      "Svetlana Z. Makova,\n",
      "Syndi Barish,\n",
      "Samir Zaidi,\n",
      "Sameet Mehta,\n",
      "Jeffrey Drozd,\n",
      "Sheng Chih Jin,\n",
      "Bruce D. Gelb,\n",
      "Christine E. Seidman,\n",
      "Wendy K. Chung,\n",
      "Richard P. Lifton,\n",
      "Mustafa K. Khokha,\n",
      "[...]Martina Brueckner,\n",
      "+10 authors\n",
      "This is an addendum toComplex viscosity of helical and doubly helical polymeric liquids from general rigid bead-rod theoryResearch ArticleJune 21, 2019Acculturation drives the evolution of intergroup conflict\n",
      "Gil J. B. Henriques,\n",
      "Burton Simon,\n",
      "Yaroslav Ispolatov,\n",
      "[...]Michael Doebeli,\n",
      "+1 authors\n",
      "This is an addendum toComplex viscosity of helical and doubly helical polymeric liquids from general rigid bead-rod theory\n",
      "Trending\n",
      "Research ArticleJuly 24, 2023ER chaperone GRP78/BiP translocates to the nucleus under stress and acts as a transcriptional regulatorEndoplasmic reticulum (ER) stress-mediated relocalization of ER chaperones to other\n",
      "cellular compartments allows cells to expand their functionality beyond the ER. Our\n",
      "finding that the ER luminal chaperone GRP78/BiP, commonly overexpressed in ...Cancer cells are commonly subjected to endoplasmic reticulum (ER) stress. To gain\n",
      "survival advantage, cancer cells exploit the adaptive aspects of the unfolded protein\n",
      "response such as upregulation of the ER luminal chaperone GRP78. The finding that\n",
      "when ...\n",
      "Ze Liu,\n",
      "Guanlin Liu,\n",
      "Dat P. Ha,\n",
      "Justin Wang,\n",
      "Min Xiong,\n",
      "[...]Amy S. Lee,\n",
      "+3 authors\n",
      "This is an addendum toComplex viscosity of helical and doubly helical polymeric liquids from general rigid bead-rod theoryPerspectiveAugust 1, 2022Climate Endgame: Exploring catastrophic climate change scenariosPrudent risk management requires consideration of bad-to-worst-case scenarios. Yet,\n",
      "for climate change, such potential futures are poorly understood. Could anthropogenic\n",
      "climate change result in worldwide societal collapse or even eventual human ...\n",
      "Luke Kemp,\n",
      "Chi Xu,\n",
      "Joanna Depledge,\n",
      "Kristie L. Ebi,\n",
      "Goodwin Gibbins,\n",
      "Timothy A. Kohler,\n",
      "Johan Rockström,\n",
      "Marten Scheffer,\n",
      "Hans Joachim Schellnhuber,\n",
      "Will Steffen,\n",
      "[...]Timothy M. Lenton,\n",
      "+8 authors\n",
      "This is an addendum toComplex viscosity of helical and doubly helical polymeric liquids from general rigid bead-rod theoryResearch ArticleMarch 9, 2010Atrazine induces complete feminization and chemical castration in male African clawed frogs (Xenopus laevis)The herbicide atrazine is one of the most commonly applied pesticides in the world.\n",
      "As a result, atrazine is the most commonly detected pesticide contaminant of ground,\n",
      "surface, and drinking water. Atrazine is also a potent endocrine disruptor that is\n",
      "...\n",
      "Tyrone B. Hayes,\n",
      "Vicky Khoury,\n",
      "Anne Narayan,\n",
      "Mariam Nazir,\n",
      "Andrew Park,\n",
      "Travis Brown,\n",
      "Lillian Adame,\n",
      "Elton Chan,\n",
      "Daniel Buchholz,\n",
      "Theresa Stueve,\n",
      "[...]Sherrie Gallipeau,\n",
      "+8 authors\n",
      "This is an addendum toComplex viscosity of helical and doubly helical polymeric liquids from general rigid bead-rod theory\n",
      "Recommended articlesResearch ArticleAugust 22, 2016pH determines the energetic efficiency of the cyanobacterial CO2 concentrating mechanismThis is an addendum toComplex viscosity of helical and doubly helical polymeric liquids from general rigid bead-rod theoryResearch ArticleApril 30, 2021Rubisco proton production can drive the elevation of CO2 within condensates and carboxysomesThis is an addendum toComplex viscosity of helical and doubly helical polymeric liquids from general rigid bead-rod theoryMay 9, 2006Despite slow catalysis and confused substrate specificity, all ribulose bisphosphate carboxylases may be nearly perfectly optimizedThis is an addendum toComplex viscosity of helical and doubly helical polymeric liquids from general rigid bead-rod theory\n",
      "Sign up for thePNAS Highlights newsletter\n",
      "Get in-depth science stories sent to your inbox twice a month.\n",
      "Subscribe\n",
      "Request Username\n",
      "x\n",
      "Close\n",
      "Can't sign in? Forgot your username?\n",
      "Enter your email address below and we will send you your username\n",
      "Email\n",
      "Submit\n",
      "If the address matches an existing account you will receive an email wit\n",
      "len(chunk):  will receive an email with instructions to retrieve your username\n",
      "Create a new account\n",
      "xClose\n",
      "Email\n",
      "Returning user\n",
      "Submit\n",
      "We use cookies on this site to enhance your user experience. By using this website, you are giving your consent for us to set cookies. Find out moreCONTINUE\n",
      "Login\n",
      "xClose\n",
      "Email*\n",
      "Password*\n",
      "Forgot your password? Reset it\n",
      "here\n",
      "Keep me logged in\n",
      "Login\n",
      "Don’t have an account? Register\n",
      "here\n",
      "Institutional Login\n",
      "Change Password\n",
      "xClose\n",
      "Enter the verification code\n",
      "Submit\n",
      "Congrats!\n",
      "Your Phone has been verified\n",
      "18001801180218031804\n",
      "Now Reading:Four amino acids define the CO2 binding pocket of enoyl-CoA carboxylases/reductasesShareFiguresTablesView figureScheme 1.Reaction catalyzed by Ccr. Carboxylation to (2S)-ethylmalonyl-CoA in the presence of CO2 and reduction to butyryl-CoA in the absence of CO2.View figureFig. 1.Structure of KsCcr complexed with NADPH and ethylmalonyl-CoA. (Left) KsCcr forms a dimer of dimers of open- and closed-form subunits. The subunits are highlighted in gray and green (closed form with NADPH and ethylmalonyl-CoA, both represented in spheres) and blue and orange (open form with NADPH only, represented in spheres). The rectangle represents close-up of the active site shown in the middle and right. (Middle) The active site with the CO2-binding residues His365, Glu171, Phe170, and Asn81 in green, ethylmalonyl-CoA (salmon), and NADPH (cyan); oxygen and nitrogen atoms are colored in red and blue, respectively. (Right) Same as Middle but rotated by 180° about the viewing direction.View figureFig. 2.Left, Middle, and Right columns represent the WT, N81L, and F170A variants respectively. (A, D, and G) Active site of KsCcr variants before the nucleophilic attack of the enolate onto CO2. An overlay of different conformations of the CO2 molecule visualizes the tumbling motion in the different enzyme variants. CoA-ester is shown in salmon and NADP+ in cyan, and the red sphere represents the conserved water molecule coordinated by His365 and Glu171. (B, E, and H) Hydration sites located within 5 Å of the Cα obtained with SSTMap. For each enzyme, representative structures of the reactant state were subjected to 1 ns of constrained quantum mechanics/molecular mechanics simulations. The different positions sampled by the water molecules allowed the determination of clusters showing the preferential location of the solvent within the active site. (C, F, and I) Minimum free-energy path from the C2-ene adduct through the enolate to the products (2S)-ethylmalonyl-CoA and NADP+. Parameter D represents the distance between Cα and C2 minus the distance between Cα and CO2.View figureFig. 3.Left and Right columns represent the E171A and H365 N variants, respectively. (A and D). Active site of KsCcr variants before the nucleophilic attack of the enolate onto CO2. The CoA-ester is shown in salmon and NADP+ in cyan. Different orientations of carbon dioxide in the simulations represent the tumbling motion of the molecule before product formation. (B and E) Hydration sites located within 5 Å of the Cα obtained with SSTMap. (C and F) Minimum free-energy paths from the C2-ene adduct through the enolate to the products (2S)-ethylmalonyl-CoA and NADP+. Parameter D represents the distance between Cα and C2 minus the distance between Cα and CO2.Table 1.Apparent steady-state parameters for KsCcr and its mutants expressed as mean value ± SEReference #1__(\"articleCrossmark.closePopup\")\n",
      "✓Thanks for sharing!AddToAnyMore…\n",
      "Chunk: 0 / 16\n",
      "seed_initial_question_decomposition_prompt \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract accurate and useful direct quotes from the source text that are relevant to answering the key questions and also output the quote's most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"<insert unique quote index>\": {\n",
      "    \"reasoning\": <one sentence explaining how this direct quote helps answer the key question>\n",
      "    \"quote\": \"<insert direct quote>\",\n",
      "    \"index\": <insert most relevant key question index>\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Source text: ContentsSignificanceAbstractResultsDiscussionMaterials and MethodsData AvailabilityAcknowledgmentsSupporting InformationReferencesInformation & AuthorsMetrics & CitationsView OptionsReferencesMediaShareSignificanceCarboxylases capture and convert CO2, which makes them key enzymes in photosynthesis and the global carbon cycle. However, the question how enzymes bind atmospheric CO2 is still unsolved. We studied enoyl-CoA carboxylases/reductases (Ecrs), the fastest CO2-fixing enzymes in nature, using structural biology, biochemistry, and advanced computational methods. Ecrs create a highly specific CO2-binding pocket with 4 amino acids at the active site. The pocket controls the fate of the gaseous molecule during catalysis and shields the catalytic center from oxygen and water. This exquisite control makes Ecrs highly efficient carboxylases outcompeting RuBisCO, the key enzyme of photosynthesis, by an order of magnitude. Our findings define the atomic framework for the future development of CO2-converting catalysts in biology and chemistry.AbstractCarboxylases are biocatalysts that capture and convert carbon dioxide (CO2) under mild conditions and atmospheric concentrations at a scale of more than 400 Gt annually. However, how these enzymes bind and control the gaseous CO2 molecule during catalysis is only poorly understood. One of the most efficient classes of carboxylating enzymes are enoyl-CoA carboxylases/reductases (Ecrs), which outcompete the plant enzyme RuBisCO in catalytic efficiency and fidelity by more than an order of magnitude. Here we investigated the interactions of CO2 within the active site of Ecr from Kitasatospora setae. Combining experimental biochemistry, protein crystallography, and advanced computer simulations we show that 4 amino acids, N81, F170, E171, and H365, are required to create a highly efficient CO2-fixing enzyme. Together, these 4 residues anchor and position the CO2 molecule for the attack by a reactive enolate created during the catalytic cycle. Notably, a highly ordered water molecule plays an important role in an active site that is otherwise carefully shielded from water, which is detrimental to CO2 fixation. Altogether, our study reveals unprecedented molecular details of selective CO2 binding and C–C-bond formation during the catalytic cycle of nature’s most efficient CO2-fixing enzyme. This knowledge provides the basis for the future development of catalytic frameworks for the capture and conversion of CO2 in biology and chemistry.\n",
      "Sign up for PNAS alerts.\n",
      "Get alerts for new articles, or get an alert when an article is cited.\n",
      "Manage alerts\n",
      "The efficient capture and conversion of atmospheric carbon dioxide (CO2) is a prerequisite to develop a carbon-neutral, circular future economy. In biology, carbon fixation is performed under mild conditions and at atmospheric concentrations of CO2 (0.04 vol.%) by enzymes called carboxylases (1). It is estimated that 400 Gt of CO2 are fixed annually by a single biocatalyst, ribulose-1,5-bisphophate-carboxylase/oxygenase (RuBisCO), the key enzyme of photosynthesis (2, 3). In comparison, the chemical conversion of CO2 in industry accounts for only 0.1 Gt of carbon annually and uses pressurized CO2 (4), which emphasizes our need to understand the molecular mechanism that allows (bio)catalysts to selectively interact with a low-concentrated CO2 molecule during catalysis.Carboxylases catalyze the formation of a C–C bond between an acceptor substrate and a CO2 molecule (1) where the latter represents the electrophile (5). To facilitate C–C bond formation, most carboxylases activate their respective nucleophilic substrate (usually a thioester, α-ketoacid, or ketone) by converting it into an enol(ate) (6). Enol(ate)s are strong nucleophiles and highly reactive. A key requirement of CO2-fixation catalysis is the tight control of the reaction between the activated acceptor substrate and CO2. Any loss of catalytic control over the enol(ate) or the CO2 molecule b\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Prompt:  \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract accurate and useful direct quotes from the source text that are relevant to answering the key questions and also output the quote's most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"<insert unique quote index>\": {\n",
      "    \"reasoning\": <one sentence explaining how this direct quote helps answer the key question>\n",
      "    \"quote\": \"<insert direct quote>\",\n",
      "    \"index\": <insert most relevant key question index>\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Source text: ContentsSignificanceAbstractResultsDiscussionMaterials and MethodsData AvailabilityAcknowledgmentsSupporting InformationReferencesInformation & AuthorsMetrics & CitationsView OptionsReferencesMediaShareSignificanceCarboxylases capture and convert CO2, which makes them key enzymes in photosynthesis and the global carbon cycle. However, the question how enzymes bind atmospheric CO2 is still unsolved. We studied enoyl-CoA carboxylases/reductases (Ecrs), the fastest CO2-fixing enzymes in nature, using structural biology, biochemistry, and advanced computational methods. Ecrs create a highly specific CO2-binding pocket with 4 amino acids at the active site. The pocket controls the fate of the gaseous molecule during catalysis and shields the catalytic center from oxygen and water. This exquisite control makes Ecrs highly efficient carboxylases outcompeting RuBisCO, the key enzyme of photosynthesis, by an order of magnitude. Our findings define the atomic framework for the future development of CO2-converting catalysts in biology and chemistry.AbstractCarboxylases are biocatalysts that capture and convert carbon dioxide (CO2) under mild conditions and atmospheric concentrations at a scale of more than 400 Gt annually. However, how these enzymes bind and control the gaseous CO2 molecule during catalysis is only poorly understood. One of the most efficient classes of carboxylating enzymes are enoyl-CoA carboxylases/reductases (Ecrs), which outcompete the plant enzyme RuBisCO in catalytic efficiency and fidelity by more than an order of magnitude. Here we investigated the interactions of CO2 within the active site of Ecr from Kitasatospora setae. Combining experimental biochemistry, protein crystallography, and advanced computer simulations we show that 4 amino acids, N81, F170, E171, and H365, are required to create a highly efficient CO2-fixing enzyme. Together, these 4 residues anchor and position the CO2 molecule for the attack by a reactive enolate created during the catalytic cycle. Notably, a highly ordered water molecule plays an important role in an active site that is otherwise carefully shielded from water, which is detrimental to CO2 fixation. Altogether, our study reveals unprecedented molecular details of selective CO2 binding and C–C-bond formation during the catalytic cycle of nature’s most efficient CO2-fixing enzyme. This knowledge provides the basis for the future development of catalytic frameworks for the capture and conversion of CO2 in biology and chemistry.\n",
      "Sign up for PNAS alerts.\n",
      "Get alerts for new articles, or get an alert when an article is cited.\n",
      "Manage alerts\n",
      "The efficient capture and conversion of atmospheric carbon dioxide (CO2) is a prerequisite to develop a carbon-neutral, circular future economy. In biology, carbon fixation is performed under mild conditions and at atmospheric concentrations of CO2 (0.04 vol.%) by enzymes called carboxylases (1). It is estimated that 400 Gt of CO2 are fixed annually by a single biocatalyst, ribulose-1,5-bisphophate-carboxylase/oxygenase (RuBisCO), the key enzyme of photosynthesis (2, 3). In comparison, the chemical conversion of CO2 in industry accounts for only 0.1 Gt of carbon annually and uses pressurized CO2 (4), which emphasizes our need to understand the molecular mechanism that allows (bio)catalysts to selectively interact with a low-concentrated CO2 molecule during catalysis.Carboxylases catalyze the formation of a C–C bond between an acceptor substrate and a CO2 molecule (1) where the latter represents the electrophile (5). To facilitate C–C bond formation, most carboxylases activate their respective nucleophilic substrate (usually a thioester, α-ketoacid, or ketone) by converting it into an enol(ate) (6). Enol(ate)s are strong nucleophiles and highly reactive. A key requirement of CO2-fixation catalysis is the tight control of the reaction between the activated acceptor substrate and CO2. Any loss of catalytic control over the enol(ate) or the CO2 molecule b\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"0\\\": {\\n    \\\"reasoning\\\": \\\"This quote provides information about the level of ECR enzyme activity in Kitsatospor setae bacteria.\\\",\\n    \\\"quote\\\": \\\"Ecrs create a highly specific CO2-binding pocket with 4 amino acids at the active site.\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"1\\\": {\\n    \\\"reasoning\\\": \\\"This quote compares the catalytic efficiency and fidelity of ECR enzymes to RuBisCO, the key enzyme of photosynthesis.\\\",\\n    \\\"quote\\\": \\\"which outcompete the plant enzyme RuBisCO in catalytic efficiency and fidelity by more than an order of magnitude.\\\",\\n    \\\"index\\\": 1\\n  },\\n  \\\"2\\\": {\\n    \\\"reasoning\\\": \\\"This quote explains the factors required to create a highly efficient CO2-fixing enzyme in Kitsatospor setae bacteria.\\\",\\n    \\\"quote\\\": \\\"4 amino acids, N81, F170, E171, and H365, are required to create a highly efficient CO2-fixing enzyme.\\\",\\n    \\\"index\\\": 2\\n  },\\n  \\\"3\\\": {\\n    \\\"reasoning\\\": \\\"This quote discusses the potential applications and implications of high ECR enzyme activity in Kitsatospor setae bacteria.\\\",\\n    \\\"quote\\\": \\\"This knowledge provides the basis for the future development of catalytic frameworks for the capture and conversion of CO2 in biology and chemistry.\\\",\\n    \\\"index\\\": 3\\n  }\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691165052,\n",
      "  \"id\": \"chatcmpl-7jrZcJdgUy4qBQUrBSqwe3mJKogRK\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 296,\n",
      "    \"prompt_tokens\": 1121,\n",
      "    \"total_tokens\": 1417\n",
      "  }\n",
      "}\n",
      "Extracted quotes:  {\n",
      "  \"0\": {\n",
      "    \"reasoning\": \"This quote provides information about the level of ECR enzyme activity in Kitsatospor setae bacteria.\",\n",
      "    \"quote\": \"Ecrs create a highly specific CO2-binding pocket with 4 amino acids at the active site.\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"1\": {\n",
      "    \"reasoning\": \"This quote compares the catalytic efficiency and fidelity of ECR enzymes to RuBisCO, the key enzyme of photosynthesis.\",\n",
      "    \"quote\": \"which outcompete the plant enzyme RuBisCO in catalytic efficiency and fidelity by more than an order of magnitude.\",\n",
      "    \"index\": 1\n",
      "  },\n",
      "  \"2\": {\n",
      "    \"reasoning\": \"This quote explains the factors required to create a highly efficient CO2-fixing enzyme in Kitsatospor setae bacteria.\",\n",
      "    \"quote\": \"4 amino acids, N81, F170, E171, and H365, are required to create a highly efficient CO2-fixing enzyme.\",\n",
      "    \"index\": 2\n",
      "  },\n",
      "  \"3\": {\n",
      "    \"reasoning\": \"This quote discusses the potential applications and implications of high ECR enzyme activity in Kitsatospor setae bacteria.\",\n",
      "    \"quote\": \"This knowledge provides the basis for the future development of catalytic frameworks for the capture and conversion of CO2 in biology and chemistry.\",\n",
      "    \"index\": 3\n",
      "  }\n",
      "}\n",
      "Chunk: 1 / 16\n",
      "seed_initial_question_decomposition_prompt \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract accurate and useful direct quotes from the source text that are relevant to answering the key questions and also output the quote's most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"<insert unique quote index>\": {\n",
      "    \"reasoning\": <one sentence explaining how this direct quote helps answer the key question>\n",
      "    \"quote\": \"<insert direct quote>\",\n",
      "    \"index\": <insert most relevant key question index>\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Source text: te) or the CO2 molecule bears the danger of side reactions and reduces the efficiency of carbon fixation (7). The most prominent example is RuBisCO, which is known to feature several side reactions, most notably an oxygenation reaction (8). One in every 5 turnovers RuBisCO will incorporate an oxygen (O2) molecule instead of CO2, which leads to the formation of 2-phosphoglycolate, a side product that is toxic to the cell and has to be recycled in an energy-demanding process, highlighting the need of carboxylases to control the reaction of the activated acceptor with CO2.Another challenge in this respect is the accessibility of water (or protic amino acids) to the active site of carboxylases. Protons are better electrophiles than the CO2 molecule, which can directly quench the enolate. As a consequence, it is not sufficient that carboxylases enrich a low abundant gaseous CO2 molecule; they also need to efficiently suppress any competing (re)protonation reactions. Altogether, these examples show that controlling the fate of CO2 at the molecular level is a crucial feature of carboxylases. However, to date only very limited biochemical, let alone structural, information on CO2 binding in carboxylases (and other proteins) is available, besides some theoretical considerations (9–14).Here, we focused on a class of carboxylases, enoyl-CoA carboxylases/reductases (Ecrs) (15, 16) that show the fastest turnover frequencies among all carboxylases to date and exclusively react with CO2 in the presence of O2. These features make Ecrs excellent model systems to understand the details of selective CO2 binding and C–C-bond formation in proteins.The best-studied Ecr is crotonyl-CoA carboxylase/reductase (Ccr) that catalyzes the NADPH-dependent reductive carboxylation of crotonyl-CoA into (2S)-ethylmalonyl-CoA. While Ccr does not show side reactivity with O2, the enzyme catalyzes the reduction of crotonyl-CoA to butyryl-CoA as a side reaction, but only in the absence of CO2 and at low catalytic efficiency (Scheme 1) (17, 18). It has been suggested that this side reaction is an evolutionary remnant of Ecrs, which are evolutionary related to enoyl thioester reductases that catalyze the ordinary reduction of enoyl-CoA esters (6, 19). Apparently, Ecrs evolved from simple reductases into reductive carboxylases by acquiring a CO2-fixation function along their evolutionary trajectory.Reaction catalyzed by Ccr. Carboxylation to (2S)-ethylmalonyl-CoA in the presence of CO2 and reduction to butyryl-CoA in the absence of CO2.Open in viewerIn a previous structural study, CinF, an Ecr from Streptomyces sp. JS360, was crystallized with NADP+ and octenoyl-CoA [Protein Data Bank (PDB); PDB ID code: 4A0S (20)]. A putative CO2 binding pocket was proposed to be composed of Asn77, Phe166, and Glu167, which are all highly conserved in Ecrs. It was suggested that CO2 is held in position by hydrogen bonding to Asn77 and Glu167 while Phe166 would undergo hydrophobic interactions with CO2. Mutation of these residues suppressed the carboxylation of octenoyl-CoA (20). However, the exact role of the individual residues directing and controlling the carboxylation reaction remains enigmatic: in particular, how the gaseous CO2 molecule is aligned at the active site and how the reduction side reaction is efficiently suppressed.Here we combine experimental biochemistry, protein crystallography, and computer simulations to define the molecular interactions of CO2 during C–C-bond formation at the active site of Ecrs. Our results suggest that 4 amino acids are sufficient to convert an ordinary reductase into a highly efficient carboxylase. Together, these 4 residues anchor and lock the CO2 molecule in a favorable position for the attack by the reactive enolate created during catalysis. Notably, a highly ordered water molecule plays an essential role in coordinating the CO2 molecule, while the active site is otherwise effectively shielded from water to suppress the reduction side re\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Prompt:  \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract accurate and useful direct quotes from the source text that are relevant to answering the key questions and also output the quote's most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"<insert unique quote index>\": {\n",
      "    \"reasoning\": <one sentence explaining how this direct quote helps answer the key question>\n",
      "    \"quote\": \"<insert direct quote>\",\n",
      "    \"index\": <insert most relevant key question index>\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Source text: te) or the CO2 molecule bears the danger of side reactions and reduces the efficiency of carbon fixation (7). The most prominent example is RuBisCO, which is known to feature several side reactions, most notably an oxygenation reaction (8). One in every 5 turnovers RuBisCO will incorporate an oxygen (O2) molecule instead of CO2, which leads to the formation of 2-phosphoglycolate, a side product that is toxic to the cell and has to be recycled in an energy-demanding process, highlighting the need of carboxylases to control the reaction of the activated acceptor with CO2.Another challenge in this respect is the accessibility of water (or protic amino acids) to the active site of carboxylases. Protons are better electrophiles than the CO2 molecule, which can directly quench the enolate. As a consequence, it is not sufficient that carboxylases enrich a low abundant gaseous CO2 molecule; they also need to efficiently suppress any competing (re)protonation reactions. Altogether, these examples show that controlling the fate of CO2 at the molecular level is a crucial feature of carboxylases. However, to date only very limited biochemical, let alone structural, information on CO2 binding in carboxylases (and other proteins) is available, besides some theoretical considerations (9–14).Here, we focused on a class of carboxylases, enoyl-CoA carboxylases/reductases (Ecrs) (15, 16) that show the fastest turnover frequencies among all carboxylases to date and exclusively react with CO2 in the presence of O2. These features make Ecrs excellent model systems to understand the details of selective CO2 binding and C–C-bond formation in proteins.The best-studied Ecr is crotonyl-CoA carboxylase/reductase (Ccr) that catalyzes the NADPH-dependent reductive carboxylation of crotonyl-CoA into (2S)-ethylmalonyl-CoA. While Ccr does not show side reactivity with O2, the enzyme catalyzes the reduction of crotonyl-CoA to butyryl-CoA as a side reaction, but only in the absence of CO2 and at low catalytic efficiency (Scheme 1) (17, 18). It has been suggested that this side reaction is an evolutionary remnant of Ecrs, which are evolutionary related to enoyl thioester reductases that catalyze the ordinary reduction of enoyl-CoA esters (6, 19). Apparently, Ecrs evolved from simple reductases into reductive carboxylases by acquiring a CO2-fixation function along their evolutionary trajectory.Reaction catalyzed by Ccr. Carboxylation to (2S)-ethylmalonyl-CoA in the presence of CO2 and reduction to butyryl-CoA in the absence of CO2.Open in viewerIn a previous structural study, CinF, an Ecr from Streptomyces sp. JS360, was crystallized with NADP+ and octenoyl-CoA [Protein Data Bank (PDB); PDB ID code: 4A0S (20)]. A putative CO2 binding pocket was proposed to be composed of Asn77, Phe166, and Glu167, which are all highly conserved in Ecrs. It was suggested that CO2 is held in position by hydrogen bonding to Asn77 and Glu167 while Phe166 would undergo hydrophobic interactions with CO2. Mutation of these residues suppressed the carboxylation of octenoyl-CoA (20). However, the exact role of the individual residues directing and controlling the carboxylation reaction remains enigmatic: in particular, how the gaseous CO2 molecule is aligned at the active site and how the reduction side reaction is efficiently suppressed.Here we combine experimental biochemistry, protein crystallography, and computer simulations to define the molecular interactions of CO2 during C–C-bond formation at the active site of Ecrs. Our results suggest that 4 amino acids are sufficient to convert an ordinary reductase into a highly efficient carboxylase. Together, these 4 residues anchor and lock the CO2 molecule in a favorable position for the attack by the reactive enolate created during catalysis. Notably, a highly ordered water molecule plays an essential role in coordinating the CO2 molecule, while the active site is otherwise effectively shielded from water to suppress the reduction side re\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"0\\\": {\\n    \\\"reasoning\\\": \\\"The source text mentions enoyl-CoA carboxylases/reductases (Ecrs) as a class of carboxylases that react with CO2 in the presence of O2.\\\",\\n    \\\"quote\\\": \\\"Here, we focused on a class of carboxylases, enoyl-CoA carboxylases/reductases (Ecrs) (15, 16) that show the fastest turnover frequencies among all carboxylases to date and exclusively react with CO2 in the presence of O2.\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"1\\\": {\\n    \\\"reasoning\\\": \\\"The source text mentions that Ecrs are excellent model systems to understand the details of selective CO2 binding and C\\u2013C-bond formation in proteins.\\\",\\n    \\\"quote\\\": \\\"These features make Ecrs excellent model systems to understand the details of selective CO2 binding and C\\u2013C-bond formation in proteins.\\\",\\n    \\\"index\\\": 1\\n  },\\n  \\\"2\\\": {\\n    \\\"reasoning\\\": \\\"The source text mentions that the exact role of individual residues directing and controlling the carboxylation reaction in Ecrs remains enigmatic.\\\",\\n    \\\"quote\\\": \\\"However, the exact role of the individual residues directing and controlling the carboxylation reaction remains enigmatic: in particular, how the gaseous CO2 molecule is aligned at the active site and how the reduction side reaction is efficiently suppressed.\\\",\\n    \\\"index\\\": 2\\n  }\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691165062,\n",
      "  \"id\": \"chatcmpl-7jrZm2SFV42ZwsFvDKoI1Ky0IcRvC\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 315,\n",
      "    \"prompt_tokens\": 1195,\n",
      "    \"total_tokens\": 1510\n",
      "  }\n",
      "}\n",
      "Extracted quotes:  {\n",
      "  \"0\": {\n",
      "    \"reasoning\": \"The source text mentions enoyl-CoA carboxylases/reductases (Ecrs) as a class of carboxylases that react with CO2 in the presence of O2.\",\n",
      "    \"quote\": \"Here, we focused on a class of carboxylases, enoyl-CoA carboxylases/reductases (Ecrs) (15, 16) that show the fastest turnover frequencies among all carboxylases to date and exclusively react with CO2 in the presence of O2.\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"1\": {\n",
      "    \"reasoning\": \"The source text mentions that Ecrs are excellent model systems to understand the details of selective CO2 binding and C–C-bond formation in proteins.\",\n",
      "    \"quote\": \"These features make Ecrs excellent model systems to understand the details of selective CO2 binding and C–C-bond formation in proteins.\",\n",
      "    \"index\": 1\n",
      "  },\n",
      "  \"2\": {\n",
      "    \"reasoning\": \"The source text mentions that the exact role of individual residues directing and controlling the carboxylation reaction in Ecrs remains enigmatic.\",\n",
      "    \"quote\": \"However, the exact role of the individual residues directing and controlling the carboxylation reaction remains enigmatic: in particular, how the gaseous CO2 molecule is aligned at the active site and how the reduction side reaction is efficiently suppressed.\",\n",
      "    \"index\": 2\n",
      "  }\n",
      "}\n",
      "Chunk: 2 / 16\n",
      "seed_initial_question_decomposition_prompt \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract accurate and useful direct quotes from the source text that are relevant to answering the key questions and also output the quote's most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"<insert unique quote index>\": {\n",
      "    \"reasoning\": <one sentence explaining how this direct quote helps answer the key question>\n",
      "    \"quote\": \"<insert direct quote>\",\n",
      "    \"index\": <insert most relevant key question index>\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Source text: ess the reduction side reaction. Altogether our computational and experimental studies reveal the details of selective CO2 binding and C–C-bond formation in the catalytic cycle of nature’s most efficient CO2-fixing enzyme.ResultsCrystal Structure of KsCcr with Ethylmalonyl-CoA and NADPH.We solved the structure of Ccr from Kitasatospora setae (KsCcr) cocrystallized with NADPH and soaked with ethylmalonyl-CoA, the product of the carboxylation reaction, at 1.7-Å resolution (Fig. 1 and SI Appendix, Table S1, PDB ID code: 6OWE). The active site of KsCcr shares similar features with the CinF homolog previously reported (20) [PDB ID codes: 4A0S (20), 4Y0K (21), and 4GI2]. In contrast to these structures that did not capture the interaction of the protein with CO2, neither as free gas or covalently bound to the acyl-CoA moiety, our structure shows densities that can be interpreted as carboxylated product (SI Appendix, Fig. S1). This enabled us to identify 4 residues that potentially interact with CO2, namely Asn81, Phe170, Glu171, and His365. His365 also interacts with NADPH via hydrogen bonding (3.0 Å) to the carboxamide group of the nicotinamide, indicating a second function of His365 in coordinating the NADPH cofactor during catalysis. Similar to previously published structures we also observed an ordered water molecule between His365 and Glu171 at a distance of 2.9 and 2.7 Å, respectively. This feature is absent in structures lacking substrate and cofactor (PDB ID codes: 3HZZ and 3KRT) suggesting a role of the water molecule in the active enzyme complex. To test the function of these residues during catalysis, we characterized different active site variants and addressed the reaction mechanism with molecular-dynamics simulations along the minimum free-energy path within the quantum mechanics/molecular mechanics methodology.Structure of KsCcr complexed with NADPH and ethylmalonyl-CoA. (Left) KsCcr forms a dimer of dimers of open- and closed-form subunits. The subunits are highlighted in gray and green (closed form with NADPH and ethylmalonyl-CoA, both represented in spheres) and blue and orange (open form with NADPH only, represented in spheres). The rectangle represents close-up of the active site shown in the middle and right. (Middle) The active site with the CO2-binding residues His365, Glu171, Phe170, and Asn81 in green, ethylmalonyl-CoA (salmon), and NADPH (cyan); oxygen and nitrogen atoms are colored in red and blue, respectively. (Right) Same as Middle but rotated by 180° about the viewing direction.Open in viewerKinetic Characterization of KsCcr Wild Type.KsCcr wild type (WT) showed an apparent turnover frequency (kcat) of 103 ± 3 s−1 which is well in line with previously reported value of 104 s−1 for the Ccr of Rhodobacter Sphaeroides (16). Apparent KM values were 21 ± 2 µM (crotonyl-CoA), 37 ± 4 µM (NADPH), and 90 ± 10 µM (CO2), respectively, and substrate inhibition for crotonyl-CoA was observed at a Ki of 3,650 ± 810 µM. Under saturating amounts of CO2, the enzyme showed 100% carboxylation activity and exclusively formed (2S)-ethylmalonyl-CoA. In the absence of CO2, the enzyme catalyzed the reduction of crotonyl-CoA to butyryl-CoA (Table 1). Stereochemical analysis of the butyryl-CoA in D2O showed that 94 ± 2% of the deuterium label was retained (Table 1), demonstrating that the reduction side reaction took place in a stereospecific manner.Apparent steady-state parameters for KsCcr and its mutants expressed as mean value ± SEEnzymeSubstrateKM, µMKi, µMkcat, s−1% EMC*% Label retention†WTCrotonyl-CoA21 ± 23,650 ± 810103 ± 310094.3 ± 1.8 NADPH37 ± 4—86 ± 2 CO2‡90 ± 10—78 ± 2N81LCrotonyl-CoAND§NDND1958.9 ± 1.1 NADPHNDNDNDF170YCrotonyl-CoA10 ± 1558 ± 8083 ± 4100ND NADPH36 ± 3—56 ± 1 CO2‡150 ± 20—56 ± 2F170ACrotonyl-CoA31 ± 6—8.3 ± 0.41787 ± 0.6 NADPH11 ± 0.6—11 ± 0.1H365NCrotonyl-CoA29.8 ± 4.2—5.0 ± 0.29363.3 ± 0.5 NADPH22 ± 2—8.1 ± 0.3 CO2‡1310 ± 220—7.4 ± 0.7E171ACrotonyl-CoA500 ± 62—5.1 ± 0.29791.4 ± 0.2 NADPH112 ± 8—6.0\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Prompt:  \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract accurate and useful direct quotes from the source text that are relevant to answering the key questions and also output the quote's most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"<insert unique quote index>\": {\n",
      "    \"reasoning\": <one sentence explaining how this direct quote helps answer the key question>\n",
      "    \"quote\": \"<insert direct quote>\",\n",
      "    \"index\": <insert most relevant key question index>\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Source text: ess the reduction side reaction. Altogether our computational and experimental studies reveal the details of selective CO2 binding and C–C-bond formation in the catalytic cycle of nature’s most efficient CO2-fixing enzyme.ResultsCrystal Structure of KsCcr with Ethylmalonyl-CoA and NADPH.We solved the structure of Ccr from Kitasatospora setae (KsCcr) cocrystallized with NADPH and soaked with ethylmalonyl-CoA, the product of the carboxylation reaction, at 1.7-Å resolution (Fig. 1 and SI Appendix, Table S1, PDB ID code: 6OWE). The active site of KsCcr shares similar features with the CinF homolog previously reported (20) [PDB ID codes: 4A0S (20), 4Y0K (21), and 4GI2]. In contrast to these structures that did not capture the interaction of the protein with CO2, neither as free gas or covalently bound to the acyl-CoA moiety, our structure shows densities that can be interpreted as carboxylated product (SI Appendix, Fig. S1). This enabled us to identify 4 residues that potentially interact with CO2, namely Asn81, Phe170, Glu171, and His365. His365 also interacts with NADPH via hydrogen bonding (3.0 Å) to the carboxamide group of the nicotinamide, indicating a second function of His365 in coordinating the NADPH cofactor during catalysis. Similar to previously published structures we also observed an ordered water molecule between His365 and Glu171 at a distance of 2.9 and 2.7 Å, respectively. This feature is absent in structures lacking substrate and cofactor (PDB ID codes: 3HZZ and 3KRT) suggesting a role of the water molecule in the active enzyme complex. To test the function of these residues during catalysis, we characterized different active site variants and addressed the reaction mechanism with molecular-dynamics simulations along the minimum free-energy path within the quantum mechanics/molecular mechanics methodology.Structure of KsCcr complexed with NADPH and ethylmalonyl-CoA. (Left) KsCcr forms a dimer of dimers of open- and closed-form subunits. The subunits are highlighted in gray and green (closed form with NADPH and ethylmalonyl-CoA, both represented in spheres) and blue and orange (open form with NADPH only, represented in spheres). The rectangle represents close-up of the active site shown in the middle and right. (Middle) The active site with the CO2-binding residues His365, Glu171, Phe170, and Asn81 in green, ethylmalonyl-CoA (salmon), and NADPH (cyan); oxygen and nitrogen atoms are colored in red and blue, respectively. (Right) Same as Middle but rotated by 180° about the viewing direction.Open in viewerKinetic Characterization of KsCcr Wild Type.KsCcr wild type (WT) showed an apparent turnover frequency (kcat) of 103 ± 3 s−1 which is well in line with previously reported value of 104 s−1 for the Ccr of Rhodobacter Sphaeroides (16). Apparent KM values were 21 ± 2 µM (crotonyl-CoA), 37 ± 4 µM (NADPH), and 90 ± 10 µM (CO2), respectively, and substrate inhibition for crotonyl-CoA was observed at a Ki of 3,650 ± 810 µM. Under saturating amounts of CO2, the enzyme showed 100% carboxylation activity and exclusively formed (2S)-ethylmalonyl-CoA. In the absence of CO2, the enzyme catalyzed the reduction of crotonyl-CoA to butyryl-CoA (Table 1). Stereochemical analysis of the butyryl-CoA in D2O showed that 94 ± 2% of the deuterium label was retained (Table 1), demonstrating that the reduction side reaction took place in a stereospecific manner.Apparent steady-state parameters for KsCcr and its mutants expressed as mean value ± SEEnzymeSubstrateKM, µMKi, µMkcat, s−1% EMC*% Label retention†WTCrotonyl-CoA21 ± 23,650 ± 810103 ± 310094.3 ± 1.8 NADPH37 ± 4—86 ± 2 CO2‡90 ± 10—78 ± 2N81LCrotonyl-CoAND§NDND1958.9 ± 1.1 NADPHNDNDNDF170YCrotonyl-CoA10 ± 1558 ± 8083 ± 4100ND NADPH36 ± 3—56 ± 1 CO2‡150 ± 20—56 ± 2F170ACrotonyl-CoA31 ± 6—8.3 ± 0.41787 ± 0.6 NADPH11 ± 0.6—11 ± 0.1H365NCrotonyl-CoA29.8 ± 4.2—5.0 ± 0.29363.3 ± 0.5 NADPH22 ± 2—8.1 ± 0.3 CO2‡1310 ± 220—7.4 ± 0.7E171ACrotonyl-CoA500 ± 62—5.1 ± 0.29791.4 ± 0.2 NADPH112 ± 8—6.0\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"0\\\": {\\n    \\\"reasoning\\\": \\\"This quote provides information about the ECR enzyme activity in Kitsatospor setae bacteria.\\\",\\n    \\\"quote\\\": \\\"KsCcr wild type (WT) showed an apparent turnover frequency (kcat) of 103 \\u00b1 3 s\\u22121 which is well in line with previously reported value of 104 s\\u22121 for the Ccr of Rhodobacter Sphaeroides (16).\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"1\\\": {\\n    \\\"reasoning\\\": \\\"This quote compares the ECR enzyme activity in Kitsatospor setae bacteria to other bacteria.\\\",\\n    \\\"quote\\\": \\\"KsCcr wild type (WT) showed an apparent turnover frequency (kcat) of 103 \\u00b1 3 s\\u22121 which is well in line with previously reported value of 104 s\\u22121 for the Ccr of Rhodobacter Sphaeroides (16).\\\",\\n    \\\"index\\\": 1\\n  },\\n  \\\"2\\\": {\\n    \\\"reasoning\\\": \\\"This quote discusses the factors that potentially influence ECR enzyme activity in Kitsatospor setae bacteria.\\\",\\n    \\\"quote\\\": \\\"Apparent KM values were 21 \\u00b1 2 \\u00b5M (crotonyl-CoA), 37 \\u00b1 4 \\u00b5M (NADPH), and 90 \\u00b1 10 \\u00b5M (CO2), respectively, and substrate inhibition for crotonyl-CoA was observed at a Ki of 3,650 \\u00b1 810 \\u00b5M.\\\",\\n    \\\"index\\\": 2\\n  },\\n  \\\"3\\\": {\\n    \\\"reasoning\\\": \\\"This quote provides information about the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria.\\\",\\n    \\\"quote\\\": \\\"In the absence of CO2, the enzyme catalyzed the reduction of crotonyl-CoA to butyryl-CoA (Table 1). Stereochemical analysis of the butyryl-CoA in D2O showed that 94 \\u00b1 2% of the deuterium label was retained (Table 1), demonstrating that the reduction side reaction took place in a stereospecific manner.\\\",\\n    \\\"index\\\": 3\\n  }\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691165075,\n",
      "  \"id\": \"chatcmpl-7jrZzNnH5xEBXm2BrW6SnkHmcJRUf\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 456,\n",
      "    \"prompt_tokens\": 1462,\n",
      "    \"total_tokens\": 1918\n",
      "  }\n",
      "}\n",
      "Extracted quotes:  {\n",
      "  \"0\": {\n",
      "    \"reasoning\": \"This quote provides information about the ECR enzyme activity in Kitsatospor setae bacteria.\",\n",
      "    \"quote\": \"KsCcr wild type (WT) showed an apparent turnover frequency (kcat) of 103 ± 3 s−1 which is well in line with previously reported value of 104 s−1 for the Ccr of Rhodobacter Sphaeroides (16).\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"1\": {\n",
      "    \"reasoning\": \"This quote compares the ECR enzyme activity in Kitsatospor setae bacteria to other bacteria.\",\n",
      "    \"quote\": \"KsCcr wild type (WT) showed an apparent turnover frequency (kcat) of 103 ± 3 s−1 which is well in line with previously reported value of 104 s−1 for the Ccr of Rhodobacter Sphaeroides (16).\",\n",
      "    \"index\": 1\n",
      "  },\n",
      "  \"2\": {\n",
      "    \"reasoning\": \"This quote discusses the factors that potentially influence ECR enzyme activity in Kitsatospor setae bacteria.\",\n",
      "    \"quote\": \"Apparent KM values were 21 ± 2 µM (crotonyl-CoA), 37 ± 4 µM (NADPH), and 90 ± 10 µM (CO2), respectively, and substrate inhibition for crotonyl-CoA was observed at a Ki of 3,650 ± 810 µM.\",\n",
      "    \"index\": 2\n",
      "  },\n",
      "  \"3\": {\n",
      "    \"reasoning\": \"This quote provides information about the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria.\",\n",
      "    \"quote\": \"In the absence of CO2, the enzyme catalyzed the reduction of crotonyl-CoA to butyryl-CoA (Table 1). Stereochemical analysis of the butyryl-CoA in D2O showed that 94 ± 2% of the deuterium label was retained (Table 1), demonstrating that the reduction side reaction took place in a stereospecific manner.\",\n",
      "    \"index\": 3\n",
      "  }\n",
      "}\n",
      "Max tokens reached in chunks!\n",
      "len(chunk):  Figure 1 - available via license: Creative Commons Attribution-NonCommercial-NoDerivatives 4.0 InternationalContent may be subject to copyright.DownloadView publicationCopy referenceCopy captionEmbed figureReaction scheme and structural organization of the K. setae ECR complex. (a) Carboxylation reaction scheme of ECR. (b) Anisotropic B-factors of the tetramer of the different ECR complexes solved in this study are shown color coded according to the B factors (blue for low values and red for high values).Source publication\n",
      "Intersubunit Coupling Enables Fast CO 2 -Fixation by Reductive CarboxylasesArticleFull-text availableApr 2022 Hasan Demirci Yashas Rao Gabriele Stoffel[...] Soichi WakatsukiEnoyl-CoA carboxylases/reductases (ECRs) are some of the most efficient CO2-fixing enzymes described to date. However, the molecular mechanisms underlying the extraordinary catalytic activity of ECRs on the level of the protein assembly remain elusive. Here we used a combination of ambient-temperature X-ray free electron laser (XFEL) and cryogenic...CiteDownload full-textContexts in source publicationContext 1... ECRs catalyze the reduction of various α,β-unsaturated enoylCoAs into the corresponding alkylmalonyl-CoA esters using the reduced form of the cofactor nicotinamide adenine dinucleotide phosphate (NADPH). Hydride transfer from NADPH to the substrate generates a reactive enolate species, which acts as a nucleophile that attacks a bound CO 2 molecule ( Figure 1a). 2,5−7 However, the structural details of the carboxylation reaction have remained elusive, in part due to the lack of high-resolution structures of ECRs containing catalytic intermediates and carboxylated products. ...View in full-textContext 2... we aimed at providing a mechanistic understanding of the fast carboxylation reaction of the K. setae ECR on the level of the overall oligomeric complex. We first determined four highresolution K. setae ECR structures in different conformational states: the apo form and three holo forms in a binary complex with the reduced cofactor NADPH, in a ternary complex with NADPH and butyryl-CoA, and a binary complex with the oxidized cofactor NADP + (Figure 1b). ...View in full-textContext 3... asymmetric unit contains one homotetramer composed of four subunits arranged in a pair of dimers geometry similar to those of previously reported binary and ternary ECR complexes, crotonyl-CoA carboxylase/reductase AntE in complex with NADP + (PDB: 4Y0K) and 2-octenoyl-CoA carboxylase/reductase CinF in complex with NADP + and 2-octenoyl-CoA, (PDB: 4A0S), respectively. Overall, the K. setae ECR tetramer shows a noncrystallographic, close to D 2 (dihedral) symmetry with four conformationally identical subunits ( Figures S1 and S2). The tetrameric oligomer state of the ECR apo enzyme is further supported by size-exclusion chromatography, which showed that the 51.2 kDa protein eluted as a single peak at 205 kDa ( Figure S3). ...View in full-textContext 4... postulate that the same crystallization conditions used for both structures might have resulted in the same space group (P2 1 2 1 2 1 ) with similar crystal contacts, which might have kept the NADP + bound ECR tetramer in the apo form. The NADP + -bound subunits are less rigid in comparison, and some residues show greater variations (Figures S1 and S2). ...View in full-textContext 5... NADPH binding breaks the dihedral D 2 symmetry observed in the apo-form tetramer, while symmetry about the y axis is retained, resulting in a noncrystallographic, almost cyclic C 2 symmetry ( Figure S1). In the NADPH*ECR binary complex, each of the two dimers, A/C and B/D, loses its 2-fold symmetry, giving rise to closed (A and B) and open (C and D) subunits ( Figure S1). ...View in full-textContext 6... NADPH binding breaks the dihedral D 2 symmetry observed in the apo-form tetramer, while symmetry about the y axis is retained, resulting in a noncrystallographic, almost cyclic C 2 symmetry ( Figure S1). In the NADPH*E\n",
      "len(chunk):  igure S1). In the NADPH*ECR binary complex, each of the two dimers, A/C and B/D, loses its 2-fold symmetry, giving rise to closed (A and B) and open (C and D) subunits ( Figure S1). In the A and B subunits, the cofactor binding pocket is compressed inward, which seals the NADPH cofactor within the catalytic domain, resulting in a \"closed-form\" state ( Figure 2b). ...View in full-textContext 7... next investigated the effects of substrate binding to the K. setae ECR tetramer complex and determined its ternary complex structure with the substrate analogue butyryl-CoA and NADPH at 1.7 Å resolution ( Figure 3). The structure is overall very similar to the conformation of the ECR*NADPH binary complex with the noncrystallographic, pseudo-C 2 cyclic symmetry ( Figure 3a and Figure S1). It comprises two sets of open-and closed-form subunits that overlay very well with those of the ECR*NADPH binary complex ( Figures S1 and S2). ...View in full-textContext 8... structure is overall very similar to the conformation of the ECR*NADPH binary complex with the noncrystallographic, pseudo-C 2 cyclic symmetry ( Figure 3a and Figure S1). It comprises two sets of open-and closed-form subunits that overlay very well with those of the ECR*NADPH binary complex ( Figures S1 and S2). The NADPH cofactor appears to be bound to all active sites; however, only the closed-form subunits A and B also contain electron density for the complete butyryl-CoA thioester, while the open-form subunits (C and D) show only the adenine group electron density of the butyrylCoA thioester (Figure 3a,b). ...View in full-textContext 9... we inspected CoA-thioester binding in the open-form subunits, we also observed the electron density in the adenine binding pocket of the neighboring closed form subunits (A or B), strongly indicating that the CoAthioester was bound (Figure 3g, right inset). However, the electron density beyond the adenine ring becomes invisible, suggesting that the part of the CoA molecule reaching into the open-form subunit's active site remains flexible and disordered, which is corroborated by the higher B-factors of the catalytic domain ( Figure 1, bottom right). ...View in full-textContext 10... characterize the dynamics, we compared the projections of all 10 dimer trajectories on PC1 and PC2. Dimer A/C opens its closed subunit first, associated with PC1, followed by a twist motion characterized by PC2 in some of the simulations ( Figure 4d and Figure S10). This twist motion rotates the catalytic domain of the empty, now open form active site in subunit A, which would push a bound crotonyl-CoA substrate toward the NADPH cofactor in the C subunit. ...View in full-textContext 11... crotonyl-CoA from subunit A, we expected the�now empty�\"closed\" active site subunit to transition to the open form, inducing the twist motion in the A/C dimer observed before, and the other closed subunit B still bearing the substrate remains closed, if the two dynamics of the pair of dimers are uncoupled. The projection of 10 100 ns trajectories on PC1 and PC2 confirmed that the empty active site (A in the A/C dimer) undergoes a conformational change to the open form (Figure 4d), whereas the dimer containing the substrate (B in the B/D dimer) remained in a closed conformation (see Figure S11 for a comparison to the simulation with two substrates maintaining the closed state). Importantly, however, none of the A/C dimer trajectories showed a significant change in PC2 associated with the twist motion observed in the binary ECR*NADPH complex, indicating that the twist motion is only initiated once the other dimer B/D adopts the open conformation. ...View in full-textContext 12... ± 25 1.1 ± 0.0 26 ± 3 0.7 ± 0.0 440 ± 40 0.9 ± 0.0 a Michaelis−Menten curves of K. setae ECR and its variants are provided in Figure S14. ...View in full-textContext 13... ± 130 38 ± 7 2980 ± 510 0.3 ± 0.0 a Michaelis−Menten curves of K. setae ECR and its variants are provided in Figure S14. synchrony: i.\n",
      "len(chunk):  Figure S14. synchrony: i.e., when one dimer switches from the open−close to the closed−open states, the other dimer would do the opposite, closed−open to open−closed states. ...View in full-textContext 14... i.e., when one dimer switches from the open−close to the closed−open states, the other dimer would do the opposite, closed−open to open−closed states. Using this model, we simulated the case with just one dimer with two subunits working in negative or positive cooperativity using the method described by Hill and Levitski 21 ( Figure S12). For a thorough investigation of how cooperativity, positive or negative, enhances the overall reaction rate of the K. setae ECR, we incorporated all the parameters in Hill and Levitski's model in our simulation. ...View in full-textContext 15... turned out that only negative cooperativity could explain our experimental observations that ECR activity in the WT tetramer is increased by 2 orders of magnitude in comparison to the triple variant E151D/N157E/N218E, in which synchronization is decoupled (Table 1). Our simulations yielded a 90% probability that both active sites are occupied with substrates/products and NADPH/NADP + (p22), while only 10% are half-occupied (p12 + p21) ( Figure S12). Almost none of the dimers are empty (p11), which agrees with the occupancy of our experimentally determined structures of the ternary complex. ...View in full-textContext 16... While our structural, biochemical, and simulation data indicate that the K. setae ECR achieves high catalytic rates by synchronizing active sites, this might not necessarily be true for other ECRs. A differentiation into a pair of dimers was not observed in NADPH-bound or ternary structures of other ECRs (e.g., PDB 4Y0K and 4A0S, respectively, which share substantial amino acid identity) ( Figure S13). Note, however, that ECRs fall into two different classes: primary ECRs (such as the K. setae ECR) that operate in central carbon metabolism and secondary ECRs (such as 4Y0K) that provide extender units for synthesizing polyketides in secondary metabolism. ...View in full-textCitations... X-ray structures with a cofactor and a substrate analogue at different stages of the catalytic cycle reveal that the active site adopts either an open or closed conformation in a dimer of dimers configuration. 14 The catalytically competent form is the closed conformation containing the substrate analogue and NADPH (see Figure S1 in SI). The other active site in the dimer is in the open conformation containing only the NADPH cofactor.\n",
      "...Covalent Adduct Formation as a Strategy for Efficient CO 2 Fixation in Crotonyl-CoA Carboxylases/ReductasesArticleApr 2023 Rodrigo Recabarren Matthias Tinzl David Adrian Saez Aharon Gomez Esteban Vöhringer-MartinezView... X-ray structures with cofactor and a substrate analogue at different stages of the catalytic cycle reveal that the active site adopts either an open or closed conformation in a dimer of dimers configuration. 14 The catalytically competent form is the closed conformation containing the substrate analogue and NADPH (see Figure S1 in SI). The other active site in the dimer is in the open conformation containing only the NADPH cofactor.\n",
      "...Covalent adduct formation as strategy for efficient CO2 fixation in crotonyl-CoA carboxylases/reductasesPreprintFull-text availableApr 2023 Rodrigo Recabarren Matthias Tinzl David Saez Aharon Gomez Esteban Vöhringer-MartinezIncreasing levels of CO2 in the atmosphere have led to a growing interest into the various ways nature transforms this greenhouse gas into valuable organic compounds. Crotonyl-CoA carboxylases/reductases (Ccr's) are the most efficient biocatalysts for CO2 fixation because of their oxygen tolerance, their high catalytic rate constants and their high fidelity. The reaction mechanism involving hydride transfer from the NADPH cofactor and addition of CO2 to the reactive enolate, however, is not completely understood. In this study, we use computer s\n",
      "len(chunk):   study, we use computer simulations in combination with high-level ab initio calculations to trace the free energy landscape along two possible reaction paths: In the direct mechanism hydride transfer is immediately followed by CO2 addition whereas in the C2 mechanism a thermodynamically stable covalent adduct between the substrate and the NADPH cofactor is formed. This C2 adduct, which has been previously characterized experimentally, serves as a stable intermediate avoiding the reduction side reaction of the reactive enolate species and it is able to react with CO2 with similar kinetics as the direct reaction mechanism as confirmed by measured kinetic isotope effects. Thus, our results show that nature's most efficient CO2-fixing enzyme uses the formation of a covalent adduct as a strategy to store the reactive enolate species. The emerging microscopic picture of the CO2-fixing mechanism confirms previous experimental observations and provides new insights into how nature handles highly reactive intermediates to fix this inert greenhouse gas.View... It is composed of a pair of dimers that take open and closed configurations, undergoing significant conformational rearrangement to achieve a catalytically active state. 1 Insight into the mode of action of ECRs, as well as the chemistry involved, has been limited by a dearth of high-resolution structures of ECRs in complex with catalytic intermediates and carboxylated products. The authors present four high-resolution structures of the K. setae ECR, one in the unliganded apo form, and three liganded forms of the enzyme: two binary complexes with reduced NADPH and oxidized NADP + , respectively, as well as a ternary complex with NADPH and butyryl-CoA.\n",
      "...A Structural View into the Complexity of Carbon Dioxide FixationArticleFull-text availableMay 2022 Robert Kourist Shina Caroline Lynn KamerlinViewOpportunities and avenues for achieving crop climate resilienceArticleJun 2023ENVIRON EXP BOT Tinashe Zenda Nan WangXiaocui YanAnyi Dong Huijun DuanViewCovalent adduct formation as strategy for efficient CO2 fixation in crotonyl-CoA carboxylases/reductasesPreprintFull-text availableDec 2022 Rodrigo Recabarren Matthias Tinzl David Saez Aharon Gomez Esteban Vöhringer-MartinezIncreasing levels of CO2 in the atmosphere have led to a growing interest into the various ways nature transforms this greenhouse gas in valuable organic compounds. Crotonyl-CoA carboxylases/reductases (Ccrs) are the most efficient biocatalysts for CO2 fixation because of their oxygen tolerance, their high catalytic rate constants and their high fidelity. The reaction mechanism involving hydride transfer from the NADPH cofactor and addition of CO2 to the reactive enolate, however, is not completely understood. In this study, we use computer simulations in combination with high-level ab initio calculations to trace the free energy landscape along two possible reaction paths and identify a thermodynamically stable covalent adduct formed between the substrate and the NADPH cofactor in agreement with experimental observation. This C2 adduct serves as a stable intermediate avoiding the reduction side reaction of the reactive enolate species and it is able to react with CO2 with similar kinetics as the direct reaction mechanism as confirmed by measured kinetic isotope effects. Thus, our results show that nature's most efficient CO2-fixing enzyme uses the formation of a covalent adduct as a strategy to store the reactive enolate species. The emerging microscopic picture of the CO2-fixing mechanism confirms previous experimental observations and provides new insights into how nature handles highly reactive intermediates to fix the inert greenhouse gas.ViewEffects of Dynamical Degrees of Freedom on Magnetic Compass Sensitivity: A Comparison of Plant and Avian CryptochromesArticleDec 2022J AM CHEM SOC Gesa Grüning Siu Ying Wong Luca Gerhards Fabian SchuhmannIlia A. Solov’yovThe magnetic compass of migratory birds is thought to \n",
      "len(chunk):  tory birds is thought to rely on a radical pair reaction inside the blue-light photoreceptor protein cryptochrome. The sensitivity of such a sensor to weak external magnetic fields is determined by a variety of magnetic interactions, including electron-nuclear hyperfine interactions. Here, we investigate the implications of thermal motion, focusing on fluctuations in the dihedral and librational angles of flavin adenine dinucleotide (FAD) and tryptophan (Trp) radicals in cryptochrome 4a from European robin (Erithacus rubecula, ErCry4a) and pigeon (Columba livia, ClCry4a) and cryptochrome 1 from the plant Arabidopsis thaliana (AtCry1). Molecular dynamics simulations and density functional theory-derived hyperfine interactions are used to calculate the quantum yield of radical pair recombination dependent on the direction of the geomagnetic field. This quantity and various dynamical parameters are compared for [FAD•- Trp•+] in ErCry4a, ClCry4a, and AtCry1, with TrpC or TrpD being the third and fourth components of the tryptophan triad/tetrad in the respective proteins. We find that (i) differences in the average dihedral angles in the radical pairs are small, (ii) the librational motions of TrpC•+ in the avian cryptochromes are appreciably smaller than in AtCry1, (iii) the rapid vibrational motions of the radicals leading to strong fluctuations in the hyperfine couplings affect the spin dynamics depending on the usage of instantaneous or time-averaged interactions. Future investigations of radical pair compass sensitivity should therefore not be based on single snapshots of the protein structure but should include the ensemble properties of the hyperfine interactions.ViewGet access to 30 million figuresJoin ResearchGate to access over 30 million figures and 160+ million publications – all in one place.Join for free\n",
      "We and our partners use cookies✕By using this site, you consent to the processing of your personal data, the storing of cookies on your device, and the use of similar technologies for personalization, ads, analytics, etc. For more information or to opt out, see our Privacy Policy\n",
      "Chunk: 0 / 5\n",
      "seed_initial_question_decomposition_prompt \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract accurate and useful direct quotes from the source text that are relevant to answering the key questions and also output the quote's most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"<insert unique quote index>\": {\n",
      "    \"reasoning\": <one sentence explaining how this direct quote helps answer the key question>\n",
      "    \"quote\": \"<insert direct quote>\",\n",
      "    \"index\": <insert most relevant key question index>\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Source text: Figure 1 - available via license: Creative Commons Attribution-NonCommercial-NoDerivatives 4.0 InternationalContent may be subject to copyright.DownloadView publicationCopy referenceCopy captionEmbed figureReaction scheme and structural organization of the K. setae ECR complex. (a) Carboxylation reaction scheme of ECR. (b) Anisotropic B-factors of the tetramer of the different ECR complexes solved in this study are shown color coded according to the B factors (blue for low values and red for high values).Source publication\n",
      "Intersubunit Coupling Enables Fast CO 2 -Fixation by Reductive CarboxylasesArticleFull-text availableApr 2022 Hasan Demirci Yashas Rao Gabriele Stoffel[...] Soichi WakatsukiEnoyl-CoA carboxylases/reductases (ECRs) are some of the most efficient CO2-fixing enzymes described to date. However, the molecular mechanisms underlying the extraordinary catalytic activity of ECRs on the level of the protein assembly remain elusive. Here we used a combination of ambient-temperature X-ray free electron laser (XFEL) and cryogenic...CiteDownload full-textContexts in source publicationContext 1... ECRs catalyze the reduction of various α,β-unsaturated enoylCoAs into the corresponding alkylmalonyl-CoA esters using the reduced form of the cofactor nicotinamide adenine dinucleotide phosphate (NADPH). Hydride transfer from NADPH to the substrate generates a reactive enolate species, which acts as a nucleophile that attacks a bound CO 2 molecule ( Figure 1a). 2,5−7 However, the structural details of the carboxylation reaction have remained elusive, in part due to the lack of high-resolution structures of ECRs containing catalytic intermediates and carboxylated products. ...View in full-textContext 2... we aimed at providing a mechanistic understanding of the fast carboxylation reaction of the K. setae ECR on the level of the overall oligomeric complex. We first determined four highresolution K. setae ECR structures in different conformational states: the apo form and three holo forms in a binary complex with the reduced cofactor NADPH, in a ternary complex with NADPH and butyryl-CoA, and a binary complex with the oxidized cofactor NADP + (Figure 1b). ...View in full-textContext 3... asymmetric unit contains one homotetramer composed of four subunits arranged in a pair of dimers geometry similar to those of previously reported binary and ternary ECR complexes, crotonyl-CoA carboxylase/reductase AntE in complex with NADP + (PDB: 4Y0K) and 2-octenoyl-CoA carboxylase/reductase CinF in complex with NADP + and 2-octenoyl-CoA, (PDB: 4A0S), respectively. Overall, the K. setae ECR tetramer shows a noncrystallographic, close to D 2 (dihedral) symmetry with four conformationally identical subunits ( Figures S1 and S2). The tetrameric oligomer state of the ECR apo enzyme is further supported by size-exclusion chromatography, which showed that the 51.2 kDa protein eluted as a single peak at 205 kDa ( Figure S3). ...View in full-textContext 4... postulate that the same crystallization conditions used for both structures might have resulted in the same space group (P2 1 2 1 2 1 ) with similar crystal contacts, which might have kept the NADP + bound ECR tetramer in the apo form. The NADP + -bound subunits are less rigid in comparison, and some residues show greater variations (Figures S1 and S2). ...View in full-textContext 5... NADPH binding breaks the dihedral D 2 symmetry observed in the apo-form tetramer, while symmetry about the y axis is retained, resulting in a noncrystallographic, almost cyclic C 2 symmetry ( Figure S1). In the NADPH*ECR binary complex, each of the two dimers, A/C and B/D, loses its 2-fold symmetry, giving rise to closed (A and B) and open (C and D) subunits ( Figure S1). ...View in full-textContext 6... NADPH binding breaks the dihedral D 2 symmetry observed in the apo-form tetramer, while symmetry about the y axis is retained, resulting in a noncrystallographic, almost cyclic C 2 symmetry ( Figure S1). In the NADPH*E\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Prompt:  \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract accurate and useful direct quotes from the source text that are relevant to answering the key questions and also output the quote's most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"<insert unique quote index>\": {\n",
      "    \"reasoning\": <one sentence explaining how this direct quote helps answer the key question>\n",
      "    \"quote\": \"<insert direct quote>\",\n",
      "    \"index\": <insert most relevant key question index>\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Source text: Figure 1 - available via license: Creative Commons Attribution-NonCommercial-NoDerivatives 4.0 InternationalContent may be subject to copyright.DownloadView publicationCopy referenceCopy captionEmbed figureReaction scheme and structural organization of the K. setae ECR complex. (a) Carboxylation reaction scheme of ECR. (b) Anisotropic B-factors of the tetramer of the different ECR complexes solved in this study are shown color coded according to the B factors (blue for low values and red for high values).Source publication\n",
      "Intersubunit Coupling Enables Fast CO 2 -Fixation by Reductive CarboxylasesArticleFull-text availableApr 2022 Hasan Demirci Yashas Rao Gabriele Stoffel[...] Soichi WakatsukiEnoyl-CoA carboxylases/reductases (ECRs) are some of the most efficient CO2-fixing enzymes described to date. However, the molecular mechanisms underlying the extraordinary catalytic activity of ECRs on the level of the protein assembly remain elusive. Here we used a combination of ambient-temperature X-ray free electron laser (XFEL) and cryogenic...CiteDownload full-textContexts in source publicationContext 1... ECRs catalyze the reduction of various α,β-unsaturated enoylCoAs into the corresponding alkylmalonyl-CoA esters using the reduced form of the cofactor nicotinamide adenine dinucleotide phosphate (NADPH). Hydride transfer from NADPH to the substrate generates a reactive enolate species, which acts as a nucleophile that attacks a bound CO 2 molecule ( Figure 1a). 2,5−7 However, the structural details of the carboxylation reaction have remained elusive, in part due to the lack of high-resolution structures of ECRs containing catalytic intermediates and carboxylated products. ...View in full-textContext 2... we aimed at providing a mechanistic understanding of the fast carboxylation reaction of the K. setae ECR on the level of the overall oligomeric complex. We first determined four highresolution K. setae ECR structures in different conformational states: the apo form and three holo forms in a binary complex with the reduced cofactor NADPH, in a ternary complex with NADPH and butyryl-CoA, and a binary complex with the oxidized cofactor NADP + (Figure 1b). ...View in full-textContext 3... asymmetric unit contains one homotetramer composed of four subunits arranged in a pair of dimers geometry similar to those of previously reported binary and ternary ECR complexes, crotonyl-CoA carboxylase/reductase AntE in complex with NADP + (PDB: 4Y0K) and 2-octenoyl-CoA carboxylase/reductase CinF in complex with NADP + and 2-octenoyl-CoA, (PDB: 4A0S), respectively. Overall, the K. setae ECR tetramer shows a noncrystallographic, close to D 2 (dihedral) symmetry with four conformationally identical subunits ( Figures S1 and S2). The tetrameric oligomer state of the ECR apo enzyme is further supported by size-exclusion chromatography, which showed that the 51.2 kDa protein eluted as a single peak at 205 kDa ( Figure S3). ...View in full-textContext 4... postulate that the same crystallization conditions used for both structures might have resulted in the same space group (P2 1 2 1 2 1 ) with similar crystal contacts, which might have kept the NADP + bound ECR tetramer in the apo form. The NADP + -bound subunits are less rigid in comparison, and some residues show greater variations (Figures S1 and S2). ...View in full-textContext 5... NADPH binding breaks the dihedral D 2 symmetry observed in the apo-form tetramer, while symmetry about the y axis is retained, resulting in a noncrystallographic, almost cyclic C 2 symmetry ( Figure S1). In the NADPH*ECR binary complex, each of the two dimers, A/C and B/D, loses its 2-fold symmetry, giving rise to closed (A and B) and open (C and D) subunits ( Figure S1). ...View in full-textContext 6... NADPH binding breaks the dihedral D 2 symmetry observed in the apo-form tetramer, while symmetry about the y axis is retained, resulting in a noncrystallographic, almost cyclic C 2 symmetry ( Figure S1). In the NADPH*E\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"0\\\": {\\n    \\\"reasoning\\\": \\\"The direct quote provides information about the carboxylation reaction scheme of ECR, which is relevant to understanding the level of ECR enzyme activity in Kitsatospor setae bacteria.\\\",\\n    \\\"quote\\\": \\\"Carboxylation reaction scheme of ECR.\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"1\\\": {\\n    \\\"reasoning\\\": \\\"The direct quote mentions that ECRs are some of the most efficient CO2-fixing enzymes described to date, which provides a basis for comparing ECR enzyme activity to other bacteria.\\\",\\n    \\\"quote\\\": \\\"Enoyl-CoA carboxylases/reductases (ECRs) are some of the most efficient CO2-fixing enzymes described to date.\\\",\\n    \\\"index\\\": 1\\n  },\\n  \\\"2\\\": {\\n    \\\"reasoning\\\": \\\"The direct quote mentions that the structural details of the carboxylation reaction have remained elusive, which suggests that there are factors influencing ECR enzyme activity in Kitsatospor setae bacteria that are not yet fully understood.\\\",\\n    \\\"quote\\\": \\\"However, the structural details of the carboxylation reaction have remained elusive, in part due to the lack of high-resolution structures of ECRs containing catalytic intermediates and carboxylated products.\\\",\\n    \\\"index\\\": 2\\n  },\\n  \\\"3\\\": {\\n    \\\"reasoning\\\": \\\"There is no direct quote that specifically addresses the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria.\\\",\\n    \\\"quote\\\": \\\"\\\",\\n    \\\"index\\\": null\\n  }\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691165091,\n",
      "  \"id\": \"chatcmpl-7jraFk54EhY7BjiUHM5OKbx6n9f9L\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 325,\n",
      "    \"prompt_tokens\": 1250,\n",
      "    \"total_tokens\": 1575\n",
      "  }\n",
      "}\n",
      "Extracted quotes:  {\n",
      "  \"0\": {\n",
      "    \"reasoning\": \"The direct quote provides information about the carboxylation reaction scheme of ECR, which is relevant to understanding the level of ECR enzyme activity in Kitsatospor setae bacteria.\",\n",
      "    \"quote\": \"Carboxylation reaction scheme of ECR.\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"1\": {\n",
      "    \"reasoning\": \"The direct quote mentions that ECRs are some of the most efficient CO2-fixing enzymes described to date, which provides a basis for comparing ECR enzyme activity to other bacteria.\",\n",
      "    \"quote\": \"Enoyl-CoA carboxylases/reductases (ECRs) are some of the most efficient CO2-fixing enzymes described to date.\",\n",
      "    \"index\": 1\n",
      "  },\n",
      "  \"2\": {\n",
      "    \"reasoning\": \"The direct quote mentions that the structural details of the carboxylation reaction have remained elusive, which suggests that there are factors influencing ECR enzyme activity in Kitsatospor setae bacteria that are not yet fully understood.\",\n",
      "    \"quote\": \"However, the structural details of the carboxylation reaction have remained elusive, in part due to the lack of high-resolution structures of ECRs containing catalytic intermediates and carboxylated products.\",\n",
      "    \"index\": 2\n",
      "  },\n",
      "  \"3\": {\n",
      "    \"reasoning\": \"There is no direct quote that specifically addresses the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria.\",\n",
      "    \"quote\": \"\",\n",
      "    \"index\": null\n",
      "  }\n",
      "}\n",
      "Chunk: 1 / 5\n",
      "seed_initial_question_decomposition_prompt \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract accurate and useful direct quotes from the source text that are relevant to answering the key questions and also output the quote's most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"<insert unique quote index>\": {\n",
      "    \"reasoning\": <one sentence explaining how this direct quote helps answer the key question>\n",
      "    \"quote\": \"<insert direct quote>\",\n",
      "    \"index\": <insert most relevant key question index>\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Source text: igure S1). In the NADPH*ECR binary complex, each of the two dimers, A/C and B/D, loses its 2-fold symmetry, giving rise to closed (A and B) and open (C and D) subunits ( Figure S1). In the A and B subunits, the cofactor binding pocket is compressed inward, which seals the NADPH cofactor within the catalytic domain, resulting in a \"closed-form\" state ( Figure 2b). ...View in full-textContext 7... next investigated the effects of substrate binding to the K. setae ECR tetramer complex and determined its ternary complex structure with the substrate analogue butyryl-CoA and NADPH at 1.7 Å resolution ( Figure 3). The structure is overall very similar to the conformation of the ECR*NADPH binary complex with the noncrystallographic, pseudo-C 2 cyclic symmetry ( Figure 3a and Figure S1). It comprises two sets of open-and closed-form subunits that overlay very well with those of the ECR*NADPH binary complex ( Figures S1 and S2). ...View in full-textContext 8... structure is overall very similar to the conformation of the ECR*NADPH binary complex with the noncrystallographic, pseudo-C 2 cyclic symmetry ( Figure 3a and Figure S1). It comprises two sets of open-and closed-form subunits that overlay very well with those of the ECR*NADPH binary complex ( Figures S1 and S2). The NADPH cofactor appears to be bound to all active sites; however, only the closed-form subunits A and B also contain electron density for the complete butyryl-CoA thioester, while the open-form subunits (C and D) show only the adenine group electron density of the butyrylCoA thioester (Figure 3a,b). ...View in full-textContext 9... we inspected CoA-thioester binding in the open-form subunits, we also observed the electron density in the adenine binding pocket of the neighboring closed form subunits (A or B), strongly indicating that the CoAthioester was bound (Figure 3g, right inset). However, the electron density beyond the adenine ring becomes invisible, suggesting that the part of the CoA molecule reaching into the open-form subunit's active site remains flexible and disordered, which is corroborated by the higher B-factors of the catalytic domain ( Figure 1, bottom right). ...View in full-textContext 10... characterize the dynamics, we compared the projections of all 10 dimer trajectories on PC1 and PC2. Dimer A/C opens its closed subunit first, associated with PC1, followed by a twist motion characterized by PC2 in some of the simulations ( Figure 4d and Figure S10). This twist motion rotates the catalytic domain of the empty, now open form active site in subunit A, which would push a bound crotonyl-CoA substrate toward the NADPH cofactor in the C subunit. ...View in full-textContext 11... crotonyl-CoA from subunit A, we expected the�now empty�\"closed\" active site subunit to transition to the open form, inducing the twist motion in the A/C dimer observed before, and the other closed subunit B still bearing the substrate remains closed, if the two dynamics of the pair of dimers are uncoupled. The projection of 10 100 ns trajectories on PC1 and PC2 confirmed that the empty active site (A in the A/C dimer) undergoes a conformational change to the open form (Figure 4d), whereas the dimer containing the substrate (B in the B/D dimer) remained in a closed conformation (see Figure S11 for a comparison to the simulation with two substrates maintaining the closed state). Importantly, however, none of the A/C dimer trajectories showed a significant change in PC2 associated with the twist motion observed in the binary ECR*NADPH complex, indicating that the twist motion is only initiated once the other dimer B/D adopts the open conformation. ...View in full-textContext 12... ± 25 1.1 ± 0.0 26 ± 3 0.7 ± 0.0 440 ± 40 0.9 ± 0.0 a Michaelis−Menten curves of K. setae ECR and its variants are provided in Figure S14. ...View in full-textContext 13... ± 130 38 ± 7 2980 ± 510 0.3 ± 0.0 a Michaelis−Menten curves of K. setae ECR and its variants are provided in Figure S14. synchrony: i.\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Prompt:  \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract accurate and useful direct quotes from the source text that are relevant to answering the key questions and also output the quote's most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"<insert unique quote index>\": {\n",
      "    \"reasoning\": <one sentence explaining how this direct quote helps answer the key question>\n",
      "    \"quote\": \"<insert direct quote>\",\n",
      "    \"index\": <insert most relevant key question index>\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Source text: igure S1). In the NADPH*ECR binary complex, each of the two dimers, A/C and B/D, loses its 2-fold symmetry, giving rise to closed (A and B) and open (C and D) subunits ( Figure S1). In the A and B subunits, the cofactor binding pocket is compressed inward, which seals the NADPH cofactor within the catalytic domain, resulting in a \"closed-form\" state ( Figure 2b). ...View in full-textContext 7... next investigated the effects of substrate binding to the K. setae ECR tetramer complex and determined its ternary complex structure with the substrate analogue butyryl-CoA and NADPH at 1.7 Å resolution ( Figure 3). The structure is overall very similar to the conformation of the ECR*NADPH binary complex with the noncrystallographic, pseudo-C 2 cyclic symmetry ( Figure 3a and Figure S1). It comprises two sets of open-and closed-form subunits that overlay very well with those of the ECR*NADPH binary complex ( Figures S1 and S2). ...View in full-textContext 8... structure is overall very similar to the conformation of the ECR*NADPH binary complex with the noncrystallographic, pseudo-C 2 cyclic symmetry ( Figure 3a and Figure S1). It comprises two sets of open-and closed-form subunits that overlay very well with those of the ECR*NADPH binary complex ( Figures S1 and S2). The NADPH cofactor appears to be bound to all active sites; however, only the closed-form subunits A and B also contain electron density for the complete butyryl-CoA thioester, while the open-form subunits (C and D) show only the adenine group electron density of the butyrylCoA thioester (Figure 3a,b). ...View in full-textContext 9... we inspected CoA-thioester binding in the open-form subunits, we also observed the electron density in the adenine binding pocket of the neighboring closed form subunits (A or B), strongly indicating that the CoAthioester was bound (Figure 3g, right inset). However, the electron density beyond the adenine ring becomes invisible, suggesting that the part of the CoA molecule reaching into the open-form subunit's active site remains flexible and disordered, which is corroborated by the higher B-factors of the catalytic domain ( Figure 1, bottom right). ...View in full-textContext 10... characterize the dynamics, we compared the projections of all 10 dimer trajectories on PC1 and PC2. Dimer A/C opens its closed subunit first, associated with PC1, followed by a twist motion characterized by PC2 in some of the simulations ( Figure 4d and Figure S10). This twist motion rotates the catalytic domain of the empty, now open form active site in subunit A, which would push a bound crotonyl-CoA substrate toward the NADPH cofactor in the C subunit. ...View in full-textContext 11... crotonyl-CoA from subunit A, we expected the�now empty�\"closed\" active site subunit to transition to the open form, inducing the twist motion in the A/C dimer observed before, and the other closed subunit B still bearing the substrate remains closed, if the two dynamics of the pair of dimers are uncoupled. The projection of 10 100 ns trajectories on PC1 and PC2 confirmed that the empty active site (A in the A/C dimer) undergoes a conformational change to the open form (Figure 4d), whereas the dimer containing the substrate (B in the B/D dimer) remained in a closed conformation (see Figure S11 for a comparison to the simulation with two substrates maintaining the closed state). Importantly, however, none of the A/C dimer trajectories showed a significant change in PC2 associated with the twist motion observed in the binary ECR*NADPH complex, indicating that the twist motion is only initiated once the other dimer B/D adopts the open conformation. ...View in full-textContext 12... ± 25 1.1 ± 0.0 26 ± 3 0.7 ± 0.0 440 ± 40 0.9 ± 0.0 a Michaelis−Menten curves of K. setae ECR and its variants are provided in Figure S14. ...View in full-textContext 13... ± 130 38 ± 7 2980 ± 510 0.3 ± 0.0 a Michaelis−Menten curves of K. setae ECR and its variants are provided in Figure S14. synchrony: i.\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"0\\\": {\\n    \\\"reasoning\\\": \\\"This quote provides information about the conformation of the ECR enzyme in Kitsatospor setae bacteria.\\\",\\n    \\\"quote\\\": \\\"In the NADPH*ECR binary complex, each of the two dimers, A/C and B/D, loses its 2-fold symmetry, giving rise to closed (A and B) and open (C and D) subunits ( Figure S1).\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"1\\\": {\\n    \\\"reasoning\\\": \\\"This quote provides information about the conformation of the ECR enzyme in Kitsatospor setae bacteria.\\\",\\n    \\\"quote\\\": \\\"The structure is overall very similar to the conformation of the ECR*NADPH binary complex with the noncrystallographic, pseudo-C 2 cyclic symmetry ( Figure 3a and Figure S1).\\\",\\n    \\\"index\\\": 1\\n  },\\n  \\\"2\\\": {\\n    \\\"reasoning\\\": \\\"This quote provides information about the binding of the substrate analogue butyryl-CoA to the ECR enzyme in Kitsatospor setae bacteria.\\\",\\n    \\\"quote\\\": \\\"The NADPH cofactor appears to be bound to all active sites; however, only the closed-form subunits A and B also contain electron density for the complete butyryl-CoA thioester, while the open-form subunits (C and D) show only the adenine group electron density of the butyrylCoA thioester (Figure 3a,b).\\\",\\n    \\\"index\\\": 2\\n  },\\n  \\\"3\\\": {\\n    \\\"reasoning\\\": \\\"This quote provides information about the dynamics of the ECR enzyme in Kitsatospor setae bacteria.\\\",\\n    \\\"quote\\\": \\\"Dimer A/C opens its closed subunit first, associated with PC1, followed by a twist motion characterized by PC2 in some of the simulations ( Figure 4d and Figure S10).\\\",\\n    \\\"index\\\": 2\\n  },\\n  \\\"4\\\": {\\n    \\\"reasoning\\\": \\\"This quote provides information about the dynamics of the ECR enzyme in Kitsatospor setae bacteria.\\\",\\n    \\\"quote\\\": \\\"The projection of 10 100 ns trajectories on PC1 and PC2 confirmed that the empty active site (A in the A/C dimer) undergoes a conformational change to the open form (Figure 4d), whereas the dimer containing the substrate (B in the B/D dimer) remained in a closed conformation.\\\",\\n    \\\"index\\\": 2\\n  }\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691165104,\n",
      "  \"id\": \"chatcmpl-7jraSdRLo0yWjZ0S7xHHRx2Hfx3Vk\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 530,\n",
      "    \"prompt_tokens\": 1253,\n",
      "    \"total_tokens\": 1783\n",
      "  }\n",
      "}\n",
      "Extracted quotes:  {\n",
      "  \"0\": {\n",
      "    \"reasoning\": \"This quote provides information about the conformation of the ECR enzyme in Kitsatospor setae bacteria.\",\n",
      "    \"quote\": \"In the NADPH*ECR binary complex, each of the two dimers, A/C and B/D, loses its 2-fold symmetry, giving rise to closed (A and B) and open (C and D) subunits ( Figure S1).\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"1\": {\n",
      "    \"reasoning\": \"This quote provides information about the conformation of the ECR enzyme in Kitsatospor setae bacteria.\",\n",
      "    \"quote\": \"The structure is overall very similar to the conformation of the ECR*NADPH binary complex with the noncrystallographic, pseudo-C 2 cyclic symmetry ( Figure 3a and Figure S1).\",\n",
      "    \"index\": 1\n",
      "  },\n",
      "  \"2\": {\n",
      "    \"reasoning\": \"This quote provides information about the binding of the substrate analogue butyryl-CoA to the ECR enzyme in Kitsatospor setae bacteria.\",\n",
      "    \"quote\": \"The NADPH cofactor appears to be bound to all active sites; however, only the closed-form subunits A and B also contain electron density for the complete butyryl-CoA thioester, while the open-form subunits (C and D) show only the adenine group electron density of the butyrylCoA thioester (Figure 3a,b).\",\n",
      "    \"index\": 2\n",
      "  },\n",
      "  \"3\": {\n",
      "    \"reasoning\": \"This quote provides information about the dynamics of the ECR enzyme in Kitsatospor setae bacteria.\",\n",
      "    \"quote\": \"Dimer A/C opens its closed subunit first, associated with PC1, followed by a twist motion characterized by PC2 in some of the simulations ( Figure 4d and Figure S10).\",\n",
      "    \"index\": 2\n",
      "  },\n",
      "  \"4\": {\n",
      "    \"reasoning\": \"This quote provides information about the dynamics of the ECR enzyme in Kitsatospor setae bacteria.\",\n",
      "    \"quote\": \"The projection of 10 100 ns trajectories on PC1 and PC2 confirmed that the empty active site (A in the A/C dimer) undergoes a conformational change to the open form (Figure 4d), whereas the dimer containing the substrate (B in the B/D dimer) remained in a closed conformation.\",\n",
      "    \"index\": 2\n",
      "  }\n",
      "}\n",
      "Chunk: 2 / 5\n",
      "seed_initial_question_decomposition_prompt \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract accurate and useful direct quotes from the source text that are relevant to answering the key questions and also output the quote's most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"<insert unique quote index>\": {\n",
      "    \"reasoning\": <one sentence explaining how this direct quote helps answer the key question>\n",
      "    \"quote\": \"<insert direct quote>\",\n",
      "    \"index\": <insert most relevant key question index>\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Source text: Figure S14. synchrony: i.e., when one dimer switches from the open−close to the closed−open states, the other dimer would do the opposite, closed−open to open−closed states. ...View in full-textContext 14... i.e., when one dimer switches from the open−close to the closed−open states, the other dimer would do the opposite, closed−open to open−closed states. Using this model, we simulated the case with just one dimer with two subunits working in negative or positive cooperativity using the method described by Hill and Levitski 21 ( Figure S12). For a thorough investigation of how cooperativity, positive or negative, enhances the overall reaction rate of the K. setae ECR, we incorporated all the parameters in Hill and Levitski's model in our simulation. ...View in full-textContext 15... turned out that only negative cooperativity could explain our experimental observations that ECR activity in the WT tetramer is increased by 2 orders of magnitude in comparison to the triple variant E151D/N157E/N218E, in which synchronization is decoupled (Table 1). Our simulations yielded a 90% probability that both active sites are occupied with substrates/products and NADPH/NADP + (p22), while only 10% are half-occupied (p12 + p21) ( Figure S12). Almost none of the dimers are empty (p11), which agrees with the occupancy of our experimentally determined structures of the ternary complex. ...View in full-textContext 16... While our structural, biochemical, and simulation data indicate that the K. setae ECR achieves high catalytic rates by synchronizing active sites, this might not necessarily be true for other ECRs. A differentiation into a pair of dimers was not observed in NADPH-bound or ternary structures of other ECRs (e.g., PDB 4Y0K and 4A0S, respectively, which share substantial amino acid identity) ( Figure S13). Note, however, that ECRs fall into two different classes: primary ECRs (such as the K. setae ECR) that operate in central carbon metabolism and secondary ECRs (such as 4Y0K) that provide extender units for synthesizing polyketides in secondary metabolism. ...View in full-textCitations... X-ray structures with a cofactor and a substrate analogue at different stages of the catalytic cycle reveal that the active site adopts either an open or closed conformation in a dimer of dimers configuration. 14 The catalytically competent form is the closed conformation containing the substrate analogue and NADPH (see Figure S1 in SI). The other active site in the dimer is in the open conformation containing only the NADPH cofactor.\n",
      "...Covalent Adduct Formation as a Strategy for Efficient CO 2 Fixation in Crotonyl-CoA Carboxylases/ReductasesArticleApr 2023 Rodrigo Recabarren Matthias Tinzl David Adrian Saez Aharon Gomez Esteban Vöhringer-MartinezView... X-ray structures with cofactor and a substrate analogue at different stages of the catalytic cycle reveal that the active site adopts either an open or closed conformation in a dimer of dimers configuration. 14 The catalytically competent form is the closed conformation containing the substrate analogue and NADPH (see Figure S1 in SI). The other active site in the dimer is in the open conformation containing only the NADPH cofactor.\n",
      "...Covalent adduct formation as strategy for efficient CO2 fixation in crotonyl-CoA carboxylases/reductasesPreprintFull-text availableApr 2023 Rodrigo Recabarren Matthias Tinzl David Saez Aharon Gomez Esteban Vöhringer-MartinezIncreasing levels of CO2 in the atmosphere have led to a growing interest into the various ways nature transforms this greenhouse gas into valuable organic compounds. Crotonyl-CoA carboxylases/reductases (Ccr's) are the most efficient biocatalysts for CO2 fixation because of their oxygen tolerance, their high catalytic rate constants and their high fidelity. The reaction mechanism involving hydride transfer from the NADPH cofactor and addition of CO2 to the reactive enolate, however, is not completely understood. In this study, we use computer s\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Prompt:  \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract accurate and useful direct quotes from the source text that are relevant to answering the key questions and also output the quote's most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"<insert unique quote index>\": {\n",
      "    \"reasoning\": <one sentence explaining how this direct quote helps answer the key question>\n",
      "    \"quote\": \"<insert direct quote>\",\n",
      "    \"index\": <insert most relevant key question index>\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Source text: Figure S14. synchrony: i.e., when one dimer switches from the open−close to the closed−open states, the other dimer would do the opposite, closed−open to open−closed states. ...View in full-textContext 14... i.e., when one dimer switches from the open−close to the closed−open states, the other dimer would do the opposite, closed−open to open−closed states. Using this model, we simulated the case with just one dimer with two subunits working in negative or positive cooperativity using the method described by Hill and Levitski 21 ( Figure S12). For a thorough investigation of how cooperativity, positive or negative, enhances the overall reaction rate of the K. setae ECR, we incorporated all the parameters in Hill and Levitski's model in our simulation. ...View in full-textContext 15... turned out that only negative cooperativity could explain our experimental observations that ECR activity in the WT tetramer is increased by 2 orders of magnitude in comparison to the triple variant E151D/N157E/N218E, in which synchronization is decoupled (Table 1). Our simulations yielded a 90% probability that both active sites are occupied with substrates/products and NADPH/NADP + (p22), while only 10% are half-occupied (p12 + p21) ( Figure S12). Almost none of the dimers are empty (p11), which agrees with the occupancy of our experimentally determined structures of the ternary complex. ...View in full-textContext 16... While our structural, biochemical, and simulation data indicate that the K. setae ECR achieves high catalytic rates by synchronizing active sites, this might not necessarily be true for other ECRs. A differentiation into a pair of dimers was not observed in NADPH-bound or ternary structures of other ECRs (e.g., PDB 4Y0K and 4A0S, respectively, which share substantial amino acid identity) ( Figure S13). Note, however, that ECRs fall into two different classes: primary ECRs (such as the K. setae ECR) that operate in central carbon metabolism and secondary ECRs (such as 4Y0K) that provide extender units for synthesizing polyketides in secondary metabolism. ...View in full-textCitations... X-ray structures with a cofactor and a substrate analogue at different stages of the catalytic cycle reveal that the active site adopts either an open or closed conformation in a dimer of dimers configuration. 14 The catalytically competent form is the closed conformation containing the substrate analogue and NADPH (see Figure S1 in SI). The other active site in the dimer is in the open conformation containing only the NADPH cofactor.\n",
      "...Covalent Adduct Formation as a Strategy for Efficient CO 2 Fixation in Crotonyl-CoA Carboxylases/ReductasesArticleApr 2023 Rodrigo Recabarren Matthias Tinzl David Adrian Saez Aharon Gomez Esteban Vöhringer-MartinezView... X-ray structures with cofactor and a substrate analogue at different stages of the catalytic cycle reveal that the active site adopts either an open or closed conformation in a dimer of dimers configuration. 14 The catalytically competent form is the closed conformation containing the substrate analogue and NADPH (see Figure S1 in SI). The other active site in the dimer is in the open conformation containing only the NADPH cofactor.\n",
      "...Covalent adduct formation as strategy for efficient CO2 fixation in crotonyl-CoA carboxylases/reductasesPreprintFull-text availableApr 2023 Rodrigo Recabarren Matthias Tinzl David Saez Aharon Gomez Esteban Vöhringer-MartinezIncreasing levels of CO2 in the atmosphere have led to a growing interest into the various ways nature transforms this greenhouse gas into valuable organic compounds. Crotonyl-CoA carboxylases/reductases (Ccr's) are the most efficient biocatalysts for CO2 fixation because of their oxygen tolerance, their high catalytic rate constants and their high fidelity. The reaction mechanism involving hydride transfer from the NADPH cofactor and addition of CO2 to the reactive enolate, however, is not completely understood. In this study, we use computer s\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"0\\\": {\\n    \\\"reasoning\\\": \\\"The quote mentions ECR activity in the WT tetramer being increased by 2 orders of magnitude compared to the triple variant E151D/N157E/N218E.\\\",\\n    \\\"quote\\\": \\\"turned out that only negative cooperativity could explain our experimental observations that ECR activity in the WT tetramer is increased by 2 orders of magnitude in comparison to the triple variant E151D/N157E/N218E, in which synchronization is decoupled (Table 1).\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"1\\\": {\\n    \\\"reasoning\\\": \\\"The quote mentions that differentiation into a pair of dimers was not observed in other ECRs, indicating that ECR enzyme activity may differ between bacteria.\\\",\\n    \\\"quote\\\": \\\"While our structural, biochemical, and simulation data indicate that the K. setae ECR achieves high catalytic rates by synchronizing active sites, this might not necessarily be true for other ECRs.\\\",\\n    \\\"index\\\": 1\\n  }\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691165119,\n",
      "  \"id\": \"chatcmpl-7jrah4NpvgLOYf7jQITlPXBGSd73M\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 216,\n",
      "    \"prompt_tokens\": 1164,\n",
      "    \"total_tokens\": 1380\n",
      "  }\n",
      "}\n",
      "Extracted quotes:  {\n",
      "  \"0\": {\n",
      "    \"reasoning\": \"The quote mentions ECR activity in the WT tetramer being increased by 2 orders of magnitude compared to the triple variant E151D/N157E/N218E.\",\n",
      "    \"quote\": \"turned out that only negative cooperativity could explain our experimental observations that ECR activity in the WT tetramer is increased by 2 orders of magnitude in comparison to the triple variant E151D/N157E/N218E, in which synchronization is decoupled (Table 1).\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"1\": {\n",
      "    \"reasoning\": \"The quote mentions that differentiation into a pair of dimers was not observed in other ECRs, indicating that ECR enzyme activity may differ between bacteria.\",\n",
      "    \"quote\": \"While our structural, biochemical, and simulation data indicate that the K. setae ECR achieves high catalytic rates by synchronizing active sites, this might not necessarily be true for other ECRs.\",\n",
      "    \"index\": 1\n",
      "  }\n",
      "}\n",
      "Max tokens reached in chunks!\n"
     ]
    }
   ],
   "source": [
    "# Go through each full text rated highly to extract facts from\n",
    "full_text_folder_path = f'autoscious_logs/{sanitize_filename(search_query)}/sources/full_text'\n",
    "\n",
    "# Loop through every file in the directory\n",
    "for filename in os.listdir(full_text_folder_path):\n",
    "    curr_tokens = 0\n",
    "    if filename.endswith('.txt'):\n",
    "        file_path = os.path.join(full_text_folder_path, filename)\n",
    "        with open(file_path, 'r', encoding='utf-8') as file:\n",
    "            title = file.readline().strip()\n",
    "            url = file.readline().strip()\n",
    "            text = file.read()\n",
    "        \n",
    "        # Break the text into chunk to extract information from\n",
    "        chunks = chunk_text(text, chunk_size, overlap)\n",
    "        for i, chunk in enumerate(chunks):\n",
    "            print(f\"Chunk: {i} / {len(chunks)}\")\n",
    "            extract_facts_from_website_text(search_query_file_safe, key_questions_list_string, title, chunk, url)\n",
    "\n",
    "            curr_tokens += len(chunk)\n",
    "            if curr_tokens > MAX_TOKENS:\n",
    "                print(\"Max tokens reached in chunks!\")\n",
    "                break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
