{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "from util import sanitize_filename\n",
    "import os\n",
    "from collections import defaultdict\n",
    "from llm import chat_openai\n",
    "from autogpt.commands.web_search import web_search_ddg\n",
    "from datetime import datetime"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "chunk_size = 4000 # How many characters to read per chunk in website text\n",
    "overlap = 25 # How much overlap between chunks\n",
    "MAX_TOKENS = 10000 # Roughly 3K tokens, $0.10 per MAX_TOKENs, max tokens to read per key question\n",
    "search_query = \"How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Can be moved to util.py\n",
    "# Get the list of key questions\n",
    "def get_key_questions_list_string(search_query_file_safe):\n",
    "    file_path = f'autoscious_logs/{search_query_file_safe}/decompositions/improved_decomposition.json'\n",
    "\n",
    "    with open(file_path, 'r') as f:\n",
    "        question_decomposition = json.load(f)\n",
    "    key_questions = question_decomposition['key_drivers']['1']['hypotheses']['1']['key_questions']\n",
    "    print(\"\\nkey_questions\\n\", key_questions)\n",
    "\n",
    "    numbered_key_questions_string = \"\"\n",
    "    for key, value in key_questions.items():\n",
    "        first_two_words = ' '.join(value.split())\n",
    "        numbered_key_questions_string += f'{int(key) - 1}. {first_two_words}\\n'\n",
    "\n",
    "    print(\"\\nnumbered_key_questions_string\\n\", numbered_key_questions_string)\n",
    "    return key_questions, numbered_key_questions_string"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "key_questions\n",
      " {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity compare to other bacteria?', '3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?', '4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}\n",
      "\n",
      "numbered_key_questions_string\n",
      " 0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "key_questions_dict {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity compare to other bacteria?', '3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?', '4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}\n"
     ]
    }
   ],
   "source": [
    "# 1) Get key questions\n",
    "search_query_file_safe = sanitize_filename(search_query)\n",
    "key_questions_dict, key_questions_list_string = get_key_questions_list_string(search_query_file_safe)\n",
    "print(\"key_questions_dict\", key_questions_dict)\n",
    "\n",
    "# 2) create facts folder and subfolders\n",
    "facts_folder_path = f'autoscious_logs/{sanitize_filename(search_query)}/facts'\n",
    "if not os.path.exists(facts_folder_path):\n",
    "    os.makedirs(facts_folder_path)\n",
    "for kq_idx in key_questions_dict:\n",
    "    kq_facts_folder_path = f'{facts_folder_path}/kq{int(kq_idx)-1}'\n",
    "    if not os.path.exists(kq_facts_folder_path):\n",
    "        os.makedirs(kq_facts_folder_path)\n",
    "    with open(f'{kq_facts_folder_path}/facts.txt', 'a') as f:\n",
    "        pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Can be broken to prompts.py and util.py\n",
    "import os\n",
    "\n",
    "def extract_facts_from_website_text(search_query_file_safe, key_questions_list_string, website_title, website_text, website_url):\n",
    "    seed_initial_question_decomposition_prompt = f'''\n",
    "Key questions (index : question): \n",
    "{key_questions_list_string}\n",
    "\n",
    "Task: \n",
    "Extract and output as many accurate direct quotes from the text that are relevant to answering the key questions and its most relevant key question index. Format as a JSON.\n",
    "```json\n",
    "{{\n",
    "  \"1\": {{\n",
    "    \"quote\": \"\",\n",
    "    \"index\": 0\n",
    "  }},\n",
    "  etc.\n",
    "}}\n",
    "```\n",
    "\n",
    "Text: {website_text}\n",
    "\n",
    "Respond only with the output, with no explanation or conversation.\n",
    "'''\n",
    "    # Ask GPT the prompt\n",
    "    print(\"seed_initial_question_decomposition_prompt\", seed_initial_question_decomposition_prompt)\n",
    "    res = chat_openai(seed_initial_question_decomposition_prompt, model=\"gpt-3.5-turbo\")\n",
    "    print(\"Extracted quotes: \", res[0])\n",
    "\n",
    "    # Save the quote to the corresponding key question index file\n",
    "    res_json = json.loads(res[0])\n",
    "    for key, value in res_json.items():\n",
    "        index = value['index']\n",
    "        quote = value['quote']\n",
    "\n",
    "        # Only log if there is a quote\n",
    "        if quote:\n",
    "          file_name = f'autoscious_logs/{search_query_file_safe}/facts/kq{index}/facts.txt'\n",
    "\n",
    "          with open(file_name, 'a', encoding='utf-8') as f:\n",
    "              f.write(quote.replace('/\"', '\"') + f\"[{website_url}]\" + os.linesep)\n",
    "\n",
    "    return"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Can be moved to util\n",
    "def chunk_text(text: str, chunk_size: int, overlap: int) -> list[str]:\n",
    "    chunks = []\n",
    "    for i in range(0, len(text), chunk_size - overlap):\n",
    "        chunks.append(text[i : i + chunk_size])\n",
    "        print(\"len(chunk): \", text[i : i + chunk_size])\n",
    "    return chunks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "len(chunk):  Back to Top\n",
      "Skip to main content\n",
      "The .gov means it’s official.\n",
      "Federal government websites often end in .gov or .mil. Before\n",
      "sharing sensitive information, make sure you’re on a federal\n",
      "government site.\n",
      "The site is secure.\n",
      "The https:// ensures that you are connecting to the\n",
      "official website and that any information you provide is encrypted\n",
      "and transmitted securely.\n",
      "Access keys\n",
      "NCBI Homepage\n",
      "MyNCBI Homepage\n",
      "Main Content\n",
      "Main Navigation\n",
      "Journal List\n",
      "Virulence\n",
      "v.8(6); 2017\n",
      "PMC5626244\n",
      "As a library, NLM provides access to scientific literature. Inclusion in an NLM database does not imply endorsement of, or agreement with,\n",
      "the contents by NLM or the National Institutes of Health.\n",
      "Learn more about our disclaimer.\n",
      "Virulence. 2017; 8(6): 938–958. Published online 2016 Oct 20.\n",
      "doi: 10.1080/21505594.2016.1250995PMCID: PMC5626244PMID: 27763824Blue light treatment of Pseudomonas aeruginosa: Strong bactericidal activity, synergism with antibiotics and inactivation of virulence factorsGrzegorz Fila,a Anna Kawiak,b,c and\n",
      "Mariusz Stanislaw GrinholcaGrzegorz FilaaLaboratory of Molecular Diagnostics, Department of Biotechnology, Intercollegiate Faculty of Biotechnology, University of Gdansk and Medical University of Gdansk, Gdansk, PolandFind articles by Grzegorz FilaAnna KawiakbDivision of Plant Protection and Biotechnology, Department of Biotechnology, Intercollegiate Faculty of Biotechnology, University of Gdansk and Medical University of Gdansk, Gdansk, PolandcLaboratory of Human Physiology, Medical University of Gdansk, Gdansk, PolandFind articles by Anna KawiakMariusz Stanislaw GrinholcaLaboratory of Molecular Diagnostics, Department of Biotechnology, Intercollegiate Faculty of Biotechnology, University of Gdansk and Medical University of Gdansk, Gdansk, PolandFind articles by Mariusz Stanislaw GrinholcAuthor information Article notes Copyright and License information PMC DisclaimeraLaboratory of Molecular Diagnostics, Department of Biotechnology, Intercollegiate Faculty of Biotechnology, University of Gdansk and Medical University of Gdansk, Gdansk, PolandbDivision of Plant Protection and Biotechnology, Department of Biotechnology, Intercollegiate Faculty of Biotechnology, University of Gdansk and Medical University of Gdansk, Gdansk, PolandcLaboratory of Human Physiology, Medical University of Gdansk, Gdansk, PolandCONTACT Mariusz Stanislaw Grinholc lp.ude.gu.hcetoib@clohnirg.zsuiram, Department of Biotechnology, Intercollegiate Faculty of Biotechnology, University of Gdansk and Medical University of Gdansk, Abrahama 58, 80-307 Gdansk, PolandColor versions of one or more of the figures in the article can be found online at www.tandfonline.com/kvir.Supplemental data for this article can be accessed on the publisher's website.Received 2016 Jul 6; Revised 2016 Oct 12; Accepted 2016 Oct 15.Copyright © 2017 The Author(s). Published with license by Taylor & FrancisThis is an Open Access article distributed under the terms of the Creative Commons Attribution-Non-Commercial License (http://creativecommons.org/licenses/by-nc/3.0/), which permits unrestricted non-commercial use, distribution, and reproduction in any medium, provided the original work is properly cited. The moral rights of the named author(s) have been asserted.See editorial \"The antimicrobial effect of blue light: What are behind?\" in\n",
      "volume 8 on page 649.Associated DataSupplementary MaterialsKVIR_S_1250995.zipkvir-08-06-1250995-s001.zip (2.2M)GUID: 0B98ED62-FFD7-4C5D-93A1-66EBC9AF0A87Go to:ABSTRACTPseudomonas aeruginosa is among the most common pathogens responsible for both acute and chronic infections of high incidence and severity. Additionally, P. aeruginosa resistance to conventional antimicrobials has increased rapidly over the past decade. Therefore, it is crucial to explore new therapeutic options, particularly options that specifically target the pathogenic mechanisms of this microbe. The ability of a pathogenic bacterium to cause disease is dependent upon the production\n",
      "len(chunk):  ndent upon the production of agents termed ‘virulence factors’, and approaches to mitigate these agents have gained increasing attention as new antibacterial strategies. Although blue light irradiation is a promising alternative approach, only limited and preliminary studies have described its effect on virulence factors. The current study aimed to investigate the effects of lethal and sub-lethal doses of blue light treatment (BLT) on P. aeruginosa virulence factors. We analyzed the inhibitory effects of blue light irradiation on the production/activity of several virulence factors. Lethal BLT inhibited the activity of pyocyanin, staphylolysin, pseudolysin and other proteases, but sub-lethal BLT did not affect the production/expression of proteases, phospholipases, and flagella- or type IV pili-associated motility. Moreover, a eukaryotic cytotoxicity test confirmed the decreased toxicity of blue light-treated extracellular P. aeruginosa fractions. Finally, the increased antimicrobial susceptibility of P. aeruginosa treated with sequential doses of sub-lethal BLT was demonstrated with a checkerboard test. Thus, this work provides evidence-based proof of the susceptibility of drug-resistant P. aeruginosa to BLT-mediated killing, accompanied by virulence factor reduction, and describes the synergy between antibiotics and sub-lethal BLT.KEYWORDS: blue light treatment, elastases, hemolysins, lipases, motility, photoinactivation, proteases, Pseudomonas aeruginosa, virulence factorsGo to:IntroductionPseudomonas aeruginosa is a Gram-negative bacterium that thrives in most natural and man-made environments. It is found in diverse habitats, including soil, water, plants and animals, and infects multiple hosts.1\n",
      "P. aeruginosa causes a wide variety of acute (i.e., short duration and typically severe) and chronic (i.e., long duration, often refractory to treatment, and of variable severity depending on the pathogen) human infections, including in patients with severe burn wounds, urinary tract infections, acquired immune deficiency syndrome (AIDS), lung cancer, chronic obstructive pulmonary disease, bronchiectasis and cystic fibrosis (CF).2-5 Adding to the problems caused by its high incidence and the severity of infection, the resistance of P. aeruginosa to conventional antimicrobial treatments has increased over the past decade.6 Despite their wide recognition, the high incidence, severity and resistance of P. aeruginosa persist, and various strategies to address these problems have been proposed, such as (i) virulence and pathogenicity factors as therapeutic targets (for a review see ref. Seven); (ii) secretion systems as therapeutic targets (for a review see ref. Eight); (iii) cell adhesion and biofilm formation as therapeutic targets (for a review see ref. Nine); or (iv) quorum sensing as a therapeutic target (for a review see ref. Ten), as well as many others (for a review see ref. Eleven)7-11. Therefore, it is crucial to explore new therapeutic options for P. aeruginosa infections, which may be achieved by specifically targeting its pathogenic mechanisms. The ability of a pathogenic bacterium to cause disease depends on the production of agents termed ‘virulence factors’, such as toxins and adhesion molecules, which actively damage host tissues. Targeting these virulence factors (e.g., inhibiting their production, delivery or function) has gained increasing attention as a potential new antibacterial strategy; in principle this would ‘disarm’ a pathogen and allow the host immune system a better chance of clearing the infection before the pathogen causes too much tissue damage. Its metabolic versatility, intrinsic and acquired antibiotic resistance, biofilm formation and the production of multiple virulence (disease-causing) factors make P. aeruginosa a formidable pathogen. The virulence machinery of P. aeruginosa comprises both cell-associated determinants (such as lipopolysaccharides, pili, and flagella) and numerous secreted factor\n",
      "len(chunk):   numerous secreted factors (such as elastases, proteases, exotoxins, pyocyanin (PCN), and extracellular polysaccharides).P. aeruginosa pathogenicity is strongly associated with its ability to move. The following 3 types of movement have been identified: (1) flagella-related swimming and (2) swarming and (3) type IV pili-dependent twitching. Another significant factor contributing to the virulence of P. aeruginosa is PCN (1-hydroxy-5-methylphenazine), a cytotoxic pigment secreted by the bacterium. Moreover, P. aeruginosa virulence is augmented by secreted factors that allow the bacterium to destroy host tissue, modulate the immune system, create biofilms and initiate the colonization process. LasA (staphylolysin) and LasB (pseudolysin) elastases are proteases secreted by P. aeruginosa.12 Protease IV is a Pseudomonas serine protease that has been shown to degrade components of the immune system as well as fibrinogen, plasmin, and plasminogen. In addition, alkaline protease is involved in the extracellular processing of proteases; it has a fairly broad substrate range and is thought to act synergistically with other proteases.13 Finally, P. aeruginosa virulence factors include hemolysins, which lyse red blood cells and release free iron from heme, allowing the bacteria to grow rapidly and leading to greater colonization of the host tissue. P. aeruginosa produces 2 elements with hemolytic properties: heat-resistant glycolipids and heat-labile phospholipase C.14,15The severity and increasing incidence of infections involving multidrug-resistant (MDR) P. aeruginosa have driven research strategies targeting virulence factors that lead to both bacterial eradication and the reduction of bacterial virulence. Major research efforts have been initiated to identify an alternative antimicrobial approach to combat bacteria without easily facilitating the development of resistance. In a recent paper published in Nature Reviews Microbiology, Karen Bush and a group of 30 scientists from academia and industry noted, “The investigation of novel non-antibiotic approaches for the prevention of and protection against infectious diseases should be encouraged, and such approaches must be high-priority research and development projects.”16 The development of light-based antimicrobial therapies, including antimicrobial photodynamic therapy (aPDT) (for a review see refs. Seventeen-18) and ultraviolet C radiation (UVC) (for a review see refs. Seventeen and 20), presents an alternative to antibiotic approaches.17-20 Light is part of the electromagnetic spectrum, which is composed of wavelengths ranging from radio waves to gamma rays. Electromagnetic radiation waves are electric and magnetic field fluctuations that transport energy from one location to another. Visible light (wavelengths ranging from 390 to 700 nm) is not inherently different from the other parts of the electromagnetic spectrum with the exception that the human eye can detect visible light waves. Electromagnetic radiation is also described as a photon stream of massless particles, each traveling at the speed of light and possessing wavelike properties. A photon is the smallest quantum of energy that can be transported.21 Light-based antimicrobial approaches have an extremely important advantage in that they demonstrate equal bactericidal effectiveness for both MDR and wildtype (drug-sensitive) isolates.22 However, the aPDT approach combining a photosensitizer (PS) and light presents several challenges, such as the introducing of a photosensitizer in close proximity to bacterial cells and/or the attachment of a PS to the cell membrane and the development of strategies leading to increased photosensitizer selectivity for microbial cells instead of the host tissue. The use of UVC irradiation also has several limitations, primarily attributable to its harmful effects on eukaryotic cells and the possibility for host tissue damage, including carcinogenesis.23 Currently, a novel light-based approa\n",
      "len(chunk):   novel light-based approach, blue light therapy (BLT), is considered an attractive alternative. BLT exerts bactericidal (> 3 log10 units reduction in viable counts, up to bacterial eradication) effects without requiring the addition of exogenous photosensitizer.24,25 Moreover, BLT is considered safer for host tissue because it has fewer detrimental effects on mammalian cells than UV irradiation.23 Blue light has already been used clinically to treat inflammatory acne.26 Moreover, the efficacy of blue light for wound infections has demonstrated.24,27 The lethal and mutagenic effects of visible light radiation have been extensively investigated in bacteria. High doses of visible light lead to lethal effects primarily via the production of reactive oxygen species (ROS), which oxidatively damage proteins and lipids with a consequent loss of cell viability.28 The exact mechanism underlying the antibacterial effects of blue light is not fully understood. One generally accepted hypothesis proposes that blue light leads to the excitation of intracellular photosensitizing compounds, i.e., porphyrins, which act as photosensitizers in classical aPDT. The absorption of a photon leads to the transfer of energy and the ultimate production of highly cytotoxic reactive oxygen species, particularly singlet oxygen.24 In our previous study examining the S. aureus wild-type reference strain NCTC 8325–4, which is capable of endogenous porphyrin production, and an isogenic knockout mutant (8325–4 ΔhemB) in which endogenous porphyrin production was blocked, we demonstrated the requirement for endogenous porphyrins for the bactericidal activity of a 405-nm light treatment.29The bactericidal effects of blue light radiation have been studied in many organisms, including Propionibacterium acnes, Helicobacter pylori, Escherichia coli, Staphylococcus aureus and P. aeruginosa.24,28,30,31 Bactericidal effects have also been observed for black-pigmented oral bacteria such as Porphyromonas gingivalis and Prevotella spp. (for a review see ref. 32) as well as Aggregatibacter actinomycetemcomitans.32,33 Maclean et al. examined the bactericidal effects of 405-nm light against specific bacterial species, including Gram-positives Staphylococcus epidermidis, Enterococcus faecalis, Streptococcus pyogenes, and Clostridium perfringens and Gram-negative Acinetobacter baumannii, Proteus vulgaris, and Klebsiella pneumoniae, which are associated with hospital-acquired infections. The majority of Gram-positive and Gram-negative species are inactivated by light treatment.34 To use blue light to treat infectious diseases, it is important to assess whether blue light causes damage to the host cells. However, to date only a few studies have assessed such damage. Recent clinical trials conducted by Kleinpenning et al. showed no significant effects of blue light on human skin.35 Blue light irradiation resulted in no observed changes in studied parameters. Therefore, the use of blue light to treat skin is considered safe. Furthermore, Wang et al. did not detect apoptotic cells in blue light-irradiated mouse skin up to 24 hours after exposure, confirming its safety for use on mammalian tissue. In addition, a group from the Wellman Center for Photomedicine (WCP) investigated the potential development of P. aeruginosa tolerance to BLT. They also tested the effectiveness of antimicrobial blue light in a mouse model of wound infection caused by P. aeruginosa. P. aeruginosa did not develop tolerance to antimicrobial blue light after 10 cycles of sub-lethal inactivation, and BLT eradicated infection in a mouse model.36 Moreover, in agreement with our previous studies, the WCP group implicated endogenous porphyrins, specifically coproporphyrin III, in the photodynamic effects of blue light.36 The most recent paper by Halstead et al. described the antibacterial efficacy of blue light against a panel of 34 bacterial isolates (clinical and type strains), demonstrating the high susceptibility\n",
      "len(chunk):  g the high susceptibility of both planktonic phase bacteria and bacterial biofilms to blue light and suggesting blue light as a novel decontamination strategy for nosocomial environments and infections.27Thus, this study sought analyze the effects of blue light irradiation on P. aeruginosa in the absence of an exogenously administered PS in the context of 2 major cell surface virulence factors (flagella and pili), secreted virulence factors (such as PCN, elastases, lipases, and hemolysins), and proteolytic activity. Phototherapy primarily acts by inducing the production of reactive oxygen species and leads to the photodestruction of target cells; thus, we hypothesized that the oxidative-related destruction of bacterial cells would significantly decrease the activity of virulence factors as well as motility-mediated structures such as flagella or pili and mitigate P. aeruginosa pathogenicity. This paper describes the high bactericidal effectiveness of BLT and its inactivation of numerous P. aeruginosa virulence factors.Go to:ResultsBlue light irradiation exerts strong bactericidal effects against P. aeruginosa strainsWe subjected 15 strains (3 wild-type reference strains and 12 clinical isolates), including both wild-type (drug-sensitive) and MDR P. aeruginosa, to blue light irradiation. Phototreatment responses of the different strains were heterogeneous, ranging from 5.2 to 8 log10-unit reductions in viable counts (colony-forming units (CFU)/ml) under similar experimental conditions (at a fluence of 50 J/cm2) (Fig. 1, panels A, B, and C). Regardless of the level of antimicrobial strain resistance, BLT reduced microbial viability by ≥ 5 log10 units, suggesting blue light inactivation is an alternative bactericidal approach in the fight against MDR and extensively drug-resistant (XDR) P. aeruginosa isolates (Fig. 1, panel C). To investigate the effects of BLT on the production and/or activity of virulence factors, lethal and sub-lethal doses of blue light irradiation were evaluated because only experiments using live microbial cells (treated with sub-lethal BLT doses) permit the effects of BLT on virulence factors production to be determined. Lethal and sub-lethal conditions are defined based on whether direct damage is sufficient or insufficient to destroy the majority of a bacterial population37. In this study, lethal and sub-lethal damage were respectively defined as viable count reductions of ≥ 3 log10 and 0.5–2 log10 unit. BLT sub-lethally inactivated P. aeruginosa strains at a fluence of up to 10 J/cm2 (Fig. 1, panel A, B, and C).\n",
      "Open in a separate windowFigure 1.In vitro BLT efficacy against P. aeruginosa. Planktonic microbial cell suspensions (108 CFU/ml) were illuminated with blue light fluences ranging from 0 to 50 J/cm2 (15.7 mW/cm2; λ 405 nm; duration of irradiation: 0 to 3184 sec). Fluences up to 10 J/cm2 reveal sub-lethal efficacy against P. aeruginosa isolates. (A) BLT efficacy against reference wild-type P. aeruginosa strains; (B) BLT activity against drug-sensitive clinical P. aeruginosa isolates; (C) BLT activity against MDR clinical P. aeruginosa isolates. MDR, multidrug-resistant; XDR, extensively drug-resistant. The values are the means of 3 separate experiments (3 independent experiments with 3 repetitions per sample), and the bars represent the log relative error according to http://faculty.washington.edu/stuve/log_error.pdf. The test was performed for reference P. aeruginosa isolates PAO1, PA14, and PAK and clinical isolates no. 1959/o, 3146/s, 3404/p, 133/K, 23/K, 3752/sz, 153/s, 2284/p, 556/K, 4190/pA, 3109/o, and 143/p.Lethal BLTLethal BLT affects PCN activity PCN was analyzed and quantified in all P. aeruginosa isolates as well as light treated supernatants from bacterial cultures to investigate the impact of BLT on PCN activity. PCN activity was significantly reduced by BLT (Fig. 2, panel A). Mutant strains that were unable to produce PCN demonstrated higher susceptibility to blue light irradiation\n",
      "len(chunk):  to blue light irradiation than the PCN-producing parent strain (Fig. 2, panel B).\n",
      "Open in a separate windowFigure 2.PCN activity during BLT. (A) PCN detection and quantification during BLT. PCN activity in P. aeruginosa extracellular fractions was examined before and upon light irradiation (fluence 50 J/cm2; 15.7 mW/cm2; λ 405 nm; duration of irradiation: 3184 sec). The values are the means of 3 separate experiments (the assay was performed for 3 independent experiments with 3 repetitions per sample), and the bars represent the standard error (SE) according to Cumming et al.81 The test was performed for the PCN-producing strains: PAO1 and PA14 (reference strains) as well as 1959/o, 3404/p, 133/K, 23/K, 153/s, 2284/p, 556/K, and 4190/pA (clinical isolates). (B) Photo protective effects of PCN. Planktonic cultures of P. aeruginosa isogenic strains (108 CFU/ml) were illuminated with blue light fluences ranging from 0 to 50 J/cm2 (15.7 mW/cm2; λ 405 nm; duration of irradiation: 0 to 3184 sec). The solid line indicates the wild-type P. aeruginosa reference strain, and the dotted lines indicate isogenic derivatives expressing different phenazines (detailed description in Table 1). The values are the means of 3 separate experiments (3 independent experiments with 3 repetitions per sample), and the bars represent the log relative error according to http://faculty.washington.edu/stuve/log_error.pdf.Elastase A activity An S. aureus-based assay was employed to investigate the effects of lethal BLT on elastase A activity. All of the isolates studied yielded similar results (Fig. 3, panel A). Based on absorbance measurements at λ 600 nm, the time required for elastase A to lyse the same amount of heat-inactivated S. aureus cells was longer for blue light-treated extracellular fractions of a P. aeruginosa culture; thus lethal BLT decreased elastase A activity. Because a wide range of delta-t values was obtained, only the endpoints for all strains are shown in Fig. 3, panel A. The time point at which approximately 80% of the maximum activity (absorbance = 0.2) was observed for non-irradiated supernatants is shown for all strains in a bar chart. Time-curves are included in the Supplementary Data. The symbol Δt indicates the change in time required for the comparable lysis of S. aureus cells for BLT-treated and non-illuminated elastase A. Although each P. aeruginosa isolate demonstrated a different ratio for elastolytic activity, longer amounts of time, ranging from from 30 to 85 min, were observed for all strains (Supplementary Data).\n",
      "Open in a separate windowFigure 3.Elastase activity during BLT. (A) Elastase (A)activity during BLT. The staphylolysin activity of elastase A was determined using a reference heat-inactivated S. aureus Newman strain treated with P. aeruginosa extracellular fractions (“dark control:” kept in the dark; “illumination:” BLT; fluence 50 J/cm2; 15.7 mW/cm2; λ 405 nm; duration of irradiation: 3184 sec), and OD600 values were measured every 5 min during incubation (37°C). The figure represents the time at which the dark control supernatants reached 80% of maximum elastolytic activity. The values are the means of 3 separate experiments (the assay was performed for 3 independent experiments with one repetition per sample), and the bars represent the SE. The maximum value of absorbance was set as 1, and all other measurements were normalized as a ratio of 1. (B) Elastase (B)activity during BLT. The pseudolysin activity of elastase B was analyzed by performing and ECR assay. The reaction buffer containing ECR was treated with P. aeruginosa extracellular fractions (“dark control:” kept in the dark; “illumination:” BLT; fluence 50 J/cm2; 15.7 mW/cm2; λ 405 nm; duration of irradiation: 3184 sec), and Congo Red release was quantified by measuring the absorbance at 495 nm, which is indicative of elastase B activity. The values are the means of 3 separate experiments (the assay was performed for 3 independent experiments with one\n",
      "len(chunk):  dent experiments with one repetition per sample), and the bars represent the SE. The test was performed for the elastase B-producing strains PAO1 and PAK (reference strains) as well as 1959/o, 3146/s, 3404/p, 3752/sz, 556/K, and 143/p (clinical isolates). *P < 0.05, **P < 0.01 (dark control vs. illumination).Elastase B activity An Elastin Congo Red (ECR) assay revealed the inactivating effects of elastase B by BLT. Optical density measurements of the irradiated and non-irradiated extracellular fractions in the presence of ECR at λ 495 nm revealed decreased elastase B activity following BLT (Fig. 3, panel B). The production of elastase B was heterogeneous in the P. aeruginosa population, and light-dependent reduction in elastase B activity was observed regardless of LasB levels (Fig. 3, panel B).Overall proteolytic activity A modified skim milk assay was performed to investigate the effects of BLT on the total proteolytic activity of the strains by measuring turbidity changes in milk during incubation with extracellular fractions of P. aeruginosa isolates. A significant decrease in proteolytic activity was observed for BLT supernatants, indicating the ability of lethal BLT to inactivate bacterial proteases. Similar results were observed for all P. aeruginosa strains (Fig. 4). The bar graph shows the endpoint results for all strains at 80% maximum activity (absorbance = 0.2) in the non-irradiated supernatants. The endpoint for 40% maximum activity is shown for strain 3109/o because 80% was not reached (Fig. 4). Time-curves showing changes in the time required for proteolytic agents to lyse the same amount of skim milk proteins are presented in the Supplementary Data. Although each P. aeruginosa isolate demonstrated a different ratio for proteolytic activity, longer amounts of time, ranging from 10 to 110 min, were observed for all strains (Supplementary Data).\n",
      "Open in a separate windowFigure 4.Proteolytic activity during BLT. The overall proteolytic activity of P. aeruginosa was determined by measuring the changes in milk turbidity during incubation with bacterial extracellular fractions (“dark control:” kept in the dark; “illumination:” BLT; fluence 50 J/cm2; 15.7 mW/cm2; λ 405 nm; duration of irradiation: 3184 sec). OD600 values were measured every 5 min during incubation (37°C and 200 rpm). The figure represents the time at which the dark control supernatants reached 80% of the maximum proteolytic activity. The values are the means of 3 separate experiments (the assay was performed for 3 independent experiments with one repetition per sample), and the bars represent the SE. The maximum value of absorbance was set as 1, and all other measurements were normalized as the ratio of 1. # indicates strains for which 80% proteolytic activity was not achieved.Hemolytic activity We used fresh sheep red blood cells (RBC) to identify and quantify hemolytic agents (i.e., phospholipase C) in all P. aeruginosa isolates. The extracellular fractions of strains that demonstrated high production of hemolytic agents were subsequently irradiated with a lethal dose of blue light to investigate the effects of BLT on hemolysin activity, after which the hemolytic activity of the irradiated supernatants was measured in an RBC assay (Fig. 5). Hemolysin production by P. aeruginosa was heterogeneous, with strains exhibiting various levels of hemolytic agents. However, BLT did not affect the hemolytic activity of the strains; thus, P. aeruginosa hemolysins were not inactivated by blue light irradiation up to a fluence of 50 J/cm2.\n",
      "Open in a separate windowFigure 5.Hemolytic activity during BLT. The hemolytic activity of P. aeruginosa was evaluated using fresh sheep RBCs treated with P. aeruginosa extracellular fractions (“dark control:” kept in the dark; “illumination:” BLT; fluence 50 J/cm2; 15.7 mW/cm2; λ 405 nm; duration of irradiation: 3184 sec). The amount of hemoglobin released was assessed by measuring the absorbance at λ 540 nm. The percentage (%)\n",
      "len(chunk):  40 nm. The percentage (%) of lysed cells was calculated as follows: % = [(X-B)/(T-B)] × 100 (where B (baseline) is a negative control corresponding to the RBCs incubated with sterile medium, and T is a positive control corresponding to the total lysis obtained by incubating RBCs with a medium supplemented with 0.1% SDS). The values are the means of 3 separate experiments (the assay was performed for 3 independent experiments with one repetition per sample), and the bars represent the SE. The test was performed for all hemolysin-producing P. aeruginosa isolates: PAO1 (reference strain) and clinical isolates no. 3146/s, 4190/pA, 1959/o, 133/K, 23/K, 153/s, 556/K, 2284/p, and 143/p.Cytotoxicity of illuminated bacterial suspensions We analyzed the extracellular fractions of bacterial cultures for all P. aeruginosa isolates to confirm significant reductions in virulence factor activity by blue light irradiation. We treated human keratinocytes with filtered bacterial supernatants that had been illuminated with a lethal dose of BLT (50 J/cm2) or left untreated and measured the amounts of viable keratinocytes after 24 hrs (Fig. 6). With this experiment, we sought to confirm the ability of phototreatment with blue light to affect the potency of secreted bacterial virulence factors. P. aeruginosa isolates exhibited a range of virulence, but for all strains, the toxicity of the extracellular fractions to human dermal keratinocytes was significantly reduced by blue light illumination (Fig. 6). The viability of human cells treated with BLT-exposed P. aeruginosa supernatants increased because BLT is associated with reduced secreted virulence factor activity suggesting additional benefits derived from the use of a light-based antibacterial strategy to treat infectious diseases.\n",
      "Open in a separate windowFigure 6.Toxicity of illuminated P. aeruginosa extracellular fractions to keratinocytes. The extracellular fractions from all P. aeruginosa isolates were collected by centrifugation, filtered through 0.22-μm syringe filters, illuminated with blue light (50 J/cm2; 15.7 mW/cm2; λ 405 nm; duration of irradiation: 3184 sec) and incubated with keratinocytes for 24 h at 37°C. Cell survival was determined by performing an MTT assay. The values are the means of 3 separate experiments (3 independent experiments with 3 repetitions for each sample), and the bars represent the SE. The test was performed for the reference P. aeruginosa isolates PAO1, PA14, and PAK and clinical isolates no. 1959/o, 3146/s, 3404/p, 133/K, 23/K, 3752/sz, 153/s, 2284/p, 556/K, 4190/pA, 3109/o, and 143/p.Sub-lethal BLTSub-lethal BLT does not affect virulence factor production/activity The application of sub-lethal BLT to all P. aeruginosa isolates permitted us to identify its effects on the production and/or activity of virulence factors including (i) proteolytic enzymes (i.e., elastases A and B, serine protease IV, and alkaline protease), (ii) lipolytic agents (phospholipase C and lipases), and (iii) bacterial motility (swimming, swarming, and twitching). Sub-lethal BLT demonstrated no effects on the production/activity of these virulence factors. The experiments were performed for all strains, but Fig. 7 shows an example of the results for isolate no. 133/K, which possesses all of the virulence factors mentioned. Semi-quantitative measurement (halo diameter) indicated no changes in virulence factor activity after BLT. For the dark control and illuminated samples, we obtained the following halo measurements: 20 mm ± 1 (for proteolytic agents); 20 mm ± 2 (for lipolytic agents); 24 mm ± 2 (for swimming motility); 19 mm ± 2 (for swarming motility), and 14 mm ± 3 (for twitching motility).\n",
      "Open in a separate windowFigure 7.Sub-lethal BLT does not affect virulence factor production/activity. By applying sub-lethal BLT to P. aeruginosa isolates, we analyzed the effects of phototreatment on the production and/or activity of virulence factors. (upper panel) Proteolytic enzymes (i.e., e\n",
      "len(chunk):  teolytic enzymes (i.e., elastases A and B, serine protease IV, and alkaline protease). Skim milk agar plates were employed to analyze the overall proteolytic activity of the bacteria. P. aeruginosa cultures (“dark control:” kept in the dark; “illumination:” sub-lethal BLT; fluence 10 J/cm2; 15.7 mW/cm2; λ 405 nm; duration of irradiation: 637 sec) were transferred onto agar plates and incubated for 18 hrs at 37°C; clear zones around the colonies were indicative of total proteolytic activity. The test was performed for all P. aeruginosa strains employed in the study, but the results for only one strain (no. 133/K) are shown in the figure. (middle panel) Lipolytic agents (phospholipase C and lipases). Lipolytic activity was analyzed using egg yolk agar plates. Bacterial cultures (“dark control:” kept in the dark; “illumination:” sub-lethal BLT; fluence 10 J/cm2; 15.7 mW/cm2; λ 405 nm; duration of irradiation: 637 sec) were transferred onto agar plates and incubated for 48 hrs at 37°C. After 24 hrs, white precipitate halos around the colonies were indicative of phospholipase and lipase activity. The test was performed for all P. aeruginosa strains employed in the study, but the results for only one strain (no. 133/K) are shown in the figure. (lower panels) Bacterial motility (swimming, swarming, and twitching). Swimming and swarming motility assays were performed using M8 agar plates, and twitching motility was analyzed on LB agar plates. After incubation, bacterial growth zones were measured. The motility test was performed for all P. aeruginosa strains used in the study, but the results for only one strain (no. 133/K) are shown in the figure. Each motility assay was performed for 3 independent experiments, with one repetition per sample.Sub-lethal BLT renders drug-resistant P. aeruginosa susceptible to antimicrobials A checkerboard assay confirmed the increased susceptibility of P. aeruginosa isolates (clinical strain no. 3404/p, 556/K, 23/K and 3752/sz) pre-treated with sub-lethal BLT to the antimicrobial agents gentamycin, ceftazidime and meropenem (Fig. 8). Sequential blue light irradiation and antibiotic administration controlled bacterial growth at minimum inhibitory concentration (MIC) values that were 2-, 8-, and even 6four-fold lower. The most pronounced effects occurred for ceftazidime (Fig. 8, panels C and D) preceded by irradiation with a blue light fluence of 12.5 J/cm2 resulting in a 6four-fold decrease in the MIC value required for P. aeruginosa. In addition, the minimal bactericidal concentrations (MBCs) of antimicrobials acting synergistically with BLT were lower for blue light pre-treated P. aeruginosa strains (Table 3). Moreover, no bacterial re-growth was observed at 24 hrs after the combination treatment.\n",
      "Open in a separate windowFigure 8.BLT renders P. aeruginosa susceptible to gentamycin, ceftazidime and meropenem. A checkerboard titration method was employed to analyze the effects of sequential BLT combined with the application of antimicrobial agents. We combined various concentrations of antimicrobial agents with a range of sub-lethal doses of blue light irradiation; antibiotics were added to P. aeruginosa cultures after BLT). The plates were incubated overnight at 37°C in the dark. After incubation, OD600 values were measured to evaluate MIC changes. The values are the means of 3 separate experiments. Raw data indicating SD and SE are included in the Supplementary Data. Panels represent results for various strains: (A) strain no. 3404/p, (B) strain no. 556/K, (C) strain no. Twenty-three/K, and (D) strain no. 3752/sz.Table 3.Minimal Bactericidal Concentration change upon blue light treatment.  Minimal Bactericidal Concentration [μg/ml]  P. aeruginosa isolateAntimicrobialsDark controlBLTa treatedBLTb treated3404/p     Gentamicin32168 Ceftazidime256168 Meropenem1684556/K     Gentamicin1024512512 Ceftazidime>1024256128 Meropenem128321623/K     Gentamicin32168 Ceftazidime>512256128 Meropenem321683752/sz     \n",
      "len(chunk):  eropenem321683752/sz     Gentamicin1024168 Ceftazidime>2566432 Meropenem1684Open in a separate windowNotes.aBacteria were pretreated with sub-lethal dose of BLT (fluence 10 J/cm2; 15.7 mW/cm2; λ 405 nm)bBacteria were pretreated with sub-lethal dose of BLT (fluence 12.5 J/cm2; 15.7 mW/cm2; λ 405 nm)Go to:DiscussionIn this study, we tested the effects of BLT on several reference and clinical isolates. Notably, all of the strains were highly susceptible to BLT and demonstrated reductions in survival above 5 log10 CFU/ml and as high as 8 log10 units. Thus, according to the American Society for Microbiology, this new approach warrants the designation “antimicrobial.“38 Moreover, high bactericidal effectiveness was achieved in both wild-type and MDR isolates, including XDR strains. This is the first report demonstrating the bactericidal effectiveness of phototherapy against XDR P. aeruginosa isolates, indicating the high potential of the BLT approach for the development of light-mediated therapeutic options to treat Pseudomonas infections.P. aeruginosa secretes a number of cytotoxic factors that exert deleterious effects in cells. One of these factors is PCN (1-hydroxy-5-methylphenazine), a blue pigment with redox properties that is important for bacterial virulence and survival. PCN has deleterious effects on a variety of host cell types, including epithelial and endothelial cells and leukocytes.39 Few effective methods to inactivate this cytotoxin are available. The exposure of PCN to Rose Bengal (RB) or riboflavin and visible light causes irreversible bleaching concomitant with partial inactivation of the pigment. Based on the observed inhibition of PCN oxidation by azide, 1O2 has been proposed as a major oxidant.40 In animal models of acute and chronic lung infection, PCN is essential to P. aeruginosa virulence. Additionally, due to its known oxidoreductive properties, PCN oxidizes glutathione, inactivates catalase in respiratory epithelial cells and thus participates in oxidative-stress related damage.41 Although no therapeutic strategies directly target PCN, several antioxidant therapies have proven useful. An antioxidant-supplemented regimen involving the delivery of glutathione via aerosolization improved the lung function of cystic fibrosis patients after 8 weeks.42 Recently, the inhibition of PCN synthesis has been suggested as a therapeutic strategy, although studies have not yet been carried out.43,44 Our results demonstrate the ability of BLT to effectively inactivate PCN ; thus, BLT represents a potential therapeutic approach to target this virulence factor, ultimately leading to reductions in P. aeruginosa virulence and the diminished severity of infection outcomes.PCN is synthesized and secreted by P. aeruginosa and appears to be a primary target for photodynamic degradation. PCN is an intensely colored blue pigment in pH neutral and alkaline solutions (λmax 660 nm) and pink in acidic buffers (λmax 520 nm).45 Thus, we hypothesized that PCN protects P. aeruginosa against photooxidative injury by absorbing incident light in the visible region of the spectrum and scavenging bactericidal species generated by photosensitization, such as 1O2; both of these processes may affect the efficacy of photodynamic treatment. Successful killing of P. aeruginosa and the suppression of its virulence may also require the elimination of PCN. A recent study unambiguously confirmed the interaction of PCN and 1O2 with a high rate constant suggesting the ability of PCN to initially photoprotect P. aeruginosa against photodynamic eradication by scavenging this toxic agent.40 However, the same reaction causes irreversible pigment bleaching implying the degradation of the phenazine chromophore. Based on the results of this study, PCN may indeed photoprotect bacteria against light-based treatment by lowering the susceptibility of P. aeruginosa to BLT. The bactericidal effectiveness of BLT for isogenic mutants lacking PCN was higher than for the\n",
      "len(chunk):  N was higher than for the wild-type (PCN-producing) strain. Thus, photo-based inactivation of PCN may suppress bacterial virulence and appears to be a useful photochemical/pharmacological approach to eradicate P. aeruginosa from infected tissue.40Additionally, the pathogenicity of P. aeruginosa depends on enzymatic activity. Secreted virulence factors allow the bacteria to destroy host tissue by cleaving proteins, affecting the immune system, creating biofilms and colonizing tissue. P. aeruginosa produces numerous extracellular substances that play roles in its pathogenesis. The elastases LasB and LasA are among the most commonly secreted P. aeruginosa proteases and are classic metalloproteases.12,46 However, the total proteolytic activity of P. aeruginosa arises not only from LasA and LasB but also other secreted proteases such as protease IV and alkaline proteases.13 This study reveals the ability of BLT to significantly inhibit the proteolytic activity of P. aeruginosa, potentially reducing its virulence and lessening the severity of an infection.Motility, arguably one of the most impressive features of microbial physiology, facilitates rapid colonization of the airway. Cellular structures responsible for bacterial motility, such as flagella or pili, are crucial during the adhesion phase of colonization and bind to asialoGM1 in the epithelial cell membrane.47,48 Furthermore, both pili-mediated adherence and twitching motility are critical to P. aeruginosa virulence. In an infant mouse model of lung infection, piliated strains of P. aeruginosa caused more severe and diffuse pneumonia than corresponding non-piliated mutants.43 Unfortunately, based on the current study, BLT does not target flagella and type IV pili because it did not affect P. aeruginosa movement. However, this effect was analyzed using a sub-lethal dose of BLT, which does not exclude the possibility that lethal BLT targets the cellular structures responsible for bacterial movement.The hemolytic activity of P. aeruginosa is another significant virulence factor. In combination with other extracellular toxins, hemolysins threaten human life by causing considerable cytopathology at the site of an infection. P. aeruginosa produces and secretes hemolytic substances such as heat-resistant glycolipids and heat-labile phospholipase C.14 Unfortunately, in this study, BLT did not inactivate the agents responsible for P. aeruginosa hemolytic activity.Notably, this paper is the first report to describe the inactivation of toxic bacterial exoproducts and virulence factors of a clinically relevant pathogen by BLT. Interestingly, the effects of blue light on bacterial virulence and motility have been observed in the plant pathogen Pseudomonas syringae.49 A small number of studies in P. aeruginosa have proposed the photoinactivation of secreted proteases and phospholipase C via photodynamic treatment with exogenously administered sensitizers.50,51 Proteases were significantly reduced by irradiation with red light in the presence of the photosensitizer toluidine blue O (TBO) in a dose-dependent manner, in terms of the light energy dose and the TBO concentration. Similarly, only a few studies have described the inactivating effects of aPDT in combination with exogenously administered PSs in other bacterial species. Tubby et al. reported the inactivation of staphylococcal virulence factors using a light-activated antimicrobial agent. Protease, α-hemolysin and sphingomyelinase activities were inhibited in a dose-dependent manner following exposure to a laser light in the presence of methylene blue.52 Recently, Bartolomeu et al. the effects of photodynamic treatment using the exogenous sensitizer 5,-10,-15,-20-tetrakis(1-methylpyridinium-4-yl)porphyrin tetra-iodide (Tetra-Py(+)-Me) on catalase activity, β hemolysis, lipases, thermonuclease, enterotoxins, and coagulase production in S. aureus.53 Kato et al. described the antimicrobial photodynamic inactivation of Candida albicans v\n",
      "len(chunk):  ion of Candida albicans virulence factors and a reduction in the in vivo pathogenicity of this microbe. Mice systemically infected with C. albicans pre-treated with aPDT demonstrated significantly increased survival compared to mice infected with untreated yeast.54 Thus, photodynamic therapy in combination with exogenously administered sensitizers reduces the harmful impacts of preformed virulence factors on the host. The current study is the first report of virulence factor inactivation by blue light alone. The blue light-mediated inactivation of virulence factors likely results from the ROS activity generated during photodynamic processes. ROS easily interacts with various biomolecules, leading to their destruction. The ROS-mediated inactivation of 2 virulence factors was observed for aPDT in the presence of an exogenous photosensitizer.50 However, even after the successful inactivation of an infecting microorganism, virulence factors may still be present and cause significant damage to host tissues. Moreover, a number of in vitro and in vivo studies have shown an increase in endotoxins and hydrolytic enzymes (proteases, alkaline phosphatase, phospholipase C, and peptidoglycan hydrolase) released after exposure to different antibiotics.55,56 These virulence factors exert deleterious effects even after the pathogen has been eradicated.57,58 BLT has potential for the treatment of topical infections; thus, if it also inactivates microbial virulence factors, it has an advantage over conventional antibiotic therapy.Notably, all experiments were performed with stationary-phase P. aeruginosa cultures in which all relevant virulence factors were already expressed. Thus, the rationale for using cell-free supernatants to analyze the BLT-mediated inactivation of virulence factors may be called into question. During photodynamic processes, bacterial inactivation occurs via the loss of cell-membrane integrity and cell lysis, and all cytoplasmic components leak into the supernatant. If we would irradiate the cell culture and collect the supernatants after the treatment one could speculate that some other than photodynamic processes could occur and result in reduction of virulence factors activity. Moreover, photoinactivation utilizes the extracellular fractions of endogenous porphyrins in cell-free supernatants. Our results therefore suggest the realization of similar virulence factor photoinactivation, even in cell-free areas of infected sites in vivo.Finally, we provide evidence of the ability of BLT to affect secreted virulence factors and decrease the toxicity of extracellular P. aeruginosa fractions to human cells, which is an additional benefit of using BLT to treat infectious diseases. We have not investigated all molecules and substances in the culture supernatant that are potentially destroyed, and thus we are unable to suggest specific molecules that are responsible for the reduced toxicity of irradiated supernatants; however, the general toxicity of the extracellular fractions was clearly reduced by photoinactivation.A significant achievement of the current study was the observed synergism of sequential BLT and antibiotics, leading to the increased susceptibility of P. aeruginosa to antimicrobial agents. To our knowledge, this is the first report providing evidence of possible interactions between BLT and antimicrobial agents in P. aeruginosa. However, synergy between antibiotics and photodynamic inactivation with exogenously administered photosensitizers has been previously described in other bacterial species. Cassidy et al. observed the higher efficacy of photodynamic antimicrobial inactivation in combination with antibiotics for the treatment of a Burkholderia cepacia complex infection.59 For Staphyloccocus, Dastgheyb et al. demonstrated the additive antimicrobial activity of an illuminated porphyrin sensitizer with chloramphenicol and tobramycin in S. aureus, as well as synergistic activity against MRSA (methicillin-resist\n",
      "len(chunk):   MRSA (methicillin-resistant S. aureus) and Staphylococcus epidermidis.60 Similarly, Ronqui et al. showed the synergistic antimicrobial effects of photodynamic therapy with methylene blue and ciprofloxacin against S. aureus and E. coli.61 Furthermore, Chibebe et al. observed the ability of photodynamic and antibiotic therapy with vancomycin to impair the pathogenesis of Enterococcus faecium in a whole-animal insect model using Galleria mellonella.62 Drug resistance is a significant virulence factor that facilitates bacterial spread and leads to infection. Based on our study, blue light irradiation decreases the resistance of P. aeruginosa to various antimicrobials, such as gentamycin, ceftazidime, and meropenem, and hypothetically to other antimicrobial agents as well. Notably, this effect was achieved with sub-lethal doses of BLT, implying realization of the same results in infection site areas for which lethal BLT is not feasible due to inadequate tissue penetration by the light.Although there were several limitations to this study, such as its focus on a single bacterial species and on planktonic rather than biofilm cultures, as well as the relatively small number of virulence factors tested, it provides valuable insights into BLT and, in particular, demonstrates the potency of BLT for reducing virulence factor activity.P. aeruginosa is an example of a bacterial pathogen whose resistance is an issue of public concern; however, the current study suggests BLT may easily eradicate this bacterium, regardless of its level of drug resistance. Moreover, BLT is not only an effective tool for bacterial inactivation, reducing cell viability up to 8 log10 units, but is also beneficial for targeting and reducing the activity of several P. aeruginosa virulence factors.MDR P. aeruginosa still represents a significant fraction of circulating strains according to the latest Antimicrobial Resistance Surveillance Report. The development of new tools to treat chronic infections caused by biologically and clinically relevant strains resistant to classical antibiotics is of great importance for preventing the spread of these variants via, for example, nosocomial transmission, as well as in community-associated settings. Although one of the most pronounced limitations of this study is its focus on planktonic rather than biofilm growth, we remain convinced of the evidence-based proof that this study provides regarding BLT-mediated P. aeruginosa eradication and virulence factor reduction, leading to the development of a new non-antibiotic therapeutic approach to protect human health.Go to:Materials and methodsStrains and culture conditionsThe reference P. aeruginosa strains used are listed in Table 1. The 12 clinical P. aeruginosa isolates, were provided by Dr. Hab Julianna Kurlenda and isolated from patients hospitalized in the Provincial Hospital in Koszalin and a hospital in Gdansk (Table 2). The microorganisms were grown in Luria-Bertani (LB) broth or on LB agar (Oxoid, United Kingdom). The isolates were categorized as susceptible, intermediate, or resistant according to the Clinical and Laboratory Standards Institute (CLSI) breakpoint guidelines (for a review see ref. 63) and drug resistance patterns were determined according to the guidelines of the European Center for Disease Prevention and Control (ECDC) and the Centers for Disease Control and Prevention (CDC).63,64Table 1.Reference P. aeruginosa strains used in the study.Strain genotypePhenotypeReferencesPA14Wild-type strain. Phenazines produced: PCA, PCN, 5-MCA, PYO82PA14 ΔphzA1-G1 ΔphzA2-G2Phenazines produced: NonePA14 ΔphzHPhenazines produced: PCA, 5-MCA, PYOPA14 ΔphzH ΔphzSPhenazines produced: PCA, 5-MCAPA14 ΔphzH ΔphzMPhenazines produced: PCAPAO1Wild-type strain83PAO1 ΔfliCMutant strain with deleted gene coding major subunit of flagella (inhibited swimming motility)PAO1 ΔpilAMutant strain with deleted gene coding major subunit of the type IV pili (inhibited twitching motility)PAKWild\n",
      "len(chunk):  witching motility)PAKWild-type strain84Open in a separate windowTable 2.Clinical P. aeruginosa isolates used in the study. P. aeruginosa isolate no.           Characteristics3404/p3146/s4190/pA1959/o3109/o133/K23/K3752/sz153/s556/K2284/p143/pSource specimen UrineWoundWoundBronchiUrineUrineUrineEarUrineWoundEarWoundAntibiograma)indication of resistance to listed antimicrobials TIMFOFPIPTIMPIPNDTIMTIMTIMTIMTIMPIP  FOF TIMFOFTIM   FOFCAZ TIM    TZP TZP    IPM TZP    LVX CAZ    MEM FEP    FOF IPM    DOR GEN      ME    GEN AMK      M    AMK TOB      DOR    NET CIP      LVX    TOB        FOF    CIP LVX           LVX  Pigmentation b) LGNDLGGNDDGGBLGGDGNDVirulence factorsPyocyanin+ND++ND++ND+++ND HemolysinsND+++ND++ND++++ Elastases++++++++++++ Proteases++++++++++++ Lipases++++++++++++MotilitySwimming++++++++ND+++ SwarmingND++NDND+++++++ TwitchingND+NDND++++++++Open in a separate windowNotes.a)Antimicrobial susceptibility testing included following antimicrobials: piperacillin (PIP); ticarcillin-clavulanic acid (TIM); piperacillin-tazobactam (TZP); cefepime (FEP); ceftazidime (CAZ); imipenem (IPM); meropenem (MEM); doripenem (DOR); gentamicin (GEN); amikacin (AMK); netilmicin (NET); tobramycin (TOB); ciprofloxacin (CIP); levofloxacin (LVX); colistin (CST); polymyxin B (PMB); fosfomycin (FOF)b)LG, light green; G, green; DG, dark green; B, brown; ND, not detected.Light sourceIllumination was provided with a single-emitter diode lamp (405 nm, 120 W; output light with a power of 11 W/12 mm2; full width at half maximum (FWHM) of 10 nm), equipped with a panel controller and PC software (SecureMedia, Poland).Visible light irradiation studiesBacterial cultures were grown overnight (16 hrs) at 37°C in LB broth and then diluted to 108/ml bacterial cells. The cells were then transferred to a 96-well microtiter plate (100 µl per well) and illuminated with the selected blue light fluence, ranging from 0 to 50 J/cm2 (15.7 mW/cm2; λ 405 nm; duration of irradiation: 0 to 3184 sec) at room temperature. Following illumination, 10-µl aliquots were obtained to determine the CFU. The contents of the wells were mixed before sampling, and the aliquots were serially diluted 10-fold in PBS (0.13 mM NaCl, 8.1 mM Na2HPO4, 2.68 mM KCl, 1.47 mM KH2PO4) to achieve final dilutions of 10−1 to 10−6, which were then streaked horizontally as previously described.65 The control consisted of bacteria incubated in the dark for the same amount of time as the bacteria exposed to light. Each experiment was carried out 3 times (3 independent experiments with 3 repetitions per sample; the time interval between the 3 replicates was 24 hrs). The survival fraction was expressed as the ratio of the CFU of bacteria treated with blue light to the CFU of untreated bacteria. The detection limit was 10 CFU/ml.Determination of lethal and sub-lethal dosesA sub-lethal dose was defined as a dose that did not eliminate the majority of the microbial population. The lethal dose was defined as the dose that eliminated ≥ 99.9% of the bacterial population.66 In this study, the lethal dose was determined under photodynamic conditions as the dose that reduced the number of bacterial cells by ≥ 3 log10. A sub-lethal dose reduced the number of bacteria by 0.5 – 2 log10. Sub-lethal doses were defined according to previous publications.36,37,67-69Preparation of P. aeruginosa cultures supernatantsBacterial supernatants were prepared to determine PCN concentrations and evaluate protease, elastase and hemolysin activities according to a previously published protocol.70 Fresh overnight liquid bacterial cultures in LB were diluted to optical density (OD)600 = 1. A total of 0.25 ml of this dilution was transferred into 25 ml of fresh LB in a 250-ml glass flask and incubated overnight at 37°C at 150 rpm. After overnight incubation, the culture was transferred into Falcon tubes and centrifuged (10,000 rpm for 10 min), and the supernatant filtered through a 0.22-µm syringe filter. The supernatant was then trans\n",
      "len(chunk):  upernatant was then transferred to a 24-well microtiter plate (600 µl per well) and illuminated with the selected blue light fluence at room temperature. Following irradiation, the supernatants were stored at −20°C for further analysis.PCN detection and quantificationThe concentration of PCN in the supernatant was determined before and after light irradiation. A total of 2.5 ml of supernatant and 1.5 ml of chloroform (Sigma-Aldrich, Germany) were mixed and centrifuged for 5 min at 4,500 rpm. The chloroform phase was transferred into a new tube and mixed with 0.5 ml of 0.2 M HCl (Avantor Performance Materials, Poland). The sample was then centrifuged at 4,500 rpm for 8 min. The pink-colored acidic phase was transferred to a 96-well plate, and the absorbance at λ 520 nm was measured.70 The assay was performed for 3 independent experiments with 3 repetitions per sample. The detection limit was 2.5 μM.71Hemolytic activityTo analyze the hemolytic activity of the bacterial supernatant, fresh sheep RBC (Innovative Research, USA) were employed using a previously described technique.15 The cells were washed 3 times in PBS (centrifuged for 5 min at 6000 rpm) and resuspended in RPMI-1640 medium (Roswell Park Memorial Institute, RPMI) (Sigma-Aldrich, Germany) to a final concentration of 2% (cell volume/medium volume). Next, 700 µl of the prepared RBC solution was mixed with 300 µl of the supernatant in an Eppendorf tube. The tubes were incubated horizontally at 37°C and 150 rpm for 25 min. After incubation, the suspension was centrifuged (10,000 rcf, 10 min, 4°C), and 240 µl of the sample was transferred into a 96 well plate. The released hemoglobin was determined by measuring the absorbance at λ 540 nm. The percentage (%) of lysed cells was calculated as follows: % = [(X-B)/(T-B)] × 100 (where B (baseline) is a negative control corresponding to RBCs incubated with 300 μl of sterile LB and T is a positive control corresponding to the total lysis obtained by incubating RBCs with 300 µl of LB supplemented with 0.1% SDS (Sigma-Aldrich, Germany)). The assay was performed for 3 independent experiments with one repetition per sample.Elastolytic activityElastase A (staphylolysin) activity Staphylococcus aureus cells were employed to analyze staphylolysin activity according to a previously described method.72 A reaction buffer consisting of 0.02 M Tris-HCl (Sigma-Aldrich, Germany) at pH 8.5 was prepared and sterilized. One colony of the S. aureus Newman strain was transferred into 250 ml of LB broth and incubated overnight (37°C, 150 rpm). The bacteria were centrifuged, washed twice with reaction buffer and resuspended in 25 ml of buffer in a heat-resistant tube. The solution was placed in a water bath and heated to 100°C for 15 min to kill all of the bacteria, then stored at −20°C. Prior to experiments, the S. aureus cells were defrosted and diluted in reaction buffer to an OD595 of 0.8–0.9. Next, 900 µl of the prepared solution was transferred into 24-well plates, and 100 µl of P. aeruginosa extracellular fractions, both untreated and treated with BLT, were added. The plates were placed immediately in an EnVision plate reader. OD600 measurements were taken every 5 min during incubation (37°C, orbital shaking mode). The assay was performed for 3 independent experiments with one repetition per sample. The maximum absorbance value was set as 1, and all other measurements were normalized at a ratio of 1.Elastase B (pseudolysin) activity Pseudolysin activity was analyzed by performing an Elastin Congo Red assay. A reaction buffer containing 0.05 M Tris-HCl supplemented with 0.5 mM CaCl2 (Avantor Performance Materials, Poland) at pH 7.5 was prepared and sterilized. A total of 10 mg of Elastin Congo Red (Sigma-Aldrich, Germany) was transferred into an Eppendorf tube, and 900 µl of reaction buffer was added. Next, 100 µl of the analyzed P. aeruginosa extracellular fractions, both untreated and treated with BLT, were added, and the tubes were incubated \n",
      "len(chunk):  the tubes were incubated horizontally at 37°C and 150 rpm for 3 and 24 hrs. The suspensions were centrifuged (10,000 rpm, 10 min, and 4°C), and 200 µl of each sample was transferred into 96-well plates. The released Congo Red was quantified by measuring the absorbance at 495 nm, which was indicative of elastase B activity.72,73 Staphylolysin also exhibited a certain degree of activity in this assay, which enhanced the elastinolytic activity of pseudolysin by several fold, indicating synergistic action by staphylolysin and pseudolysin.74 However, because the increased absorbance at 495 nm resulted from the liberation of soluble dye fragments due to pseudolysin action, this assay only measured elastase B activity.72 The assay was performed for 3 independent experiments with one repetition per sample. The detection limit was 4.0 μg of elastase per mg of total protein.75Proteolytic activityTo analyze the overall proteolytic activity of the bacteria, skim milk agar plates were employed as previously described.72 A regular LB 1.5% agar solution was sterilized, cooled to 50°C and then supplemented with 1.25% (w/v) sterile skim milk. A total of 20 ml of the solution was poured aseptically into a Petri dish to solidify at room temperature. A total of 2 µl of each of the analyzed cultures, both untreated and treated with a sub-lethal dose of BLT, were transferred onto the agar plates and incubated for 18 hrs at 37°C. The clear zones around the colonies were indicative of total proteolytic activity. For lethal BLT, total secreted proteases activity was determined using a modified skim milk assay based on the turbidity changes in the milk during protease incubation.70 A total of 900 µl of 1.5% skim milk was transferred into a transparent 24-well plate. Next, 100 µl of the prepared supernatants were added, and the plate was analyzed using an EnVision plate reader. Finally, OD600 measurements were taken every 5 min during incubation (37°C and 200 rpm). The assay was performed for 3 independent experiments with one repetition per sample.Lipolytic activityLipolytic activity was analyzed using egg yolk agar plates according to a previously described method.76 To prepare the egg yolk emulsion, chicken eggs were washed carefully and left in 70% ethanol for 5 min. The shells were crushed, the albumen was separated, and the yolks were transferred into a sterile graduated cylinder and mixed with ddH2O at a ratio of 2:3. LB supplemented with agar (15 g/l) was autoclaved and cooled to 50°C. The medium was mixed thoroughly with a freshly prepared egg yolk emulsion (10% v/v). A total of 20 ml of solution was poured into appropriate plates to solidify for at least 20 min. A total of 2 µl of the analyzed P. aeruginosa cultures, both untreated and treated with BLT, were transferred onto the agar plates and incubated for 24 and 48 hrs at 37°C. After 24 hrs, a white precipitation halo around the colonies was observed as the result of phospholipase and lipase activity. The assay was performed for 3 independent experiments with one repetition per sample.Motility assaysSwimming Swimming motility assays were performed with M8 agar plates supplemented with glucose, casamino acids, and MgSO4 (Avantor Performance Materials, Poland) following a previously described method with modifications.77 Five-fold concentrated M8 medium (30 g/l Na2HPO4; 15 g/l KH2PO4; and 2.5 g/l NaCl) was autoclaved. A 2% glucose solution, a 2% casamino acids solution and a 1 M MgSO4 solution were filtered separately using 0.22-µm syringe filters. Next, 3 g of agar was mixed with 765 ml of water, autoclaved and cooled to 50°C. Then, 200 ml of 5x M8 solution, 10 ml of 2% glucose, 25 ml of 2% casamino acids and 1 ml of 1 M MgSO4 solution were thoroughly mixed. A total of 25 ml of the prepared medium was poured into appropriate plates to solidify and dry for 2 hrs. The tips of sterile toothpicks were immersed in individual analyzed bacterial cultures, both untreated and treated with BLT, and th\n",
      "len(chunk):   treated with BLT, and then inserted into a swimming agar plate (avoiding piercing the agar completely down to the bottom of the Petri dish). The plates were incubated with the lids up for 18 hrs at 37°C, and bacterial growth zones were measured.Swarming Swarming motility assays were performed with M8 agar plates supplemented with glucose, casamino acids, and MgSO4 following a previously described method with modifications.78 The 5x M8 medium previously utilized to analyze swimming motility was autoclaved. A 10% glucose solution, a 10% casamino acids solution and a 1 M MgSO4 solution were filtered separately through 0.22-µm syringe filters. Next, 8 g of agar was mixed with 765 ml of water, autoclaved and cooled to 50°C. Then, 200 ml of 5x M8 solution, 10 ml of 10% glucose, 25 ml of 10% casamino acids and 1 ml of 1 M MgSO4 solution were mixed thoroughly. A total of 25 ml of the prepared medium was poured into appropriate plates to solidify and dry for 2 hrs. A total of 2 µl of the analyzed bacterial cultures, both untreated and treated with BLT, were transferred onto the agar plates. The plates were incubated with the lids up for 18 hrs at 37°C, and bacterial growth zones were measured.Twitching Twitching motility was analyzed on LB agar plates according to a previously described method.79 LB medium at 20 g/l and agar at 10 g/l were mixed and autoclaved. Ten milliliters of the prepared solution was poured into appropriate plates to solidify and dry for 24 hrs. The tips of sterile toothpicks were immersed in individual analyzed bacterial cultures, both untreated and treated with BLT, and inserted deep into the twitching agar plates to reach the bottom. The plates were placed with lids down in a humidifying chamber and incubated for 18 hrs at 37°C. To visualize the results, a TM solution (50% v/v methanol and 10% v/v glacial acetic acid) (Avantor Performance Materials, Poland) was dropped on top of the agar plate and incubated for at least 5 min. Then, the TM solution was carefully washed out, and bacterial growth zones were measured.Each motility assay was performed for 3 independent experiments with one repetition per sample.Checkerboard analysis of the interaction between gentamycin, ceftazidime, meropenem and blue light irradiationTo evaluate potential synergistic cooperation between antimicrobials and BLT, we analyzed various combinations of antibiotics and BLT using a checkerboard titration method.80 We combined different concentrations of antimicrobials with a range of sub-lethal doses of blue light irradiation (the antibiotic was added to the P. aeruginosa culture after the BLT treatment). The plates were incubated overnight at 37°C in the dark. After incubation, OD600 values were measured to determine MIC changes. All measurements were carried out 3 times (3 independent experiments).Eukaryotic cell cultureA human keratinocyte (HaCaT) cell line (purchased from Cell Line Services Eppelheim, Germany) was cultured in Dulbecco's modified Eagle's medium (DMEM) supplemented with 10% fetal bovine serum, 2 mM glutamine, 100 units/ml penicillin and 100 μg/ml streptomycin. The cultures were maintained in a humidified atmosphere containing 5% CO2 at 37°C.Cytotoxicity assay with extracellular fractionsP. aeruginosa strain supernatants were collected as previously described. Keratinocytes (3 × 104) in a 200-μl volume were seeded into 96-well plates and allowed to adhere overnight. Bacterial cell supernatants (both illuminated and kept in dark), grown for 48 hrs and filtered, were then added to the medium in a volume of 10 µl, and cells were incubated for a further 24 hrs. Cell survival was determined by performing an MTT (3-(4,5-dimethylthiazol-2-yl)-2,5-diphenyltetrazolium bromide) assay. Briefly, MTT (Sigma-Aldrich, Germany) was added to a final concentration of 0.5 mg/ml and incubated for 3 hrs at 37°C. Cells were lysed with DMSO (Sigma-Aldrich, Germany) and the absorbance of the formazan solution was measured at 550 nm with a plat\n",
      "len(chunk):  red at 550 nm with a plate reader (Victor, 1420 multilabel counter). All measurements were carried out 3 times (3 independent experiments with 3 repetitions for each sample).Statistical methodsStatistical analyses were performed using the statistical suite STATISTICA (data analysis software system) version 10.0 (StatSoft. Inc. (2011), www.statsoft.com) and Excel. Quantitative variables were characterized by the arithmetic mean of standard deviation. To test the homogeneity of the variances, the Leven (Brown-Forsythe) test was utilized. The statistical significance of differences between 2 groups was determined with Student's t-test. The significance level was P < 0.05.Go to:Supplementary MaterialKVIR_S_1250995.zip:Click here to view.(2.2M, zip)Go to:Disclosure of potential conflicts of interestNo potential conflicts of interest were disclosed.Go to:AcknowledgmentsThe isogenic pairs of isolates with affected phenazines production were kindly provided by Lars Dietrich from Department of Biological Sciences Columbia University New York. P. aeruginosa strains with disrupted movement systems were provided by Joanne N. Engel from Department of Medicine, University of California San Francisco, United States of America. Wild-type strain PAK was kindly provided by Reuben Ramphal from Institut Pasteur Paris, FranceGo to:FundingThe studies investigating blue light treatment of P. aeruginosa and its synergy with antimicrobials were supported by grant no. 2015/19/B/NZ7/02487 from the National Science Center (M.G.). Publication costs were covered by the EU FP7 project MOBI4Health, GA. 316094. The funders had no role in study design, data collection and analysis, decision to publish, or preparation of the manuscript.Go to:Author contributionsGF did the experimental work, participated in conception of the study and has been involved in drafting manuscript. AK performed studies concerning eukaryotic cells. MG participated in the experimental work, has been involved in the coordination, conception, and design of the study and wrote the manuscript. All of the authors have read and approved the final manuscript.Go to:Reference[1] Balasubramanian D, Schneper L, Kumari H, Mathee K.\n",
      "A dynamic and intricate regulatory network determines Pseudomonas aeruginosa virulence. Nucleic Acids Res\n",
      "2013; 41:1-20; PMID:23143271; http://dx.doi.org/ 10.1093/nar/gks1039 [PMC free article] [PubMed] [CrossRef] [Google Scholar][2] Rybtke M, Hultqvist LD, Givskov M, Tolker-Nielsen T.\n",
      "Pseudomonas aeruginosa Biofilm Infections: Community Structure, Antimicrobial Tolerance and Immune Response. J Mol Biol\n",
      "2015; 427:3628-45; PMID:26319792; http://dx.doi.org/ 10.1016/j.jmb.2015.08.016 [PubMed] [CrossRef] [Google Scholar][3] Serra R, Grande R, Butrico L, Rossi A, Settimio UF, Caroleo B, Amato B, Gallelli L, de FS.\n",
      "Chronic wound infections: the role of Pseudomonas aeruginosa and Staphylococcus aureus. Expert Rev Anti Infect Ther\n",
      "2015; 13:605-13; PMID:25746414; http://dx.doi.org/ 10.1586/14787210.2015.1023291 [PubMed] [CrossRef] [Google Scholar][4] Mayaud C, Parrot A, Cadranel J.\n",
      "Pyogenic bacterial lower respiratory tract infection in human immunodeficiency virus-infected patients. Eur Respir J Suppl\n",
      "2002; 36:28s-39s; PMID:12168745; http://dx.doi.org/ 10.1183/09031936.02.00400602 [PubMed] [CrossRef] [Google Scholar][5] Vento S, Cainelli F, Temesgen Z.\n",
      "Lung infections after cancer chemotherapy. Lancet Oncol\n",
      "2008; 9:982-92; PMID:19071255; http://dx.doi.org/ 10.1016/S1470-2045(08)70255-9 [PubMed] [CrossRef] [Google Scholar][6] Potron A, Poirel L, Nordmann P.\n",
      "Emerging broad-spectrum resistance in Pseudomonas aeruginosa and Acinetobacter baumannii: Mechanisms and epidemiology. Int J Antimicrob Agents\n",
      "2015; 45:568-85; PMID:25857949; http://dx.doi.org/ 10.1016/j.ijantimicag.2015.03.001 [PubMed] [CrossRef] [Google Scholar][7] Clatworthy AE, Pierson E, Hung DT.\n",
      "Targeting virulence: a new paradigm for antimicrobial therapy. Nat Chem Biol\n",
      "2007; 3:541-8; PMID:17710100; http://dx.doi.org/ 1\n",
      "len(chunk):  100; http://dx.doi.org/ 10.1038/nchembio.2007.24 [PubMed] [CrossRef] [Google Scholar][8] Merrell DS, Falkow S.\n",
      "Frontal and stealth attack strategies in microbial pathogenesis. Nature\n",
      "2004; 430:250-6; PMID:15241423; http://dx.doi.org/ 10.1038/nature02760 [PubMed] [CrossRef] [Google Scholar][9] Sommer R, Joachim I, Wagner S, Titz A.\n",
      "New approaches to control infections: anti-biofilm strategies against gram-negative bacteria. Chimia (Aarau)\n",
      "2013; 67:286-90; PMID:23967708; http://dx.doi.org/ 10.2533/chimia.2013.286 [PubMed] [CrossRef] [Google Scholar][10] Rasmussen TB, Givskov M.\n",
      "Quorum-sensing inhibitors as anti-pathogenic drugs. Int J Med Microbiol\n",
      "2006; 296:149-61; PMID:16503194; http://dx.doi.org/ 10.1016/j.ijmm.2006.02.005 [PubMed] [CrossRef] [Google Scholar][11] Wagner S, Sommer R, Hinsberger S, Lu C, Hartmann RW, Empting M, Titz A.\n",
      "Novel Strategies for the Treatment of Pseudomonas aeruginosa Infections. J Med Chem\n",
      "2016; 59:5929-69; PMID:26804741; http://dx.doi.org/ 10.1021/acs.jmedchem.5b01698 [PubMed] [CrossRef] [Google Scholar][12] Kuang Z, Hao Y, Walling BE, Jeffries JL, Ohman DE, Lau GW.\n",
      "Pseudomonas aeruginosa elastase provides an escape from phagocytosis by degrading the pulmonary surfactant protein-A. PLoS One\n",
      "2011; 6:e27091; PMID:22069491; http://dx.doi.org/ 10.1371/journal.pone.0027091 [PMC free article] [PubMed] [CrossRef] [Google Scholar][13] Caballero AR, Moreau JM, Engel LS, Marquart ME, Hill JM, O'Callaghan RJ.\n",
      "Pseudomonas aeruginosa protease IV enzyme assays and comparison to other Pseudomonas proteases. Anal Biochem\n",
      "2001; 290:330-7; PMID:11237336; http://dx.doi.org/ 10.1006/abio.2001.4999 [PubMed] [CrossRef] [Google Scholar][14] Berka RM, Vasil ML.\n",
      "Phospholipase C (heat-labile hemolysin) of Pseudomonas aeruginosa: purification and preliminary characterization. J Bacteriol\n",
      "1982; 152:239-45; PMID:6811552 [PMC free article] [PubMed] [Google Scholar][15] Rossignol G, Merieau A, Guerillon J, Veron W, Lesouhaitier O, Feuilloley MG, Orange N.\n",
      "Involvement of a phospholipase C in the hemolytic activity of a clinical strain of Pseudomonas fluorescens. BMC Microbiol\n",
      "2008; 8:189; PMID:18973676; http://dx.doi.org/ 10.1186/1471-2180-8-189 [PMC free article] [PubMed] [CrossRef] [Google Scholar][16] Bush K, Courvalin P, Dantas G, Davies J, Eisenstein B, Huovinen P, Jacoby GA, Kishony R, Kreiswirth BN, Kutter E, et al..\n",
      "Tackling antibiotic resistance. Nat Rev Microbiol\n",
      "2011; 9:894-6; PMID:22048738; http://dx.doi.org/ 10.1038/nrmicro2693 [PMC free article] [PubMed] [CrossRef] [Google Scholar][17] Dai T, Tegos GP, Zhiyentayev T, Mylonakis E, Hamblin MR.\n",
      "Photodynamic therapy for methicillin-resistant Staphylococcus aureus infection in a mouse skin abrasion model. Lasers Surg Med\n",
      "2010; 42:38-44; PMID:20077489; http://dx.doi.org/ 10.1002/lsm.20887 [PMC free article] [PubMed] [CrossRef] [Google Scholar][18] Dai T, Tegos GP, Lu Z, Huang L, Zhiyentayev T, Franklin MJ, Baer DG, Hamblin MR.\n",
      "Photodynamic therapy for Acinetobacter baumannii burn infections in mice. Antimicrob Agents Chemother\n",
      "2009; 53:3929-34; PMID:19564369; http://dx.doi.org/ 10.1128/AAC.00027-09 [PMC free article] [PubMed] [CrossRef] [Google Scholar][19] Dai T, Garcia B, Murray CK, Vrahas MS, Hamblin MR.\n",
      "UVC light prophylaxis for cutaneous wound infections in mice. Antimicrob Agents Chemother\n",
      "2012; 56:3841-8; PMID:22564833; http://dx.doi.org/ 10.1128/AAC.00161-12 [PMC free article] [PubMed] [CrossRef] [Google Scholar][20] Dai T, Kharkwal GB, Zhao J, St Denis TG, Wu Q, Xia Y, Huang L, Sharma SK, d'Enfert C, Hamblin MR.\n",
      "Ultraviolet-C light for treatment of Candida albicans burn infection in mice. Photochem Photobiol\n",
      "2011; 87:342-9; PMID:21208209; http://dx.doi.org/ 10.1111/j.1751-1097.2011.00886.x [PMC free article] [PubMed] [CrossRef] [Google Scholar][21] Waldman G.\n",
      "Introduction to light : the physics of light, vision, and color. 193. Mineola: Dover Publications, 2002 [Google Scholar][22] Grinholc M, Rapacka-Zdonczyk A, Rybak B, Szabados F, Bielawski KP.\n",
      "Multiresistant \n",
      "len(chunk):  awski KP.\n",
      "Multiresistant Strains Are as Susceptible to Photodynamic Inactivation as Their Naive Counterparts: Protoporphyrin IX-Mediated Photoinactivation Reveals Differences Between Methicillin-Resistant and Methicillin-Sensitive Staphylococcus aureus Strains. Photomed Laser Surg\n",
      "2014; 32:121-9; PMID:24527879; http://dx.doi.org/ 10.1089/pho.2013.3663 [PMC free article] [PubMed] [CrossRef] [Google Scholar][23] Dai T, Vrahas MS, Murray CK, Hamblin MR.\n",
      "Ultraviolet C irradiation: an alternative antimicrobial approach to localized infections?\n",
      "Expert Rev Anti Infect Ther\n",
      "2012; 10:185-95; PMID:22339192; http://dx.doi.org/ 10.1586/eri.11.166 [PMC free article] [PubMed] [CrossRef] [Google Scholar][24] Dai T, Gupta A, Huang YY, Yin R, Murray CK, Vrahas MS, Sherwood ME, Tegos GP, Hamblin MR.\n",
      "Blue light rescues mice from potentially fatal Pseudomonas aeruginosa burn infection: efficacy, safety, and mechanism of action. Antimicrob Agents Chemother\n",
      "2013; 57:1238-45; PMID:23262998; http://dx.doi.org/ 10.1128/AAC.01652-12 [PMC free article] [PubMed] [CrossRef] [Google Scholar][25] Enwemeka CS, Williams D, Hollosi S, Yens D, Enwemeka SK.\n",
      "Visible 405 nm SLD light photo-destroys methicillin-resistant Staphylococcus aureus (MRSA) in vitro. Lasers Surg Med\n",
      "2008; 40:734-7; PMID:19065556; http://dx.doi.org/ 10.1002/lsm.20724 [PubMed] [CrossRef] [Google Scholar][26] Wheeland RG, Dhawan S.\n",
      "Evaluation of self-treatment of mild-to-moderate facial acne with a blue light treatment system. J Drugs Dermatol\n",
      "2011; 10:596-602; PMID:21637900 [PubMed] [Google Scholar][27] Halstead FD, Thwaite JE, Burt R, Laws TR, Raguse M, Moeller R, Webber MA, Oppenheim BA.\n",
      "The antibacterial activity of blue light against nosocomial wound pathogens growing planktonically and as mature biofilms. Appl Environ Microbiol\n",
      "2016; PMID:27129967 [PMC free article] [PubMed] [Google Scholar][28] Dai T, Gupta A, Murray CK, Vrahas MS, Tegos GP, Hamblin MR.\n",
      "Blue light for infectious diseases: Propionibacterium acnes, Helicobacter pylori, and beyond?\n",
      "Drug Resist Updat\n",
      "2012; 15:223-36; PMID:22846406; http://dx.doi.org/ 10.1016/j.drup.2012.07.001 [PMC free article] [PubMed] [CrossRef] [Google Scholar][29] Grinholc M, Rodziewicz A, Forys K, Rapacka-Zdonczyk A, Kawiak A, Domachowska A, Golunski G, Wolz C, Mesak L, Becker K, et al..\n",
      "Fine-tuning recA expression in Staphylococcus aureus for antimicrobial photoinactivation: importance of photo-induced DNA damage in the photoinactivation mechanism. Appl Microbiol Biotechnol\n",
      "2015; 99:9161-76; PMID:26252968; http://dx.doi.org/ 10.1007/s00253-015-6863-z [PMC free article] [PubMed] [CrossRef] [Google Scholar][30] Dadras S, Mohajerani E, Eftekhar F, Hosseini M.\n",
      "Different photoresponses of Staphylococcus aureus and Pseudomonas aeruginosa to 514, 532, and 633 nm low level lasers in vitro. Curr Microbiol\n",
      "2006; 53:282-6; PMID:16941244; http://dx.doi.org/ 10.1007/s00284-005-0490-3 [PubMed] [CrossRef] [Google Scholar][31] Kim S, Kim J, Lim W, Jeon S, Kim O, Koh JT, Kim CS, Choi H, Kim O.\n",
      "In vitro bactericidal effects of 625, 525, and 425 nm wavelength (red, green, and blue) light-emitting diode irradiation. Photomed Laser Surg\n",
      "2013; 31:554-62; PMID:24138193; http://dx.doi.org/ 10.1089/pho.2012.3343 [PMC free article] [PubMed] [CrossRef] [Google Scholar][32] Cieplik F, Spath A, Leibl C, Gollmer A, Regensburger J, Tabenski L, Hiller KA, Maisch T, Schmalz G.\n",
      "Blue light kills Aggregatibacter actinomycetemcomitans due to its endogenous photosensitizers. Clin Oral Investig\n",
      "2014; 18:1763-9; PMID:24297656; http://dx.doi.org/ 10.1007/s00784-013-1151-8 [PubMed] [CrossRef] [Google Scholar][33] Soukos NS, Som S, Abernethy AD, Ruggiero K, Dunham J, Lee C, Doukas AG, Goodson JM.\n",
      "Phototargeting oral black-pigmented bacteria. Antimicrob Agents Chemother\n",
      "2005; 49:1391-6; PMID:15793117; http://dx.doi.org/ 10.1128/AAC.49.4.1391-1396.2005 [PMC free article] [PubMed] [CrossRef] [Google Scholar][34] Maclean M, McKenzie K, Anderson JG, Gettinby G, MacGregor SJ.\n",
      "405 nm light technolo\n",
      "len(chunk):  SJ.\n",
      "405 nm light technology for the inactivation of pathogens and its potential role for environmental disinfection and infection control. J Hosp Infect\n",
      "2014; 88:1-11; PMID:25066049; http://dx.doi.org/ 10.1016/j.jhin.2014.06.004 [PubMed] [CrossRef] [Google Scholar][35] Kleinpenning MM, Smits T, Frunt MH, van Erp PE, van de Kerkhof PC, Gerritsen RM.\n",
      "Clinical and histological effects of blue light on normal skin. Photodermatol Photoimmunol Photomed\n",
      "2010; 26:16-21; PMID:20070834; http://dx.doi.org/ 10.1111/j.1600-0781.2009.00474.x [PubMed] [CrossRef] [Google Scholar][36] Amin RM, Bhayana B, Hamblin MR, Dai T.\n",
      "Antimicrobial blue light inactivation of Pseudomonas aeruginosa by photo-excitation of endogenous porphyrins: In vitro and in vivo studies. Lasers Surg Med\n",
      "2016; PMID:26891084 [PMC free article] [PubMed] [Google Scholar][37] Dodd CER, Sharman RL, Bloomfield SF, Booth IR, Stewart GSAB.\n",
      "Inimical processes: Bacterial self-destruction and sub-lethal injury. Trends Food Sci Technol\n",
      "1997; 8:238-41; http://dx.doi.org/ 10.1016/S0924-2244(97)01043-1 [CrossRef] [Google Scholar][38] ASM\n",
      "Antimicrobial Agents and Chemotherapy, Instructions to Authors. http://aac.asm.org/site/misc/2016AprilAACITA.pdf. 2016 [Google Scholar][39] Lau GW, Ran H, Kong F, Hassett DJ, Mavrodi D.\n",
      "Pseudomonas aeruginosa pyocyanin is critical for lung infection in mice. Infect Immun\n",
      "2004; 72:4275-8; PMID:15213173; http://dx.doi.org/ 10.1128/IAI.72.7.4275-4278.2004 [PMC free article] [PubMed] [CrossRef] [Google Scholar][40] Reszka KJ, Bilski PJ, Britigan BE.\n",
      "Quenching of singlet oxygen by pyocyanin and related phenazines. Photochem Photobiol\n",
      "2010; 86:742-6; PMID:20408986; http://dx.doi.org/ 10.1111/j.1751-1097.2010.00728.x [PubMed] [CrossRef] [Google Scholar][41] Hawdon NA, Aval PS, Barnes RJ, Gravelle SK, Rosengren J, Khan S, Ciofu O, Johansen HK, Hoiby N, Ulanova M.\n",
      "Cellular responses of A549 alveolar epithelial cells to serially collected Pseudomonas aeruginosa from cystic fibrosis patients at different stages of pulmonary infection. FEMS Immunol Med Microbiol\n",
      "2010; 59:207-20; PMID:20528926; http://dx.doi.org/ 10.1111/j.1574-695X.2010.00693.x [PubMed] [CrossRef] [Google Scholar][42] Griese M, Kappler M, Eismann C, Ballmann M, Junge S, Rietschel E, van Koningsbruggen-Rietschel S, Staab D, Rolinck-Werninghaus C, Mellies U, et al..\n",
      "Inhalation treatment with glutathione in patients with cystic fibrosis. A randomized clinical trial. Am J Respir Crit Care Med\n",
      "2013; 188:83-9; PMID:23631796; http://dx.doi.org/ 10.1164/rccm.201303-0427OC [PubMed] [CrossRef] [Google Scholar][43] Kipnis E, Sawa T, Wiener-Kronish J.\n",
      "Targeting mechanisms of Pseudomonas aeruginosa pathogenesis. Med Mal Infect\n",
      "2006; 36:78-91; PMID:16427231; http://dx.doi.org/ 10.1016/j.medmal.2005.10.007 [PubMed] [CrossRef] [Google Scholar][44] Lau GW, Hassett DJ, Ran H, Kong F.\n",
      "The role of pyocyanin in Pseudomonas aeruginosa infection. Trends Mol Med\n",
      "2004; 10:599-606; PMID:15567330; http://dx.doi.org/ 10.1016/j.molmed.2004.10.002 [PubMed] [CrossRef] [Google Scholar][45] Das T, Kutty SK, Kumar N, Manefield M.\n",
      "Pyocyanin facilitates extracellular DNA binding to Pseudomonas aeruginosa influencing cell surface properties and aggregation. PLoS One\n",
      "2013; 8:e58299; PMID:23505483; http://dx.doi.org/ 10.1371/journal.pone.0058299 [PMC free article] [PubMed] [CrossRef] [Google Scholar][46] Das MC, Sandhu P, Gupta P, Rudrapaul P, De UC, Tribedi P, Akhter Y, Bhattacharjee S.\n",
      "Attenuation of Pseudomonas aeruginosa biofilm formation by Vitexin: A combinatorial study with azithromycin and gentamicin. Sci Rep\n",
      "2016; 6:23347; PMID:27000525; http://dx.doi.org/ 10.1038/srep23347 [PMC free article] [PubMed] [CrossRef] [Google Scholar][47] Comolli JC, Waite LL, Mostov KE, Engel JN.\n",
      "Pili binding to asialo-GM1 on epithelial cells can mediate cytotoxicity or bacterial internalization by Pseudomonas aeruginosa. Infect Immun\n",
      "1999; 67:3207-14; PMID:10377092 [PMC free article] [PubMed] [Google Scholar][48] de BS, Roger P, Dupuit F, Bajolet-L\n",
      "len(chunk):  er P, Dupuit F, Bajolet-Laudinat O, Fuchey C, Plotkowski MC, Puchelle E.\n",
      "Asialo GM1 is a receptor for Pseudomonas aeruginosa adherence to regenerating respiratory epithelial cells. Infect Immun\n",
      "1996; 64:1582-8; PMID:8613364 [PMC free article] [PubMed] [Google Scholar][49] Rio-Alvarez I, Rodriguez-Herva JJ, Martinez PM, Gonzalez-Melendi P, Garcia-Casado G, Rodriguez-Palenzuela P, Lopez-Solanilla E.\n",
      "Light regulates motility, attachment and virulence in the plant pathogen Pseudomonas syringae pv tomato DC3000. Environ Microbiol\n",
      "2014; 16:2072-85; PMID:24033935; http://dx.doi.org/ 10.1111/1462-2920.12240 [PubMed] [CrossRef] [Google Scholar][50] Komerik N, Wilson M, Poole S.\n",
      "The effect of photodynamic action on two virulence factors of gram-negative bacteria. Photochem Photobiol\n",
      "2000; 72:676-80; PMID:11107854; http://dx.doi.org/ 10.1562/0031-8655(2000)072%3c0676:TEOPAO%3e2.0.CO;2 [PubMed] [CrossRef] [Google Scholar][51] Sharma M, Bansal H, Gupta PK.\n",
      "Virulence of Pseudomonas aeruginosa cells surviving photodynamic treatment with toluidine blue. Curr Microbiol\n",
      "2005; 50:277-80; PMID:15886909; http://dx.doi.org/ 10.1007/s00284-005-4473-1 [PubMed] [CrossRef] [Google Scholar][52] Tubby S, Wilson M, Nair SP.\n",
      "Inactivation of staphylococcal virulence factors using a light-activated antimicrobial agent. BMC Microbiol\n",
      "2009; 9:211; PMID:19804627; http://dx.doi.org/ 10.1186/1471-2180-9-211 [PMC free article] [PubMed] [CrossRef] [Google Scholar][53] Bartolomeu M, Rocha S, Cunha A, Neves MG, Faustino MA, Almeida A.\n",
      "Effect of Photodynamic Therapy on the Virulence Factors of Staphylococcus aureus. Front Microbiol\n",
      "2016; 7:267; PMID:27014198; http://dx.doi.org/ 10.3389/fmicb.2016.00267 [PMC free article] [PubMed] [CrossRef] [Google Scholar][54] Kato IT, Prates RA, Sabino CP, Fuchs BB, Tegos GP, Mylonakis E, Hamblin MR, Ribeiro MS.\n",
      "Antimicrobial photodynamic inactivation inhibits Candida albicans virulence factors and reduces in vivo pathogenicity. Antimicrob Agents Chemother\n",
      "2013; 57:445-51; PMID:23129051; http://dx.doi.org/ 10.1128/AAC.01451-12 [PMC free article] [PubMed] [CrossRef] [Google Scholar][55] Kadurugamuwa JL, Beveridge TJ.\n",
      "Virulence factors are released from Pseudomonas aeruginosa in association with membrane vesicles during normal growth and exposure to gentamicin: a novel mechanism of enzyme secretion. J Bacteriol\n",
      "1995; 177:3998-4008; PMID:7608073 [PMC free article] [PubMed] [Google Scholar][56] Kadurugamuwa JL, Beveridge TJ.\n",
      "Natural release of virulence factors in membrane vesicles by Pseudomonas aeruginosa and the effect of aminoglycoside antibiotics on their release. J Antimicrob Chemother\n",
      "1997; 40:615-21; PMID:9421308; http://dx.doi.org/ 10.1093/jac/40.5.615 [PubMed] [CrossRef] [Google Scholar][57] Crosby HA, Bion JF, Penn CW, Elliott TS.\n",
      "Antibiotic-induced release of endotoxin from bacteria in vitro. J Med Microbiol\n",
      "1994; 40:23-30; PMID:8289210; http://dx.doi.org/ 10.1099/00222615-40-1-23 [PubMed] [CrossRef] [Google Scholar][58] Prins JM, van Deventer SJ, Kuijper EJ, Speelman P.\n",
      "Clinical relevance of antibiotic-induced endotoxin release. Antimicrob Agents Chemother\n",
      "1994; 38:1211-8; PMID:8092816; http://dx.doi.org/ 10.1128/AAC.38.6.1211 [PMC free article] [PubMed] [CrossRef] [Google Scholar][59] Cassidy CM, Donnelly RF, Elborn JS, Magee ND, Tunney MM.\n",
      "Photodynamic Antimicrobial Chemotherapy (PACT) in combination with antibiotics for treatment of Burkholderia cepacia complex infection. J Photochem Photobiol B\n",
      "2012; 106:95-100; PMID:22079165; http://dx.doi.org/ 10.1016/j.jphotobiol.2011.10.010 [PubMed] [CrossRef] [Google Scholar][60] Dastgheyb SS, Eckmann DM, Composto RJ, Hickok NJ.\n",
      "Photo-activated porphyrin in combination with antibiotics: therapies against Staphylococci. J Photochem Photobiol B\n",
      "2013; 129:27-35; PMID:24148969; http://dx.doi.org/ 10.1016/j.jphotobiol.2013.09.006 [PMC free article] [PubMed] [CrossRef] [Google Scholar][61] Ronqui MR, de Aguiar Coletti TM, de Freitas LM, Miranda ET, Fontana CR.\n",
      "Synergistic antimicrobial e\n",
      "len(chunk):  nergistic antimicrobial effect of photodynamic therapy and ciprofloxacin. J Photochem Photobiol B\n",
      "2016; 158:122-9; PMID:26971277; http://dx.doi.org/ 10.1016/j.jphotobiol.2016.02.036 [PubMed] [CrossRef] [Google Scholar][62] Chibebe JJ, Fuchs BB, Sabino CP, Junqueira JC, Jorge AO, Ribeiro MS, Gilmore MS, Rice LB, Tegos GP, Hamblin MR, et al..\n",
      "Photodynamic and antibiotic therapy impair the pathogenesis of Enterococcus faecium in a whole animal insect model. PLoS One\n",
      "2013; 8:e55926; PMID:23457486; http://dx.doi.org/ 10.1371/journal.pone.0055926 [PMC free article] [PubMed] [CrossRef] [Google Scholar][63] Magiorakos AP, Srinivasan A, Carey RB, Carmeli Y, Falagas ME, Giske CG, Harbarth S, Hindler JF, Kahlmeter G, Olsson-Liljequist B, et al..\n",
      "Multidrug-resistant, extensively drug-resistant and pandrug-resistant bacteria: an international expert proposal for interim standard definitions for acquired resistance. Clin Microbiol Infect\n",
      "2012; 18:268-281; PMID:21793988; http://dx.doi.org/ 10.1111/j.1469-0691.2011.03570.x [PubMed] [CrossRef] [Google Scholar][64] CLSI\n",
      "Performance standards for antimicrobial susceptibility testing\n",
      "CLSI approved standard M100-S22. Wayne, PA: Clinical and Laboratory Standards Institute, 2012 [Google Scholar][65] Jett BD, Hatter KL, Huycke MM, Gilmore MS.\n",
      "Simplified agar plate method for quantifying viable bacteria. Biotechniques\n",
      "1997; 23:648-50; PMID:9343684 [PubMed] [Google Scholar][66] Barry AL, Lasner RA.\n",
      "In-vitro methods for determining minimal lethal concentrations of antimicrobial agents. Am J Clin Pathol\n",
      "1979; 71:88-92; PMID:105628; http://dx.doi.org/ 10.1093/ajcp/71.1.88 [PubMed] [CrossRef] [Google Scholar][67] Kohanski MA, DePristo MA, Collins JJ.\n",
      "Sublethal antibiotic treatment leads to multidrug resistance via radical-induced mutagenesis. Mol Cell\n",
      "2010; 37:311-20; PMID:20159551; http://dx.doi.org/ 10.1016/j.molcel.2010.01.003 [PMC free article] [PubMed] [CrossRef] [Google Scholar][68] Latimer J, Forbes S, McBain AJ.\n",
      "Attenuated virulence and biofilm formation in Staphylococcus aureus following sublethal exposure to triclosan. Antimicrob Agents Chemother\n",
      "2012; 56:3092-100; PMID:22430975; http://dx.doi.org/ 10.1128/AAC.05904-11 [PMC free article] [PubMed] [CrossRef] [Google Scholar][69] Andersson DI, Hughes D.\n",
      "Microbiological effects of sublethal levels of antibiotics. Nat Rev Microbiol\n",
      "2014; 12:465-78; PMID:24861036; http://dx.doi.org/ 10.1038/nrmicro3270 [PubMed] [CrossRef] [Google Scholar][70] El-Mowafy SA, bd El Galil KH, El-Messery SM, Shaaban MI.\n",
      "Aspirin is an efficient inhibitor of quorum sensing, virulence and toxins in Pseudomonas aeruginosa. Microb Pathog\n",
      "2014; 74:25-32; PMID:25088031; http://dx.doi.org/ 10.1016/j.micpath.2014.07.008 [PubMed] [CrossRef] [Google Scholar][71] Alatraktchi FA, Andersen SB, Johansen HK, Molin S, Svendsen WE.\n",
      "Fast Selective Detection of Pyocyanin Using Cyclic Voltammetry. Sensors (Basel)\n",
      "2016; 16; PMID:27007376; http://dx.doi.org/ 10.3390/s16030408 [PMC free article] [PubMed] [CrossRef] [Google Scholar][72] Kessler E, Safrin M.\n",
      "Elastinolytic and proteolytic enzymes. Filloux A, and Ramos J. L.\n",
      "135-69. 2014.\n",
      "Humana Press. Pseudomonas. Methods and Protocols [Google Scholar][73] Ohman DE, Cryz SJ, Iglewski BH.\n",
      "Isolation and characterization of Pseudomonas aeruginosa PAO mutant that produces altered elastase. J Bacteriol\n",
      "1980; 142:836-42; PMID:6769912 [PMC free article] [PubMed] [Google Scholar][74] Kessler E, Safrin M, Abrams WR, Rosenbloom J, Ohman DE.\n",
      "Inhibitors and specificity of Pseudomonas aeruginosa LasA. J Biol Chem\n",
      "1997; 272:9884-9; PMID:9092525; http://dx.doi.org/ 10.1074/jbc.272.15.9884 [PubMed] [CrossRef] [Google Scholar][75] Hamood AN, Griswold J, Colmer J.\n",
      "Characterization of elastase-deficient clinical isolates of Pseudomonas aeruginosa. Infect Immun\n",
      "1996; 64:3154-60; PMID:8757847 [PMC free article] [PubMed] [Google Scholar][76] Kashid SG, Ghosh JS.\n",
      "Production, Isolation and Characterization of Exotoxin Produced by Bacillus cereus NCIM-2156 and Bacillus lich\n",
      "len(chunk):  IM-2156 and Bacillus licheniformis NCIM-5343. Br J Pharmacol Toxicol\n",
      "2010; 1:50-5 [Google Scholar][77] Ha D, Kuchma SL, O'Toole GA.\n",
      "Plate-Based assay for swimming motility in Pseudomonas aeruginosa. Filloux A, and Ramos J. L.\n",
      "59-65. 2014.\n",
      "Humana Press. Pseudomonas. Methods and Protocols [PMC free article] [PubMed] [Google Scholar][78] Ha D, Kuchma SL, O'Toole GA.\n",
      "Plate-Based assay for swarming motility in Pseudomonas aeruginosa\n",
      "Filloux A, and Ramos J. L.\n",
      "67-72. 2014.\n",
      "Humana Press. Pseudomonas. Methods and Protocols [Google Scholar][79] Turnbull L, Whitchurch CB.\n",
      "Motility assay: twitching motility. Filloux A, and Ramos J. L.\n",
      "73-86. 2014.\n",
      "Humana Press. Pseudomonas. Methods and Protocols [PubMed] [Google Scholar][80] Thornsberry C.\n",
      "Antimicrobial susceptibility testing of anaerobic bacteria: review, comments, and opinions. Ann Otol Rhinol Laryngol Suppl\n",
      "1991; 154:7-10; PMID:1952686 [PubMed] [Google Scholar][81] Cumming G, Fidler F, Vaux DL.\n",
      "Error bars in experimental biology. J Cell Biol\n",
      "2007; 177:7-11; PMID:17420288; http://dx.doi.org/ 10.1083/jcb.200611141 [PMC free article] [PubMed] [CrossRef] [Google Scholar][82] Bellin DL, Sakhtah H, Rosenstein JK, Levine PM, Thimot J, Emmett K, Dietrich LE, Shepard KL.\n",
      "Integrated circuit-based electrochemical sensor for spatially resolved detection of redox-active metabolites in biofilms. Nat Commun\n",
      "2014; 5:3256; PMID:24510163; http://dx.doi.org/ 10.1038/ncomms4256 [PMC free article] [PubMed] [CrossRef] [Google Scholar][83] Bucior I, Pielage JF, Engel JN.\n",
      "Pseudomonas aeruginosa pili and flagella mediate distinct binding and signaling events at the apical and basolateral surface of airway epithelium. PLoS Pathog\n",
      "2012; 8:e1002616; PMID:22496644; http://dx.doi.org/ 10.1371/journal.ppat.1002616 [PMC free article] [PubMed] [CrossRef] [Google Scholar][84] Ramphal R, Balloy V, Jyot J, Verma A, Si-Tahar M, Chignard M.\n",
      "Control of Pseudomonas aeruginosa in the lung requires the recognition of either lipopolysaccharide or flagellin. J Immunol\n",
      "2008; 181:586-92; PMID:18566425; http://dx.doi.org/ 10.4049/jimmunol.181.1.586 [PMC free article] [PubMed] [CrossRef] [Google Scholar]ABSTRACTIntroductionResultsDiscussionMaterials and methodsSupplementary MaterialDisclosure of potential conflicts of interestAcknowledgmentsFundingAuthor contributionsReferenceArticles from Virulence are provided here courtesy of Taylor & Francis[1] Balasubramanian D, Schneper L, Kumari H, Mathee K.\n",
      "A dynamic and intricate regulatory network determines Pseudomonas aeruginosa virulence. Nucleic Acids Res\n",
      "2013; 41:1-20; PMID:23143271; http://dx.doi.org/ 10.1093/nar/gks1039 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][2] Rybtke M, Hultqvist LD, Givskov M, Tolker-Nielsen T.\n",
      "Pseudomonas aeruginosa Biofilm Infections: Community Structure, Antimicrobial Tolerance and Immune Response. J Mol Biol\n",
      "2015; 427:3628-45; PMID:26319792; http://dx.doi.org/ 10.1016/j.jmb.2015.08.016 [PubMed] [CrossRef] [Google Scholar] [Ref list][6] Potron A, Poirel L, Nordmann P.\n",
      "Emerging broad-spectrum resistance in Pseudomonas aeruginosa and Acinetobacter baumannii: Mechanisms and epidemiology. Int J Antimicrob Agents\n",
      "2015; 45:568-85; PMID:25857949; http://dx.doi.org/ 10.1016/j.ijantimicag.2015.03.001 [PubMed] [CrossRef] [Google Scholar] [Ref list][7] Clatworthy AE, Pierson E, Hung DT.\n",
      "Targeting virulence: a new paradigm for antimicrobial therapy. Nat Chem Biol\n",
      "2007; 3:541-8; PMID:17710100; http://dx.doi.org/ 10.1038/nchembio.2007.24 [PubMed] [CrossRef] [Google Scholar] [Ref list][8] Merrell DS, Falkow S.\n",
      "Frontal and stealth attack strategies in microbial pathogenesis. Nature\n",
      "2004; 430:250-6; PMID:15241423; http://dx.doi.org/ 10.1038/nature02760 [PubMed] [CrossRef] [Google Scholar] [Ref list][9] Sommer R, Joachim I, Wagner S, Titz A.\n",
      "New approaches to control infections: anti-biofilm strategies against gram-negative bacteria. Chimia (Aarau)\n",
      "2013; 67:286-90; PMID:23967708; http://dx.doi.org/ 10.2533/chimia.2013.286 [PubMed] [CrossRef] [Google \n",
      "len(chunk):  bMed] [CrossRef] [Google Scholar] [Ref list][10] Rasmussen TB, Givskov M.\n",
      "Quorum-sensing inhibitors as anti-pathogenic drugs. Int J Med Microbiol\n",
      "2006; 296:149-61; PMID:16503194; http://dx.doi.org/ 10.1016/j.ijmm.2006.02.005 [PubMed] [CrossRef] [Google Scholar] [Ref list][11] Wagner S, Sommer R, Hinsberger S, Lu C, Hartmann RW, Empting M, Titz A.\n",
      "Novel Strategies for the Treatment of Pseudomonas aeruginosa Infections. J Med Chem\n",
      "2016; 59:5929-69; PMID:26804741; http://dx.doi.org/ 10.1021/acs.jmedchem.5b01698 [PubMed] [CrossRef] [Google Scholar] [Ref list][12] Kuang Z, Hao Y, Walling BE, Jeffries JL, Ohman DE, Lau GW.\n",
      "Pseudomonas aeruginosa elastase provides an escape from phagocytosis by degrading the pulmonary surfactant protein-A. PLoS One\n",
      "2011; 6:e27091; PMID:22069491; http://dx.doi.org/ 10.1371/journal.pone.0027091 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][13] Caballero AR, Moreau JM, Engel LS, Marquart ME, Hill JM, O'Callaghan RJ.\n",
      "Pseudomonas aeruginosa protease IV enzyme assays and comparison to other Pseudomonas proteases. Anal Biochem\n",
      "2001; 290:330-7; PMID:11237336; http://dx.doi.org/ 10.1006/abio.2001.4999 [PubMed] [CrossRef] [Google Scholar] [Ref list][14] Berka RM, Vasil ML.\n",
      "Phospholipase C (heat-labile hemolysin) of Pseudomonas aeruginosa: purification and preliminary characterization. J Bacteriol\n",
      "1982; 152:239-45; PMID:6811552 [PMC free article] [PubMed] [Google Scholar] [Ref list][16] Bush K, Courvalin P, Dantas G, Davies J, Eisenstein B, Huovinen P, Jacoby GA, Kishony R, Kreiswirth BN, Kutter E, et al..\n",
      "Tackling antibiotic resistance. Nat Rev Microbiol\n",
      "2011; 9:894-6; PMID:22048738; http://dx.doi.org/ 10.1038/nrmicro2693 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][17] Dai T, Tegos GP, Zhiyentayev T, Mylonakis E, Hamblin MR.\n",
      "Photodynamic therapy for methicillin-resistant Staphylococcus aureus infection in a mouse skin abrasion model. Lasers Surg Med\n",
      "2010; 42:38-44; PMID:20077489; http://dx.doi.org/ 10.1002/lsm.20887 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][21] Waldman G.\n",
      "Introduction to light : the physics of light, vision, and color. 193. Mineola: Dover Publications, 2002 [Google Scholar] [Ref list][22] Grinholc M, Rapacka-Zdonczyk A, Rybak B, Szabados F, Bielawski KP.\n",
      "Multiresistant Strains Are as Susceptible to Photodynamic Inactivation as Their Naive Counterparts: Protoporphyrin IX-Mediated Photoinactivation Reveals Differences Between Methicillin-Resistant and Methicillin-Sensitive Staphylococcus aureus Strains. Photomed Laser Surg\n",
      "2014; 32:121-9; PMID:24527879; http://dx.doi.org/ 10.1089/pho.2013.3663 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][23] Dai T, Vrahas MS, Murray CK, Hamblin MR.\n",
      "Ultraviolet C irradiation: an alternative antimicrobial approach to localized infections?\n",
      "Expert Rev Anti Infect Ther\n",
      "2012; 10:185-95; PMID:22339192; http://dx.doi.org/ 10.1586/eri.11.166 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][24] Dai T, Gupta A, Huang YY, Yin R, Murray CK, Vrahas MS, Sherwood ME, Tegos GP, Hamblin MR.\n",
      "Blue light rescues mice from potentially fatal Pseudomonas aeruginosa burn infection: efficacy, safety, and mechanism of action. Antimicrob Agents Chemother\n",
      "2013; 57:1238-45; PMID:23262998; http://dx.doi.org/ 10.1128/AAC.01652-12 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][26] Wheeland RG, Dhawan S.\n",
      "Evaluation of self-treatment of mild-to-moderate facial acne with a blue light treatment system. J Drugs Dermatol\n",
      "2011; 10:596-602; PMID:21637900 [PubMed] [Google Scholar] [Ref list][28] Dai T, Gupta A, Murray CK, Vrahas MS, Tegos GP, Hamblin MR.\n",
      "Blue light for infectious diseases: Propionibacterium acnes, Helicobacter pylori, and beyond?\n",
      "Drug Resist Updat\n",
      "2012; 15:223-36; PMID:22846406; http://dx.doi.org/ 10.1016/j.drup.2012.07.001 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][29] Grinholc M, Rodziewicz A, Forys K, Rapacka-Zdonczyk A, Kawiak A, Do\n",
      "len(chunk):  -Zdonczyk A, Kawiak A, Domachowska A, Golunski G, Wolz C, Mesak L, Becker K, et al..\n",
      "Fine-tuning recA expression in Staphylococcus aureus for antimicrobial photoinactivation: importance of photo-induced DNA damage in the photoinactivation mechanism. Appl Microbiol Biotechnol\n",
      "2015; 99:9161-76; PMID:26252968; http://dx.doi.org/ 10.1007/s00253-015-6863-z [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][32] Cieplik F, Spath A, Leibl C, Gollmer A, Regensburger J, Tabenski L, Hiller KA, Maisch T, Schmalz G.\n",
      "Blue light kills Aggregatibacter actinomycetemcomitans due to its endogenous photosensitizers. Clin Oral Investig\n",
      "2014; 18:1763-9; PMID:24297656; http://dx.doi.org/ 10.1007/s00784-013-1151-8 [PubMed] [CrossRef] [Google Scholar] [Ref list][34] Maclean M, McKenzie K, Anderson JG, Gettinby G, MacGregor SJ.\n",
      "405 nm light technology for the inactivation of pathogens and its potential role for environmental disinfection and infection control. J Hosp Infect\n",
      "2014; 88:1-11; PMID:25066049; http://dx.doi.org/ 10.1016/j.jhin.2014.06.004 [PubMed] [CrossRef] [Google Scholar] [Ref list][35] Kleinpenning MM, Smits T, Frunt MH, van Erp PE, van de Kerkhof PC, Gerritsen RM.\n",
      "Clinical and histological effects of blue light on normal skin. Photodermatol Photoimmunol Photomed\n",
      "2010; 26:16-21; PMID:20070834; http://dx.doi.org/ 10.1111/j.1600-0781.2009.00474.x [PubMed] [CrossRef] [Google Scholar] [Ref list][36] Amin RM, Bhayana B, Hamblin MR, Dai T.\n",
      "Antimicrobial blue light inactivation of Pseudomonas aeruginosa by photo-excitation of endogenous porphyrins: In vitro and in vivo studies. Lasers Surg Med\n",
      "2016; PMID:26891084 [PMC free article] [PubMed] [Google Scholar] [Ref list][27] Halstead FD, Thwaite JE, Burt R, Laws TR, Raguse M, Moeller R, Webber MA, Oppenheim BA.\n",
      "The antibacterial activity of blue light against nosocomial wound pathogens growing planktonically and as mature biofilms. Appl Environ Microbiol\n",
      "2016; PMID:27129967 [PMC free article] [PubMed] [Google Scholar] [Ref list][37] Dodd CER, Sharman RL, Bloomfield SF, Booth IR, Stewart GSAB.\n",
      "Inimical processes: Bacterial self-destruction and sub-lethal injury. Trends Food Sci Technol\n",
      "1997; 8:238-41; http://dx.doi.org/ 10.1016/S0924-2244(97)01043-1 [CrossRef] [Google Scholar] [Ref list][81] Cumming G, Fidler F, Vaux DL.\n",
      "Error bars in experimental biology. J Cell Biol\n",
      "2007; 177:7-11; PMID:17420288; http://dx.doi.org/ 10.1083/jcb.200611141 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][38] ASM\n",
      "Antimicrobial Agents and Chemotherapy, Instructions to Authors. http://aac.asm.org/site/misc/2016AprilAACITA.pdf. 2016 [Google Scholar] [Ref list][39] Lau GW, Ran H, Kong F, Hassett DJ, Mavrodi D.\n",
      "Pseudomonas aeruginosa pyocyanin is critical for lung infection in mice. Infect Immun\n",
      "2004; 72:4275-8; PMID:15213173; http://dx.doi.org/ 10.1128/IAI.72.7.4275-4278.2004 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][40] Reszka KJ, Bilski PJ, Britigan BE.\n",
      "Quenching of singlet oxygen by pyocyanin and related phenazines. Photochem Photobiol\n",
      "2010; 86:742-6; PMID:20408986; http://dx.doi.org/ 10.1111/j.1751-1097.2010.00728.x [PubMed] [CrossRef] [Google Scholar] [Ref list][41] Hawdon NA, Aval PS, Barnes RJ, Gravelle SK, Rosengren J, Khan S, Ciofu O, Johansen HK, Hoiby N, Ulanova M.\n",
      "Cellular responses of A549 alveolar epithelial cells to serially collected Pseudomonas aeruginosa from cystic fibrosis patients at different stages of pulmonary infection. FEMS Immunol Med Microbiol\n",
      "2010; 59:207-20; PMID:20528926; http://dx.doi.org/ 10.1111/j.1574-695X.2010.00693.x [PubMed] [CrossRef] [Google Scholar] [Ref list][42] Griese M, Kappler M, Eismann C, Ballmann M, Junge S, Rietschel E, van Koningsbruggen-Rietschel S, Staab D, Rolinck-Werninghaus C, Mellies U, et al..\n",
      "Inhalation treatment with glutathione in patients with cystic fibrosis. A randomized clinical trial. Am J Respir Crit Care Med\n",
      "2013; 188:83-9; PMID:23631796; http://dx.doi.org/ 10.1164/rccm.201303-0427OC [PubMed] [Cro\n",
      "len(chunk):  1303-0427OC [PubMed] [CrossRef] [Google Scholar] [Ref list][43] Kipnis E, Sawa T, Wiener-Kronish J.\n",
      "Targeting mechanisms of Pseudomonas aeruginosa pathogenesis. Med Mal Infect\n",
      "2006; 36:78-91; PMID:16427231; http://dx.doi.org/ 10.1016/j.medmal.2005.10.007 [PubMed] [CrossRef] [Google Scholar] [Ref list][45] Das T, Kutty SK, Kumar N, Manefield M.\n",
      "Pyocyanin facilitates extracellular DNA binding to Pseudomonas aeruginosa influencing cell surface properties and aggregation. PLoS One\n",
      "2013; 8:e58299; PMID:23505483; http://dx.doi.org/ 10.1371/journal.pone.0058299 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][47] Comolli JC, Waite LL, Mostov KE, Engel JN.\n",
      "Pili binding to asialo-GM1 on epithelial cells can mediate cytotoxicity or bacterial internalization by Pseudomonas aeruginosa. Infect Immun\n",
      "1999; 67:3207-14; PMID:10377092 [PMC free article] [PubMed] [Google Scholar] [Ref list][49] Rio-Alvarez I, Rodriguez-Herva JJ, Martinez PM, Gonzalez-Melendi P, Garcia-Casado G, Rodriguez-Palenzuela P, Lopez-Solanilla E.\n",
      "Light regulates motility, attachment and virulence in the plant pathogen Pseudomonas syringae pv tomato DC3000. Environ Microbiol\n",
      "2014; 16:2072-85; PMID:24033935; http://dx.doi.org/ 10.1111/1462-2920.12240 [PubMed] [CrossRef] [Google Scholar] [Ref list][50] Komerik N, Wilson M, Poole S.\n",
      "The effect of photodynamic action on two virulence factors of gram-negative bacteria. Photochem Photobiol\n",
      "2000; 72:676-80; PMID:11107854; http://dx.doi.org/ 10.1562/0031-8655(2000)072%3c0676:TEOPAO%3e2.0.CO;2 [PubMed] [CrossRef] [Google Scholar] [Ref list][52] Tubby S, Wilson M, Nair SP.\n",
      "Inactivation of staphylococcal virulence factors using a light-activated antimicrobial agent. BMC Microbiol\n",
      "2009; 9:211; PMID:19804627; http://dx.doi.org/ 10.1186/1471-2180-9-211 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][53] Bartolomeu M, Rocha S, Cunha A, Neves MG, Faustino MA, Almeida A.\n",
      "Effect of Photodynamic Therapy on the Virulence Factors of Staphylococcus aureus. Front Microbiol\n",
      "2016; 7:267; PMID:27014198; http://dx.doi.org/ 10.3389/fmicb.2016.00267 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][54] Kato IT, Prates RA, Sabino CP, Fuchs BB, Tegos GP, Mylonakis E, Hamblin MR, Ribeiro MS.\n",
      "Antimicrobial photodynamic inactivation inhibits Candida albicans virulence factors and reduces in vivo pathogenicity. Antimicrob Agents Chemother\n",
      "2013; 57:445-51; PMID:23129051; http://dx.doi.org/ 10.1128/AAC.01451-12 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][55] Kadurugamuwa JL, Beveridge TJ.\n",
      "Virulence factors are released from Pseudomonas aeruginosa in association with membrane vesicles during normal growth and exposure to gentamicin: a novel mechanism of enzyme secretion. J Bacteriol\n",
      "1995; 177:3998-4008; PMID:7608073 [PMC free article] [PubMed] [Google Scholar] [Ref list][57] Crosby HA, Bion JF, Penn CW, Elliott TS.\n",
      "Antibiotic-induced release of endotoxin from bacteria in vitro. J Med Microbiol\n",
      "1994; 40:23-30; PMID:8289210; http://dx.doi.org/ 10.1099/00222615-40-1-23 [PubMed] [CrossRef] [Google Scholar] [Ref list][59] Cassidy CM, Donnelly RF, Elborn JS, Magee ND, Tunney MM.\n",
      "Photodynamic Antimicrobial Chemotherapy (PACT) in combination with antibiotics for treatment of Burkholderia cepacia complex infection. J Photochem Photobiol B\n",
      "2012; 106:95-100; PMID:22079165; http://dx.doi.org/ 10.1016/j.jphotobiol.2011.10.010 [PubMed] [CrossRef] [Google Scholar] [Ref list][60] Dastgheyb SS, Eckmann DM, Composto RJ, Hickok NJ.\n",
      "Photo-activated porphyrin in combination with antibiotics: therapies against Staphylococci. J Photochem Photobiol B\n",
      "2013; 129:27-35; PMID:24148969; http://dx.doi.org/ 10.1016/j.jphotobiol.2013.09.006 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][61] Ronqui MR, de Aguiar Coletti TM, de Freitas LM, Miranda ET, Fontana CR.\n",
      "Synergistic antimicrobial effect of photodynamic therapy and ciprofloxacin. J Photochem Photobiol B\n",
      "2016; 158:122-9; PMID:26971277; http\n",
      "len(chunk):  22-9; PMID:26971277; http://dx.doi.org/ 10.1016/j.jphotobiol.2016.02.036 [PubMed] [CrossRef] [Google Scholar] [Ref list][62] Chibebe JJ, Fuchs BB, Sabino CP, Junqueira JC, Jorge AO, Ribeiro MS, Gilmore MS, Rice LB, Tegos GP, Hamblin MR, et al..\n",
      "Photodynamic and antibiotic therapy impair the pathogenesis of Enterococcus faecium in a whole animal insect model. PLoS One\n",
      "2013; 8:e55926; PMID:23457486; http://dx.doi.org/ 10.1371/journal.pone.0055926 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][63] Magiorakos AP, Srinivasan A, Carey RB, Carmeli Y, Falagas ME, Giske CG, Harbarth S, Hindler JF, Kahlmeter G, Olsson-Liljequist B, et al..\n",
      "Multidrug-resistant, extensively drug-resistant and pandrug-resistant bacteria: an international expert proposal for interim standard definitions for acquired resistance. Clin Microbiol Infect\n",
      "2012; 18:268-281; PMID:21793988; http://dx.doi.org/ 10.1111/j.1469-0691.2011.03570.x [PubMed] [CrossRef] [Google Scholar] [Ref list][82] Bellin DL, Sakhtah H, Rosenstein JK, Levine PM, Thimot J, Emmett K, Dietrich LE, Shepard KL.\n",
      "Integrated circuit-based electrochemical sensor for spatially resolved detection of redox-active metabolites in biofilms. Nat Commun\n",
      "2014; 5:3256; PMID:24510163; http://dx.doi.org/ 10.1038/ncomms4256 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][83] Bucior I, Pielage JF, Engel JN.\n",
      "Pseudomonas aeruginosa pili and flagella mediate distinct binding and signaling events at the apical and basolateral surface of airway epithelium. PLoS Pathog\n",
      "2012; 8:e1002616; PMID:22496644; http://dx.doi.org/ 10.1371/journal.ppat.1002616 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][84] Ramphal R, Balloy V, Jyot J, Verma A, Si-Tahar M, Chignard M.\n",
      "Control of Pseudomonas aeruginosa in the lung requires the recognition of either lipopolysaccharide or flagellin. J Immunol\n",
      "2008; 181:586-92; PMID:18566425; http://dx.doi.org/ 10.4049/jimmunol.181.1.586 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][65] Jett BD, Hatter KL, Huycke MM, Gilmore MS.\n",
      "Simplified agar plate method for quantifying viable bacteria. Biotechniques\n",
      "1997; 23:648-50; PMID:9343684 [PubMed] [Google Scholar] [Ref list][66] Barry AL, Lasner RA.\n",
      "In-vitro methods for determining minimal lethal concentrations of antimicrobial agents. Am J Clin Pathol\n",
      "1979; 71:88-92; PMID:105628; http://dx.doi.org/ 10.1093/ajcp/71.1.88 [PubMed] [CrossRef] [Google Scholar] [Ref list][70] El-Mowafy SA, bd El Galil KH, El-Messery SM, Shaaban MI.\n",
      "Aspirin is an efficient inhibitor of quorum sensing, virulence and toxins in Pseudomonas aeruginosa. Microb Pathog\n",
      "2014; 74:25-32; PMID:25088031; http://dx.doi.org/ 10.1016/j.micpath.2014.07.008 [PubMed] [CrossRef] [Google Scholar] [Ref list][71] Alatraktchi FA, Andersen SB, Johansen HK, Molin S, Svendsen WE.\n",
      "Fast Selective Detection of Pyocyanin Using Cyclic Voltammetry. Sensors (Basel)\n",
      "2016; 16; PMID:27007376; http://dx.doi.org/ 10.3390/s16030408 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][15] Rossignol G, Merieau A, Guerillon J, Veron W, Lesouhaitier O, Feuilloley MG, Orange N.\n",
      "Involvement of a phospholipase C in the hemolytic activity of a clinical strain of Pseudomonas fluorescens. BMC Microbiol\n",
      "2008; 8:189; PMID:18973676; http://dx.doi.org/ 10.1186/1471-2180-8-189 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][72] Kessler E, Safrin M.\n",
      "Elastinolytic and proteolytic enzymes. Filloux A, and Ramos J. L.\n",
      "135-69. 2014.\n",
      "Humana Press. Pseudomonas. Methods and Protocols [Google Scholar] [Ref list][74] Kessler E, Safrin M, Abrams WR, Rosenbloom J, Ohman DE.\n",
      "Inhibitors and specificity of Pseudomonas aeruginosa LasA. J Biol Chem\n",
      "1997; 272:9884-9; PMID:9092525; http://dx.doi.org/ 10.1074/jbc.272.15.9884 [PubMed] [CrossRef] [Google Scholar] [Ref list][75] Hamood AN, Griswold J, Colmer J.\n",
      "Characterization of elastase-deficient clinical isolates of Pseudomonas aeruginosa. Infect Immun\n",
      "1996; 64:3154-60; PMID:8757847 [PM\n",
      "len(chunk):  3154-60; PMID:8757847 [PMC free article] [PubMed] [Google Scholar] [Ref list][76] Kashid SG, Ghosh JS.\n",
      "Production, Isolation and Characterization of Exotoxin Produced by Bacillus cereus NCIM-2156 and Bacillus licheniformis NCIM-5343. Br J Pharmacol Toxicol\n",
      "2010; 1:50-5 [Google Scholar] [Ref list][77] Ha D, Kuchma SL, O'Toole GA.\n",
      "Plate-Based assay for swimming motility in Pseudomonas aeruginosa. Filloux A, and Ramos J. L.\n",
      "59-65. 2014.\n",
      "Humana Press. Pseudomonas. Methods and Protocols [PMC free article] [PubMed] [Google Scholar] [Ref list][78] Ha D, Kuchma SL, O'Toole GA.\n",
      "Plate-Based assay for swarming motility in Pseudomonas aeruginosa\n",
      "Filloux A, and Ramos J. L.\n",
      "67-72. 2014.\n",
      "Humana Press. Pseudomonas. Methods and Protocols [Google Scholar] [Ref list][79] Turnbull L, Whitchurch CB.\n",
      "Motility assay: twitching motility. Filloux A, and Ramos J. L.\n",
      "73-86. 2014.\n",
      "Humana Press. Pseudomonas. Methods and Protocols [PubMed] [Google Scholar] [Ref list][80] Thornsberry C.\n",
      "Antimicrobial susceptibility testing of anaerobic bacteria: review, comments, and opinions. Ann Otol Rhinol Laryngol Suppl\n",
      "1991; 154:7-10; PMID:1952686 [PubMed] [Google Scholar] [Ref list]\n",
      "Other Formats\n",
      "PubReader\n",
      "PDF (1.7M)\n",
      "Actions\n",
      "Cite\n",
      "Collections\n",
      "Add to Collections\n",
      "Create a new collection\n",
      "Add to an existing collection\n",
      "Name your collection:\n",
      "Name must be less than\n",
      "characters\n",
      "Choose a collection:\n",
      "Unable to load your collection due to an error\n",
      "Please try again\n",
      "Add\n",
      "Cancel\n",
      "Share\n",
      "Permalink\n",
      "Copy\n",
      "RESOURCES\n",
      "Similar articles\n",
      "Cited by other articles\n",
      "Links to NCBI Databases\n",
      "[x]\n",
      "Cite\n",
      "Copy\n",
      "Download .nbib\n",
      ".nbib\n",
      "Format:\n",
      "AMA\n",
      "APA\n",
      "MLA\n",
      "NLM\n",
      "External link. Please review our privacy policy.\n",
      "Chunk: 0 / 29\n",
      "seed_initial_question_decomposition_prompt \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract and output as many accurate direct quotes from the text that are relevant to answering the key questions and its most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"1\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Text: Back to Top\n",
      "Skip to main content\n",
      "The .gov means it’s official.\n",
      "Federal government websites often end in .gov or .mil. Before\n",
      "sharing sensitive information, make sure you’re on a federal\n",
      "government site.\n",
      "The site is secure.\n",
      "The https:// ensures that you are connecting to the\n",
      "official website and that any information you provide is encrypted\n",
      "and transmitted securely.\n",
      "Access keys\n",
      "NCBI Homepage\n",
      "MyNCBI Homepage\n",
      "Main Content\n",
      "Main Navigation\n",
      "Journal List\n",
      "Virulence\n",
      "v.8(6); 2017\n",
      "PMC5626244\n",
      "As a library, NLM provides access to scientific literature. Inclusion in an NLM database does not imply endorsement of, or agreement with,\n",
      "the contents by NLM or the National Institutes of Health.\n",
      "Learn more about our disclaimer.\n",
      "Virulence. 2017; 8(6): 938–958. Published online 2016 Oct 20.\n",
      "doi: 10.1080/21505594.2016.1250995PMCID: PMC5626244PMID: 27763824Blue light treatment of Pseudomonas aeruginosa: Strong bactericidal activity, synergism with antibiotics and inactivation of virulence factorsGrzegorz Fila,a Anna Kawiak,b,c and\n",
      "Mariusz Stanislaw GrinholcaGrzegorz FilaaLaboratory of Molecular Diagnostics, Department of Biotechnology, Intercollegiate Faculty of Biotechnology, University of Gdansk and Medical University of Gdansk, Gdansk, PolandFind articles by Grzegorz FilaAnna KawiakbDivision of Plant Protection and Biotechnology, Department of Biotechnology, Intercollegiate Faculty of Biotechnology, University of Gdansk and Medical University of Gdansk, Gdansk, PolandcLaboratory of Human Physiology, Medical University of Gdansk, Gdansk, PolandFind articles by Anna KawiakMariusz Stanislaw GrinholcaLaboratory of Molecular Diagnostics, Department of Biotechnology, Intercollegiate Faculty of Biotechnology, University of Gdansk and Medical University of Gdansk, Gdansk, PolandFind articles by Mariusz Stanislaw GrinholcAuthor information Article notes Copyright and License information PMC DisclaimeraLaboratory of Molecular Diagnostics, Department of Biotechnology, Intercollegiate Faculty of Biotechnology, University of Gdansk and Medical University of Gdansk, Gdansk, PolandbDivision of Plant Protection and Biotechnology, Department of Biotechnology, Intercollegiate Faculty of Biotechnology, University of Gdansk and Medical University of Gdansk, Gdansk, PolandcLaboratory of Human Physiology, Medical University of Gdansk, Gdansk, PolandCONTACT Mariusz Stanislaw Grinholc lp.ude.gu.hcetoib@clohnirg.zsuiram, Department of Biotechnology, Intercollegiate Faculty of Biotechnology, University of Gdansk and Medical University of Gdansk, Abrahama 58, 80-307 Gdansk, PolandColor versions of one or more of the figures in the article can be found online at www.tandfonline.com/kvir.Supplemental data for this article can be accessed on the publisher's website.Received 2016 Jul 6; Revised 2016 Oct 12; Accepted 2016 Oct 15.Copyright © 2017 The Author(s). Published with license by Taylor & FrancisThis is an Open Access article distributed under the terms of the Creative Commons Attribution-Non-Commercial License (http://creativecommons.org/licenses/by-nc/3.0/), which permits unrestricted non-commercial use, distribution, and reproduction in any medium, provided the original work is properly cited. The moral rights of the named author(s) have been asserted.See editorial \"The antimicrobial effect of blue light: What are behind?\" in\n",
      "volume 8 on page 649.Associated DataSupplementary MaterialsKVIR_S_1250995.zipkvir-08-06-1250995-s001.zip (2.2M)GUID: 0B98ED62-FFD7-4C5D-93A1-66EBC9AF0A87Go to:ABSTRACTPseudomonas aeruginosa is among the most common pathogens responsible for both acute and chronic infections of high incidence and severity. Additionally, P. aeruginosa resistance to conventional antimicrobials has increased rapidly over the past decade. Therefore, it is crucial to explore new therapeutic options, particularly options that specifically target the pathogenic mechanisms of this microbe. The ability of a pathogenic bacterium to cause disease is dependent upon the production\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Prompt:  \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract and output as many accurate direct quotes from the text that are relevant to answering the key questions and its most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"1\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Text: Back to Top\n",
      "Skip to main content\n",
      "The .gov means it’s official.\n",
      "Federal government websites often end in .gov or .mil. Before\n",
      "sharing sensitive information, make sure you’re on a federal\n",
      "government site.\n",
      "The site is secure.\n",
      "The https:// ensures that you are connecting to the\n",
      "official website and that any information you provide is encrypted\n",
      "and transmitted securely.\n",
      "Access keys\n",
      "NCBI Homepage\n",
      "MyNCBI Homepage\n",
      "Main Content\n",
      "Main Navigation\n",
      "Journal List\n",
      "Virulence\n",
      "v.8(6); 2017\n",
      "PMC5626244\n",
      "As a library, NLM provides access to scientific literature. Inclusion in an NLM database does not imply endorsement of, or agreement with,\n",
      "the contents by NLM or the National Institutes of Health.\n",
      "Learn more about our disclaimer.\n",
      "Virulence. 2017; 8(6): 938–958. Published online 2016 Oct 20.\n",
      "doi: 10.1080/21505594.2016.1250995PMCID: PMC5626244PMID: 27763824Blue light treatment of Pseudomonas aeruginosa: Strong bactericidal activity, synergism with antibiotics and inactivation of virulence factorsGrzegorz Fila,a Anna Kawiak,b,c and\n",
      "Mariusz Stanislaw GrinholcaGrzegorz FilaaLaboratory of Molecular Diagnostics, Department of Biotechnology, Intercollegiate Faculty of Biotechnology, University of Gdansk and Medical University of Gdansk, Gdansk, PolandFind articles by Grzegorz FilaAnna KawiakbDivision of Plant Protection and Biotechnology, Department of Biotechnology, Intercollegiate Faculty of Biotechnology, University of Gdansk and Medical University of Gdansk, Gdansk, PolandcLaboratory of Human Physiology, Medical University of Gdansk, Gdansk, PolandFind articles by Anna KawiakMariusz Stanislaw GrinholcaLaboratory of Molecular Diagnostics, Department of Biotechnology, Intercollegiate Faculty of Biotechnology, University of Gdansk and Medical University of Gdansk, Gdansk, PolandFind articles by Mariusz Stanislaw GrinholcAuthor information Article notes Copyright and License information PMC DisclaimeraLaboratory of Molecular Diagnostics, Department of Biotechnology, Intercollegiate Faculty of Biotechnology, University of Gdansk and Medical University of Gdansk, Gdansk, PolandbDivision of Plant Protection and Biotechnology, Department of Biotechnology, Intercollegiate Faculty of Biotechnology, University of Gdansk and Medical University of Gdansk, Gdansk, PolandcLaboratory of Human Physiology, Medical University of Gdansk, Gdansk, PolandCONTACT Mariusz Stanislaw Grinholc lp.ude.gu.hcetoib@clohnirg.zsuiram, Department of Biotechnology, Intercollegiate Faculty of Biotechnology, University of Gdansk and Medical University of Gdansk, Abrahama 58, 80-307 Gdansk, PolandColor versions of one or more of the figures in the article can be found online at www.tandfonline.com/kvir.Supplemental data for this article can be accessed on the publisher's website.Received 2016 Jul 6; Revised 2016 Oct 12; Accepted 2016 Oct 15.Copyright © 2017 The Author(s). Published with license by Taylor & FrancisThis is an Open Access article distributed under the terms of the Creative Commons Attribution-Non-Commercial License (http://creativecommons.org/licenses/by-nc/3.0/), which permits unrestricted non-commercial use, distribution, and reproduction in any medium, provided the original work is properly cited. The moral rights of the named author(s) have been asserted.See editorial \"The antimicrobial effect of blue light: What are behind?\" in\n",
      "volume 8 on page 649.Associated DataSupplementary MaterialsKVIR_S_1250995.zipkvir-08-06-1250995-s001.zip (2.2M)GUID: 0B98ED62-FFD7-4C5D-93A1-66EBC9AF0A87Go to:ABSTRACTPseudomonas aeruginosa is among the most common pathogens responsible for both acute and chronic infections of high incidence and severity. Additionally, P. aeruginosa resistance to conventional antimicrobials has increased rapidly over the past decade. Therefore, it is crucial to explore new therapeutic options, particularly options that specifically target the pathogenic mechanisms of this microbe. The ability of a pathogenic bacterium to cause disease is dependent upon the production\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"0\\\": {\\n    \\\"quote\\\": \\\"Pseudomonas aeruginosa is among the most common pathogens responsible for both acute and chronic infections of high incidence and severity.\\\",\\n    \\\"index\\\": 1\\n  },\\n  \\\"1\\\": {\\n    \\\"quote\\\": \\\"P. aeruginosa resistance to conventional antimicrobials has increased rapidly over the past decade.\\\",\\n    \\\"index\\\": 1\\n  }\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691117585,\n",
      "  \"id\": \"chatcmpl-7jfE1tVFpbTJjtWiTMo2o4dudtD0g\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 86,\n",
      "    \"prompt_tokens\": 1181,\n",
      "    \"total_tokens\": 1267\n",
      "  }\n",
      "}\n",
      "Extracted quotes:  {\n",
      "  \"0\": {\n",
      "    \"quote\": \"Pseudomonas aeruginosa is among the most common pathogens responsible for both acute and chronic infections of high incidence and severity.\",\n",
      "    \"index\": 1\n",
      "  },\n",
      "  \"1\": {\n",
      "    \"quote\": \"P. aeruginosa resistance to conventional antimicrobials has increased rapidly over the past decade.\",\n",
      "    \"index\": 1\n",
      "  }\n",
      "}\n",
      "Chunk: 1 / 29\n",
      "seed_initial_question_decomposition_prompt \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract and output as many accurate direct quotes from the text that are relevant to answering the key questions and its most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"1\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Text: ndent upon the production of agents termed ‘virulence factors’, and approaches to mitigate these agents have gained increasing attention as new antibacterial strategies. Although blue light irradiation is a promising alternative approach, only limited and preliminary studies have described its effect on virulence factors. The current study aimed to investigate the effects of lethal and sub-lethal doses of blue light treatment (BLT) on P. aeruginosa virulence factors. We analyzed the inhibitory effects of blue light irradiation on the production/activity of several virulence factors. Lethal BLT inhibited the activity of pyocyanin, staphylolysin, pseudolysin and other proteases, but sub-lethal BLT did not affect the production/expression of proteases, phospholipases, and flagella- or type IV pili-associated motility. Moreover, a eukaryotic cytotoxicity test confirmed the decreased toxicity of blue light-treated extracellular P. aeruginosa fractions. Finally, the increased antimicrobial susceptibility of P. aeruginosa treated with sequential doses of sub-lethal BLT was demonstrated with a checkerboard test. Thus, this work provides evidence-based proof of the susceptibility of drug-resistant P. aeruginosa to BLT-mediated killing, accompanied by virulence factor reduction, and describes the synergy between antibiotics and sub-lethal BLT.KEYWORDS: blue light treatment, elastases, hemolysins, lipases, motility, photoinactivation, proteases, Pseudomonas aeruginosa, virulence factorsGo to:IntroductionPseudomonas aeruginosa is a Gram-negative bacterium that thrives in most natural and man-made environments. It is found in diverse habitats, including soil, water, plants and animals, and infects multiple hosts.1\n",
      "P. aeruginosa causes a wide variety of acute (i.e., short duration and typically severe) and chronic (i.e., long duration, often refractory to treatment, and of variable severity depending on the pathogen) human infections, including in patients with severe burn wounds, urinary tract infections, acquired immune deficiency syndrome (AIDS), lung cancer, chronic obstructive pulmonary disease, bronchiectasis and cystic fibrosis (CF).2-5 Adding to the problems caused by its high incidence and the severity of infection, the resistance of P. aeruginosa to conventional antimicrobial treatments has increased over the past decade.6 Despite their wide recognition, the high incidence, severity and resistance of P. aeruginosa persist, and various strategies to address these problems have been proposed, such as (i) virulence and pathogenicity factors as therapeutic targets (for a review see ref. Seven); (ii) secretion systems as therapeutic targets (for a review see ref. Eight); (iii) cell adhesion and biofilm formation as therapeutic targets (for a review see ref. Nine); or (iv) quorum sensing as a therapeutic target (for a review see ref. Ten), as well as many others (for a review see ref. Eleven)7-11. Therefore, it is crucial to explore new therapeutic options for P. aeruginosa infections, which may be achieved by specifically targeting its pathogenic mechanisms. The ability of a pathogenic bacterium to cause disease depends on the production of agents termed ‘virulence factors’, such as toxins and adhesion molecules, which actively damage host tissues. Targeting these virulence factors (e.g., inhibiting their production, delivery or function) has gained increasing attention as a potential new antibacterial strategy; in principle this would ‘disarm’ a pathogen and allow the host immune system a better chance of clearing the infection before the pathogen causes too much tissue damage. Its metabolic versatility, intrinsic and acquired antibiotic resistance, biofilm formation and the production of multiple virulence (disease-causing) factors make P. aeruginosa a formidable pathogen. The virulence machinery of P. aeruginosa comprises both cell-associated determinants (such as lipopolysaccharides, pili, and flagella) and numerous secreted factor\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Prompt:  \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract and output as many accurate direct quotes from the text that are relevant to answering the key questions and its most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"1\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Text: ndent upon the production of agents termed ‘virulence factors’, and approaches to mitigate these agents have gained increasing attention as new antibacterial strategies. Although blue light irradiation is a promising alternative approach, only limited and preliminary studies have described its effect on virulence factors. The current study aimed to investigate the effects of lethal and sub-lethal doses of blue light treatment (BLT) on P. aeruginosa virulence factors. We analyzed the inhibitory effects of blue light irradiation on the production/activity of several virulence factors. Lethal BLT inhibited the activity of pyocyanin, staphylolysin, pseudolysin and other proteases, but sub-lethal BLT did not affect the production/expression of proteases, phospholipases, and flagella- or type IV pili-associated motility. Moreover, a eukaryotic cytotoxicity test confirmed the decreased toxicity of blue light-treated extracellular P. aeruginosa fractions. Finally, the increased antimicrobial susceptibility of P. aeruginosa treated with sequential doses of sub-lethal BLT was demonstrated with a checkerboard test. Thus, this work provides evidence-based proof of the susceptibility of drug-resistant P. aeruginosa to BLT-mediated killing, accompanied by virulence factor reduction, and describes the synergy between antibiotics and sub-lethal BLT.KEYWORDS: blue light treatment, elastases, hemolysins, lipases, motility, photoinactivation, proteases, Pseudomonas aeruginosa, virulence factorsGo to:IntroductionPseudomonas aeruginosa is a Gram-negative bacterium that thrives in most natural and man-made environments. It is found in diverse habitats, including soil, water, plants and animals, and infects multiple hosts.1\n",
      "P. aeruginosa causes a wide variety of acute (i.e., short duration and typically severe) and chronic (i.e., long duration, often refractory to treatment, and of variable severity depending on the pathogen) human infections, including in patients with severe burn wounds, urinary tract infections, acquired immune deficiency syndrome (AIDS), lung cancer, chronic obstructive pulmonary disease, bronchiectasis and cystic fibrosis (CF).2-5 Adding to the problems caused by its high incidence and the severity of infection, the resistance of P. aeruginosa to conventional antimicrobial treatments has increased over the past decade.6 Despite their wide recognition, the high incidence, severity and resistance of P. aeruginosa persist, and various strategies to address these problems have been proposed, such as (i) virulence and pathogenicity factors as therapeutic targets (for a review see ref. Seven); (ii) secretion systems as therapeutic targets (for a review see ref. Eight); (iii) cell adhesion and biofilm formation as therapeutic targets (for a review see ref. Nine); or (iv) quorum sensing as a therapeutic target (for a review see ref. Ten), as well as many others (for a review see ref. Eleven)7-11. Therefore, it is crucial to explore new therapeutic options for P. aeruginosa infections, which may be achieved by specifically targeting its pathogenic mechanisms. The ability of a pathogenic bacterium to cause disease depends on the production of agents termed ‘virulence factors’, such as toxins and adhesion molecules, which actively damage host tissues. Targeting these virulence factors (e.g., inhibiting their production, delivery or function) has gained increasing attention as a potential new antibacterial strategy; in principle this would ‘disarm’ a pathogen and allow the host immune system a better chance of clearing the infection before the pathogen causes too much tissue damage. Its metabolic versatility, intrinsic and acquired antibiotic resistance, biofilm formation and the production of multiple virulence (disease-causing) factors make P. aeruginosa a formidable pathogen. The virulence machinery of P. aeruginosa comprises both cell-associated determinants (such as lipopolysaccharides, pili, and flagella) and numerous secreted factor\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"0\\\": {\\n    \\\"quote\\\": \\\"The ability of a pathogenic bacterium to cause disease depends on the production of agents termed \\u2018virulence factors\\u2019, such as toxins and adhesion molecules, which actively damage host tissues.\\\",\\n    \\\"index\\\": 2\\n  },\\n  \\\"1\\\": {\\n    \\\"quote\\\": \\\"Targeting these virulence factors (e.g., inhibiting their production, delivery or function) has gained increasing attention as a potential new antibacterial strategy; in principle this would \\u2018disarm\\u2019 a pathogen and allow the host immune system a better chance of clearing the infection before the pathogen causes too much tissue damage.\\\",\\n    \\\"index\\\": 2\\n  }\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691117588,\n",
      "  \"id\": \"chatcmpl-7jfE4345qAcGI9JN16giRp1ZS2D5E\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 142,\n",
      "    \"prompt_tokens\": 1045,\n",
      "    \"total_tokens\": 1187\n",
      "  }\n",
      "}\n",
      "Extracted quotes:  {\n",
      "  \"0\": {\n",
      "    \"quote\": \"The ability of a pathogenic bacterium to cause disease depends on the production of agents termed ‘virulence factors’, such as toxins and adhesion molecules, which actively damage host tissues.\",\n",
      "    \"index\": 2\n",
      "  },\n",
      "  \"1\": {\n",
      "    \"quote\": \"Targeting these virulence factors (e.g., inhibiting their production, delivery or function) has gained increasing attention as a potential new antibacterial strategy; in principle this would ‘disarm’ a pathogen and allow the host immune system a better chance of clearing the infection before the pathogen causes too much tissue damage.\",\n",
      "    \"index\": 2\n",
      "  }\n",
      "}\n",
      "Chunk: 2 / 29\n",
      "seed_initial_question_decomposition_prompt \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract and output as many accurate direct quotes from the text that are relevant to answering the key questions and its most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"1\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Text:  numerous secreted factors (such as elastases, proteases, exotoxins, pyocyanin (PCN), and extracellular polysaccharides).P. aeruginosa pathogenicity is strongly associated with its ability to move. The following 3 types of movement have been identified: (1) flagella-related swimming and (2) swarming and (3) type IV pili-dependent twitching. Another significant factor contributing to the virulence of P. aeruginosa is PCN (1-hydroxy-5-methylphenazine), a cytotoxic pigment secreted by the bacterium. Moreover, P. aeruginosa virulence is augmented by secreted factors that allow the bacterium to destroy host tissue, modulate the immune system, create biofilms and initiate the colonization process. LasA (staphylolysin) and LasB (pseudolysin) elastases are proteases secreted by P. aeruginosa.12 Protease IV is a Pseudomonas serine protease that has been shown to degrade components of the immune system as well as fibrinogen, plasmin, and plasminogen. In addition, alkaline protease is involved in the extracellular processing of proteases; it has a fairly broad substrate range and is thought to act synergistically with other proteases.13 Finally, P. aeruginosa virulence factors include hemolysins, which lyse red blood cells and release free iron from heme, allowing the bacteria to grow rapidly and leading to greater colonization of the host tissue. P. aeruginosa produces 2 elements with hemolytic properties: heat-resistant glycolipids and heat-labile phospholipase C.14,15The severity and increasing incidence of infections involving multidrug-resistant (MDR) P. aeruginosa have driven research strategies targeting virulence factors that lead to both bacterial eradication and the reduction of bacterial virulence. Major research efforts have been initiated to identify an alternative antimicrobial approach to combat bacteria without easily facilitating the development of resistance. In a recent paper published in Nature Reviews Microbiology, Karen Bush and a group of 30 scientists from academia and industry noted, “The investigation of novel non-antibiotic approaches for the prevention of and protection against infectious diseases should be encouraged, and such approaches must be high-priority research and development projects.”16 The development of light-based antimicrobial therapies, including antimicrobial photodynamic therapy (aPDT) (for a review see refs. Seventeen-18) and ultraviolet C radiation (UVC) (for a review see refs. Seventeen and 20), presents an alternative to antibiotic approaches.17-20 Light is part of the electromagnetic spectrum, which is composed of wavelengths ranging from radio waves to gamma rays. Electromagnetic radiation waves are electric and magnetic field fluctuations that transport energy from one location to another. Visible light (wavelengths ranging from 390 to 700 nm) is not inherently different from the other parts of the electromagnetic spectrum with the exception that the human eye can detect visible light waves. Electromagnetic radiation is also described as a photon stream of massless particles, each traveling at the speed of light and possessing wavelike properties. A photon is the smallest quantum of energy that can be transported.21 Light-based antimicrobial approaches have an extremely important advantage in that they demonstrate equal bactericidal effectiveness for both MDR and wildtype (drug-sensitive) isolates.22 However, the aPDT approach combining a photosensitizer (PS) and light presents several challenges, such as the introducing of a photosensitizer in close proximity to bacterial cells and/or the attachment of a PS to the cell membrane and the development of strategies leading to increased photosensitizer selectivity for microbial cells instead of the host tissue. The use of UVC irradiation also has several limitations, primarily attributable to its harmful effects on eukaryotic cells and the possibility for host tissue damage, including carcinogenesis.23 Currently, a novel light-based approa\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Prompt:  \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract and output as many accurate direct quotes from the text that are relevant to answering the key questions and its most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"1\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Text:  numerous secreted factors (such as elastases, proteases, exotoxins, pyocyanin (PCN), and extracellular polysaccharides).P. aeruginosa pathogenicity is strongly associated with its ability to move. The following 3 types of movement have been identified: (1) flagella-related swimming and (2) swarming and (3) type IV pili-dependent twitching. Another significant factor contributing to the virulence of P. aeruginosa is PCN (1-hydroxy-5-methylphenazine), a cytotoxic pigment secreted by the bacterium. Moreover, P. aeruginosa virulence is augmented by secreted factors that allow the bacterium to destroy host tissue, modulate the immune system, create biofilms and initiate the colonization process. LasA (staphylolysin) and LasB (pseudolysin) elastases are proteases secreted by P. aeruginosa.12 Protease IV is a Pseudomonas serine protease that has been shown to degrade components of the immune system as well as fibrinogen, plasmin, and plasminogen. In addition, alkaline protease is involved in the extracellular processing of proteases; it has a fairly broad substrate range and is thought to act synergistically with other proteases.13 Finally, P. aeruginosa virulence factors include hemolysins, which lyse red blood cells and release free iron from heme, allowing the bacteria to grow rapidly and leading to greater colonization of the host tissue. P. aeruginosa produces 2 elements with hemolytic properties: heat-resistant glycolipids and heat-labile phospholipase C.14,15The severity and increasing incidence of infections involving multidrug-resistant (MDR) P. aeruginosa have driven research strategies targeting virulence factors that lead to both bacterial eradication and the reduction of bacterial virulence. Major research efforts have been initiated to identify an alternative antimicrobial approach to combat bacteria without easily facilitating the development of resistance. In a recent paper published in Nature Reviews Microbiology, Karen Bush and a group of 30 scientists from academia and industry noted, “The investigation of novel non-antibiotic approaches for the prevention of and protection against infectious diseases should be encouraged, and such approaches must be high-priority research and development projects.”16 The development of light-based antimicrobial therapies, including antimicrobial photodynamic therapy (aPDT) (for a review see refs. Seventeen-18) and ultraviolet C radiation (UVC) (for a review see refs. Seventeen and 20), presents an alternative to antibiotic approaches.17-20 Light is part of the electromagnetic spectrum, which is composed of wavelengths ranging from radio waves to gamma rays. Electromagnetic radiation waves are electric and magnetic field fluctuations that transport energy from one location to another. Visible light (wavelengths ranging from 390 to 700 nm) is not inherently different from the other parts of the electromagnetic spectrum with the exception that the human eye can detect visible light waves. Electromagnetic radiation is also described as a photon stream of massless particles, each traveling at the speed of light and possessing wavelike properties. A photon is the smallest quantum of energy that can be transported.21 Light-based antimicrobial approaches have an extremely important advantage in that they demonstrate equal bactericidal effectiveness for both MDR and wildtype (drug-sensitive) isolates.22 However, the aPDT approach combining a photosensitizer (PS) and light presents several challenges, such as the introducing of a photosensitizer in close proximity to bacterial cells and/or the attachment of a PS to the cell membrane and the development of strategies leading to increased photosensitizer selectivity for microbial cells instead of the host tissue. The use of UVC irradiation also has several limitations, primarily attributable to its harmful effects on eukaryotic cells and the possibility for host tissue damage, including carcinogenesis.23 Currently, a novel light-based approa\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"0\\\": {\\n    \\\"quote\\\": \\\"\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"1\\\": {\\n    \\\"quote\\\": \\\"\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"2\\\": {\\n    \\\"quote\\\": \\\"\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"3\\\": {\\n    \\\"quote\\\": \\\"\\\",\\n    \\\"index\\\": 0\\n  }\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691117592,\n",
      "  \"id\": \"chatcmpl-7jfE8zBHeVh9pWVzyXED5HowToXsN\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 78,\n",
      "    \"prompt_tokens\": 1034,\n",
      "    \"total_tokens\": 1112\n",
      "  }\n",
      "}\n",
      "Extracted quotes:  {\n",
      "  \"0\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"1\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"2\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"3\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  }\n",
      "}\n",
      "Max tokens reached in chunks!\n",
      "len(chunk):  Back to Top\n",
      "Skip to main content\n",
      "The .gov means it’s official.\n",
      "Federal government websites often end in .gov or .mil. Before\n",
      "sharing sensitive information, make sure you’re on a federal\n",
      "government site.\n",
      "The site is secure.\n",
      "The https:// ensures that you are connecting to the\n",
      "official website and that any information you provide is encrypted\n",
      "and transmitted securely.\n",
      "Access keys\n",
      "NCBI Homepage\n",
      "MyNCBI Homepage\n",
      "Main Content\n",
      "Main Navigation\n",
      "Journal List\n",
      "ACS Cent Sci\n",
      "v.8(8); 2022 Aug 24\n",
      "PMC9413435\n",
      "As a library, NLM provides access to scientific literature. Inclusion in an NLM database does not imply endorsement of, or agreement with,\n",
      "the contents by NLM or the National Institutes of Health.\n",
      "Learn more about our disclaimer.\n",
      "ACS Cent Sci. 2022 Aug 24; 8(8): 1091–1101. Published online 2022 Apr 25.\n",
      "doi: 10.1021/acscentsci.2c00057PMCID: PMC9413435PMID: 36032767Intersubunit Coupling Enables Fast CO2-Fixation\n",
      "by Reductive CarboxylasesHasan DeMirci,*†‡§▼ Yashas Rao,†⊥▼ Gabriele M. Stoffel,∥▼ Bastian Vögeli,∥▼ Kristina Schell,∥▼ Aharon Gomez,⊥▼ Alexander Batyuk,# Cornelius Gati,†∇ Raymond G. Sierra,# Mark S. Hunter,# E. Han Dao,†‡ Halil I. Ciftci,‡ Brandon Hayes,# Fredric Poitevin,# Po-Nan Li,†○ Manat Kaur,∇ Kensuke Tono,●□ David Adrian Saez,⊥■ Samuel Deutsch,△ Yasuo Yoshikuni,△ Helmut Grubmüller,▲ Tobias J. Erb,*∥▽ Esteban Vöhringer-Martinez,*⊥ and\n",
      "Soichi Wakatsuki*†∇Hasan DeMirci†Biosciences\n",
      "Division, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States‡PULSE\n",
      "Institute, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States§Department\n",
      "of Molecular Biology and Genetics, Koc University, 34450 Sariyer/Istanbul, TurkeyFind articles by Hasan DeMirciYashas Rao†Biosciences\n",
      "Division, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States⊥Departamento\n",
      "de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, ChileFind articles by Yashas RaoGabriele M. Stoffel∥Department\n",
      "of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, GermanyFind articles by Gabriele M. StoffelBastian Vögeli∥Department\n",
      "of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, GermanyFind articles by Bastian VögeliKristina Schell∥Department\n",
      "of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, GermanyFind articles by Kristina SchellAharon Gomez⊥Departamento\n",
      "de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, ChileFind articles by Aharon GomezAlexander Batyuk#Linac Coherent\n",
      "Light Source, SLAC National Accelerator\n",
      "Laboratory,\n",
      "Menlo Park, California 94025, United StatesFind articles by Alexander BatyukCornelius Gati†Biosciences\n",
      "Division, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States∇Structural\n",
      "Biology Department, Stanford University,\n",
      "Stanford, California 94305, United StatesFind articles by Cornelius GatiRaymond G. Sierra#Linac Coherent\n",
      "Light Source, SLAC National Accelerator\n",
      "Laboratory,\n",
      "Menlo Park, California 94025, United StatesFind articles by Raymond G. SierraMark S. Hunter#Linac Coherent\n",
      "Light Source, SLAC National Accelerator\n",
      "Laboratory,\n",
      "Menlo Park, California 94025, United StatesFind articles by Mark S. HunterE. Han Dao†Biosciences\n",
      "Division, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States‡PULSE\n",
      "Institute, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United StatesFind articles by E. Han DaoHalil I. Ciftci‡PULSE\n",
      "Institute, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United StatesFind articles by Halil I. CiftciBrandon Hayes#Linac Coherent\n",
      "Light Source, SLAC National Accelerator\n",
      "Laboratory,\n",
      "Menlo Park, California 94025, United StatesFind articles by Brandon \n",
      "len(chunk):  Find articles by Brandon HayesFredric Poitevin#Linac Coherent\n",
      "Light Source, SLAC National Accelerator\n",
      "Laboratory,\n",
      "Menlo Park, California 94025, United StatesFind articles by Fredric PoitevinPo-Nan Li†Biosciences\n",
      "Division, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States○Electrical\n",
      "Engineering Department, Stanford University,\n",
      "Stanford, California 94305, United StatesFind articles by Po-Nan LiManat Kaur∇Structural\n",
      "Biology Department, Stanford University,\n",
      "Stanford, California 94305, United StatesFind articles by Manat KaurKensuke Tono●RIKEN\n",
      "SPring-8 Center, Sayo, Hyogo 679-5148, Japan□Japan Synchrotron Radiation Research Institute, Sayo, Hyogo 679-5198, JapanFind articles by Kensuke TonoDavid Adrian Saez⊥Departamento\n",
      "de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, Chile■Departamento\n",
      "de Farmacia, Facultad de Farmacia, Universidad\n",
      "de Concepción, Concepción 00000, ChileFind articles by David Adrian SaezSamuel Deutsch△U.S.\n",
      "Department of Energy Joint Genome Institute, Lawrence Berkeley National Laboratory, Walnut Creek, California 94720, United StatesFind articles by Samuel DeutschYasuo Yoshikuni△U.S.\n",
      "Department of Energy Joint Genome Institute, Lawrence Berkeley National Laboratory, Walnut Creek, California 94720, United StatesFind articles by Yasuo YoshikuniHelmut Grubmüller▲Department\n",
      "of Theoretical and Computational Biophysics, Max Planck Institute for Multidisciplinary Sciences, Am Fassberg 11, 37077 Göttingen, GermanyFind articles by Helmut GrubmüllerTobias J. Erb∥Department\n",
      "of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, Germany▽LOEWE Center for Synthetic Microbiology (SYNMIKRO), 35032 Marburg, GermanyFind articles by Tobias J. ErbEsteban Vöhringer-Martinez⊥Departamento\n",
      "de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, ChileFind articles by Esteban Vöhringer-MartinezSoichi Wakatsuki†Biosciences\n",
      "Division, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States∇Structural\n",
      "Biology Department, Stanford University,\n",
      "Stanford, California 94305, United StatesFind articles by Soichi WakatsukiAuthor information Copyright and License information PMC Disclaimer†Biosciences\n",
      "Division, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States‡PULSE\n",
      "Institute, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States§Department\n",
      "of Molecular Biology and Genetics, Koc University, 34450 Sariyer/Istanbul, Turkey∥Department\n",
      "of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, Germany⊥Departamento\n",
      "de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, Chile#Linac Coherent\n",
      "Light Source, SLAC National Accelerator\n",
      "Laboratory,\n",
      "Menlo Park, California 94025, United States∇Structural\n",
      "Biology Department, Stanford University,\n",
      "Stanford, California 94305, United States○Electrical\n",
      "Engineering Department, Stanford University,\n",
      "Stanford, California 94305, United States●RIKEN\n",
      "SPring-8 Center, Sayo, Hyogo 679-5148, Japan□Japan Synchrotron Radiation Research Institute, Sayo, Hyogo 679-5198, Japan■Departamento\n",
      "de Farmacia, Facultad de Farmacia, Universidad\n",
      "de Concepción, Concepción 00000, Chile△U.S.\n",
      "Department of Energy Joint Genome Institute, Lawrence Berkeley National Laboratory, Walnut Creek, California 94720, United States▲Department\n",
      "of Theoretical and Computational Biophysics, Max Planck Institute for Multidisciplinary Sciences, Am Fassberg 11, 37077 Göttingen, Germany▽LOEWE Center for Synthetic Microbiology (SYNMIKRO), 35032 Marburg, GermanyCorresponding author.*Email for H.D.: rt.ude.uk@icrimedh.*Email for T.J.E.: ed.gpm.grubram-ipm@breot.*Email for E.V.-M.: lc.cedu@regnirhove.*Email for S.W.: ude.drofnats@ikustakaw.ihcios.Copyright © 2022 The Authors\n",
      "len(chunk):  yright © 2022 The Authors. Published by American Chemical SocietyPermits non-commercial access and re-use, provided that author attribution and integrity are maintained; but does not permit creation of adaptations or other derivative works (https://creativecommons.org/licenses/by-nc-nd/4.0/).Associated DataSupplementary Materialsoc2c00057_si_001.pdfoc2c00057_si_001.pdf (17M)GUID: 3DAFD832-2754-4387-9475-06B0D4C1EC12oc2c00057_si_002.movoc2c00057_si_002.mov (9.0M)GUID: BE8629BC-6773-425C-97E3-CA37E22A1673oc2c00057_si_003.movoc2c00057_si_003.mov (8.2M)GUID: A7E8C0A4-AE12-4912-AC31-AB45D940046Boc2c00057_si_004.movoc2c00057_si_004.mov (9.0M)GUID: 43D17EE6-B4F7-47EA-B452-11A1274C2A28oc2c00057_si_005.movoc2c00057_si_005.mov (8.1M)GUID: 650EFC78-525F-40C6-84C0-68935E145692oc2c00057_si_006.movoc2c00057_si_006.mov (8.1M)GUID: F8DC6B1B-9299-40EB-821F-D8239E059778oc2c00057_si_007.movoc2c00057_si_007.mov (8.2M)GUID: 9006AB25-3285-4E55-ADD4-67300E3D6A1Coc2c00057_si_008.movoc2c00057_si_008.mov (8.2M)GUID: 633ED857-A1B2-449F-8BC6-4BF29F625C37oc2c00057_si_009.movoc2c00057_si_009.mov (8.2M)GUID: 08FFAF3F-F1C5-4C12-92CE-68628A714916oc2c00057_si_010.movoc2c00057_si_010.mov (6.4M)GUID: C9DB1821-A4B8-4664-AF89-09878ED6B3CEoc2c00057_si_011.movoc2c00057_si_011.mov (8.2M)GUID: 91B081EB-30FD-4129-9F0E-4CB086CEB2EDoc2c00057_si_012.movoc2c00057_si_012.mov (8.7M)GUID: 8FE88134-6B28-4058-A773-085B05F0354Eoc2c00057_si_013.movoc2c00057_si_013.mov (8.7M)GUID: 7BE15129-0BCA-48B0-921C-21552979FFFBoc2c00057_si_014.pdfoc2c00057_si_014.pdf (242K)GUID: 00BA5186-3ABB-48EB-89B5-FCD4099EA78CGo to:AbstractEnoyl-CoA carboxylases/reductases\n",
      "(ECRs) are some of the most efficient\n",
      "CO2-fixing enzymes described to date. However, the molecular\n",
      "mechanisms underlying the extraordinary catalytic activity of ECRs\n",
      "on the level of the protein assembly remain elusive. Here we used\n",
      "a combination of ambient-temperature X-ray free electron laser (XFEL)\n",
      "and cryogenic synchrotron experiments to study the structural organization\n",
      "of the ECR from Kitasatospora setae. The K. setae ECR is a homotetramer\n",
      "that differentiates into a pair of dimers of open- and closed-form\n",
      "subunits in the catalytically active state. Using molecular dynamics\n",
      "simulations and structure-based mutagenesis, we show that catalysis\n",
      "is synchronized in the K. setae ECR\n",
      "across the pair of dimers. This conformational coupling of catalytic\n",
      "domains is conferred by individual amino acids to achieve high CO2-fixation rates. Our results provide unprecedented insights\n",
      "into the dynamic organization and synchronized inter- and intrasubunit\n",
      "communications of this remarkably efficient CO2-fixing\n",
      "enzyme during catalysis.Go to:Short abstractIntersubunit\n",
      "coupling drives fast carboxylation by enoyl-CoA\n",
      "carboxylases/reductases, providing the basis for engineering highly\n",
      "efficient CO2-fixing enzymes for bioenergy and bioproduct\n",
      "applications.Go to:IntroductionThe\n",
      "capture and conversion of atmospheric CO2 remains\n",
      "a challenging task for chemistry, resulting in an ever-increasing\n",
      "interest to understand and exploit CO2 fixation mechanisms\n",
      "offered by biology. Nature has evolved different enzymes capable of\n",
      "capturing and/or converting CO2, including biotin-dependent\n",
      "CoA carboxylases, which incorporate CO2 into different\n",
      "alkyl-CoA esters, formate dehydrogenases, which directly reduce CO2 into the single-carbon compound formate, and RubisCO, the\n",
      "central enzyme in the Calvin–Benson–Bassham cycle of\n",
      "photosynthesis that produces two molecules of 3-phosphoglycerate from\n",
      "ribulose-1,5-bisphosphate and CO2.1−4 The recently discovered family\n",
      "of enoyl-CoA carboxylases/reductases (ECRs) encompasses some of the\n",
      "most efficient CO2-fixing enzymes found in nature to date.1 With respect to the catalytic rate, for instance,\n",
      "ECR family members from primary metabolism outcompete RubisCO by more\n",
      "than 1 order of magnitude, which makes them exquisite models to study\n",
      "the molecular basis of efficient CO2 catalysis.2,5ECRs catalyze the reduction of various α,β-unsaturated\n",
      "e\n",
      "len(chunk):  various α,β-unsaturated\n",
      "enoyl-CoAs into the corresponding alkylmalonyl-CoA esters using the\n",
      "reduced form of the cofactor nicotinamide adenine dinucleotide phosphate\n",
      "(NADPH). Hydride transfer from NADPH to the substrate generates a\n",
      "reactive enolate species, which acts as a nucleophile that attacks\n",
      "a bound CO2 molecule (Figure ​Figure11a).2,5−7 However, the structural details of the carboxylation reaction have\n",
      "remained elusive, in part due to the lack of high-resolution structures\n",
      "of ECRs containing catalytic intermediates and carboxylated products.Open in a separate windowFigure 1Reaction\n",
      "scheme and structural organization of the K. setae ECR complex. (a) Carboxylation reaction\n",
      "scheme of ECR. (b) Anisotropic B-factors of the tetramer of the different\n",
      "ECR complexes solved in this study are shown color coded according\n",
      "to the B factors (blue for low values and red for high values).Until recently, five structures of ECRs have been\n",
      "reported. These\n",
      "ECRs all have different substrate specificities, ranging from short\n",
      "(PDB: 3HZZ, 3KRT)- to long-chain\n",
      "(4A0S8) and aromatic enoyl-CoA substrates (4Y0K9), and are from different biological backgrounds, including\n",
      "primary (i.e., central carbon) metabolism (PDB: 4GI2) and secondary metabolism\n",
      "(PDB: 4A0S, 4Y0K). However, most\n",
      "of them cocrystallized with only NADPH or NADP+ and do\n",
      "not contain CO2, enoyl-CoA substrates, or acyl-CoA products.In a recent study, we reported the ternary structure of the ECR\n",
      "(EC 1.3.1.85) from K. setae in a complex\n",
      "with ethylmalonyl-CoA and NADPH (PDB: 6OWE) and identified four active site residues\n",
      "critical to binding and accommodation of CO2 at the monomer\n",
      "level.10 Yet, the molecular mechanism of\n",
      "the extraordinary rate enhancement of CO2 fixation by ECR\n",
      "tetramers remains enigmatic. This is even more notable, as ECR complexes\n",
      "from primary metabolism, such as the K. setae ECR, show on average almost 30-fold higher CO2 fixation\n",
      "rates in comparison to those ECR complexes from secondary metabolism,\n",
      "despite having very high structural similarities. Overall, this work\n",
      "indicated that ECRs from primary metabolism must use specific mechanisms\n",
      "that allow them to enhance CO2 fixation dramatically in\n",
      "comparison to their structurally closely related homologues from secondary\n",
      "metabolism.Here, we aimed at providing a mechanistic understanding\n",
      "of the\n",
      "fast carboxylation reaction of the K. setae ECR on the level of the overall oligomeric complex. We first determined\n",
      "four high-resolution K. setae ECR structures\n",
      "in different conformational states: the apo form and three holo forms\n",
      "in a binary complex with the reduced cofactor NADPH, in a ternary\n",
      "complex with NADPH and butyryl-CoA, and a binary complex with the\n",
      "oxidized cofactor NADP+ (Figure ​Figure11b).Our high-resolution structures\n",
      "show that the tetrameric complex\n",
      "assumes a pair of dimers geometry. In this complex, the central oligomerization\n",
      "domains of the K. setae ECR remain\n",
      "essentially unchanged, while the peripheral catalytic domains move\n",
      "significantly to provide two sets of active site conformations—open\n",
      "and closed forms—upon binding of the NADPH cofactors in the\n",
      "presence of substrates. This coordinated motion is enabled by a tight\n",
      "coupling of catalytic domains across the pairs of dimers, which we\n",
      "could trace to the role of individual amino acids far away from the\n",
      "active site. Using computer simulations and kinetic experiments, we\n",
      "provide compelling evidence that synchronization across the pair of\n",
      "dimers through “swing and twist motions” is crucial\n",
      "for the K. setae ECR to achieve high\n",
      "catalytic rates. We further demonstrate that subunit communication\n",
      "within a dimer is essential to synchronize open and closed states.\n",
      "Altogether, our results unveil a dynamic intersubunit synchronization\n",
      "of the K. setae ECR complex that is\n",
      "essential for the functional organization of one of nature’s\n",
      "most efficient CO2-fixing enzymes during catalysis.Go to:Results\n",
      "and DiscussionApo ECR Is a Symmetric Homotetramer, Readily\n",
      "Accessible for\n",
      "Cofactor and Sub\n",
      "len(chunk):  ible for\n",
      "Cofactor and Substrate BindingWe first determined the apo\n",
      "form of the ECR crystal structure from K. setae at 1.8 Å resolution using synchrotron X-ray crystallography\n",
      "at cryogenic temperature (Table S1 in the\n",
      "Supporting Information). The asymmetric unit contains one homotetramer\n",
      "composed of four subunits arranged in a pair of dimers geometry similar\n",
      "to those of previously reported binary and ternary ECR complexes,\n",
      "crotonyl-CoA carboxylase/reductase AntE in complex with NADP+ (PDB: 4Y0K) and 2-octenoyl-CoA carboxylase/reductase CinF in complex with NADP+ and 2-octenoyl-CoA, (PDB: 4A0S), respectively. Overall, the K. setae ECR tetramer shows a noncrystallographic,\n",
      "close to D2 (dihedral) symmetry with four\n",
      "conformationally identical subunits (Figures S1 and S2). The tetrameric oligomer state of the ECR apo enzyme\n",
      "is further supported by size-exclusion chromatography, which showed\n",
      "that the 51.2 kDa protein eluted as a single peak at 205 kDa (Figure S3).Each ECR subunit consists of\n",
      "two domains—a larger catalytic domain formed by residues 1–212\n",
      "and 364–445, and a smaller oligomerization domain formed by\n",
      "residues 212–363 (Figure S4A). The\n",
      "oligomerization domain comprises a Rossmann fold11 with repeating αβ-motifs that form a 6-stranded\n",
      "β-sheet (β12−β17). The 6-stranded β-sheets\n",
      "of two neighboring subunits are combined into one 12-stranded β-sheet,\n",
      "forming the core of one dimer, subunits A/C and B/D. Two of these\n",
      "Rossmann fold domains form the tetrameric complex’s core (Figure S4B). The catalytic domains of the K. setae ECR are located at the periphery of the\n",
      "tetrameric complex. The active site cavities in the apo form are open\n",
      "and accessible for both cofactors and substrates. We also determined\n",
      "the structure of NADP+-bound ECR at cryogenic temperature\n",
      "and found that it is rather similar to that of the apo form structure,\n",
      "although NADP+ molecules are bound in all four subunits.\n",
      "We postulate that the same crystallization conditions used for both\n",
      "structures might have resulted in the same space group (P212121) with similar crystal contacts,\n",
      "which might have kept the NADP+ bound ECR tetramer in the\n",
      "apo form. The NADP+-bound subunits are less rigid in comparison,\n",
      "and some residues show greater variations (Figures S1 and S2).Cofactor Binding Breaks Tetramer Symmetry\n",
      "and Induces Formation\n",
      "of a Pair of DimersIn order to understand the effects of\n",
      "binding of NADPH, a cofactor before reaction as opposed to the spent\n",
      "cofactor NADP+ described above, we then determined the K. setae ECR-NADPH binary complex’s crystal\n",
      "structure at 2.4 Å resolution using serial femtosecond X-ray\n",
      "crystallography (SFX) at ambient temperature (Figure ​Figure22 and Table S1).12−15 In all four subunits, NADPH binds with its adenine moiety in the\n",
      "oligomerization domain and spans the catalytic domain, where its nicotinamide\n",
      "moiety is located (Figure S5).Open in a separate windowFigure 2Binding of\n",
      "NADPH results in global and local conformational changes\n",
      "in K. setae ECR. (a) NADPH-bound tetramer\n",
      "complex that is organized as pair of dimers, a pair of closed (green)\n",
      "and open (orange) subunits, and another pair containing closed (gray)\n",
      "and open (blue) subunits. (b) The foreground dimer with open (orange)-\n",
      "and closed-form (green) subunits rotated by 30° from the view\n",
      "in (a). Each subunit is composed of a catalytic and an oligomerization\n",
      "domain. (c) Comparison of the putative substrate binding sites between\n",
      "the open- and closed-form subunits.Notably, NADPH binding breaks the dihedral D2 symmetry observed in the apo-form tetramer, while symmetry\n",
      "about the y axis is retained, resulting in a noncrystallographic,\n",
      "almost cyclic C2 symmetry (Figure S1). In the NADPH*ECR binary complex,\n",
      "each of the two dimers, A/C and B/D, loses its 2-fold symmetry, giving\n",
      "rise to closed (A and B) and open (C and D) subunits (Figure ​Figure22a,b and Figure S1). In the A and B subunits, the cofactor binding\n",
      "pocket is compressed inward, which seals the NADPH cofactor within\n",
      "the catalytic domain, re\n",
      "len(chunk):  \n",
      "the catalytic domain, resulting in a “closed-form” state\n",
      "(Figure ​Figure22b). On the\n",
      "other hand, the C and D subunits show open cofactor-binding pockets,\n",
      "referred to hereafter as an “open-form” state. The substrate-binding\n",
      "pocket in the open-form C or D subunit is more than 5 Å wider\n",
      "than the closed A or B subunit (Figure ​Figure22c). When they were taken together, these structures\n",
      "indicated that during catalysis the enzyme differentiates into a pair\n",
      "of dimers (A/C and B/D dimers, respectively), with one closed (A or\n",
      "B) and one open subunit (C or D) per dimer.Cofactor–Substrate\n",
      "Complex Indicates Half-Site Reactivity\n",
      "within Each DimerWe next investigated the effects of substrate\n",
      "binding to the K. setae ECR tetramer\n",
      "complex and determined its ternary complex structure with the substrate\n",
      "analogue butyryl-CoA and NADPH at 1.7 Å resolution (Figure ​Figure33). The structure\n",
      "is overall very similar to the conformation of the ECR*NADPH binary\n",
      "complex with the noncrystallographic, pseudo-C2 cyclic symmetry (Figure ​Figure33a and Figure S1). It comprises\n",
      "two sets of open- and closed-form subunits that overlay very well\n",
      "with those of the ECR*NADPH binary complex (Figures S1 and S2). The NADPH cofactor appears to be bound to all active\n",
      "sites; however, only the closed-form subunits A and B also contain\n",
      "electron density for the complete butyryl-CoA thioester, while the\n",
      "open-form subunits (C and D) show only the adenine group electron\n",
      "density of the butyryl-CoA thioester (Figure ​Figure33a,b). The presence of the intact butyryl-CoA\n",
      "thioester in the active site of the closed-form subunits strongly\n",
      "suggests that the A and B conformations represent the Michaelis complex\n",
      "in which the substrate and cofactor are positioned for catalysis.\n",
      "In contrast, the open-form subunits (C and D) represent catalytically\n",
      "incompetent ternary complexes.Open in a separate windowFigure 3Structure of the ternary ECR complex.\n",
      "(a) ECR tetramer in a complex\n",
      "with NADPH and butyryl-CoA organized as a pair of dimers, the foreground\n",
      "dimer with one closed subunit (green) with NADPH and butyryl-CoA,\n",
      "an open (orange) subunit containing NADPH, and another pair in the\n",
      "background with one closed (gray) and one open (blue) subunit. (b)\n",
      "The foreground dimer with closed (green) and open (orange) subunits,\n",
      "rotated by 30° from the view in (a). Butyryl-CoA and NADPH atoms\n",
      "are represented as spheres. (c) Comparison of the product binding\n",
      "sites between the open- and closed-form subunits. (d) Cartoon and\n",
      "stick representation of the closed-form subunit active site. In (d–g),\n",
      "simple 2Fo – Fc density contoured at the 1.5 σ level is shown\n",
      "for butyryl-CoA, or pa ortion thereof, and NADPH within 3 Å from\n",
      "the molecules. (e) Cartoon and stick representation of the open-form\n",
      "subunit active site. (f) Superposition of the open-form subunit onto\n",
      "the closed-form subunit with a stick representation of the residues\n",
      "surrounding butyryl-CoA. (g) Butyryl-CoA binding in both open- and\n",
      "closed-form subunits with the electron density of the bound butyryl-CoA\n",
      "and NADPH at the active site of the closed subunit (green) and the\n",
      "adenine ring of butyryl-CoA at the active site of the open subunit\n",
      "(only the adenine ring electron density is visible). Left inset: the\n",
      "adenine binding pocket of the open-form subunit stabilizing the adenine\n",
      "ring of butyryl-CoA that stretches into the adjacent closed-form subunit.\n",
      "Right inset: the adenine binding pocket of the closed-form subunit\n",
      "holding the adenine ring of butyryl-CoA. Note that only the adenine\n",
      "ring of butyryl-CoA has visible electron density, while the rest of\n",
      "the molecule is disordered, and hence is indicated by a transparent\n",
      "stick model. In both cases, three residues of the adjacent subunits,\n",
      "K296, R303, and Y328, together hold the adenine ring.The binding of the cofactor and substrates in the closed-form\n",
      "ECR\n",
      "subunit is achieved by loops 37–44, 88–94, 338–350\n",
      "and helices 6, 7, and 21 of the catalytic domains, which creates multiple\n",
      "interactions of the protein with the\n",
      "len(chunk):  s of the protein with the NADPH and butyryl-CoA (Figure ​Figure33d–f). Notably,\n",
      "the CoA-thioester extends from the closed-form active site out to\n",
      "the oligomerization domain of the neighboring open-form subunit in\n",
      "the same dimer pair (from A to C in the A/C dimer, or from B to D\n",
      "in the B/D dimer), where R352 and Y353 from the open-form subunit\n",
      "(C or D) interact with the phosphate backbone of the CoA. The adenosine\n",
      "tail of the CoA extending from the closed-form subunit interacts with\n",
      "the three residues Y328, K296, and R303 that form an adenine binding\n",
      "pocket at the neighboring open-form subunit (Figure ​Figure33g). When we inspected CoA-thioester binding\n",
      "in the open-form subunits, we also observed the electron density in\n",
      "the adenine binding pocket of the neighboring closed form subunits\n",
      "(A or B), strongly indicating that the CoA-thioester was bound (Figure ​Figure33g, right inset).\n",
      "However, the electron density beyond the adenine ring becomes invisible,\n",
      "suggesting that the part of the CoA molecule reaching into the open-form\n",
      "subunit’s active site remains flexible and disordered, which\n",
      "is corroborated by the higher B-factors of the catalytic domain (Figure ​Figure11, bottom right).To evaluate the substrate’s flexibility in the open- and\n",
      "closed-form subunits, quantum mechanical/molecular mechanics (QM/MM)\n",
      "simulations on a dimer of subunits A and C that account for the electronic\n",
      "degrees of freedom of the substrate were combined with MM simulations\n",
      "of the pair of dimers (tetramer) with substrates initially modeled\n",
      "in the open and bound closed subunits to reach longer time scales.\n",
      "These simulations showed that the substrate in the closed subunit\n",
      "had significantly lower temperature factors (B-factors) in comparison\n",
      "to that in the open subunit (Figures S6–S8 and Movies S1a–e and S2a–e). In the open subunit, the acyl moiety shows a high degree\n",
      "of flexibility in the active site, in agreement with the higher B-factors\n",
      "observed in crystal structures.Together, crystallographic structures\n",
      "and QM/MM simulations indicate\n",
      "that upon substrate binding the enzyme complex splits into two catalytically\n",
      "competent (A or B) and incompetent subunits (C or D) per dimer, suggesting\n",
      "that the enzyme tetramer operates with half-site reactivity, in which\n",
      "catalytically competent and incompetent active sites alternate during\n",
      "catalysis.16−19Swing and Twist Motions of Peripheral Catalytic Domains Are\n",
      "Coupled within and across a Pair of Dimers to Promote CatalysisTo study the switching between the open and closed forms of the\n",
      "ternary complex, we used computer simulations. Starting from the ternary\n",
      "crystal structure with NADPH and butyryl-CoA, we modeled the natural\n",
      "substrate crotonyl-CoA to carry out all-atom molecular dynamics simulations\n",
      "of the catalytic active enzyme (see Supplementary Methods). In our simulations, the closed subunit remained closed\n",
      "and maintained crotonyl-CoA in the active site when the dynamics of\n",
      "the substrate were restrained to maintain specific protein–substrate/cofactor\n",
      "interactions. However, in simulations without substrate restraints,\n",
      "the acyl moiety of the substrate left the active site, resulting in\n",
      "a conformational change of the closed subunit to the open form.To study this conformational change in detail, we eliminated the\n",
      "crotonyl-CoA substrate from the ternary complex, creating an ECR*NADPH\n",
      "binary complex with two closed and two open subunits. In all 10 100\n",
      "ns simulations of the tetramers, the closed subunits underwent a fast\n",
      "conformational change during the first tens of nanoseconds (the same\n",
      "was observed when we started from the X-ray binary ECR*NADPH structure).\n",
      "To further describe this conformational change, we used principal\n",
      "component analysis (PCA) of the backbone Cα atom’s positions\n",
      "of 10 combined 100 ns trajectories of both dimers. The 100 ns time\n",
      "scale was enough to describe the primary steps of the conformational\n",
      "changes, as evidenced by the convergence analysis (Figures S9 and S10). The first principal component (PC1) \n",
      "len(chunk):  rincipal component (PC1) represents\n",
      "a transition of the closed to the open form, and the second principal\n",
      "component (PC2) describes a twist motion of one subunit to the other\n",
      "subunit in each dimer. In both cases the catalytic domains move relative\n",
      "to the oligomerization domains (Figure ​Figure44a–c and Movies S3 and S4). The rotation axes of PC1 and\n",
      "PC2 cross with each other with close to 90° inside the catalytic\n",
      "domain in each subunit. The two PC1 axes of subunits A and C are nearly\n",
      "parallel, corresponding to the closed-open motions, while the PC2\n",
      "axes show the twisting motions of the catalytic domains (Figure ​Figure44b,c).Open in a separate windowFigure 4Molecular dynamics simulations\n",
      "and principal component analyses\n",
      "reveal that the absence of the substrate induces a transition between\n",
      "the open and closed forms of the catalytic domain in the ECR tetramer\n",
      "and a twist in the subunits, which would move a bound substrate toward\n",
      "the NADPH cofactor. (a) Representation of the swing motion of the\n",
      "catalytic domain in dimer AC described by the first principal component\n",
      "(PC1) and the twist motion of one subunit to the other by PC2. This\n",
      "twist motion would bring a bound substrate closer to the NADPH cofactor.\n",
      "PC1 and PC2 axes of A (closed) and C (open) subunits are superimposed\n",
      "on the ternary tetramer structure in the same orientation as in Figure ​Figure33a, showing the functional\n",
      "relevance of PC1 as opening–closing and PC2 as twisting of\n",
      "the catalytic domains. (b, c) Sets of vectors between initial and\n",
      "final Cα positions of the catalytic domains are shown viewed\n",
      "along the PC1 (b) and PC2 (c) axes of subunit A as shown on the bottom\n",
      "left of (a). (d) Projection of 10 100 ns trajectories on the first\n",
      "two principal components of the wild type binary system with the NADPH\n",
      "cofactor ECRwt·NADPH (crosses and diamonds represent\n",
      "the open- and closed-form subunits from the X-ray structure, respectively).\n",
      "The middle panels show the projection on the same PCs for the ECRwt ternary complex bearing only one substrate in the B subunit\n",
      "of the BD dimer ([B·Crot-CoA]D). The right panels represent the\n",
      "dynamics of the E151D/N157E/N218E triple variant with the NADPH cofactor\n",
      "(ECRTriple.Mut·NADPH) on the same eigenvectors. Colors\n",
      "of each points represent the time frame according to the scale bar\n",
      "at the bottom. (e) Mean values and 1.5 times the standard deviation\n",
      "of the principal components of 10 trajectories as a function of simulation\n",
      "time for each dimer in the WT (orange) and E151D/N157E/N218E triple\n",
      "variant (blue) ECR·NADPH complex. Time traces of PC1 are shown\n",
      "on the left and those of PC2 on the right. In the middle, free energy\n",
      "profiles as a function of PC1 and PC2 were derived for the wild type\n",
      "(orange) and variant (blue) on the basis of the observed kinetics.\n",
      "The triple variant with slower opening kinetics (PC1) is consistent\n",
      "with a barrier in the transition to the open form, and its larger\n",
      "standard deviation suggests a shallower minimum in the closed form.\n",
      "The free energy profile of PC2 corresponds to a broad minimum for\n",
      "the triple variant because of the larger standard deviation observed\n",
      "in the kinetics and a large barrier to reach the twisted conformation\n",
      "observed in the WT dynamics (brown trajectory shown in dimer AC for\n",
      "PC2 on the right).To characterize the dynamics,\n",
      "we compared the projections of all\n",
      "10 dimer trajectories on PC1 and PC2. Dimer A/C opens its closed subunit\n",
      "first, associated with PC1, followed by a twist motion characterized\n",
      "by PC2 in some of the simulations (Figure ​Figure44d and Figure S10). This twist motion rotates the catalytic domain of the empty, now\n",
      "open form active site in subunit A, which would push a bound crotonyl-CoA\n",
      "substrate toward the NADPH cofactor in the C subunit. The B/D dimer’s\n",
      "dynamics, on the other hand, were mainly governed by the closed subunit\n",
      "opening represented by PC1, likely because of the initial structural\n",
      "differences of the two dimers.After characterizing the conformational\n",
      "transitions associated\n",
      "\n",
      "len(chunk):  l\n",
      "transitions associated\n",
      "with the presence or absence of CoA substrate, we asked whether the\n",
      "dynamics of one dimer in the tetramer complex is coupled to that of\n",
      "the other dimer. Computer simulations allowed us to address this question\n",
      "by studying the dynamics of a hypothetical system that lacks the substrate\n",
      "in only one active site. Removing crotonyl-CoA from subunit A, we\n",
      "expected the—now empty—“closed” active\n",
      "site subunit to transition to the open form, inducing the twist motion\n",
      "in the A/C dimer observed before, and the other closed subunit B still\n",
      "bearing the substrate remains closed, if the two dynamics of the pair\n",
      "of dimers are uncoupled. The projection of 10 100 ns trajectories\n",
      "on PC1 and PC2 confirmed that the empty active site (A in the A/C\n",
      "dimer) undergoes a conformational change to the open form (Figure ​Figure44d), whereas the dimer\n",
      "containing the substrate (B in the B/D dimer) remained in a closed\n",
      "conformation (see Figure S11 for a comparison\n",
      "to the simulation with two substrates maintaining the closed state).\n",
      "Importantly, however, none of the A/C dimer trajectories showed a\n",
      "significant change in PC2 associated with the twist motion observed\n",
      "in the binary ECR*NADPH complex, indicating that the twist motion\n",
      "is only initiated once the other dimer B/D adopts the open conformation.In summary, our molecular dynamics simulations indicate that the\n",
      "absence or release of the substrate triggers a conformational transition\n",
      "from the closed to the open conformation and that coupled dynamics\n",
      "of the two dimer pairs induce an additional twist motion after opening\n",
      "the closed subunit of the other dimer pair.Three Remote Amino Acids\n",
      "Couple Catalysis between Pairs of DimersGiven the coordinated\n",
      "motions of the catalytic domains during catalysis,\n",
      "how is catalysis synchronized across the enzyme complex (i.e., across\n",
      "dimers A/C and B/D)? One intriguing aspect of the ECR tetramer structures\n",
      "is that the catalytic domains share a common interface of 1636 Å2 between the pairs of dimers (Figure ​Figure55a,b), suggesting that they move together\n",
      "as rigid bodies.Open in a separate windowFigure 5Inter- and intradimer communications drive fast CO2 fixation\n",
      "by K. setae ECR. (a) Two distinct sets\n",
      "of communications: interdimer interactions between the catalytic domains\n",
      "from two dimers (purple arrows) and intradimer communication between\n",
      "the open and closed subunits within each dimer (brown arrows). (b)\n",
      "Interdimer catalytic domain interface and positions of selected amino\n",
      "acids that were mutated in this study to affect the interface between\n",
      "the two catalytic domains (open-form subunit in orange and closed\n",
      "form subunit in gray). The right panel shows the mutual H-bonding\n",
      "interaction between N218 and N157 from open- and closed-form subunits\n",
      "and H-bonding between E151 and the N atom from the protein backbone.\n",
      "(c) Alignment of ECR protein sequences from the primary (upper row)\n",
      "and the secondary (lower row) metabolisms represented as sequence\n",
      "logos. The numbering of residues, above the first row, is according\n",
      "to their position in the K. setae ECR.\n",
      "(d) Communication between the closed (green) and open (orange) subunits\n",
      "across the two dimers of the K. setae ECR. In the closed conformation the contacts between NADPH-H365-E165\n",
      "and K332 of the adjacent open subunit allow for the correct intradimer\n",
      "communication. In the open conformation the communication network\n",
      "is compromised, as indicated by the increased distances between the\n",
      "amino acid side chains that cause the incorrect positioning of the\n",
      "nicotinamide ring of NADPH.What are the molecular determinants that synchronize catalysis\n",
      "across the pair of open/closed form dimers? The intercatalytic domain\n",
      "interface is predominantly hydrophobic with some electrostatic interactions\n",
      "(Figure ​Figure55b). Most notable\n",
      "are N218, which forms a hydrogen bond to N157 of the adjacent subunit\n",
      "of the other dimer, and E151 interacting with the backbone nitrogen\n",
      "atom of N133 (and/or A134) of the neighboring subunit of the other\n",
      "d\n",
      "len(chunk):  ng subunit of the other\n",
      "dimer (Figure ​Figure55b).\n",
      "A multiple sequence alignment using CLUSTAL W20 showed that E151, N218, and N157 are highly conserved in ECRs from\n",
      "primary (i.e., central carbon) metabolism with faster CO2-fixation kinetics (average kcat 28 s–1), but not in ECRs from secondary metabolism with\n",
      "an average kcat of 1.2 s–1 (Figure ​Figure55c), which\n",
      "raised the question about their potential role in promoting catalysis\n",
      "in primary metabolic ECRs.Notably, mutation of these residues\n",
      "that are more than 20 Å\n",
      "away from the active site dramatically affected the kinetic parameters\n",
      "of K. setae ECR (Table 1). In the E151D variant, the kcat value decreased by a factor of 5, demonstrating that\n",
      "weakening the interaction between catalytic domains has profound effects\n",
      "on the enzyme’s catalytic rate. Mutations that targeted the\n",
      "asparagine interaction network also showed substantial effects on\n",
      "the catalytic rate and affected the KM value of substrate binding. Most notable were a N218E single variant\n",
      "and E151D/N157E/N218E triple variants that decreased the kcat values by more than 25- and 100-fold, respectively,\n",
      "highlighting that communication through the interface of the catalytic\n",
      "domains of the pair of dimers is an essential determinant of the catalytic\n",
      "rate in K. setae ECR.Table 1Steady -State Analysis of K. setae ECR and Variants Targeting the Catalytic\n",
      "Domain Interface between the Pair of Dimersa crotonyl-CoANADPHCO2enzymeKM (μM)Ki (μM)kcat (s–1)KM (μM)kcat (s–1)KM (μM)kcat (s–1)wild type21 ± 23650 ± 810103 ± 337 ± 486 ± 290 ± 1078 ± 2E151D28 ± 21960 ± 25020 ± 172 ± 1117 ± 180 ± 1021 ± 1N157E515 ± 75 22 ± 1105 ± 2717.1 ± 0.440 ± 614 ± 0N218E272 ± 37 3.7 ± 0.266 ± 93.3 ± 0.0.2840 ± 1503.3 ± 0.2E151D/N157E/N218E245 ± 25 1.1 ± 0.026 ± 30.7 ± 0.0440 ± 400.9 ± 0.0Open in a separate windowaMichaelis–Menten curves\n",
      "of K. setae ECR and its variants are\n",
      "provided in Figure S14.To confirm that the oligomerization\n",
      "state of the different complexes\n",
      "was not altered through these mutations, we used gel filtration and\n",
      "native gel analysis (Figure S3). Gel filtration\n",
      "under the same conditions as for our kinetic measurements showed that\n",
      "all enzyme variants remained tetramers, while a native gel analysis\n",
      "conducted under more disruptive conditions showed slight increases\n",
      "in the dimer and monomer fractions. Overall, our mutational and kinetic\n",
      "data supported the hypothesis that synchronization of catalytic domains\n",
      "contributes to the catalytic rate and is conferred through the hydrogen-bond\n",
      "network at the pair of dimers interfaces.To study the effect\n",
      "of the E151D/N157E/N218E mutations on the synchronization\n",
      "of catalytic domains and the conformational transitions of the binary\n",
      "NADPH*ECR complex, we used computer simulations as discussed above.\n",
      "The projection of 10 100 ns trajectories on the previously defined\n",
      "eigenvectors displayed the same opening transition of both dimers\n",
      "in the initially closed binary NADPH*ECR triple variant complex (Figure ​Figure44d). We analyzed the\n",
      "time traces of their first two principal components (mean ± 1\n",
      "std) of both dimers in the tetramer to spot the differences between\n",
      "the WT enzyme and the triple variant (see Figure ​Figure44e). Notably, the triple variant opens the\n",
      "closed form more slowly than the WT enzyme and presents a broader\n",
      "distribution of values in both dimers during transition. The observed\n",
      "kinetics are consistent with a free energy profile along PC1, according\n",
      "to which the variant must cross a higher free energy barrier to reach\n",
      "the open form (scheme in Figure ​Figure44e). The broader distribution during the transition\n",
      "is caused by a shallower energy minimum of the closed form of the\n",
      "triple variant. For PC2, the dynamics for the WT and the triple variant\n",
      "also differ. The triple variant presents a broader distribution associated\n",
      "with a shallow free energy minimum and shifted minima, causing the\n",
      "two systems to evolve in opposing directions at the beginning of the\n",
      "simulation. Interestingly, only the WT can pass th\n",
      "len(chunk):  , only the WT can pass the free energy barrier\n",
      "to a twisted state in some of the trajectories (shown in brown in Figure ​Figure44e). Overall, these\n",
      "results confirm that the E151D/N157E/N218E mutation at the dimer interfaces\n",
      "changes the free energy landscape of conformational transitions considerably.\n",
      "The communication between the dimers is reduced, as the triple variant\n",
      "does not share the WT’s concerted dynamics that are characterized\n",
      "first by an opening motion and then a twist of one subunit to the\n",
      "other. This twist motion of the open subunit will push the bound substrate\n",
      "to the NADPH cofactor and might be directly coupled to the first step\n",
      "of the ECR reaction: i.e., the hydride transfer.In summary,\n",
      "computer simulations and kinetic data of the triple\n",
      "variant indicate that coupling the dynamics of the two dimers is conferred\n",
      "by three conserved amino acids in ECRs from primary metabolism (E151,\n",
      "N157, and N218) and is essential to reach fast CO2 fixation\n",
      "kinetics in comparison to their counterparts from secondary metabolism\n",
      "that lack those residues.CoA Binding Synchronizes Open and Closed\n",
      "Subunits within a DimerIn addition to an interdimer domain\n",
      "interaction, the two subunits\n",
      "in each dimer bind different parts of the substrate in the open and\n",
      "closed forms through the adenine binding pocket and the active site\n",
      "(intradimer interaction, Figure ​Figure55a). We hypothesized that this shared substrate binding\n",
      "between neighboring subunits could also contribute to the mechanism\n",
      "of fast, synchronized catalysis.To understand the role of substrate\n",
      "adenine binding in catalysis, we characterized the kinetics of different\n",
      "single, double, and triple variants of the adenine binding site (Figure ​Figure33g and Table 2). Mutations in the adenine\n",
      "binding pocket, particularly of R303, strongly increased the apparent KM value of the CoA substrate as expected and\n",
      "decreased the enzyme’s apparent kcat value by a factor of 2–3. Notably, when we used crotonyl-panthetheine,\n",
      "a truncated substrate lacking the adenosine moiety, a comparable decrease\n",
      "in kcat value was also observed for the\n",
      "WT enzyme, suggesting that substrate interactions at the adenine binding\n",
      "pocket indeed contribute to catalysis. However, when we compared the\n",
      "kinetics of the WT enzyme with a truncated substrate to that of triple\n",
      "variant K296A/R303A/Y328F, in which adenosine binding was completely\n",
      "disturbed, the two variants behaved almost identically, strongly supporting\n",
      "the argument that binding of the adenine residue of the CoA substrate\n",
      "in the WT is important for high catalytic activity.Table 2Apparent Michaelis–Menten Parameters\n",
      "of K. setae ECR and Variants Targeting\n",
      "the Adenine Binding Pocket as Mean Values ± Standard Errora crotonyl-CoAcrotonyl-pantetheineenzymeKM (μM)Ki (μM)kcat (s–1)KM (μM)kcat (s–1)wild type21 ± 23650 ± 810103 ± 38660 ± 53037 ± 1K296A107 ± 11 68 ± 2  Y328F11 ± 24671 ± 169380 ± 3  K296A/Y328F190 ± 30 39 ± 2  K296A/R303A/Y328F2180 ± 280 29 ± 27770 ± 11042 ± 3K296A/R303 K/Y328F830 ± 140 53 ± 2  Q165A27 ± 4 56 ± 35630 ± 5208.6 ± 0.3K332A450 ± 130 38 ± 72980 ± 5100.3 ± 0.0Open in a separate windowaMichaelis–Menten\n",
      "curves\n",
      "of K. setae ECR and its variants are\n",
      "provided in Figure S14.We noticed that the substrate’s\n",
      "adenine binding pocket is\n",
      "directly followed by a loop, which carries a lysine residue (K332)\n",
      "that interacts with the neighboring subunit’s active site.\n",
      "K332 from the open-form subunit engages in a hydrogen-bonding network\n",
      "with the nicotinamide of the NADPH cofactor bound to the closed-form\n",
      "subunit through Q165 and H365 of the neighboring subunit (Figure ​Figure55d). These interactions\n",
      "are not observed in the active site of the open-form subunit (Figure ​Figure55d), raising the question\n",
      "of whether the hydrogen-bonding network connected to the adenine binding\n",
      "pocket might be necessary for catalysis.In variants K332A and\n",
      "Q165A, kcat was\n",
      "decreased by 2–3-fold, and for K332A, an additional increase\n",
      "in the apparent KM for crotonyl-CoA was\n",
      "observed, indicating some eff\n",
      "len(chunk):  rved, indicating some effect of these residues on the catalytic\n",
      "rate (as well as substrate binding in case of K332) (Table 2). When we tested these variants\n",
      "again with the truncated substrate analogue crotonyl-pantetheine,\n",
      "the catalytic activity in the Q165A variant was reduced by 4-fold,\n",
      "while that of the K332A variant was reduced by more than 2 orders\n",
      "of magnitude in comparison to the WT with a truncated substrate. More\n",
      "notably, the KM had not changed much between\n",
      "the WT and the two different variants (where it has even slightly\n",
      "improved), strongly suggesting that Q165 and K332 are not involved\n",
      "in substrate accommodation but rather synchronization of catalysis\n",
      "between open and closed subunits within the dimer. In conclusion,\n",
      "adenine binding and the loop carrying the residues K332 and Q165 are\n",
      "necessary to synchronize catalysis between the two subunits within\n",
      "the dimer to drive fast CO2 fixation in the K. setae ECR, providing the molecular basis of how\n",
      "catalysis might be synchronized between the open- and closed-form\n",
      "subunits within one dimer.Half-Site Reactivity with Almost Fullly Occupied\n",
      "Active Sites\n",
      "Drives Fast CO2 FixationOur structural studies\n",
      "of the K. setae ECR revealed unprecedented\n",
      "details on the functional organization of one of nature’s most\n",
      "efficient and fastest CO2-fixing enzymes. During catalysis,\n",
      "the enzyme complex differentiates into distinct functional subunits.\n",
      "The binding of the NADPH cofactor and substrates forces the homotetrameric\n",
      "apo enzyme into a pair of dimers. Each dimer is constituted of open-\n",
      "and closed-form subunits. In the closed-form subunits, the NADPH cofactor\n",
      "and CoA substrate are aligned with each other, suggesting that this\n",
      "is the catalytically competent state. The open-form subunits bind\n",
      "the cofactor and the substrate adenine rings, but the rest of the\n",
      "acyl-CoA substrate remains flexible and invisible in the active site.\n",
      "Thus, the open-subunit active sites represent a catalytically incompetent\n",
      "state that is preorganized for a subsequent round of catalysis. Overall,\n",
      "this structural reorganization of the ECR complex strongly supports\n",
      "the idea that the enzyme operates with “half-site reactivity”.16−18 Catalysis is synchronized across the enzyme tetramer and alternates\n",
      "between the open- and closed-form subunits to increase the overall\n",
      "catalytic efficiency of the complex.It is worth noting that\n",
      "half-site reactivity does not assume (or require) half-site occupancy.\n",
      "Indeed, all four subunits of the K. setae ECR tetramer ternary complex contain both NADPH and substrates (products).\n",
      "However, in the open-form subunits, only electron density for the\n",
      "adenine groups of the CoA thioester is observed (Figure ​Figure33g). The “half-site reactivity”\n",
      "is also consistent with a negative, rather than positive, cooperativity,\n",
      "as almost all our kinetics data indicate.To understand if the\n",
      "conventional half-site reactivity theory can\n",
      "be applied to the ECR tetramer case, we created a model for the enzyme\n",
      "kinetics assuming positive or negative cooperativity between the pairs\n",
      "of dimers. Instead of treating the four subunits as independent entities,\n",
      "we hypothesized that the two pairs of ECR dimers would cycle through\n",
      "the CO2-fixing reactions in synchrony: i.e., when one dimer\n",
      "switches from the open–close to the closed–open states,\n",
      "the other dimer would do the opposite, closed–open to open–closed\n",
      "states. Using this model, we simulated the case with just one dimer\n",
      "with two subunits working in negative or positive cooperativity using\n",
      "the method described by Hill and Levitski21 (Figure S12). For a thorough investigation\n",
      "of how cooperativity, positive or negative, enhances the overall reaction\n",
      "rate of the K. setae ECR, we incorporated\n",
      "all the parameters in Hill and Levitski’s model in our simulation.\n",
      "It turned out that only negative cooperativity could explain our experimental\n",
      "observations that ECR activity in the WT tetramer is increased by\n",
      "2 orders of magnitude in comparison to the triple variant E151D/N157E/N218E,\n",
      "in w\n",
      "len(chunk):  t E151D/N157E/N218E,\n",
      "in which synchronization is decoupled (Table 1). Our simulations yielded a 90% probability\n",
      "that both active sites are occupied with substrates/products and NADPH/NADP+ (p22), while only 10% are half-occupied (p12 + p21) (Figure S12). Almost none of the dimers are empty\n",
      "(p11), which agrees with the occupancy of our experimentally determined\n",
      "structures of the ternary complex.To synchronize half-site\n",
      "reactivity, the interaction of the catalytic\n",
      "domains across and within the pairs of dimers is crucial. When the\n",
      "latter is perturbed in the K. setae ECR (e.g., through triple mutant E151D/N157E/N218E or by disturbance\n",
      "of the adenosine binding pocket), the catalytic rate of the enzyme\n",
      "is severely diminished. This observation is consistent with other\n",
      "theoretical and experimental data on half-site reactivity. The synchronization\n",
      "of distant catalytic subunits has been calculated to enhance the catalytic\n",
      "rate of enzymes by up to 20-fold.21,22 Mutation of\n",
      "a single amino acid that couples the two catalytic sites of heptose\n",
      "isomerase GmhA reduced the catalytic rate to 6% of wild-type activity.23 Perturbing the interaction network in Escherichia coli thymidylate synthase, another example\n",
      "of an enzyme with half-site reactivity,24 caused a 400-fold decrease in kcat,25,26 demonstrating that domain interactions are essential in promoting\n",
      "enzyme catalysis.27While our structural,\n",
      "biochemical, and simulation data indicate\n",
      "that the K. setae ECR achieves high\n",
      "catalytic rates by synchronizing active sites, this might not necessarily\n",
      "be true for other ECRs. A differentiation into a pair of dimers was\n",
      "not observed in NADPH-bound or ternary structures of other ECRs (e.g.,\n",
      "PDB 4Y0K and 4A0S, respectively, which\n",
      "share substantial amino acid identity) (Figure S13). Note, however, that ECRs fall into two different classes:\n",
      "primary ECRs (such as the K. setae ECR)\n",
      "that operate in central carbon metabolism and secondary ECRs (such\n",
      "as 4Y0K) that\n",
      "provide extender units for synthesizing polyketides in secondary metabolism.\n",
      "Whereas primary ECRs are under intense evolutionary pressure and show\n",
      "on average kcat values of 28 s–1,28 secondary ECRs are not selected for\n",
      "high catalytic rates, which is also reflected by a 1 order of magnitude\n",
      "smaller average kcat value (kcat = 1.2 s–1).28 Thus, it might be tempting to speculate that\n",
      "secondary ECRs are not selected for high turnover rates during catalysis\n",
      "and, therefore, might not exhibit synchronized half-site reactivity,\n",
      "which is further supported by the observation that the residues E151,\n",
      "N218, and N157, conferring synchronization at the complex interface,\n",
      "are not conserved in secondary ECRs (Figure ​Figure55c).Go to:ConclusionsAltogether,\n",
      "our work revealed the structural-dynamics determinants\n",
      "explaining the substantial rate enhancement achieved by enoyl-CoA\n",
      "carboxylase/reductase through coordinated catalytic domain motions\n",
      "in comparison to unsynchronized catalytic reactions (e.g., in the\n",
      "four monomers of the triple variant E151D/N157E/N218E). The X-ray\n",
      "structures of the apo-form, binary and ternary ECR complexes provide\n",
      "the end states of these coordinated catalytic domain motions. A PCA\n",
      "analysis of extensive molecular dynamics simulations delineates an\n",
      "atomistic picture of the coupled motions, with one opening and closing\n",
      "motion of the active site and a twist motion that might aid the substrate\n",
      "binding and product release. A further analysis shows that negative\n",
      "cooperativity and half-site reactivity are the governing principles\n",
      "of the substantial enhancement of the overall reaction rate. Active\n",
      "site synchronization can be traced down to individual amino acids\n",
      "at the complex interface and the adenine binding pocket of the CoA\n",
      "substrate. These residues associated with the coupling of the catalytic\n",
      "domains across the different dimers are conserved among highly efficient\n",
      "ECRs from primary metabolism, enhancing the overall catalytic rates\n",
      "by more than a factor of 30 in comparison to their counterparts in\n",
      "se\n",
      "len(chunk):   their counterparts in\n",
      "secondary metabolism.Go to:MethodsNo unexpected or unusually\n",
      "high safety hazards were encountered.\n",
      "Details of experimental and computational methods are described in Supplementary Methods. Briefly, these contain\n",
      "methods of the amplification and cloning of the K.\n",
      "setae ECR, site-directed mutagenesis of its residues\n",
      "at the subunit interface, adenine binding residues and intradimer\n",
      "communication residues, as well as cell lysis, protein purification,\n",
      "characterization of the oligomeric state of the enzyme, and spectrophotometric\n",
      "enzyme assays. The chemical synthesis of CoA-esters and an analysis\n",
      "of ethylmalonyl-CoA stability are also described. Finally, we include\n",
      "detailed methods on the crystallization of K. setae ECR complexes, data collection, processing, and structure determination,\n",
      "QM/MM simulations of NADPH and substrate in the open and closed subunits\n",
      "of the ECR dimer, and molecular dynamics simulations of the conformational\n",
      "changes in the ECR tetramer.Go to:AcknowledgmentsThe authors acknowledge Takanori Nakane from the\n",
      "University of Tokyo for his help with calibration of XFEL data from\n",
      "SACLA, RIKEN, Japan. We thank Eriko Nango, Rie Tanaka, and the RIKEN\n",
      "SPring-8 Center for their help with data collection at SACLA, RIKEN,\n",
      "Japan. H.D. acknowledges support from NSF Science and Technology Center\n",
      "grant NSF-1231306 (Biology with X-ray Lasers, BioXFEL). This publication\n",
      "has been produced benefiting from the 2232 International Fellowship\n",
      "for Outstanding Researchers Program and the 1001 Scientific and Technological\n",
      "Research Projects Funding Program of the Scientific and Technological\n",
      "Research Council of Turkey (TÜBİTAK) (Project Nos. 118C270\n",
      "and 120Z520). However, the entire responsibility of the publication\n",
      "belongs to the authors of the publication. The financial support received\n",
      "from TÜBİTAK does not mean that the content of the publication\n",
      "is approved in a scientific sense by TÜBİTAK. The XFEL\n",
      "experiments were performed at BL3 of SACLA with the approval of the\n",
      "Japan Synchrotron Radiation Research Institute (JASRI) (Proposal No.\n",
      "2017A8055). The authors thank the beamline staff of the Structural\n",
      "Molecular Biology Group, SSRL, SLAC, and GM/CA CAT, Advance Photon\n",
      "Source, ANL, for assistance with data collection. Y.R., R.G.S., M.S.H.,\n",
      "and B.H. were supported by the U.S. Department of Energy, Office of\n",
      "Science, Office of Basic Energy Sciences under Contract No. DE-AC02-76SF00515.\n",
      "T.J.E. and G.M.S. received support from the Max Planck Society, the\n",
      "European Research Council (ERC 637675 “SYBORG”), and\n",
      "the U.S. Department of Energy Joint Genome Institute, a DOE Office\n",
      "of Science User Facility, under Contract No. DE-AC02-05CH11231. D.A.S.,\n",
      "A.G., H.G., and E.V.-M. thank the Max-Planck Society for funding as\n",
      "a Max-Planck-Partner group and CONICYT PCI MPG190003. A.G. acknowledges\n",
      "CONICYT Doctorado Nacional grant 21190262. D.A.S. thanks the Fondo\n",
      "Nacional de Desarrollo Científico y Tecnológico (Fondecyt)\n",
      "for his postdoctoral fellowship No. 3190579. H.D., Y.R., and S.W.\n",
      "are supported by the DOE Office of Science, Biological Environmental\n",
      "Research, and the National Institutes of Health, NIGMS. C.G. and S.W.\n",
      "were supported by the National Science Foundation, Major Research\n",
      "Instrument grant. H.D. and S.W. were partially supported by the Stanford\n",
      "PRECOURT Institute. Gregory M. Stewart of SLAC and Moe Wakatsuki are\n",
      "thanked for graphics work. Coordinates of the four ECR structures\n",
      "have been deposited in the Protein Data Bank under accession codes 6NA3 (apo), 6NA4 (Butyryl-CoA/NADPH\n",
      "bound), 6NA5 (NADP+ bound), and 6NA6 (NADPH-bound).Go to:NotesThis paper was published\n",
      "on April 25, 2022, with an incorrect bond pictured in the Abstract\n",
      "graphic. The corrected version was reposted on May 4, 2022.Go to:Supporting Information AvailableThe Supporting Information is available\n",
      "free of charge at https://pubs.acs.org/doi/10.1021/acscentsci.2c00057.Additional figures, discussions,\n",
      "methods and a table\n",
      "as desribed in the text (PDF)Movie as de\n",
      "len(chunk):  the text (PDF)Movie as described in the text (MOV)Movie as described in the\n",
      "text (MOV)Movie as described in the text (MOV)Movie as described in the text (MOV)Movie as described\n",
      "in the text (MOV)Movie as described in the text (MOV)Movie as described in the text (MOV)Movie as described\n",
      "in the text (MOV)Movie as described in the text (MOV)Movie as described in the text (MOV)Movie as described\n",
      "in the text (MOV)Movie as described in the text (MOV)Transparent Peer Review report available\n",
      "(PDF)Go to:Author Contributions▼ H.D.,\n",
      "Y.R., G.M.S., B.V., K.S., and A.G. contributed equally.Go to:NotesThe authors declare no\n",
      "competing financial interest.Go to:Supplementary Materialoc2c00057_si_001.pdf(17M, pdf)oc2c00057_si_002.mov(9.0M, mov)oc2c00057_si_003.mov(8.2M, mov)oc2c00057_si_004.mov(9.0M, mov)oc2c00057_si_005.mov(8.1M, mov)oc2c00057_si_006.mov(8.1M, mov)oc2c00057_si_007.mov(8.2M, mov)oc2c00057_si_008.mov(8.2M, mov)oc2c00057_si_009.mov(8.2M, mov)oc2c00057_si_010.mov(6.4M, mov)oc2c00057_si_011.mov(8.2M, mov)oc2c00057_si_012.mov(8.7M, mov)oc2c00057_si_013.mov(8.7M, mov)oc2c00057_si_014.pdf(242K, pdf)Go to:ReferencesSchwander T.; Schada von Borzyskowski L.; Burgener S.; Cortina N. S.; Erb T. J.\n",
      "A synthetic pathway\n",
      "for the fixation of carbon dioxide\n",
      "in vitro. Science (80-.).\n",
      "2016, 354, 900–904. 10.1126/science.aah5237. [PMC free article] [PubMed] [CrossRef] [Google Scholar]Erb T. J.; et al. Synthesis of C5-dicarboxylic\n",
      "acids from C2-units involving crotonyl-CoA\n",
      "carboxylase/reductase: the ethylmalonyl-CoA pathway. Proc. Natl. Acad. Sci. U. S. A.\n",
      "2007, 104, 10631–6. 10.1073/pnas.0702791104. [PMC free article] [PubMed] [CrossRef] [Google Scholar]Schada\n",
      "von Borzyskowski L.; Rosenthal R. G.; Erb T. J.\n",
      "Evolutionary history\n",
      "and biotechnological future of carboxylases. J. Biotechnol.\n",
      "2013, 168, 243–251. 10.1016/j.jbiotec.2013.05.007. [PubMed] [CrossRef] [Google Scholar]Amao Y.\n",
      "Formate dehydrogenase\n",
      "for CO2 utilization and its application. J. of CO22 Utilization\n",
      "2018, 26, 623–641. 10.1016/j.jcou.2018.06.022. [CrossRef] [Google Scholar]Erb T. J.; Brecht V.; Fuchs G.; Muller M.; Alber B. E.\n",
      "Carboxylation\n",
      "mechanism and stereochemistry of crotonyl-CoA carboxylase/reductase,\n",
      "a carboxylating enoyl-thioester reductase. Proc.\n",
      "Natl. Acad. Sci. U. S. A.\n",
      "2009, 106, 8871–8876. 10.1073/pnas.0903939106. [PMC free article] [PubMed] [CrossRef] [Google Scholar]Erb T. J.; Zarzycki J.\n",
      "Biochemical and synthetic biology approaches to improve\n",
      "photosynthetic CO2 -fixation. Curr.\n",
      "Opin. Chem. Biol.\n",
      "2016, 34, 72–79. 10.1016/j.cbpa.2016.06.026. [PMC free article] [PubMed] [CrossRef] [Google Scholar]Rosenthal R. G.; et al. Direct evidence for a covalent ene adduct intermediate\n",
      "in NAD(P)H-dependent\n",
      "enzymes. Nat. Chem. Biol.\n",
      "2014, 10, 50–55. 10.1038/nchembio.1385. [PubMed] [CrossRef] [Google Scholar]Quade N.; Huo L.; Rachid S.; Heinz D. W.; Müller R.\n",
      "Unusual carbon\n",
      "fixation gives rise to diverse polyketide extender units. Nat. Chem. Biol.\n",
      "2012, 8, 117–124. 10.1038/nchembio.734. [PubMed] [CrossRef] [Google Scholar]Zhang L.; et al. Rational Control of Polyketide Extender Units by Structure-Based\n",
      "Engineering of a Crotonyl-CoA Carboxylase/Reductase in Antimycin Biosynthesis. Angew. Chemie Int. Ed.\n",
      "2015, 54, 13462–13465. 10.1002/anie.201506899. [PubMed] [CrossRef] [Google Scholar]Stoffel G. M.; et al. Four amino acids define the CO2 binding pocket of enoyl-CoA\n",
      "carboxylases/reductases. Proc. Natl. Acad. Sci.\n",
      "U. S. A.\n",
      "2019, 116, 13964–9. 10.1073/pnas.1901471116. [PMC free article] [PubMed] [CrossRef] [Google Scholar]Rao S. T.; Rossmann M. G.\n",
      "Comparison\n",
      "of super-secondary structures in proteins. J.\n",
      "Mol. Biol.\n",
      "1973, 76, 241–56. 10.1016/0022-2836(73)90388-4. [PubMed] [CrossRef] [Google Scholar]Tono K.; et al. Diverse application platform for hard X-ray diffraction in SACLA\n",
      "(DAPHNIS): application to serial protein crystallography using an\n",
      "X-ray free-electron laser. J. Synchrotron Radiat.\n",
      "2015, 22, 532–537. 10.1107/S1600577515004464. [PMC free article] [PubMed] [CrossRef] [Google Scholar]Ch\n",
      "len(chunk):  ssRef] [Google Scholar]Chapman H. N.; et al. Femtosecond X-ray protein\n",
      "nanocrystallography. Nature\n",
      "2011, 470, 73–7. 10.1038/nature09750. [PMC free article] [PubMed] [CrossRef] [Google Scholar]Schlichting I.; Miao J.\n",
      "Emerging opportunities\n",
      "in structural biology with X-ray free-electron\n",
      "lasers. Curr. Opin. Struct. Biol.\n",
      "2012, 22, 613–26. 10.1016/j.sbi.2012.07.015. [PMC free article] [PubMed] [CrossRef] [Google Scholar]Helliwell J. R.\n",
      "How to\n",
      "solve protein structures with an X-ray laser. Science\n",
      "2013, 339, 146–7. 10.1126/science.1233209. [PubMed] [CrossRef] [Google Scholar]Bernhard S. A.; MacQuarrie R. A.\n",
      "Half-site reactivity and the”“”\n",
      "“induced-fit”“” hypothesis. J. Mol. Biol.\n",
      "1973, 74, 73–78. 10.1016/0022-2836(73)90356-2. [PubMed] [CrossRef] [Google Scholar]Seydoux F.; Malhotra O. P.; Bernhard S. A.; Stark G.\n",
      "Half-Site\n",
      "Reactivit. CRC Crit. Rev. Biochem.\n",
      "1974, 2, 227–257. 10.3109/10409237409105448. [PubMed] [CrossRef] [Google Scholar]Levitzki A.; Stallcup W. B.; Koshland D. E.\n",
      "Half-of-the-sites\n",
      "reactivity and\n",
      "conformational states of cytidine triphosphate synthetase. Biochemistry\n",
      "1971, 10, 3371–3378. 10.1021/bi00794a009. [PubMed] [CrossRef] [Google Scholar]Fersht A. R.; Mulvey R. S.; Koch G. L. E.\n",
      "Ligand binding and enzymic catalysis\n",
      "coupled through subunits in tyrosyl-tRNA synthetase. Biochemistry\n",
      "1975, 14, 13–18. 10.1021/bi00672a003. [PubMed] [CrossRef] [Google Scholar]Thompson J. D.; Higgins D. G.; Gibson T. J.\n",
      "CLUSTAL W: improving the sensitivity\n",
      "of progressive multiple sequence alignment through sequence weighting,\n",
      "position-specific gap penalties and weight matrix choice. Nucleic Acids Res.\n",
      "1994, 22 (22), 4673–4680. 10.1093/nar/22.22.4673. [PMC free article] [PubMed] [CrossRef] [Google Scholar]Hill T. L.; Levitzki A.\n",
      "Subunit neighbor interactions\n",
      "in enzyme kinetics: half-of-the-sites\n",
      "reactivity in a dimer. Proc. Natl. Acad. Sci.\n",
      "U. S. A.\n",
      "1980, 77, 5741–5745. 10.1073/pnas.77.10.5741. [PMC free article] [PubMed] [CrossRef] [Google Scholar]Levitzki A.; Koshland D. E.\n",
      "The Role of Negative Cooperativity and Half-of-the-Sites\n",
      "Reactivity in Enzyme Regulation. Current Topics\n",
      "in Cellular Regulation\n",
      "1976, 10, 1–40. 10.1016/B978-0-12-152810-2.50008-5. [PubMed] [CrossRef] [Google Scholar]Vivoli M.; Pang J.; Harmer N. J.\n",
      "A half-site\n",
      "multimeric enzyme achieves\n",
      "its cooperativity without conformational changes. Sci. Rep.\n",
      "2017, 7, 16529. 10.1038/s41598-017-16421-2. [PMC free article] [PubMed] [CrossRef] [Google Scholar]Danenberg K. D.; Danenberg P. V.\n",
      "Evidence\n",
      "for a sequential interaction of the subunits\n",
      "of thymidylate synthetase. J. Biol. Chem.\n",
      "1979, 254, 4345. 10.1016/S0021-9258(17)30013-3. [PubMed] [CrossRef] [Google Scholar]Anderson A. C.; O’Neil R. H.; DeLano W. L.; Stroud R. M.\n",
      "The Structural Mechanism\n",
      "for Half-the-Sites Reactivity in an Enzyme, Thymidylate Synthase,\n",
      "Involves a Relay of Changes between Subunits. Biochemistry\n",
      "1999, 38, 13829–13836. 10.1021/bi991610i. [PubMed] [CrossRef] [Google Scholar]Finer-Moore J. S.; Lee T. T.; Stroud R. M.\n",
      "A Single Mutation Traps a Half-Sites\n",
      "Reactive Enzyme in Midstream, Explaining Asymmetry in Hydride Transfer. Biochemistry\n",
      "2018, 57, 2786–2795. 10.1021/acs.biochem.8b00176. [PMC free article] [PubMed] [CrossRef] [Google Scholar]Hammes G. G.; Benkovic S. J.; Hammes-Schiffer S.\n",
      "Flexibility,\n",
      "Diversity, and Cooperativity:\n",
      "Pillars of Enzyme Catalysis. Biochemistry\n",
      "2011, 50, 10422–10430. 10.1021/bi201486f. [PMC free article] [PubMed] [CrossRef] [Google Scholar]Peter D. M. Doctoral Thesis. ETH Zurich: 2016. [Google Scholar]AbstractShort abstractIntroductionResults\n",
      "and DiscussionConclusionsMethodsAcknowledgmentsNotesSupporting Information AvailableAuthor ContributionsNotesSupplementary MaterialReferencesArticles from ACS Central Science are provided here courtesy of American Chemical SocietySchwander T.; Schada von Borzyskowski L.; Burgener S.; Cortina N. S.; Erb T. J.\n",
      "A synthetic pathway\n",
      "for the fixation of carbon dioxide\n",
      "in vitro. Science (80-.).\n",
      "2016, 354, 900–904. 10.1126/science.aah5237. [PMC free ar\n",
      "len(chunk):  nce.aah5237. [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list]Amao Y.\n",
      "Formate dehydrogenase\n",
      "for CO2 utilization and its application. J. of CO22 Utilization\n",
      "2018, 26, 623–641. 10.1016/j.jcou.2018.06.022. [CrossRef] [Google Scholar] [Ref list]Erb T. J.; et al. Synthesis of C5-dicarboxylic\n",
      "acids from C2-units involving crotonyl-CoA\n",
      "carboxylase/reductase: the ethylmalonyl-CoA pathway. Proc. Natl. Acad. Sci. U. S. A.\n",
      "2007, 104, 10631–6. 10.1073/pnas.0702791104. [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list]Erb T. J.; Brecht V.; Fuchs G.; Muller M.; Alber B. E.\n",
      "Carboxylation\n",
      "mechanism and stereochemistry of crotonyl-CoA carboxylase/reductase,\n",
      "a carboxylating enoyl-thioester reductase. Proc.\n",
      "Natl. Acad. Sci. U. S. A.\n",
      "2009, 106, 8871–8876. 10.1073/pnas.0903939106. [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list]Rosenthal R. G.; et al. Direct evidence for a covalent ene adduct intermediate\n",
      "in NAD(P)H-dependent\n",
      "enzymes. Nat. Chem. Biol.\n",
      "2014, 10, 50–55. 10.1038/nchembio.1385. [PubMed] [CrossRef] [Google Scholar] [Ref list]Quade N.; Huo L.; Rachid S.; Heinz D. W.; Müller R.\n",
      "Unusual carbon\n",
      "fixation gives rise to diverse polyketide extender units. Nat. Chem. Biol.\n",
      "2012, 8, 117–124. 10.1038/nchembio.734. [PubMed] [CrossRef] [Google Scholar] [Ref list]Zhang L.; et al. Rational Control of Polyketide Extender Units by Structure-Based\n",
      "Engineering of a Crotonyl-CoA Carboxylase/Reductase in Antimycin Biosynthesis. Angew. Chemie Int. Ed.\n",
      "2015, 54, 13462–13465. 10.1002/anie.201506899. [PubMed] [CrossRef] [Google Scholar] [Ref list]Stoffel G. M.; et al. Four amino acids define the CO2 binding pocket of enoyl-CoA\n",
      "carboxylases/reductases. Proc. Natl. Acad. Sci.\n",
      "U. S. A.\n",
      "2019, 116, 13964–9. 10.1073/pnas.1901471116. [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list]Rao S. T.; Rossmann M. G.\n",
      "Comparison\n",
      "of super-secondary structures in proteins. J.\n",
      "Mol. Biol.\n",
      "1973, 76, 241–56. 10.1016/0022-2836(73)90388-4. [PubMed] [CrossRef] [Google Scholar] [Ref list]Tono K.; et al. Diverse application platform for hard X-ray diffraction in SACLA\n",
      "(DAPHNIS): application to serial protein crystallography using an\n",
      "X-ray free-electron laser. J. Synchrotron Radiat.\n",
      "2015, 22, 532–537. 10.1107/S1600577515004464. [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list]Helliwell J. R.\n",
      "How to\n",
      "solve protein structures with an X-ray laser. Science\n",
      "2013, 339, 146–7. 10.1126/science.1233209. [PubMed] [CrossRef] [Google Scholar] [Ref list]Bernhard S. A.; MacQuarrie R. A.\n",
      "Half-site reactivity and the”“”\n",
      "“induced-fit”“” hypothesis. J. Mol. Biol.\n",
      "1973, 74, 73–78. 10.1016/0022-2836(73)90356-2. [PubMed] [CrossRef] [Google Scholar] [Ref list]Fersht A. R.; Mulvey R. S.; Koch G. L. E.\n",
      "Ligand binding and enzymic catalysis\n",
      "coupled through subunits in tyrosyl-tRNA synthetase. Biochemistry\n",
      "1975, 14, 13–18. 10.1021/bi00672a003. [PubMed] [CrossRef] [Google Scholar] [Ref list]Thompson J. D.; Higgins D. G.; Gibson T. J.\n",
      "CLUSTAL W: improving the sensitivity\n",
      "of progressive multiple sequence alignment through sequence weighting,\n",
      "position-specific gap penalties and weight matrix choice. Nucleic Acids Res.\n",
      "1994, 22 (22), 4673–4680. 10.1093/nar/22.22.4673. [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list]Levitzki A.; Stallcup W. B.; Koshland D. E.\n",
      "Half-of-the-sites\n",
      "reactivity and\n",
      "conformational states of cytidine triphosphate synthetase. Biochemistry\n",
      "1971, 10, 3371–3378. 10.1021/bi00794a009. [PubMed] [CrossRef] [Google Scholar] [Ref list]Hill T. L.; Levitzki A.\n",
      "Subunit neighbor interactions\n",
      "in enzyme kinetics: half-of-the-sites\n",
      "reactivity in a dimer. Proc. Natl. Acad. Sci.\n",
      "U. S. A.\n",
      "1980, 77, 5741–5745. 10.1073/pnas.77.10.5741. [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list]Levitzki A.; Koshland D. E.\n",
      "The Role of Negative Cooperativity and Half-of-the-Sites\n",
      "Reactivity in Enzyme Regulation. Current Topics\n",
      "in Cellular Regulation\n",
      "1976, 10, 1–40. 10.1016/B978-0-12-152810-2.50008-5. [PubMed] [Cros\n",
      "len(chunk):  2.50008-5. [PubMed] [CrossRef] [Google Scholar] [Ref list]Vivoli M.; Pang J.; Harmer N. J.\n",
      "A half-site\n",
      "multimeric enzyme achieves\n",
      "its cooperativity without conformational changes. Sci. Rep.\n",
      "2017, 7, 16529. 10.1038/s41598-017-16421-2. [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list]Danenberg K. D.; Danenberg P. V.\n",
      "Evidence\n",
      "for a sequential interaction of the subunits\n",
      "of thymidylate synthetase. J. Biol. Chem.\n",
      "1979, 254, 4345. 10.1016/S0021-9258(17)30013-3. [PubMed] [CrossRef] [Google Scholar] [Ref list]Anderson A. C.; O’Neil R. H.; DeLano W. L.; Stroud R. M.\n",
      "The Structural Mechanism\n",
      "for Half-the-Sites Reactivity in an Enzyme, Thymidylate Synthase,\n",
      "Involves a Relay of Changes between Subunits. Biochemistry\n",
      "1999, 38, 13829–13836. 10.1021/bi991610i. [PubMed] [CrossRef] [Google Scholar] [Ref list]Finer-Moore J. S.; Lee T. T.; Stroud R. M.\n",
      "A Single Mutation Traps a Half-Sites\n",
      "Reactive Enzyme in Midstream, Explaining Asymmetry in Hydride Transfer. Biochemistry\n",
      "2018, 57, 2786–2795. 10.1021/acs.biochem.8b00176. [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list]Hammes G. G.; Benkovic S. J.; Hammes-Schiffer S.\n",
      "Flexibility,\n",
      "Diversity, and Cooperativity:\n",
      "Pillars of Enzyme Catalysis. Biochemistry\n",
      "2011, 50, 10422–10430. 10.1021/bi201486f. [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list]Peter D. M. Doctoral Thesis. ETH Zurich: 2016. [Google Scholar] [Ref list]\n",
      "Other Formats\n",
      "PubReader\n",
      "PDF (3.1M)\n",
      "Actions\n",
      "Cite\n",
      "Collections\n",
      "Add to Collections\n",
      "Create a new collection\n",
      "Add to an existing collection\n",
      "Name your collection:\n",
      "Name must be less than\n",
      "characters\n",
      "Choose a collection:\n",
      "Unable to load your collection due to an error\n",
      "Please try again\n",
      "Add\n",
      "Cancel\n",
      "Share\n",
      "Permalink\n",
      "Copy\n",
      "RESOURCES\n",
      "Similar articles\n",
      "Cited by other articles\n",
      "Links to NCBI Databases\n",
      "[x]\n",
      "Cite\n",
      "Copy\n",
      "Download .nbib\n",
      ".nbib\n",
      "Format:\n",
      "AMA\n",
      "APA\n",
      "MLA\n",
      "NLM\n",
      "External link. Please review our privacy policy.\n",
      "Chunk: 0 / 18\n",
      "seed_initial_question_decomposition_prompt \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract and output as many accurate direct quotes from the text that are relevant to answering the key questions and its most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"1\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Text: Back to Top\n",
      "Skip to main content\n",
      "The .gov means it’s official.\n",
      "Federal government websites often end in .gov or .mil. Before\n",
      "sharing sensitive information, make sure you’re on a federal\n",
      "government site.\n",
      "The site is secure.\n",
      "The https:// ensures that you are connecting to the\n",
      "official website and that any information you provide is encrypted\n",
      "and transmitted securely.\n",
      "Access keys\n",
      "NCBI Homepage\n",
      "MyNCBI Homepage\n",
      "Main Content\n",
      "Main Navigation\n",
      "Journal List\n",
      "ACS Cent Sci\n",
      "v.8(8); 2022 Aug 24\n",
      "PMC9413435\n",
      "As a library, NLM provides access to scientific literature. Inclusion in an NLM database does not imply endorsement of, or agreement with,\n",
      "the contents by NLM or the National Institutes of Health.\n",
      "Learn more about our disclaimer.\n",
      "ACS Cent Sci. 2022 Aug 24; 8(8): 1091–1101. Published online 2022 Apr 25.\n",
      "doi: 10.1021/acscentsci.2c00057PMCID: PMC9413435PMID: 36032767Intersubunit Coupling Enables Fast CO2-Fixation\n",
      "by Reductive CarboxylasesHasan DeMirci,*†‡§▼ Yashas Rao,†⊥▼ Gabriele M. Stoffel,∥▼ Bastian Vögeli,∥▼ Kristina Schell,∥▼ Aharon Gomez,⊥▼ Alexander Batyuk,# Cornelius Gati,†∇ Raymond G. Sierra,# Mark S. Hunter,# E. Han Dao,†‡ Halil I. Ciftci,‡ Brandon Hayes,# Fredric Poitevin,# Po-Nan Li,†○ Manat Kaur,∇ Kensuke Tono,●□ David Adrian Saez,⊥■ Samuel Deutsch,△ Yasuo Yoshikuni,△ Helmut Grubmüller,▲ Tobias J. Erb,*∥▽ Esteban Vöhringer-Martinez,*⊥ and\n",
      "Soichi Wakatsuki*†∇Hasan DeMirci†Biosciences\n",
      "Division, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States‡PULSE\n",
      "Institute, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States§Department\n",
      "of Molecular Biology and Genetics, Koc University, 34450 Sariyer/Istanbul, TurkeyFind articles by Hasan DeMirciYashas Rao†Biosciences\n",
      "Division, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States⊥Departamento\n",
      "de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, ChileFind articles by Yashas RaoGabriele M. Stoffel∥Department\n",
      "of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, GermanyFind articles by Gabriele M. StoffelBastian Vögeli∥Department\n",
      "of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, GermanyFind articles by Bastian VögeliKristina Schell∥Department\n",
      "of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, GermanyFind articles by Kristina SchellAharon Gomez⊥Departamento\n",
      "de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, ChileFind articles by Aharon GomezAlexander Batyuk#Linac Coherent\n",
      "Light Source, SLAC National Accelerator\n",
      "Laboratory,\n",
      "Menlo Park, California 94025, United StatesFind articles by Alexander BatyukCornelius Gati†Biosciences\n",
      "Division, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States∇Structural\n",
      "Biology Department, Stanford University,\n",
      "Stanford, California 94305, United StatesFind articles by Cornelius GatiRaymond G. Sierra#Linac Coherent\n",
      "Light Source, SLAC National Accelerator\n",
      "Laboratory,\n",
      "Menlo Park, California 94025, United StatesFind articles by Raymond G. SierraMark S. Hunter#Linac Coherent\n",
      "Light Source, SLAC National Accelerator\n",
      "Laboratory,\n",
      "Menlo Park, California 94025, United StatesFind articles by Mark S. HunterE. Han Dao†Biosciences\n",
      "Division, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States‡PULSE\n",
      "Institute, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United StatesFind articles by E. Han DaoHalil I. Ciftci‡PULSE\n",
      "Institute, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United StatesFind articles by Halil I. CiftciBrandon Hayes#Linac Coherent\n",
      "Light Source, SLAC National Accelerator\n",
      "Laboratory,\n",
      "Menlo Park, California 94025, United StatesFind articles by Brandon \n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Prompt:  \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract and output as many accurate direct quotes from the text that are relevant to answering the key questions and its most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"1\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Text: Back to Top\n",
      "Skip to main content\n",
      "The .gov means it’s official.\n",
      "Federal government websites often end in .gov or .mil. Before\n",
      "sharing sensitive information, make sure you’re on a federal\n",
      "government site.\n",
      "The site is secure.\n",
      "The https:// ensures that you are connecting to the\n",
      "official website and that any information you provide is encrypted\n",
      "and transmitted securely.\n",
      "Access keys\n",
      "NCBI Homepage\n",
      "MyNCBI Homepage\n",
      "Main Content\n",
      "Main Navigation\n",
      "Journal List\n",
      "ACS Cent Sci\n",
      "v.8(8); 2022 Aug 24\n",
      "PMC9413435\n",
      "As a library, NLM provides access to scientific literature. Inclusion in an NLM database does not imply endorsement of, or agreement with,\n",
      "the contents by NLM or the National Institutes of Health.\n",
      "Learn more about our disclaimer.\n",
      "ACS Cent Sci. 2022 Aug 24; 8(8): 1091–1101. Published online 2022 Apr 25.\n",
      "doi: 10.1021/acscentsci.2c00057PMCID: PMC9413435PMID: 36032767Intersubunit Coupling Enables Fast CO2-Fixation\n",
      "by Reductive CarboxylasesHasan DeMirci,*†‡§▼ Yashas Rao,†⊥▼ Gabriele M. Stoffel,∥▼ Bastian Vögeli,∥▼ Kristina Schell,∥▼ Aharon Gomez,⊥▼ Alexander Batyuk,# Cornelius Gati,†∇ Raymond G. Sierra,# Mark S. Hunter,# E. Han Dao,†‡ Halil I. Ciftci,‡ Brandon Hayes,# Fredric Poitevin,# Po-Nan Li,†○ Manat Kaur,∇ Kensuke Tono,●□ David Adrian Saez,⊥■ Samuel Deutsch,△ Yasuo Yoshikuni,△ Helmut Grubmüller,▲ Tobias J. Erb,*∥▽ Esteban Vöhringer-Martinez,*⊥ and\n",
      "Soichi Wakatsuki*†∇Hasan DeMirci†Biosciences\n",
      "Division, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States‡PULSE\n",
      "Institute, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States§Department\n",
      "of Molecular Biology and Genetics, Koc University, 34450 Sariyer/Istanbul, TurkeyFind articles by Hasan DeMirciYashas Rao†Biosciences\n",
      "Division, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States⊥Departamento\n",
      "de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, ChileFind articles by Yashas RaoGabriele M. Stoffel∥Department\n",
      "of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, GermanyFind articles by Gabriele M. StoffelBastian Vögeli∥Department\n",
      "of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, GermanyFind articles by Bastian VögeliKristina Schell∥Department\n",
      "of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, GermanyFind articles by Kristina SchellAharon Gomez⊥Departamento\n",
      "de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, ChileFind articles by Aharon GomezAlexander Batyuk#Linac Coherent\n",
      "Light Source, SLAC National Accelerator\n",
      "Laboratory,\n",
      "Menlo Park, California 94025, United StatesFind articles by Alexander BatyukCornelius Gati†Biosciences\n",
      "Division, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States∇Structural\n",
      "Biology Department, Stanford University,\n",
      "Stanford, California 94305, United StatesFind articles by Cornelius GatiRaymond G. Sierra#Linac Coherent\n",
      "Light Source, SLAC National Accelerator\n",
      "Laboratory,\n",
      "Menlo Park, California 94025, United StatesFind articles by Raymond G. SierraMark S. Hunter#Linac Coherent\n",
      "Light Source, SLAC National Accelerator\n",
      "Laboratory,\n",
      "Menlo Park, California 94025, United StatesFind articles by Mark S. HunterE. Han Dao†Biosciences\n",
      "Division, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States‡PULSE\n",
      "Institute, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United StatesFind articles by E. Han DaoHalil I. Ciftci‡PULSE\n",
      "Institute, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United StatesFind articles by Halil I. CiftciBrandon Hayes#Linac Coherent\n",
      "Light Source, SLAC National Accelerator\n",
      "Laboratory,\n",
      "Menlo Park, California 94025, United StatesFind articles by Brandon \n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"0\\\": {\\n    \\\"quote\\\": \\\"Intersubunit Coupling Enables Fast CO2-Fixation by Reductive Carboxylases\\\",\\n    \\\"index\\\": 1\\n  },\\n  \\\"1\\\": {\\n    \\\"quote\\\": \\\"Hasan DeMirci\\u2020Biosciences Division, SLAC National Accelerator Laboratory, Menlo Park, California 94025, United States\\u2021PULSE Institute, SLAC National Accelerator Laboratory, Menlo Park, California 94025, United States\\u00a7Department of Molecular Biology and Genetics, Koc University, 34450 Sariyer/Istanbul, TurkeyFind articles by Hasan DeMirci\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"2\\\": {\\n    \\\"quote\\\": \\\"Yashas Rao\\u2020Biosciences Division, SLAC National Accelerator Laboratory, Menlo Park, California 94025, United States\\u22a5Departamento de F\\u00edsico Qu\\u00edmica, Facultad de Ciencias Qu\\u00edmicas, Universidad de Concepci\\u00f3n, Concepci\\u00f3n 4030000, ChileFind articles by Yashas Rao\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"3\\\": {\\n    \\\"quote\\\": \\\"Gabriele M. Stoffel\\u2225Department of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Stra\\u00dfe 10, D-35043 Marburg, GermanyFind articles by Gabriele M. Stoffel\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"4\\\": {\\n    \\\"quote\\\": \\\"Bastian V\\u00f6geli\\u2225Department of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Stra\\u00dfe 10, D-35043 Marburg, GermanyFind articles by Bastian V\\u00f6geli\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"5\\\": {\\n    \\\"quote\\\": \\\"Kristina Schell\\u2225Department of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Stra\\u00dfe 10, D-35043 Marburg, GermanyFind articles by Kristina Schell\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"6\\\": {\\n    \\\"quote\\\": \\\"Aharon Gomez\\u22a5Departamento de F\\u00edsico Qu\\u00edmica, Facultad de Ciencias Qu\\u00edmicas, Universidad de Concepci\\u00f3n, Concepci\\u00f3n 4030000, ChileFind articles by Aharon Gomez\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"7\\\": {\\n    \\\"quote\\\": \\\"Alexander Batyuk#Linac Coherent Light Source, SLAC National Accelerator Laboratory, Menlo Park, California 94025, United StatesFind articles by Alexander Batyuk\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"8\\\": {\\n    \\\"quote\\\": \\\"Cornelius Gati\\u2020Biosciences Division, SLAC National Accelerator Laboratory, Menlo Park, California 94025, United States\\u2207Structural Biology Department, Stanford University, Stanford, California 94305, United StatesFind articles by Cornelius Gati\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"9\\\": {\\n    \\\"quote\\\": \\\"Raymond G. Sierra#Linac Coherent Light Source, SLAC National Accelerator Laboratory, Menlo Park, California 94025, United StatesFind articles by Raymond G. Sierra\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"10\\\": {\\n    \\\"quote\\\": \\\"Mark S. Hunter#Linac Coherent Light Source, SLAC National Accelerator Laboratory, Menlo Park, California 94025, United StatesFind articles by Mark S. Hunter\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"11\\\": {\\n    \\\"quote\\\": \\\"E. Han Dao\\u2020Biosciences Division, SLAC National Accelerator Laboratory, Menlo Park, California 94025, United States\\u2021PULSE Institute, SLAC National Accelerator Laboratory, Menlo Park, California 94025, United StatesFind articles by E. Han Dao\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"12\\\": {\\n    \\\"quote\\\": \\\"Halil I. Ciftci\\u2021PULSE Institute, SLAC National Accelerator Laboratory, Menlo Park, California 94025, United StatesFind articles by Halil I. Ciftci\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"13\\\": {\\n    \\\"quote\\\": \\\"Brandon Hayes#Linac Coherent Light Source, SLAC National Accelerator Laboratory, Menlo Park, California 94025, United StatesFind articles by Brandon\\\",\\n    \\\"index\\\": 0\\n  }\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691117595,\n",
      "  \"id\": \"chatcmpl-7jfEBH4Qrg3QmQxL89IXNQWTRdUlw\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 975,\n",
      "    \"prompt_tokens\": 1362,\n",
      "    \"total_tokens\": 2337\n",
      "  }\n",
      "}\n",
      "Extracted quotes:  {\n",
      "  \"0\": {\n",
      "    \"quote\": \"Intersubunit Coupling Enables Fast CO2-Fixation by Reductive Carboxylases\",\n",
      "    \"index\": 1\n",
      "  },\n",
      "  \"1\": {\n",
      "    \"quote\": \"Hasan DeMirci†Biosciences Division, SLAC National Accelerator Laboratory, Menlo Park, California 94025, United States‡PULSE Institute, SLAC National Accelerator Laboratory, Menlo Park, California 94025, United States§Department of Molecular Biology and Genetics, Koc University, 34450 Sariyer/Istanbul, TurkeyFind articles by Hasan DeMirci\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"2\": {\n",
      "    \"quote\": \"Yashas Rao†Biosciences Division, SLAC National Accelerator Laboratory, Menlo Park, California 94025, United States⊥Departamento de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, ChileFind articles by Yashas Rao\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"3\": {\n",
      "    \"quote\": \"Gabriele M. Stoffel∥Department of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, GermanyFind articles by Gabriele M. Stoffel\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"4\": {\n",
      "    \"quote\": \"Bastian Vögeli∥Department of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, GermanyFind articles by Bastian Vögeli\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"5\": {\n",
      "    \"quote\": \"Kristina Schell∥Department of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, GermanyFind articles by Kristina Schell\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"6\": {\n",
      "    \"quote\": \"Aharon Gomez⊥Departamento de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, ChileFind articles by Aharon Gomez\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"7\": {\n",
      "    \"quote\": \"Alexander Batyuk#Linac Coherent Light Source, SLAC National Accelerator Laboratory, Menlo Park, California 94025, United StatesFind articles by Alexander Batyuk\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"8\": {\n",
      "    \"quote\": \"Cornelius Gati†Biosciences Division, SLAC National Accelerator Laboratory, Menlo Park, California 94025, United States∇Structural Biology Department, Stanford University, Stanford, California 94305, United StatesFind articles by Cornelius Gati\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"9\": {\n",
      "    \"quote\": \"Raymond G. Sierra#Linac Coherent Light Source, SLAC National Accelerator Laboratory, Menlo Park, California 94025, United StatesFind articles by Raymond G. Sierra\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"10\": {\n",
      "    \"quote\": \"Mark S. Hunter#Linac Coherent Light Source, SLAC National Accelerator Laboratory, Menlo Park, California 94025, United StatesFind articles by Mark S. Hunter\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"11\": {\n",
      "    \"quote\": \"E. Han Dao†Biosciences Division, SLAC National Accelerator Laboratory, Menlo Park, California 94025, United States‡PULSE Institute, SLAC National Accelerator Laboratory, Menlo Park, California 94025, United StatesFind articles by E. Han Dao\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"12\": {\n",
      "    \"quote\": \"Halil I. Ciftci‡PULSE Institute, SLAC National Accelerator Laboratory, Menlo Park, California 94025, United StatesFind articles by Halil I. Ciftci\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"13\": {\n",
      "    \"quote\": \"Brandon Hayes#Linac Coherent Light Source, SLAC National Accelerator Laboratory, Menlo Park, California 94025, United StatesFind articles by Brandon\",\n",
      "    \"index\": 0\n",
      "  }\n",
      "}\n",
      "Chunk: 1 / 18\n",
      "seed_initial_question_decomposition_prompt \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract and output as many accurate direct quotes from the text that are relevant to answering the key questions and its most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"1\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Text: Find articles by Brandon HayesFredric Poitevin#Linac Coherent\n",
      "Light Source, SLAC National Accelerator\n",
      "Laboratory,\n",
      "Menlo Park, California 94025, United StatesFind articles by Fredric PoitevinPo-Nan Li†Biosciences\n",
      "Division, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States○Electrical\n",
      "Engineering Department, Stanford University,\n",
      "Stanford, California 94305, United StatesFind articles by Po-Nan LiManat Kaur∇Structural\n",
      "Biology Department, Stanford University,\n",
      "Stanford, California 94305, United StatesFind articles by Manat KaurKensuke Tono●RIKEN\n",
      "SPring-8 Center, Sayo, Hyogo 679-5148, Japan□Japan Synchrotron Radiation Research Institute, Sayo, Hyogo 679-5198, JapanFind articles by Kensuke TonoDavid Adrian Saez⊥Departamento\n",
      "de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, Chile■Departamento\n",
      "de Farmacia, Facultad de Farmacia, Universidad\n",
      "de Concepción, Concepción 00000, ChileFind articles by David Adrian SaezSamuel Deutsch△U.S.\n",
      "Department of Energy Joint Genome Institute, Lawrence Berkeley National Laboratory, Walnut Creek, California 94720, United StatesFind articles by Samuel DeutschYasuo Yoshikuni△U.S.\n",
      "Department of Energy Joint Genome Institute, Lawrence Berkeley National Laboratory, Walnut Creek, California 94720, United StatesFind articles by Yasuo YoshikuniHelmut Grubmüller▲Department\n",
      "of Theoretical and Computational Biophysics, Max Planck Institute for Multidisciplinary Sciences, Am Fassberg 11, 37077 Göttingen, GermanyFind articles by Helmut GrubmüllerTobias J. Erb∥Department\n",
      "of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, Germany▽LOEWE Center for Synthetic Microbiology (SYNMIKRO), 35032 Marburg, GermanyFind articles by Tobias J. ErbEsteban Vöhringer-Martinez⊥Departamento\n",
      "de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, ChileFind articles by Esteban Vöhringer-MartinezSoichi Wakatsuki†Biosciences\n",
      "Division, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States∇Structural\n",
      "Biology Department, Stanford University,\n",
      "Stanford, California 94305, United StatesFind articles by Soichi WakatsukiAuthor information Copyright and License information PMC Disclaimer†Biosciences\n",
      "Division, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States‡PULSE\n",
      "Institute, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States§Department\n",
      "of Molecular Biology and Genetics, Koc University, 34450 Sariyer/Istanbul, Turkey∥Department\n",
      "of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, Germany⊥Departamento\n",
      "de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, Chile#Linac Coherent\n",
      "Light Source, SLAC National Accelerator\n",
      "Laboratory,\n",
      "Menlo Park, California 94025, United States∇Structural\n",
      "Biology Department, Stanford University,\n",
      "Stanford, California 94305, United States○Electrical\n",
      "Engineering Department, Stanford University,\n",
      "Stanford, California 94305, United States●RIKEN\n",
      "SPring-8 Center, Sayo, Hyogo 679-5148, Japan□Japan Synchrotron Radiation Research Institute, Sayo, Hyogo 679-5198, Japan■Departamento\n",
      "de Farmacia, Facultad de Farmacia, Universidad\n",
      "de Concepción, Concepción 00000, Chile△U.S.\n",
      "Department of Energy Joint Genome Institute, Lawrence Berkeley National Laboratory, Walnut Creek, California 94720, United States▲Department\n",
      "of Theoretical and Computational Biophysics, Max Planck Institute for Multidisciplinary Sciences, Am Fassberg 11, 37077 Göttingen, Germany▽LOEWE Center for Synthetic Microbiology (SYNMIKRO), 35032 Marburg, GermanyCorresponding author.*Email for H.D.: rt.ude.uk@icrimedh.*Email for T.J.E.: ed.gpm.grubram-ipm@breot.*Email for E.V.-M.: lc.cedu@regnirhove.*Email for S.W.: ude.drofnats@ikustakaw.ihcios.Copyright © 2022 The Authors\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Prompt:  \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract and output as many accurate direct quotes from the text that are relevant to answering the key questions and its most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"1\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Text: Find articles by Brandon HayesFredric Poitevin#Linac Coherent\n",
      "Light Source, SLAC National Accelerator\n",
      "Laboratory,\n",
      "Menlo Park, California 94025, United StatesFind articles by Fredric PoitevinPo-Nan Li†Biosciences\n",
      "Division, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States○Electrical\n",
      "Engineering Department, Stanford University,\n",
      "Stanford, California 94305, United StatesFind articles by Po-Nan LiManat Kaur∇Structural\n",
      "Biology Department, Stanford University,\n",
      "Stanford, California 94305, United StatesFind articles by Manat KaurKensuke Tono●RIKEN\n",
      "SPring-8 Center, Sayo, Hyogo 679-5148, Japan□Japan Synchrotron Radiation Research Institute, Sayo, Hyogo 679-5198, JapanFind articles by Kensuke TonoDavid Adrian Saez⊥Departamento\n",
      "de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, Chile■Departamento\n",
      "de Farmacia, Facultad de Farmacia, Universidad\n",
      "de Concepción, Concepción 00000, ChileFind articles by David Adrian SaezSamuel Deutsch△U.S.\n",
      "Department of Energy Joint Genome Institute, Lawrence Berkeley National Laboratory, Walnut Creek, California 94720, United StatesFind articles by Samuel DeutschYasuo Yoshikuni△U.S.\n",
      "Department of Energy Joint Genome Institute, Lawrence Berkeley National Laboratory, Walnut Creek, California 94720, United StatesFind articles by Yasuo YoshikuniHelmut Grubmüller▲Department\n",
      "of Theoretical and Computational Biophysics, Max Planck Institute for Multidisciplinary Sciences, Am Fassberg 11, 37077 Göttingen, GermanyFind articles by Helmut GrubmüllerTobias J. Erb∥Department\n",
      "of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, Germany▽LOEWE Center for Synthetic Microbiology (SYNMIKRO), 35032 Marburg, GermanyFind articles by Tobias J. ErbEsteban Vöhringer-Martinez⊥Departamento\n",
      "de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, ChileFind articles by Esteban Vöhringer-MartinezSoichi Wakatsuki†Biosciences\n",
      "Division, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States∇Structural\n",
      "Biology Department, Stanford University,\n",
      "Stanford, California 94305, United StatesFind articles by Soichi WakatsukiAuthor information Copyright and License information PMC Disclaimer†Biosciences\n",
      "Division, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States‡PULSE\n",
      "Institute, SLAC National Accelerator Laboratory,\n",
      "Menlo Park, California 94025, United States§Department\n",
      "of Molecular Biology and Genetics, Koc University, 34450 Sariyer/Istanbul, Turkey∥Department\n",
      "of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, Germany⊥Departamento\n",
      "de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, Chile#Linac Coherent\n",
      "Light Source, SLAC National Accelerator\n",
      "Laboratory,\n",
      "Menlo Park, California 94025, United States∇Structural\n",
      "Biology Department, Stanford University,\n",
      "Stanford, California 94305, United States○Electrical\n",
      "Engineering Department, Stanford University,\n",
      "Stanford, California 94305, United States●RIKEN\n",
      "SPring-8 Center, Sayo, Hyogo 679-5148, Japan□Japan Synchrotron Radiation Research Institute, Sayo, Hyogo 679-5198, Japan■Departamento\n",
      "de Farmacia, Facultad de Farmacia, Universidad\n",
      "de Concepción, Concepción 00000, Chile△U.S.\n",
      "Department of Energy Joint Genome Institute, Lawrence Berkeley National Laboratory, Walnut Creek, California 94720, United States▲Department\n",
      "of Theoretical and Computational Biophysics, Max Planck Institute for Multidisciplinary Sciences, Am Fassberg 11, 37077 Göttingen, Germany▽LOEWE Center for Synthetic Microbiology (SYNMIKRO), 35032 Marburg, GermanyCorresponding author.*Email for H.D.: rt.ude.uk@icrimedh.*Email for T.J.E.: ed.gpm.grubram-ipm@breot.*Email for E.V.-M.: lc.cedu@regnirhove.*Email for S.W.: ude.drofnats@ikustakaw.ihcios.Copyright © 2022 The Authors\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"0\\\": {\\n    \\\"quote\\\": \\\"\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"1\\\": {\\n    \\\"quote\\\": \\\"\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"2\\\": {\\n    \\\"quote\\\": \\\"\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"3\\\": {\\n    \\\"quote\\\": \\\"\\\",\\n    \\\"index\\\": 0\\n  }\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691117626,\n",
      "  \"id\": \"chatcmpl-7jfEgdww3nvAEwWl5DeNCU7QdQl0I\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 78,\n",
      "    \"prompt_tokens\": 1293,\n",
      "    \"total_tokens\": 1371\n",
      "  }\n",
      "}\n",
      "Extracted quotes:  {\n",
      "  \"0\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"1\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"2\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"3\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  }\n",
      "}\n",
      "Chunk: 2 / 18\n",
      "seed_initial_question_decomposition_prompt \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract and output as many accurate direct quotes from the text that are relevant to answering the key questions and its most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"1\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Text: yright © 2022 The Authors. Published by American Chemical SocietyPermits non-commercial access and re-use, provided that author attribution and integrity are maintained; but does not permit creation of adaptations or other derivative works (https://creativecommons.org/licenses/by-nc-nd/4.0/).Associated DataSupplementary Materialsoc2c00057_si_001.pdfoc2c00057_si_001.pdf (17M)GUID: 3DAFD832-2754-4387-9475-06B0D4C1EC12oc2c00057_si_002.movoc2c00057_si_002.mov (9.0M)GUID: BE8629BC-6773-425C-97E3-CA37E22A1673oc2c00057_si_003.movoc2c00057_si_003.mov (8.2M)GUID: A7E8C0A4-AE12-4912-AC31-AB45D940046Boc2c00057_si_004.movoc2c00057_si_004.mov (9.0M)GUID: 43D17EE6-B4F7-47EA-B452-11A1274C2A28oc2c00057_si_005.movoc2c00057_si_005.mov (8.1M)GUID: 650EFC78-525F-40C6-84C0-68935E145692oc2c00057_si_006.movoc2c00057_si_006.mov (8.1M)GUID: F8DC6B1B-9299-40EB-821F-D8239E059778oc2c00057_si_007.movoc2c00057_si_007.mov (8.2M)GUID: 9006AB25-3285-4E55-ADD4-67300E3D6A1Coc2c00057_si_008.movoc2c00057_si_008.mov (8.2M)GUID: 633ED857-A1B2-449F-8BC6-4BF29F625C37oc2c00057_si_009.movoc2c00057_si_009.mov (8.2M)GUID: 08FFAF3F-F1C5-4C12-92CE-68628A714916oc2c00057_si_010.movoc2c00057_si_010.mov (6.4M)GUID: C9DB1821-A4B8-4664-AF89-09878ED6B3CEoc2c00057_si_011.movoc2c00057_si_011.mov (8.2M)GUID: 91B081EB-30FD-4129-9F0E-4CB086CEB2EDoc2c00057_si_012.movoc2c00057_si_012.mov (8.7M)GUID: 8FE88134-6B28-4058-A773-085B05F0354Eoc2c00057_si_013.movoc2c00057_si_013.mov (8.7M)GUID: 7BE15129-0BCA-48B0-921C-21552979FFFBoc2c00057_si_014.pdfoc2c00057_si_014.pdf (242K)GUID: 00BA5186-3ABB-48EB-89B5-FCD4099EA78CGo to:AbstractEnoyl-CoA carboxylases/reductases\n",
      "(ECRs) are some of the most efficient\n",
      "CO2-fixing enzymes described to date. However, the molecular\n",
      "mechanisms underlying the extraordinary catalytic activity of ECRs\n",
      "on the level of the protein assembly remain elusive. Here we used\n",
      "a combination of ambient-temperature X-ray free electron laser (XFEL)\n",
      "and cryogenic synchrotron experiments to study the structural organization\n",
      "of the ECR from Kitasatospora setae. The K. setae ECR is a homotetramer\n",
      "that differentiates into a pair of dimers of open- and closed-form\n",
      "subunits in the catalytically active state. Using molecular dynamics\n",
      "simulations and structure-based mutagenesis, we show that catalysis\n",
      "is synchronized in the K. setae ECR\n",
      "across the pair of dimers. This conformational coupling of catalytic\n",
      "domains is conferred by individual amino acids to achieve high CO2-fixation rates. Our results provide unprecedented insights\n",
      "into the dynamic organization and synchronized inter- and intrasubunit\n",
      "communications of this remarkably efficient CO2-fixing\n",
      "enzyme during catalysis.Go to:Short abstractIntersubunit\n",
      "coupling drives fast carboxylation by enoyl-CoA\n",
      "carboxylases/reductases, providing the basis for engineering highly\n",
      "efficient CO2-fixing enzymes for bioenergy and bioproduct\n",
      "applications.Go to:IntroductionThe\n",
      "capture and conversion of atmospheric CO2 remains\n",
      "a challenging task for chemistry, resulting in an ever-increasing\n",
      "interest to understand and exploit CO2 fixation mechanisms\n",
      "offered by biology. Nature has evolved different enzymes capable of\n",
      "capturing and/or converting CO2, including biotin-dependent\n",
      "CoA carboxylases, which incorporate CO2 into different\n",
      "alkyl-CoA esters, formate dehydrogenases, which directly reduce CO2 into the single-carbon compound formate, and RubisCO, the\n",
      "central enzyme in the Calvin–Benson–Bassham cycle of\n",
      "photosynthesis that produces two molecules of 3-phosphoglycerate from\n",
      "ribulose-1,5-bisphosphate and CO2.1−4 The recently discovered family\n",
      "of enoyl-CoA carboxylases/reductases (ECRs) encompasses some of the\n",
      "most efficient CO2-fixing enzymes found in nature to date.1 With respect to the catalytic rate, for instance,\n",
      "ECR family members from primary metabolism outcompete RubisCO by more\n",
      "than 1 order of magnitude, which makes them exquisite models to study\n",
      "the molecular basis of efficient CO2 catalysis.2,5ECRs catalyze the reduction of various α,β-unsaturated\n",
      "e\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Prompt:  \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity compare to other bacteria?\n",
      "2. What factors influence ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "3. What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract and output as many accurate direct quotes from the text that are relevant to answering the key questions and its most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"1\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Text: yright © 2022 The Authors. Published by American Chemical SocietyPermits non-commercial access and re-use, provided that author attribution and integrity are maintained; but does not permit creation of adaptations or other derivative works (https://creativecommons.org/licenses/by-nc-nd/4.0/).Associated DataSupplementary Materialsoc2c00057_si_001.pdfoc2c00057_si_001.pdf (17M)GUID: 3DAFD832-2754-4387-9475-06B0D4C1EC12oc2c00057_si_002.movoc2c00057_si_002.mov (9.0M)GUID: BE8629BC-6773-425C-97E3-CA37E22A1673oc2c00057_si_003.movoc2c00057_si_003.mov (8.2M)GUID: A7E8C0A4-AE12-4912-AC31-AB45D940046Boc2c00057_si_004.movoc2c00057_si_004.mov (9.0M)GUID: 43D17EE6-B4F7-47EA-B452-11A1274C2A28oc2c00057_si_005.movoc2c00057_si_005.mov (8.1M)GUID: 650EFC78-525F-40C6-84C0-68935E145692oc2c00057_si_006.movoc2c00057_si_006.mov (8.1M)GUID: F8DC6B1B-9299-40EB-821F-D8239E059778oc2c00057_si_007.movoc2c00057_si_007.mov (8.2M)GUID: 9006AB25-3285-4E55-ADD4-67300E3D6A1Coc2c00057_si_008.movoc2c00057_si_008.mov (8.2M)GUID: 633ED857-A1B2-449F-8BC6-4BF29F625C37oc2c00057_si_009.movoc2c00057_si_009.mov (8.2M)GUID: 08FFAF3F-F1C5-4C12-92CE-68628A714916oc2c00057_si_010.movoc2c00057_si_010.mov (6.4M)GUID: C9DB1821-A4B8-4664-AF89-09878ED6B3CEoc2c00057_si_011.movoc2c00057_si_011.mov (8.2M)GUID: 91B081EB-30FD-4129-9F0E-4CB086CEB2EDoc2c00057_si_012.movoc2c00057_si_012.mov (8.7M)GUID: 8FE88134-6B28-4058-A773-085B05F0354Eoc2c00057_si_013.movoc2c00057_si_013.mov (8.7M)GUID: 7BE15129-0BCA-48B0-921C-21552979FFFBoc2c00057_si_014.pdfoc2c00057_si_014.pdf (242K)GUID: 00BA5186-3ABB-48EB-89B5-FCD4099EA78CGo to:AbstractEnoyl-CoA carboxylases/reductases\n",
      "(ECRs) are some of the most efficient\n",
      "CO2-fixing enzymes described to date. However, the molecular\n",
      "mechanisms underlying the extraordinary catalytic activity of ECRs\n",
      "on the level of the protein assembly remain elusive. Here we used\n",
      "a combination of ambient-temperature X-ray free electron laser (XFEL)\n",
      "and cryogenic synchrotron experiments to study the structural organization\n",
      "of the ECR from Kitasatospora setae. The K. setae ECR is a homotetramer\n",
      "that differentiates into a pair of dimers of open- and closed-form\n",
      "subunits in the catalytically active state. Using molecular dynamics\n",
      "simulations and structure-based mutagenesis, we show that catalysis\n",
      "is synchronized in the K. setae ECR\n",
      "across the pair of dimers. This conformational coupling of catalytic\n",
      "domains is conferred by individual amino acids to achieve high CO2-fixation rates. Our results provide unprecedented insights\n",
      "into the dynamic organization and synchronized inter- and intrasubunit\n",
      "communications of this remarkably efficient CO2-fixing\n",
      "enzyme during catalysis.Go to:Short abstractIntersubunit\n",
      "coupling drives fast carboxylation by enoyl-CoA\n",
      "carboxylases/reductases, providing the basis for engineering highly\n",
      "efficient CO2-fixing enzymes for bioenergy and bioproduct\n",
      "applications.Go to:IntroductionThe\n",
      "capture and conversion of atmospheric CO2 remains\n",
      "a challenging task for chemistry, resulting in an ever-increasing\n",
      "interest to understand and exploit CO2 fixation mechanisms\n",
      "offered by biology. Nature has evolved different enzymes capable of\n",
      "capturing and/or converting CO2, including biotin-dependent\n",
      "CoA carboxylases, which incorporate CO2 into different\n",
      "alkyl-CoA esters, formate dehydrogenases, which directly reduce CO2 into the single-carbon compound formate, and RubisCO, the\n",
      "central enzyme in the Calvin–Benson–Bassham cycle of\n",
      "photosynthesis that produces two molecules of 3-phosphoglycerate from\n",
      "ribulose-1,5-bisphosphate and CO2.1−4 The recently discovered family\n",
      "of enoyl-CoA carboxylases/reductases (ECRs) encompasses some of the\n",
      "most efficient CO2-fixing enzymes found in nature to date.1 With respect to the catalytic rate, for instance,\n",
      "ECR family members from primary metabolism outcompete RubisCO by more\n",
      "than 1 order of magnitude, which makes them exquisite models to study\n",
      "the molecular basis of efficient CO2 catalysis.2,5ECRs catalyze the reduction of various α,β-unsaturated\n",
      "e\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"0\\\": {\\n    \\\"quote\\\": \\\"The K. setae ECR is a homotetramer that differentiates into a pair of dimers of open- and closed-form subunits in the catalytically active state.\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"1\\\": {\\n    \\\"quote\\\": \\\"ECRs catalyze the reduction of various \\u03b1,\\u03b2-unsaturated\\\",\\n    \\\"index\\\": 2\\n  }\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691117628,\n",
      "  \"id\": \"chatcmpl-7jfEiLoGQi8ZhBYmlikLRPQYTzxCg\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 94,\n",
      "    \"prompt_tokens\": 1531,\n",
      "    \"total_tokens\": 1625\n",
      "  }\n",
      "}\n",
      "Extracted quotes:  {\n",
      "  \"0\": {\n",
      "    \"quote\": \"The K. setae ECR is a homotetramer that differentiates into a pair of dimers of open- and closed-form subunits in the catalytically active state.\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"1\": {\n",
      "    \"quote\": \"ECRs catalyze the reduction of various α,β-unsaturated\",\n",
      "    \"index\": 2\n",
      "  }\n",
      "}\n",
      "Max tokens reached in chunks!\n"
     ]
    }
   ],
   "source": [
    "# Go through each full text rated highly to extract facts from\n",
    "full_text_folder_path = f'autoscious_logs/{sanitize_filename(search_query)}/sources/full_text'\n",
    "\n",
    "# Loop through every file in the directory\n",
    "for filename in os.listdir(full_text_folder_path):\n",
    "    curr_tokens = 0\n",
    "    if filename.endswith('.txt'):\n",
    "        file_path = os.path.join(full_text_folder_path, filename)\n",
    "        with open(file_path, 'r', encoding='utf-8') as file:\n",
    "            title = file.readline().strip()\n",
    "            url = file.readline().strip()\n",
    "            text = file.read()\n",
    "        \n",
    "        # Break the text into chunk to extract information from\n",
    "        chunks = chunk_text(text, chunk_size, overlap)\n",
    "        for i, chunk in enumerate(chunks):\n",
    "            print(f\"Chunk: {i} / {len(chunks)}\")\n",
    "            extract_facts_from_website_text(search_query_file_safe, key_questions_list_string, title, chunk, url)\n",
    "\n",
    "            curr_tokens += len(chunk)\n",
    "            if curr_tokens > MAX_TOKENS:\n",
    "                print(\"Max tokens reached in chunks!\")\n",
    "                break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
