{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from llm import chat_openai\n",
    "import json\n",
    "from datetime import datetime"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "\n",
    "def sanitize_filename(filename):\n",
    "    return re.sub(r'[^a-zA-Z0-9]', '_', filename)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "search_query = \"How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "search_query_file_safe = sanitize_filename(search_query)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## MECE question decomposition"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "seed_initial_question_decomposition_prompt = f'''\n",
    "Research project question: {search_query}\n",
    "\n",
    "Task: Decompose this research question into 1) project objective: what should be the outcome of the project, 2) key drivers: what are the key problem drivers to achieve the project objective, 3) hypotheses: what can already be hypothesized about the key drivers, 4) key questions: what analyses need to be conducted to verify, falsify, or change the hypotheses?\n",
    "\n",
    "Rules: Be as mutually exclusive, completely exhaustive (MECE) as possible. The output should be in nested JSON format of \n",
    "```json\n",
    "{{\n",
    "  \"project_objective\": \"\",\n",
    "  \"key_drivers\": {{\n",
    "    \"1\": {{\n",
    "      \"driver\": \"\",\n",
    "      \"hypotheses\": {{\n",
    "          \"1\": {{\n",
    "            \"hypothesis\": \"\",\n",
    "            \"key_questions\": {{\n",
    "                \"1\": \"\", \n",
    "                \"2\": \"\",\n",
    "                etc.\n",
    "            }},\n",
    "          }},\n",
    "          etc.\n",
    "      }},\n",
    "    \"2\" : {{}},\n",
    "    etc.\n",
    "    }}\"\",\n",
    "  }},\n",
    "}}\n",
    "```\n",
    "Respond only with the output, with no explanation or conversation.\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "seed_initial_question_decomposition_prompt \n",
      "Research project question: How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?\n",
      "\n",
      "Task: Decompose this research question into 1) project objective: what should be the outcome of the project, 2) key drivers: what are the key problem drivers to achieve the project objective, 3) hypotheses: what can already be hypothesized about the key drivers, 4) key questions: what analyses need to be conducted to verify, falsify, or change the hypotheses?\n",
      "\n",
      "Rules: Be as mutually exclusive, completely exhaustive (MECE) as possible. The output should be in nested JSON format of \n",
      "```json\n",
      "{\n",
      "  \"project_objective\": \"\",\n",
      "  \"key_drivers\": {\n",
      "    \"1\": {\n",
      "      \"driver\": \"\",\n",
      "      \"hypotheses\": {\n",
      "          \"1\": {\n",
      "            \"hypothesis\": \"\",\n",
      "            \"key_questions\": {\n",
      "                \"1\": \"\", \n",
      "                \"2\": \"\",\n",
      "                etc.\n",
      "            },\n",
      "          },\n",
      "          etc.\n",
      "      },\n",
      "    \"2\" : {},\n",
      "    etc.\n",
      "    }\"\",\n",
      "  },\n",
      "}\n",
      "```\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(\"seed_initial_question_decomposition_prompt\", seed_initial_question_decomposition_prompt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Prompt:  \n",
      "Research project question: How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?\n",
      "\n",
      "Task: Decompose this research question into 1) project objective: what should be the outcome of the project, 2) key drivers: what are the key problem drivers to achieve the project objective, 3) hypotheses: what can already be hypothesized about the key drivers, 4) key questions: what analyses need to be conducted to verify, falsify, or change the hypotheses?\n",
      "\n",
      "Rules: Be as mutually exclusive, completely exhaustive (MECE) as possible. The output should be in nested JSON format of \n",
      "```json\n",
      "{\n",
      "  \"project_objective\": \"\",\n",
      "  \"key_drivers\": {\n",
      "    \"1\": {\n",
      "      \"driver\": \"\",\n",
      "      \"hypotheses\": {\n",
      "          \"1\": {\n",
      "            \"hypothesis\": \"\",\n",
      "            \"key_questions\": {\n",
      "                \"1\": \"\", \n",
      "                \"2\": \"\",\n",
      "                etc.\n",
      "            },\n",
      "          },\n",
      "          etc.\n",
      "      },\n",
      "    \"2\" : {},\n",
      "    etc.\n",
      "    }\"\",\n",
      "  },\n",
      "}\n",
      "```\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"project_objective\\\": \\\"To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria\\\",\\n  \\\"key_drivers\\\": {\\n    \\\"1\\\": {\\n      \\\"driver\\\": \\\"ECR enzyme activity\\\",\\n      \\\"hypotheses\\\": {\\n        \\\"1\\\": {\\n          \\\"hypothesis\\\": \\\"ECR enzyme activity is high in Kitsatospor setae bacteria\\\",\\n          \\\"key_questions\\\": {\\n            \\\"1\\\": \\\"What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\\\",\\n            \\\"2\\\": \\\"How does ECR enzyme activity vary under different conditions?\\\",\\n            \\\"3\\\": \\\"Are there any factors that enhance or inhibit ECR enzyme activity?\\\"\\n          }\\n        }\\n      }\\n    }\\n  }\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1690744745,\n",
      "  \"id\": \"chatcmpl-7i6ETD80Zlsi2RIIvZyM7UuZzfl1q\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 152,\n",
      "    \"prompt_tokens\": 277,\n",
      "    \"total_tokens\": 429\n",
      "  }\n",
      "}\n",
      "{\n",
      "  \"project_objective\": \"To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria\",\n",
      "  \"key_drivers\": {\n",
      "    \"1\": {\n",
      "      \"driver\": \"ECR enzyme activity\",\n",
      "      \"hypotheses\": {\n",
      "        \"1\": {\n",
      "          \"hypothesis\": \"ECR enzyme activity is high in Kitsatospor setae bacteria\",\n",
      "          \"key_questions\": {\n",
      "            \"1\": \"What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\",\n",
      "            \"2\": \"How does ECR enzyme activity vary under different conditions?\",\n",
      "            \"3\": \"Are there any factors that enhance or inhibit ECR enzyme activity?\"\n",
      "          }\n",
      "        }\n",
      "      }\n",
      "    }\n",
      "  }\n",
      "}\n"
     ]
    }
   ],
   "source": [
    "res = chat_openai(seed_initial_question_decomposition_prompt, model=\"gpt-3.5-turbo\")\n",
    "print(res[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "question_decomposition = json.loads(res[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "question_decomposition['search_query'] = search_query"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity vary under different conditions?', '3': 'Are there any factors that enhance or inhibit ECR enzyme activity?'}}}}}, 'search_query': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?'}\n"
     ]
    }
   ],
   "source": [
    "print(question_decomposition)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity vary under different conditions?', '3': 'Are there any factors that enhance or inhibit ECR enzyme activity?'}\n"
     ]
    }
   ],
   "source": [
    "key_questions = question_decomposition['key_drivers']['1']['hypotheses']['1']['key_questions']\n",
    "print(key_questions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: 'autoscious_logs/decompositions/How_efficiently_do_the_ECR_enzymes_work__especially_in_Kitsatospor_setae_bacteria_.json'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[47], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m \u001b[39mwith\u001b[39;00m \u001b[39mopen\u001b[39;49m(\u001b[39mf\u001b[39;49m\u001b[39m'\u001b[39;49m\u001b[39mautoscious_logs/decompositions/\u001b[39;49m\u001b[39m{\u001b[39;49;00msearch_query_file_safe\u001b[39m}\u001b[39;49;00m\u001b[39m.json\u001b[39;49m\u001b[39m'\u001b[39;49m, \u001b[39m'\u001b[39;49m\u001b[39mw\u001b[39;49m\u001b[39m'\u001b[39;49m) \u001b[39mas\u001b[39;00m f:\n\u001b[0;32m      2\u001b[0m     json\u001b[39m.\u001b[39mdump(question_decomposition, f, indent\u001b[39m=\u001b[39m\u001b[39m2\u001b[39m)\n",
      "File \u001b[1;32mc:\\Users\\1kevi\\Desktop\\projects\\Research\\autoscious-carbon-capture\\venv\\lib\\site-packages\\IPython\\core\\interactiveshell.py:284\u001b[0m, in \u001b[0;36m_modified_open\u001b[1;34m(file, *args, **kwargs)\u001b[0m\n\u001b[0;32m    277\u001b[0m \u001b[39mif\u001b[39;00m file \u001b[39min\u001b[39;00m {\u001b[39m0\u001b[39m, \u001b[39m1\u001b[39m, \u001b[39m2\u001b[39m}:\n\u001b[0;32m    278\u001b[0m     \u001b[39mraise\u001b[39;00m \u001b[39mValueError\u001b[39;00m(\n\u001b[0;32m    279\u001b[0m         \u001b[39mf\u001b[39m\u001b[39m\"\u001b[39m\u001b[39mIPython won\u001b[39m\u001b[39m'\u001b[39m\u001b[39mt let you open fd=\u001b[39m\u001b[39m{\u001b[39;00mfile\u001b[39m}\u001b[39;00m\u001b[39m by default \u001b[39m\u001b[39m\"\u001b[39m\n\u001b[0;32m    280\u001b[0m         \u001b[39m\"\u001b[39m\u001b[39mas it is likely to crash IPython. If you know what you are doing, \u001b[39m\u001b[39m\"\u001b[39m\n\u001b[0;32m    281\u001b[0m         \u001b[39m\"\u001b[39m\u001b[39myou can use builtins\u001b[39m\u001b[39m'\u001b[39m\u001b[39m open.\u001b[39m\u001b[39m\"\u001b[39m\n\u001b[0;32m    282\u001b[0m     )\n\u001b[1;32m--> 284\u001b[0m \u001b[39mreturn\u001b[39;00m io_open(file, \u001b[39m*\u001b[39margs, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwargs)\n",
      "\u001b[1;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: 'autoscious_logs/decompositions/How_efficiently_do_the_ECR_enzymes_work__especially_in_Kitsatospor_setae_bacteria_.json'"
     ]
    }
   ],
   "source": [
    "with open(f'autoscious_logs/decompositions/{search_query_file_safe}.json', 'w') as f:\n",
    "    json.dump(question_decomposition, f, indent=2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity vary under different conditions?', '3': 'Are there any factors that enhance or inhibit ECR enzyme activity?'}\n"
     ]
    }
   ],
   "source": [
    "with open(r'C:\\Users\\1kevi\\Desktop\\projects\\Research\\autoscious-carbon-capture\\question_answering\\autoscious_logs\\decompositions\\How_efficiently_do_the_ECR_enzymes_work__especially_in_Kitsatospor_setae_bacteria_.json', 'r') as f:\n",
    "        question_decomposition = json.load(f)\n",
    "key_questions = question_decomposition['key_drivers']['1']['hypotheses']['1']['key_questions']\n",
    "print(key_questions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?'"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "key_questions['1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import time\n",
    "from itertools import islice\n",
    "from duckduckgo_search import DDGS\n",
    "DUCKDUCKGO_MAX_ATTEMPTS = 3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Taken from AutoGPT\n",
    "\n",
    "def safe_google_results(results: str | list) -> str:\n",
    "    \"\"\"\n",
    "        Return the results of a Google search in a safe format.\n",
    "\n",
    "    Args:\n",
    "        results (str | list): The search results.\n",
    "\n",
    "    Returns:\n",
    "        str: The results of the search.\n",
    "    \"\"\"\n",
    "    if isinstance(results, list):\n",
    "        safe_message = json.dumps(\n",
    "            [result.encode(\"utf-8\", \"ignore\").decode(\"utf-8\") for result in results]\n",
    "        )\n",
    "    else:\n",
    "        safe_message = results.encode(\"utf-8\", \"ignore\").decode(\"utf-8\")\n",
    "    return safe_message\n",
    "\n",
    "def web_search_ddg(query: str, num_results: int = 8) -> str:\n",
    "    print(\"searching from duck duck go!\")\n",
    "    \"\"\"Return the results of a Google search\n",
    "\n",
    "    Args:\n",
    "        query (str): The search query.\n",
    "        num_results (int): The number of results to return.\n",
    "\n",
    "    Returns:\n",
    "        str: The results of the search.\n",
    "    \"\"\"\n",
    "    search_results = []\n",
    "    attempts = 0\n",
    "\n",
    "    while attempts < DUCKDUCKGO_MAX_ATTEMPTS:\n",
    "        if not query:\n",
    "            return json.dumps(search_results)\n",
    "\n",
    "        results = DDGS().text(query)\n",
    "        search_results = list(islice(results, num_results))\n",
    "\n",
    "        if search_results:\n",
    "            break\n",
    "\n",
    "        time.sleep(1)\n",
    "        attempts += 1\n",
    "\n",
    "    results = json.dumps(search_results, ensure_ascii=False, indent=4)\n",
    "    return safe_google_results(results)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "searching from duck duck go!\n"
     ]
    }
   ],
   "source": [
    "web_search_res = web_search_ddg(key_questions['1'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'[\\n    {\\n        \"title\": \"Intersubunit Coupling Enables Fast CO2-Fixation by Reductive ...\",\\n        \"href\": \"https://pubs.acs.org/doi/10.1021/acscentsci.2c00057\",\\n        \"body\": \"Here we used a combination of ambient-temperature X-ray free electron laser (XFEL) and cryogenic synchrotron experiments to study the structural organization of the ECR from Kitasatospora setae. The K. setae ECR is a homotetramer that differentiates into a pair of dimers of open- and closed-form subunits in the catalytically active state. Using ...\"\\n    },\\n    {\\n        \"title\": \"In the active centre of carbon dioxide conversion\",\\n        \"href\": \"https://www.mpg.de/13672605/0705-terr-138345-nature-s-fantastic-four\",\\n        \"body\": \"They analyzed the ECR from the bacterium Kitasatospora setae, currently the fastest known carboxylase. Using a combined approach of structural biology, biochemistry and computer simulations, they were able to understand for the first time how the enzyme binds and converts CO 2. Teamwork in the active centre\"\\n    },\\n    {\\n        \"title\": \"Significance of serum enzyme changes after cardiac catheterization and ...\",\\n        \"href\": \"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC482977/\",\\n        \"body\": \"Five of these patients whose serum enzyme levels were higher than normal before the procedure were excluded from the study. Forty-one of the remaining 51 patients had left ventriculography and also selective coronary arteriography. In these 41 patients (groups 1 and 2--see below), the mean serum CK levels increased after the procedure to exceed ...\"\\n    },\\n    {\\n        \"title\": \"Bacterial Enzyme Converts CO2 Into Carbon Compounds 20x ... - SciTechDaily\",\\n        \"href\": \"https://scitechdaily.com/bacterial-enzyme-converts-co2-into-carbon-compounds-20x-faster-than-photosynthesis/\",\\n        \"body\": \"Improving on nature The enzyme the team studied is part of a family called enoyl-CoA carboxylases/reductases, or ECRs. It comes from soil bacteria called Kitasatospora setae, which in addition to their carbon-fixing skills can also produce antibiotics.\"\\n    },\\n    {\\n        \"title\": \"Awakening the Sleeping Carboxylase Function of Enzymes: Engineering the ...\",\\n        \"href\": \"https://pubs.acs.org/doi/10.1021/jacs.9b03431\",\\n        \"body\": \"However, for enoyl-CoA carboxylase/reductase from Kitasatospora setae (ECR Ks), four conserved amino acids that form a CO 2-binding pocket at the active site were described recently (Figure 1a). These four amino acids anchor and position the CO 2 molecule during catalysis, in which a reactive enolate is formed that attacks the CO 2. Figure 1\"\\n    },\\n    {\\n        \"title\": \"Four amino acids define the CO 2 binding pocket of enoyl-CoA ... - PNAS\",\\n        \"href\": \"https://www.pnas.org/doi/10.1073/pnas.1901471116\",\\n        \"body\": \"Abstract Carboxylases are biocatalysts that capture and convert carbon dioxide (CO 2) under mild conditions and atmospheric concentrations at a scale of more than 400 Gt annually. However, how these enzymes bind and control the gaseous CO 2 molecule during catalysis is only poorly understood.\"\\n    },\\n    {\\n        \"title\": \"Taxonomy browser (Kitasatospora setae KM-6054) - National Center for ...\",\\n        \"href\": \"https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=452652\",\\n        \"body\": \"levels using filter: Kitasatospora setae KM-6054 Taxonomy ID: 452652 (for references in articles please use NCBI:txid452652) current name. Kitasatospora setae KM-6054. equivalent: ... Kitasatospora setae strain KM-6054. NCBI BLAST name: high G+C Gram-positive bacteria Rank: strain Genetic code: Translation table 11 (Bacterial, Archaeal and ...\"\\n    },\\n    {\\n        \"title\": \"Protease Activity Analysis: A Toolkit for Analyzing Enzyme Activity ...\",\\n        \"href\": \"https://pubs.acs.org/doi/10.1021/acsomega.2c01559\",\\n        \"body\": \"Analyzing the activity of proteases and their substrates is critical to defining the biological functions of these enzymes and to designing new diagnostics and therapeutics that target protease dysregulation in disease. While a wide range of databases and algorithms have been created to better predict protease cleavage sites, there is a dearth of computational tools to automate analysis of in ...\"\\n    }\\n]'"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "web_search_res"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save to main\n",
    "file_name_safe = sanitize_filename(key_questions['1'])\n",
    "\n",
    "with open(f'autoscious_logs/web_search/{file_name_safe}.json', 'w') as f:\n",
    "    json.dump(web_search_res, f, indent=2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Browse websites"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "ename": "NoConsoleScreenBufferError",
     "evalue": "Found xterm-color, while expecting a Windows console. Maybe try to run this program using \"winpty\" or run it in cmd.exe instead. Or otherwise, in case of Cygwin, use the Python executable that is compiled for Cygwin.",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNoConsoleScreenBufferError\u001b[0m                Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[46], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mautogpt\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mcommands\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mweb_selenium\u001b[39;00m \u001b[39mimport\u001b[39;00m browse_website\n",
      "File \u001b[1;32mc:\\Users\\1kevi\\Desktop\\projects\\Research\\autoscious-carbon-capture\\question_answering\\autogpt\\commands\\web_selenium.py:30\u001b[0m\n\u001b[0;32m     27\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mwebdriver_manager\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mfirefox\u001b[39;00m \u001b[39mimport\u001b[39;00m GeckoDriverManager\n\u001b[0;32m     28\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mwebdriver_manager\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mmicrosoft\u001b[39;00m \u001b[39mimport\u001b[39;00m EdgeChromiumDriverManager \u001b[39mas\u001b[39;00m EdgeDriverManager\n\u001b[1;32m---> 30\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mautogpt\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39magents\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39magent\u001b[39;00m \u001b[39mimport\u001b[39;00m Agent\n\u001b[0;32m     31\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mautogpt\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mcommand_decorator\u001b[39;00m \u001b[39mimport\u001b[39;00m command\n\u001b[0;32m     32\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mautogpt\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mlogs\u001b[39;00m \u001b[39mimport\u001b[39;00m logger\n",
      "File \u001b[1;32mc:\\Users\\1kevi\\Desktop\\projects\\Research\\autoscious-carbon-capture\\question_answering\\autogpt\\agents\\__init__.py:1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39m.\u001b[39;00m\u001b[39magent\u001b[39;00m \u001b[39mimport\u001b[39;00m Agent\n\u001b[0;32m      2\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39m.\u001b[39;00m\u001b[39mbase\u001b[39;00m \u001b[39mimport\u001b[39;00m AgentThoughts, BaseAgent, CommandArgs, CommandName\n\u001b[0;32m      4\u001b[0m __all__ \u001b[39m=\u001b[39m [\u001b[39m\"\u001b[39m\u001b[39mBaseAgent\u001b[39m\u001b[39m\"\u001b[39m, \u001b[39m\"\u001b[39m\u001b[39mAgent\u001b[39m\u001b[39m\"\u001b[39m, \u001b[39m\"\u001b[39m\u001b[39mCommandName\u001b[39m\u001b[39m\"\u001b[39m, \u001b[39m\"\u001b[39m\u001b[39mCommandArgs\u001b[39m\u001b[39m\"\u001b[39m, \u001b[39m\"\u001b[39m\u001b[39mAgentThoughts\u001b[39m\u001b[39m\"\u001b[39m]\n",
      "File \u001b[1;32mc:\\Users\\1kevi\\Desktop\\projects\\Research\\autoscious-carbon-capture\\question_answering\\autogpt\\agents\\agent.py:27\u001b[0m\n\u001b[0;32m     19\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mautogpt\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mlogs\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mlog_cycle\u001b[39;00m \u001b[39mimport\u001b[39;00m (\n\u001b[0;32m     20\u001b[0m     FULL_MESSAGE_HISTORY_FILE_NAME,\n\u001b[0;32m     21\u001b[0m     NEXT_ACTION_FILE_NAME,\n\u001b[0;32m     22\u001b[0m     USER_INPUT_FILE_NAME,\n\u001b[0;32m     23\u001b[0m     LogCycleHandler,\n\u001b[0;32m     24\u001b[0m )\n\u001b[0;32m     25\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mautogpt\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mworkspace\u001b[39;00m \u001b[39mimport\u001b[39;00m Workspace\n\u001b[1;32m---> 27\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39m.\u001b[39;00m\u001b[39mbase\u001b[39;00m \u001b[39mimport\u001b[39;00m AgentThoughts, BaseAgent, CommandArgs, CommandName\n\u001b[0;32m     30\u001b[0m \u001b[39mclass\u001b[39;00m \u001b[39mAgent\u001b[39;00m(BaseAgent):\n\u001b[0;32m     31\u001b[0m \u001b[39m    \u001b[39m\u001b[39m\"\"\"Agent class for interacting with Auto-GPT.\"\"\"\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\1kevi\\Desktop\\projects\\Research\\autoscious-carbon-capture\\question_answering\\autogpt\\agents\\base.py:16\u001b[0m\n\u001b[0;32m     14\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mautogpt\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mlogs\u001b[39;00m \u001b[39mimport\u001b[39;00m logger\n\u001b[0;32m     15\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mautogpt\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mmemory\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mmessage_history\u001b[39;00m \u001b[39mimport\u001b[39;00m MessageHistory\n\u001b[1;32m---> 16\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mautogpt\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mprompts\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mprompt\u001b[39;00m \u001b[39mimport\u001b[39;00m DEFAULT_TRIGGERING_PROMPT\n\u001b[0;32m     18\u001b[0m CommandName \u001b[39m=\u001b[39m \u001b[39mstr\u001b[39m\n\u001b[0;32m     19\u001b[0m CommandArgs \u001b[39m=\u001b[39m \u001b[39mdict\u001b[39m[\u001b[39mstr\u001b[39m, \u001b[39mstr\u001b[39m]\n",
      "File \u001b[1;32mc:\\Users\\1kevi\\Desktop\\projects\\Research\\autoscious-carbon-capture\\question_answering\\autogpt\\prompts\\prompt.py:2\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mautogpt\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mconfig\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mconfig\u001b[39;00m \u001b[39mimport\u001b[39;00m Config\n\u001b[1;32m----> 2\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mautogpt\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mconfig\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mprompt_config\u001b[39;00m \u001b[39mimport\u001b[39;00m PromptConfig\n\u001b[0;32m      3\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mautogpt\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mprompts\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mgenerator\u001b[39;00m \u001b[39mimport\u001b[39;00m PromptGenerator\n\u001b[0;32m      5\u001b[0m DEFAULT_TRIGGERING_PROMPT \u001b[39m=\u001b[39m \u001b[39m\"\u001b[39m\u001b[39mDetermine exactly one command to use, and respond using the JSON schema specified previously:\u001b[39m\u001b[39m\"\u001b[39m\n",
      "File \u001b[1;32mc:\\Users\\1kevi\\Desktop\\projects\\Research\\autoscious-carbon-capture\\question_answering\\autogpt\\config\\prompt_config.py:8\u001b[0m\n\u001b[0;32m      5\u001b[0m \u001b[39mimport\u001b[39;00m \u001b[39myaml\u001b[39;00m\n\u001b[0;32m      6\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mcolorama\u001b[39;00m \u001b[39mimport\u001b[39;00m Fore\n\u001b[1;32m----> 8\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mautogpt\u001b[39;00m \u001b[39mimport\u001b[39;00m utils\n\u001b[0;32m      9\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mautogpt\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mlogs\u001b[39;00m \u001b[39mimport\u001b[39;00m logger\n\u001b[0;32m     12\u001b[0m \u001b[39mclass\u001b[39;00m \u001b[39mPromptConfig\u001b[39;00m:\n",
      "File \u001b[1;32mc:\\Users\\1kevi\\Desktop\\projects\\Research\\autoscious-carbon-capture\\question_answering\\autogpt\\utils.py:14\u001b[0m\n\u001b[0;32m     11\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mautogpt\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mconfig\u001b[39;00m \u001b[39mimport\u001b[39;00m Config\n\u001b[0;32m     12\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mautogpt\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mlogs\u001b[39;00m \u001b[39mimport\u001b[39;00m logger\n\u001b[1;32m---> 14\u001b[0m session \u001b[39m=\u001b[39m PromptSession(history\u001b[39m=\u001b[39;49mInMemoryHistory())\n\u001b[0;32m     17\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mbatch\u001b[39m(iterable, max_batch_length: \u001b[39mint\u001b[39m, overlap: \u001b[39mint\u001b[39m \u001b[39m=\u001b[39m \u001b[39m0\u001b[39m):\n\u001b[0;32m     18\u001b[0m \u001b[39m    \u001b[39m\u001b[39m\"\"\"Batch data from iterable into slices of length N. The last batch may be shorter.\"\"\"\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\1kevi\\Desktop\\projects\\Research\\autoscious-carbon-capture\\venv\\lib\\site-packages\\prompt_toolkit\\shortcuts\\prompt.py:466\u001b[0m, in \u001b[0;36mPromptSession.__init__\u001b[1;34m(self, message, multiline, wrap_lines, is_password, vi_mode, editing_mode, complete_while_typing, validate_while_typing, enable_history_search, search_ignore_case, lexer, enable_system_prompt, enable_suspend, enable_open_in_editor, validator, completer, complete_in_thread, reserve_space_for_menu, complete_style, auto_suggest, style, style_transformation, swap_light_and_dark_colors, color_depth, cursor, include_default_pygments_style, history, clipboard, prompt_continuation, rprompt, bottom_toolbar, mouse_support, input_processors, placeholder, key_bindings, erase_when_done, tempfile_suffix, tempfile, refresh_interval, input, output)\u001b[0m\n\u001b[0;32m    464\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39msearch_buffer \u001b[39m=\u001b[39m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_create_search_buffer()\n\u001b[0;32m    465\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mlayout \u001b[39m=\u001b[39m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_create_layout()\n\u001b[1;32m--> 466\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mapp \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_create_application(editing_mode, erase_when_done)\n",
      "File \u001b[1;32mc:\\Users\\1kevi\\Desktop\\projects\\Research\\autoscious-carbon-capture\\venv\\lib\\site-packages\\prompt_toolkit\\shortcuts\\prompt.py:717\u001b[0m, in \u001b[0;36mPromptSession._create_application\u001b[1;34m(self, editing_mode, erase_when_done)\u001b[0m\n\u001b[0;32m    714\u001b[0m prompt_bindings \u001b[39m=\u001b[39m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_create_prompt_bindings()\n\u001b[0;32m    716\u001b[0m \u001b[39m# Create application\u001b[39;00m\n\u001b[1;32m--> 717\u001b[0m application: Application[_T] \u001b[39m=\u001b[39m Application(\n\u001b[0;32m    718\u001b[0m     layout\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mlayout,\n\u001b[0;32m    719\u001b[0m     style\u001b[39m=\u001b[39;49mDynamicStyle(\u001b[39mlambda\u001b[39;49;00m: \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mstyle),\n\u001b[0;32m    720\u001b[0m     style_transformation\u001b[39m=\u001b[39;49mmerge_style_transformations(\n\u001b[0;32m    721\u001b[0m         [\n\u001b[0;32m    722\u001b[0m             DynamicStyleTransformation(\u001b[39mlambda\u001b[39;49;00m: \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mstyle_transformation),\n\u001b[0;32m    723\u001b[0m             ConditionalStyleTransformation(\n\u001b[0;32m    724\u001b[0m                 SwapLightAndDarkStyleTransformation(),\n\u001b[0;32m    725\u001b[0m                 dyncond(\u001b[39m\"\u001b[39;49m\u001b[39mswap_light_and_dark_colors\u001b[39;49m\u001b[39m\"\u001b[39;49m),\n\u001b[0;32m    726\u001b[0m             ),\n\u001b[0;32m    727\u001b[0m         ]\n\u001b[0;32m    728\u001b[0m     ),\n\u001b[0;32m    729\u001b[0m     include_default_pygments_style\u001b[39m=\u001b[39;49mdyncond(\u001b[39m\"\u001b[39;49m\u001b[39minclude_default_pygments_style\u001b[39;49m\u001b[39m\"\u001b[39;49m),\n\u001b[0;32m    730\u001b[0m     clipboard\u001b[39m=\u001b[39;49mDynamicClipboard(\u001b[39mlambda\u001b[39;49;00m: \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mclipboard),\n\u001b[0;32m    731\u001b[0m     key_bindings\u001b[39m=\u001b[39;49mmerge_key_bindings(\n\u001b[0;32m    732\u001b[0m         [\n\u001b[0;32m    733\u001b[0m             merge_key_bindings(\n\u001b[0;32m    734\u001b[0m                 [\n\u001b[0;32m    735\u001b[0m                     auto_suggest_bindings,\n\u001b[0;32m    736\u001b[0m                     ConditionalKeyBindings(\n\u001b[0;32m    737\u001b[0m                         open_in_editor_bindings,\n\u001b[0;32m    738\u001b[0m                         dyncond(\u001b[39m\"\u001b[39;49m\u001b[39menable_open_in_editor\u001b[39;49m\u001b[39m\"\u001b[39;49m)\n\u001b[0;32m    739\u001b[0m                         \u001b[39m&\u001b[39;49m has_focus(DEFAULT_BUFFER),\n\u001b[0;32m    740\u001b[0m                     ),\n\u001b[0;32m    741\u001b[0m                     prompt_bindings,\n\u001b[0;32m    742\u001b[0m                 ]\n\u001b[0;32m    743\u001b[0m             ),\n\u001b[0;32m    744\u001b[0m             DynamicKeyBindings(\u001b[39mlambda\u001b[39;49;00m: \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mkey_bindings),\n\u001b[0;32m    745\u001b[0m         ]\n\u001b[0;32m    746\u001b[0m     ),\n\u001b[0;32m    747\u001b[0m     mouse_support\u001b[39m=\u001b[39;49mdyncond(\u001b[39m\"\u001b[39;49m\u001b[39mmouse_support\u001b[39;49m\u001b[39m\"\u001b[39;49m),\n\u001b[0;32m    748\u001b[0m     editing_mode\u001b[39m=\u001b[39;49mediting_mode,\n\u001b[0;32m    749\u001b[0m     erase_when_done\u001b[39m=\u001b[39;49merase_when_done,\n\u001b[0;32m    750\u001b[0m     reverse_vi_search_direction\u001b[39m=\u001b[39;49m\u001b[39mTrue\u001b[39;49;00m,\n\u001b[0;32m    751\u001b[0m     color_depth\u001b[39m=\u001b[39;49m\u001b[39mlambda\u001b[39;49;00m: \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mcolor_depth,\n\u001b[0;32m    752\u001b[0m     cursor\u001b[39m=\u001b[39;49mDynamicCursorShapeConfig(\u001b[39mlambda\u001b[39;49;00m: \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mcursor),\n\u001b[0;32m    753\u001b[0m     refresh_interval\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mrefresh_interval,\n\u001b[0;32m    754\u001b[0m     \u001b[39minput\u001b[39;49m\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_input,\n\u001b[0;32m    755\u001b[0m     output\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_output,\n\u001b[0;32m    756\u001b[0m )\n\u001b[0;32m    758\u001b[0m \u001b[39m# During render time, make sure that we focus the right search control\u001b[39;00m\n\u001b[0;32m    759\u001b[0m \u001b[39m# (if we are searching). - This could be useful if people make the\u001b[39;00m\n\u001b[0;32m    760\u001b[0m \u001b[39m# 'multiline' property dynamic.\u001b[39;00m\n\u001b[0;32m    761\u001b[0m \u001b[39m\u001b[39m\u001b[39m\"\"\"\u001b[39;00m\n\u001b[0;32m    762\u001b[0m \u001b[39mdef on_render(app):\u001b[39;00m\n\u001b[0;32m    763\u001b[0m \u001b[39m    multiline = is_true(self.multiline)\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    775\u001b[0m \u001b[39mapp.on_render += on_render\u001b[39;00m\n\u001b[0;32m    776\u001b[0m \u001b[39m\"\"\"\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\1kevi\\Desktop\\projects\\Research\\autoscious-carbon-capture\\venv\\lib\\site-packages\\prompt_toolkit\\application\\application.py:265\u001b[0m, in \u001b[0;36mApplication.__init__\u001b[1;34m(self, layout, style, include_default_pygments_style, style_transformation, key_bindings, clipboard, full_screen, color_depth, mouse_support, enable_page_navigation_bindings, paste_mode, editing_mode, erase_when_done, reverse_vi_search_direction, min_redraw_interval, max_render_postpone_time, refresh_interval, terminal_size_polling_interval, cursor, on_reset, on_invalidate, before_render, after_render, input, output)\u001b[0m\n\u001b[0;32m    263\u001b[0m \u001b[39m# I/O.\u001b[39;00m\n\u001b[0;32m    264\u001b[0m session \u001b[39m=\u001b[39m get_app_session()\n\u001b[1;32m--> 265\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39moutput \u001b[39m=\u001b[39m output \u001b[39mor\u001b[39;00m session\u001b[39m.\u001b[39;49moutput\n\u001b[0;32m    266\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39minput \u001b[39m=\u001b[39m \u001b[39minput\u001b[39m \u001b[39mor\u001b[39;00m session\u001b[39m.\u001b[39minput\n\u001b[0;32m    268\u001b[0m \u001b[39m# List of 'extra' functions to execute before a Application.run.\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\1kevi\\Desktop\\projects\\Research\\autoscious-carbon-capture\\venv\\lib\\site-packages\\prompt_toolkit\\application\\current.py:68\u001b[0m, in \u001b[0;36mAppSession.output\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m     65\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_output \u001b[39mis\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n\u001b[0;32m     66\u001b[0m     \u001b[39mfrom\u001b[39;00m \u001b[39mprompt_toolkit\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39moutput\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mdefaults\u001b[39;00m \u001b[39mimport\u001b[39;00m create_output\n\u001b[1;32m---> 68\u001b[0m     \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_output \u001b[39m=\u001b[39m create_output()\n\u001b[0;32m     69\u001b[0m \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_output\n",
      "File \u001b[1;32mc:\\Users\\1kevi\\Desktop\\projects\\Research\\autoscious-carbon-capture\\venv\\lib\\site-packages\\prompt_toolkit\\output\\defaults.py:87\u001b[0m, in \u001b[0;36mcreate_output\u001b[1;34m(stdout, always_prefer_tty)\u001b[0m\n\u001b[0;32m     83\u001b[0m         \u001b[39mreturn\u001b[39;00m cast(\n\u001b[0;32m     84\u001b[0m             Output, ConEmuOutput(stdout, default_color_depth\u001b[39m=\u001b[39mcolor_depth_from_env)\n\u001b[0;32m     85\u001b[0m         )\n\u001b[0;32m     86\u001b[0m     \u001b[39melse\u001b[39;00m:\n\u001b[1;32m---> 87\u001b[0m         \u001b[39mreturn\u001b[39;00m Win32Output(stdout, default_color_depth\u001b[39m=\u001b[39;49mcolor_depth_from_env)\n\u001b[0;32m     88\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[0;32m     89\u001b[0m     \u001b[39mfrom\u001b[39;00m \u001b[39m.\u001b[39;00m\u001b[39mvt100\u001b[39;00m \u001b[39mimport\u001b[39;00m Vt100_Output\n",
      "File \u001b[1;32mc:\\Users\\1kevi\\Desktop\\projects\\Research\\autoscious-carbon-capture\\venv\\lib\\site-packages\\prompt_toolkit\\output\\win32.py:116\u001b[0m, in \u001b[0;36mWin32Output.__init__\u001b[1;34m(self, stdout, use_complete_width, default_color_depth)\u001b[0m\n\u001b[0;32m    113\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mcolor_lookup_table \u001b[39m=\u001b[39m ColorLookupTable()\n\u001b[0;32m    115\u001b[0m \u001b[39m# Remember the default console colors.\u001b[39;00m\n\u001b[1;32m--> 116\u001b[0m info \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mget_win32_screen_buffer_info()\n\u001b[0;32m    117\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mdefault_attrs \u001b[39m=\u001b[39m info\u001b[39m.\u001b[39mwAttributes \u001b[39mif\u001b[39;00m info \u001b[39melse\u001b[39;00m \u001b[39m15\u001b[39m\n\u001b[0;32m    119\u001b[0m \u001b[39mif\u001b[39;00m _DEBUG_RENDER_OUTPUT:\n",
      "File \u001b[1;32mc:\\Users\\1kevi\\Desktop\\projects\\Research\\autoscious-carbon-capture\\venv\\lib\\site-packages\\prompt_toolkit\\output\\win32.py:220\u001b[0m, in \u001b[0;36mWin32Output.get_win32_screen_buffer_info\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    218\u001b[0m     \u001b[39mreturn\u001b[39;00m sbinfo\n\u001b[0;32m    219\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[1;32m--> 220\u001b[0m     \u001b[39mraise\u001b[39;00m NoConsoleScreenBufferError\n",
      "\u001b[1;31mNoConsoleScreenBufferError\u001b[0m: Found xterm-color, while expecting a Windows console. Maybe try to run this program using \"winpty\" or run it in cmd.exe instead. Or otherwise, in case of Cygwin, use the Python executable that is compiled for Cygwin."
     ]
    }
   ],
   "source": [
    "from autogpt.commands.web_selenium import browse_website"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract relevant facts and evidence for key question"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Testing and reading website code is in browse_websites.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity vary under different conditions?', '3': 'Are there any factors that enhance or inhibit ECR enzyme activity?'}\n"
     ]
    }
   ],
   "source": [
    "with open(r'C:\\Users\\1kevi\\Desktop\\projects\\Research\\autoscious-carbon-capture\\question_answering\\autoscious_logs\\decompositions\\How_efficiently_do_the_ECR_enzymes_work__especially_in_Kitsatospor_setae_bacteria_.json', 'r') as f:\n",
    "        question_decomposition = json.load(f)\n",
    "key_questions = question_decomposition['key_drivers']['1']['hypotheses']['1']['key_questions']\n",
    "print(key_questions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity vary under different conditions?\n",
      "2. Are there any factors that enhance or inhibit ECR enzyme activity?\n",
      "\n"
     ]
    }
   ],
   "source": [
    "numbered_key_questions_string = \"\"\n",
    "# Iterate over each item in the dictionary\n",
    "for key, value in key_questions.items():\n",
    "    # Extract the first two words from the question\n",
    "    first_two_words = ' '.join(value.split())\n",
    "    # Add the new line to the string, using the key as the number and the first two words as the question\n",
    "    numbered_key_questions_string += f'{int(key) - 1}. {first_two_words}\\n'\n",
    "\n",
    "print(numbered_key_questions_string)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "key_question = \"What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "In order to overcome the climate crisis, two measures are required: The reduction of carbon dioxide (CO2) emissions, and removal of CO2 from the earth atmosphere. The latter is the goal of Tobias Erb's research group at the Max Planck Institute for Terrestrial Microbiology in Marburg. Their approaches not only aim to benefit climate protection, but also to secure sustainability in the long term: to filter CO2 from the air and make it usable for technology.\n",
      "The Carboxylase active site: Four amino acids are important for CO2 binding in highly efficient CO2 fixing enzymes.\n",
      "© Max-Planck-Institute for Terrestrial Microbiology/Erb\n",
      "The Carboxylase active site: Four amino acids are important for CO2 binding in highly efficient CO2 fixing enzymes.\n",
      "© Max-Planck-Institute for Terrestrial Microbiology/Erb\n",
      "Filtering CO2 efficiently from the air - nature can do this through photosynthesis, converting CO2 into biomass. Unlike industrial technologies which can only use the gas in a highly concentrated form (which in turn consumes fossil energy), photosynthesis works directly with ambient air containing only 0.4% gaseous carbon dioxide. Its secret lies in the enzymes, proteins that act as catalysts to mediate specific chemical reactions, such as the fixation of CO2. In photosynthesis, this reaction is driven by the enzyme RubisCO. However, the efficiency of natural photosynthesis is not very high: in more than a fourth of all cases, RubisCO metabolizes oxygen from the air, which is a strong competitor of CO2 in this reaction.ECR enzymes are faster and more precise than RubisCoTherefore, the Max Planck researchers in Marburg have decided on alternatives to RubisCO. Enoyl-CoA Carboxylase/Reductase enzymes (ECRs) are much more efficient than RubisCO and do not make mistakes with oxygen. After many years of scientific work in order to understand this capacity, the scientists have succeeded in building a functioning process in the test tube that fixes CO2 better than nature itself. Robustness and energy efficiency are the qualities that the team would like to bestow to their artificial photosynthesis. \"Learn from the best\" is the motto: nature itself serves as a model for molecular biology.What is the reason for the high efficiency of ECRs? What is the magic spell to create a turbo CO2 fixator? Max Planck junior researchers Gabriele Stoffel and Iria Bernhardsgrütter pursued this question together with colleagues from Chile and the USA. They analyzed the ECR from the bacterium Kitasatospora setae, currently the fastest known carboxylase. Using a combined approach of structural biology, biochemistry and computer simulations, they were able to understand for the first time how the enzyme binds and converts CO2.Teamwork in the active centre\n",
      "A methionine residue shields the active site from the competing water molecules.\n",
      "© Max-Planck-Institute for Terrestrial Microbiology/Erb\n",
      "A methionine residue shields the active site from the competing water molecules.\n",
      "© Max-Planck-Institute for Terrestrial Microbiology/Erb\n",
      "\"We were surprised to learn that only four amino acids are sufficient to provide control over the CO2 molecule”, explains Gabriele Stoffel, postdoctoral fellow in the Erb department and first author of the study. “Three amino acids – one asparagine, glutamate and a histidine – hold the CO2 in place from two sides. Another amino acid, a phenylalanine, shields the bound CO2 from water, which would inhibit the reaction”, says Stoffel.These findings open up new paths for researchers.  \"We wanted to transfer the capability of binding CO2 to other enzymes. This would offer us much greater possibilities for optimizing photosynthesis,\" says Iria Bernhardsgrütter, PhD student in the research group. In another study, Bernhardsgrütter focused on two candidates for the protein scaffold: Propionyl-CoA synthase (PCS) and Archaeal Enoyl-CoA reductase (AER).Enhancing CO2 fixation capacityBoth enzymes were already able to use CO2, but only with an efficiency of about five percent and with concentrated CO2. Computational models revealed that those enzymes only possessed some of the four amino acids required and those were also misaligned. Iria Bernhardsgrütter succeeded by exchanging amino acids to correct the \"miscasts\" in PCS. Immediately, the efficiency of CO2 increased to around 20 percent. Now the second aspect was targeted, namely shielding the binding site from water. Iria Bernhardsgrütter was also able to solve this problem: another amino acid replacement blocked the water's access to the binding site. The combination of both changes led to a carboxylation rate of almost 95%. Similar experiments with AER increased CO2-conversion efficiencies to almost 90%.This knowledge of the exact requirements of CO2-fixing enzymes and its successful application has brought research a decisive step closer to its high goals: on the one hand, being able to filter CO2 efficiently from the atmosphere, on the other hand, integrating CO2 into sustainable use – towards the recycling of valuable substances following nature's example.\n",
      "You can find this video on YouTube. Click on the image to be redirected there.\n",
      "©\n",
      "MPI f. Terrestrial Mikrobiology/ Erb\n",
      "You can find this video on YouTube. Click on the image to be redirected there.\n",
      "The correct positioning of the gaseous carbon dioxide molecule is essential for the success of the carboxylation reaction.\n",
      "https://www.youtube.com/watch?v=JB7hekpOFFo\n",
      "Science Magazine\n",
      "Events\n",
      "Images of Science\n",
      "On Location\n",
      "Infographics\n",
      "Press Newsletter\n",
      "Contact\n",
      "Prof. Dr. Tobias Erb\n",
      "Director\n",
      "Max Planck Institute for Terrestrial Microbiology, Marburg\n",
      "+49 6421 178-700\n",
      "toerb@mpi-marburg.mpg.de\n",
      "Dr. Virginia Geisel\n",
      "PR Representative\n",
      "Max Planck Institute for Terrestrial Microbiology, Marburg\n",
      "+49 160 91387362\n",
      "virginia.geisel@mpi-marburg.mpg.de\n",
      "Original publications\n",
      "Stoffel, G. M. M.; Saez, D. A.; DeMirci, H.; Vögeli, B.; Rao, Y.; Zarzycki, J.; Yoshikuni, Y.;  Wakatsuki, S.;  Vöhringer-Martinez, E.; Erb, T. J.\n",
      "Four amino acids define the CO2 binding pocket of enoyl-CoA carboxylases/reductases\n",
      "Proceedings of the National Academy of Sciences 201901471 (2019)\n",
      "DOI\n",
      "Bernhardsgrütter, I.; Schell, K.; Peter, D. M.; Borijan, F.; Saez, D. A.; Vöhringer-Martinez, E.; Erb, T.\n",
      "Awakening the Sleeping Carboxylase Function of Enzymes:Engineering the Natural CO2‑Binding Potential of Reductases\n",
      "Journal of the American Chemical Society 141, 9778 −9782 (2019)\n",
      "DOI\n",
      "Related articles\n",
      "“We are trying to develop CO2 as a source of carbon”\n",
      "November 25, 2016\n",
      "Tobias Erb discusses a synthetic metabolic pathway that fixes carbon dioxide and synthetic biology\n",
      "more\n",
      "You can find this video on YouTube. Click on the image to be redirected there.\n",
      "You can find this video on YouTube. Click on the image to be redirected there.\n",
      "Engineering a more efficient system for harnessing CO2 faster than plants\n",
      "Taking inspiration from nature: Tobias Erb from the MPI for Terrestrial Microbiology on designing a more efficient and faster system for harnessing carbon dioxide.\n",
      "Other Interesting Articles\n",
      "Previous\n",
      "Germany’s President visits German-Brazilian research station\n",
      "January 03, 2023\n",
      "Climate\n",
      "Ecology (E&C)\n",
      "Research Policy\n",
      "Frank-Walter Steinmeier traveled with German Environment Minister Steffi Lemke to ATTO in the Brazilian Amazon rainforest\n",
      "Slime for the climate, delivered by brown algae\n",
      "December 26, 2022\n",
      "Climate\n",
      "Plant Research (E&C)\n",
      "Brown algae could remove up to 0.55 gigatons of carbon dioxide from the atmosphere every year\n",
      "Iron for energy storage\n",
      "December 02, 2022\n",
      "Climate\n",
      "Energy\n",
      "Materials Sciences (M&T)\n",
      "In the futuere the metal could store energy from renewable sources, for example for transportation\n",
      "The role of ice sheets for the climate\n",
      "July 21, 2023\n",
      "Climate\n",
      "Today, ice sheets cover an area that is almost as large as South America. Even though this equals only about three percent of the Earth’s surface, they play, similar to Arctic sea ice, a major role for the climate. In order to investigate ice sheets and their interaction with the climate, we are developing a climate model in which the extent of the ice sheets changes over time. Such changes and interactions have so far been insufficiently represented in conventional climate models.\n",
      "Locating the Anthropocene\n",
      "July 11, 2023\n",
      "Climate\n",
      "Social Sciences\n",
      "Researchers announce major step towards defining a new geological epoch\n",
      "African smoke over the Amazon\n",
      "May 17, 2023\n",
      "Climate\n",
      "Ecology (E&C)\n",
      "At certain times in the year, more soot particles reach the Amazon rainforest from bush fires in Africa than from regional fires.\n",
      "With formic acid towards CO2 neutrality\n",
      "May 12, 2023\n",
      "Climate\n",
      "Synthetic Biology\n",
      "Researchers develop a new method for the sustainable use of carbon dioxide\n",
      "More power from waste heat\n",
      "April 28, 2023\n",
      "Climate\n",
      "Energy\n",
      "Materials Sciences (M&T)\n",
      "An addition of titanium makes a thermoelectric material more efficient\n",
      "Wiggly proteins guard the genome\n",
      "April 27, 2023\n",
      "Cell Biology\n",
      "Proteins\n",
      "Dynamic network in the pores of the nuclear envelope blocks dangerous invaders\n",
      "Green steel produced with ammonia\n",
      "April 17, 2023\n",
      "Climate\n",
      "Energy\n",
      "Materials Sciences (M&T)\n",
      "Ammonia synthesized in sun-rich countries could facilitate sustainable iron- and steelmaking\n",
      "Major advance in super-resolution fluorescence microscopy\n",
      "March 09, 2023\n",
      "Proteins\n",
      "Pushing the MINFLUX technique to higher spatial and temporal precision allows protein dynamics to be observed under physiological conditions\n",
      "Seabirds in the eye of the storm\n",
      "February 23, 2023\n",
      "Climate\n",
      "Tolerance to strong winds and storm avoidance strategy differs among seabird species\n",
      "Germany’s President visits German-Brazilian research station\n",
      "January 03, 2023\n",
      "Climate\n",
      "Ecology (E&C)\n",
      "Research Policy\n",
      "Frank-Walter Steinmeier traveled with German Environment Minister Steffi Lemke to ATTO in the Brazilian Amazon rainforest\n",
      "Slime for the climate, delivered by brown algae\n",
      "December 26, 2022\n",
      "Climate\n",
      "Plant Research (E&C)\n",
      "Brown algae could remove up to 0.55 gigatons of carbon dioxide from the atmosphere every year\n",
      "Iron for energy storage\n",
      "December 02, 2022\n",
      "Climate\n",
      "Energy\n",
      "Materials Sciences (M&T)\n",
      "In the futuere the metal could store energy from renewable sources, for example for transportation\n",
      "The role of ice sheets for the climate\n",
      "July 21, 2023\n",
      "Climate\n",
      "Today, ice sheets cover an area that is almost as large as South America. Even though this equals only about three percent of the Earth’s surface, they play, similar to Arctic sea ice, a major role for the climate. In order to investigate ice sheets and their interaction with the climate, we are developing a climate model in which the extent of the ice sheets changes over time. Such changes and interactions have so far been insufficiently represented in conventional climate models.\n",
      "Locating the Anthropocene\n",
      "July 11, 2023\n",
      "Climate\n",
      "Social Sciences\n",
      "Researchers announce major step towards defining a new geological epoch\n",
      "African smoke over the Amazon\n",
      "May 17, 2023\n",
      "Climate\n",
      "Ecology (E&C)\n",
      "At certain times in the year, more soot particles reach the Amazon rainforest from bush fires in Africa than from regional fires.\n",
      "With formic acid towards CO2 neutrality\n",
      "May 12, 2023\n",
      "Climate\n",
      "Synthetic Biology\n",
      "Researchers develop a new method for the sustainable use of carbon dioxide\n",
      "More power from waste heat\n",
      "April 28, 2023\n",
      "Climate\n",
      "Energy\n",
      "Materials Sciences (M&T)\n",
      "An addition of titanium makes a thermoelectric material more efficient\n",
      "Wiggly proteins guard the genome\n",
      "April 27, 2023\n",
      "Cell Biology\n",
      "Proteins\n",
      "Dynamic network in the pores of the nuclear envelope blocks dangerous invaders\n",
      "Green steel produced with ammonia\n",
      "April 17, 2023\n",
      "Climate\n",
      "Energy\n",
      "Materials Sciences (M&T)\n",
      "Ammonia synthesized in sun-rich countries could facilitate sustainable iron- and steelmaking\n",
      "Major advance in super-resolution fluorescence microscopy\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Open the file in read mode ('r')\n",
    "with open(r'C:\\Users\\1kevi\\Desktop\\projects\\Research\\autoscious-carbon-capture\\question_answering\\autoscious_logs\\web_search\\What_is_the_level_of_ECR_enzyme_activity_in_Kitsatospor_setae_bacteria_\\In_the_active_centre_of_carbon_dioxide_conversion_complete_text.txt', 'r', encoding='utf-8') as file:\n",
    "    # Read the entire contents of the file\n",
    "    website_text = file.read()\n",
    "\n",
    "# Now, 'text' contains the contents of the file\n",
    "print(website_text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [],
   "source": [
    "seed_initial_question_decomposition_prompt = f'''\n",
    "Key questions (index : question): \n",
    "{numbered_key_questions_string}\n",
    "\n",
    "Task: \n",
    "Extract and output as many and any accurate direct quotes from the text that are relevant to the key questions and its most relevant key question index. Rank from most relevant to least. Format as a JSON.\n",
    "```json\n",
    "{{\n",
    "  \"1\": {{\n",
    "    \"quote\": \"\",\n",
    "    \"index\": 0\n",
    "  }},\n",
    "  etc.\n",
    "}}\n",
    "```\n",
    "\n",
    "Text: {website_text}\n",
    "\n",
    "Respond only with the output, with no explanation or conversation.\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Prompt:  \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity vary under different conditions?\n",
      "2. Are there any factors that enhance or inhibit ECR enzyme activity?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract and output as many and any accurate direct quotes from the text that are relevant to the key questions and its most relevant key question index. Rank from most relevant to least. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"1\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Text: In order to overcome the climate crisis, two measures are required: The reduction of carbon dioxide (CO2) emissions, and removal of CO2 from the earth atmosphere. The latter is the goal of Tobias Erb's research group at the Max Planck Institute for Terrestrial Microbiology in Marburg. Their approaches not only aim to benefit climate protection, but also to secure sustainability in the long term: to filter CO2 from the air and make it usable for technology.\n",
      "The Carboxylase active site: Four amino acids are important for CO2 binding in highly efficient CO2 fixing enzymes.\n",
      "© Max-Planck-Institute for Terrestrial Microbiology/Erb\n",
      "The Carboxylase active site: Four amino acids are important for CO2 binding in highly efficient CO2 fixing enzymes.\n",
      "© Max-Planck-Institute for Terrestrial Microbiology/Erb\n",
      "Filtering CO2 efficiently from the air - nature can do this through photosynthesis, converting CO2 into biomass. Unlike industrial technologies which can only use the gas in a highly concentrated form (which in turn consumes fossil energy), photosynthesis works directly with ambient air containing only 0.4% gaseous carbon dioxide. Its secret lies in the enzymes, proteins that act as catalysts to mediate specific chemical reactions, such as the fixation of CO2. In photosynthesis, this reaction is driven by the enzyme RubisCO. However, the efficiency of natural photosynthesis is not very high: in more than a fourth of all cases, RubisCO metabolizes oxygen from the air, which is a strong competitor of CO2 in this reaction.ECR enzymes are faster and more precise than RubisCoTherefore, the Max Planck researchers in Marburg have decided on alternatives to RubisCO. Enoyl-CoA Carboxylase/Reductase enzymes (ECRs) are much more efficient than RubisCO and do not make mistakes with oxygen. After many years of scientific work in order to understand this capacity, the scientists have succeeded in building a functioning process in the test tube that fixes CO2 better than nature itself. Robustness and energy efficiency are the qualities that the team would like to bestow to their artificial photosynthesis. \"Learn from the best\" is the motto: nature itself serves as a model for molecular biology.What is the reason for the high efficiency of ECRs? What is the magic spell to create a turbo CO2 fixator? Max Planck junior researchers Gabriele Stoffel and Iria Bernhardsgrütter pursued this question together with colleagues from Chile and the USA. They analyzed the ECR from the bacterium Kitasatospora setae, currently the fastest known carboxylase. Using a combined approach of structural biology, biochemistry and computer simulations, they were able to understand for the first time how the enzyme binds and converts CO2.Teamwork in the active centre\n",
      "A methionine residue shields the active site from the competing water molecules.\n",
      "© Max-Planck-Institute for Terrestrial Microbiology/Erb\n",
      "A methionine residue shields the active site from the competing water molecules.\n",
      "© Max-Planck-Institute for Terrestrial Microbiology/Erb\n",
      "\"We were surprised to learn that only four amino acids are sufficient to provide control over the CO2 molecule”, explains Gabriele Stoffel, postdoctoral fellow in the Erb department and first author of the study. “Three amino acids – one asparagine, glutamate and a histidine – hold the CO2 in place from two sides. Another amino acid, a phenylalanine, shields the bound CO2 from water, which would inhibit the reaction”, says Stoffel.These findings open up new paths for researchers.  \"We wanted to transfer the capability of binding CO2 to other enzymes. This would offer us much greater possibilities for optimizing photosynthesis,\" says Iria Bernhardsgrütter, PhD student in the research group. In another study, Bernhardsgrütter focused on two candidates for the protein scaffold: Propionyl-CoA synthase (PCS) and Archaeal Enoyl-CoA reductase (AER).Enhancing CO2 fixation capacityBoth enzymes were already able to use CO2, but only with an efficiency of about five percent and with concentrated CO2. Computational models revealed that those enzymes only possessed some of the four amino acids required and those were also misaligned. Iria Bernhardsgrütter succeeded by exchanging amino acids to correct the \"miscasts\" in PCS. Immediately, the efficiency of CO2 increased to around 20 percent. Now the second aspect was targeted, namely shielding the binding site from water. Iria Bernhardsgrütter was also able to solve this problem: another amino acid replacement blocked the water's access to the binding site. The combination of both changes led to a carboxylation rate of almost 95%. Similar experiments with AER increased CO2-conversion efficiencies to almost 90%.This knowledge of the exact requirements of CO2-fixing enzymes and its successful application has brought research a decisive step closer to its high goals: on the one hand, being able to filter CO2 efficiently from the atmosphere, on the other hand, integrating CO2 into sustainable use – towards the recycling of valuable substances following nature's example.\n",
      "You can find this video on YouTube. Click on the image to be redirected there.\n",
      "©\n",
      "MPI f. Terrestrial Mikrobiology/ Erb\n",
      "You can find this video on YouTube. Click on the image to be redirected there.\n",
      "The correct positioning of the gaseous carbon dioxide molecule is essential for the success of the carboxylation reaction.\n",
      "https://www.youtube.com/watch?v=JB7hekpOFFo\n",
      "Science Magazine\n",
      "Events\n",
      "Images of Science\n",
      "On Location\n",
      "Infographics\n",
      "Press Newsletter\n",
      "Contact\n",
      "Prof. Dr. Tobias Erb\n",
      "Director\n",
      "Max Planck Institute for Terrestrial Microbiology, Marburg\n",
      "+49 6421 178-700\n",
      "toerb@mpi-marburg.mpg.de\n",
      "Dr. Virginia Geisel\n",
      "PR Representative\n",
      "Max Planck Institute for Terrestrial Microbiology, Marburg\n",
      "+49 160 91387362\n",
      "virginia.geisel@mpi-marburg.mpg.de\n",
      "Original publications\n",
      "Stoffel, G. M. M.; Saez, D. A.; DeMirci, H.; Vögeli, B.; Rao, Y.; Zarzycki, J.; Yoshikuni, Y.;  Wakatsuki, S.;  Vöhringer-Martinez, E.; Erb, T. J.\n",
      "Four amino acids define the CO2 binding pocket of enoyl-CoA carboxylases/reductases\n",
      "Proceedings of the National Academy of Sciences 201901471 (2019)\n",
      "DOI\n",
      "Bernhardsgrütter, I.; Schell, K.; Peter, D. M.; Borijan, F.; Saez, D. A.; Vöhringer-Martinez, E.; Erb, T.\n",
      "Awakening the Sleeping Carboxylase Function of Enzymes:Engineering the Natural CO2‑Binding Potential of Reductases\n",
      "Journal of the American Chemical Society 141, 9778 −9782 (2019)\n",
      "DOI\n",
      "Related articles\n",
      "“We are trying to develop CO2 as a source of carbon”\n",
      "November 25, 2016\n",
      "Tobias Erb discusses a synthetic metabolic pathway that fixes carbon dioxide and synthetic biology\n",
      "more\n",
      "You can find this video on YouTube. Click on the image to be redirected there.\n",
      "You can find this video on YouTube. Click on the image to be redirected there.\n",
      "Engineering a more efficient system for harnessing CO2 faster than plants\n",
      "Taking inspiration from nature: Tobias Erb from the MPI for Terrestrial Microbiology on designing a more efficient and faster system for harnessing carbon dioxide.\n",
      "Other Interesting Articles\n",
      "Previous\n",
      "Germany’s President visits German-Brazilian research station\n",
      "January 03, 2023\n",
      "Climate\n",
      "Ecology (E&C)\n",
      "Research Policy\n",
      "Frank-Walter Steinmeier traveled with German Environment Minister Steffi Lemke to ATTO in the Brazilian Amazon rainforest\n",
      "Slime for the climate, delivered by brown algae\n",
      "December 26, 2022\n",
      "Climate\n",
      "Plant Research (E&C)\n",
      "Brown algae could remove up to 0.55 gigatons of carbon dioxide from the atmosphere every year\n",
      "Iron for energy storage\n",
      "December 02, 2022\n",
      "Climate\n",
      "Energy\n",
      "Materials Sciences (M&T)\n",
      "In the futuere the metal could store energy from renewable sources, for example for transportation\n",
      "The role of ice sheets for the climate\n",
      "July 21, 2023\n",
      "Climate\n",
      "Today, ice sheets cover an area that is almost as large as South America. Even though this equals only about three percent of the Earth’s surface, they play, similar to Arctic sea ice, a major role for the climate. In order to investigate ice sheets and their interaction with the climate, we are developing a climate model in which the extent of the ice sheets changes over time. Such changes and interactions have so far been insufficiently represented in conventional climate models.\n",
      "Locating the Anthropocene\n",
      "July 11, 2023\n",
      "Climate\n",
      "Social Sciences\n",
      "Researchers announce major step towards defining a new geological epoch\n",
      "African smoke over the Amazon\n",
      "May 17, 2023\n",
      "Climate\n",
      "Ecology (E&C)\n",
      "At certain times in the year, more soot particles reach the Amazon rainforest from bush fires in Africa than from regional fires.\n",
      "With formic acid towards CO2 neutrality\n",
      "May 12, 2023\n",
      "Climate\n",
      "Synthetic Biology\n",
      "Researchers develop a new method for the sustainable use of carbon dioxide\n",
      "More power from waste heat\n",
      "April 28, 2023\n",
      "Climate\n",
      "Energy\n",
      "Materials Sciences (M&T)\n",
      "An addition of titanium makes a thermoelectric material more efficient\n",
      "Wiggly proteins guard the genome\n",
      "April 27, 2023\n",
      "Cell Biology\n",
      "Proteins\n",
      "Dynamic network in the pores of the nuclear envelope blocks dangerous invaders\n",
      "Green steel produced with ammonia\n",
      "April 17, 2023\n",
      "Climate\n",
      "Energy\n",
      "Materials Sciences (M&T)\n",
      "Ammonia synthesized in sun-rich countries could facilitate sustainable iron- and steelmaking\n",
      "Major advance in super-resolution fluorescence microscopy\n",
      "March 09, 2023\n",
      "Proteins\n",
      "Pushing the MINFLUX technique to higher spatial and temporal precision allows protein dynamics to be observed under physiological conditions\n",
      "Seabirds in the eye of the storm\n",
      "February 23, 2023\n",
      "Climate\n",
      "Tolerance to strong winds and storm avoidance strategy differs among seabird species\n",
      "Germany’s President visits German-Brazilian research station\n",
      "January 03, 2023\n",
      "Climate\n",
      "Ecology (E&C)\n",
      "Research Policy\n",
      "Frank-Walter Steinmeier traveled with German Environment Minister Steffi Lemke to ATTO in the Brazilian Amazon rainforest\n",
      "Slime for the climate, delivered by brown algae\n",
      "December 26, 2022\n",
      "Climate\n",
      "Plant Research (E&C)\n",
      "Brown algae could remove up to 0.55 gigatons of carbon dioxide from the atmosphere every year\n",
      "Iron for energy storage\n",
      "December 02, 2022\n",
      "Climate\n",
      "Energy\n",
      "Materials Sciences (M&T)\n",
      "In the futuere the metal could store energy from renewable sources, for example for transportation\n",
      "The role of ice sheets for the climate\n",
      "July 21, 2023\n",
      "Climate\n",
      "Today, ice sheets cover an area that is almost as large as South America. Even though this equals only about three percent of the Earth’s surface, they play, similar to Arctic sea ice, a major role for the climate. In order to investigate ice sheets and their interaction with the climate, we are developing a climate model in which the extent of the ice sheets changes over time. Such changes and interactions have so far been insufficiently represented in conventional climate models.\n",
      "Locating the Anthropocene\n",
      "July 11, 2023\n",
      "Climate\n",
      "Social Sciences\n",
      "Researchers announce major step towards defining a new geological epoch\n",
      "African smoke over the Amazon\n",
      "May 17, 2023\n",
      "Climate\n",
      "Ecology (E&C)\n",
      "At certain times in the year, more soot particles reach the Amazon rainforest from bush fires in Africa than from regional fires.\n",
      "With formic acid towards CO2 neutrality\n",
      "May 12, 2023\n",
      "Climate\n",
      "Synthetic Biology\n",
      "Researchers develop a new method for the sustainable use of carbon dioxide\n",
      "More power from waste heat\n",
      "April 28, 2023\n",
      "Climate\n",
      "Energy\n",
      "Materials Sciences (M&T)\n",
      "An addition of titanium makes a thermoelectric material more efficient\n",
      "Wiggly proteins guard the genome\n",
      "April 27, 2023\n",
      "Cell Biology\n",
      "Proteins\n",
      "Dynamic network in the pores of the nuclear envelope blocks dangerous invaders\n",
      "Green steel produced with ammonia\n",
      "April 17, 2023\n",
      "Climate\n",
      "Energy\n",
      "Materials Sciences (M&T)\n",
      "Ammonia synthesized in sun-rich countries could facilitate sustainable iron- and steelmaking\n",
      "Major advance in super-resolution fluorescence microscopy\n",
      "\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"0\\\": {\\n    \\\"quote\\\": \\\"They analyzed the ECR from the bacterium Kitasatospora setae, currently the fastest known carboxylase.\\\",\\n    \\\"index\\\": 1\\n  },\\n  \\\"1\\\": {\\n    \\\"quote\\\": \\\"\\\\\\\"We were surprised to learn that only four amino acids are sufficient to provide control over the CO2 molecule\\u201d, explains Gabriele Stoffel, postdoctoral fellow in the Erb department and first author of the study. \\u201cThree amino acids \\u2013 one asparagine, glutamate and a histidine \\u2013 hold the CO2 in place from two sides. Another amino acid, a phenylalanine, shields the bound CO2 from water, which would inhibit the reaction\\u201d, says Stoffel.\\\",\\n    \\\"index\\\": 2\\n  },\\n  \\\"2\\\": {\\n    \\\"quote\\\": \\\"Both enzymes were already able to use CO2, but only with an efficiency of about five percent and with concentrated CO2. Computational models revealed that those enzymes only possessed some of the four amino acids required and those were also misaligned. Iria Bernhardsgr\\u00fctter succeeded by exchanging amino acids to correct the \\\\\\\"miscasts\\\\\\\" in PCS. Immediately, the efficiency of CO2 increased to around 20 percent. Now the second aspect was targeted, namely shielding the binding site from water. Iria Bernhardsgr\\u00fctter was also able to solve this problem: another amino acid replacement blocked the water's access to the binding site. The combination of both changes led to a carboxylation rate of almost 95%. Similar experiments with AER increased CO2-conversion efficiencies to almost 90%.\\\",\\n    \\\"index\\\": 2\\n  }\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1690762725,\n",
      "  \"id\": \"chatcmpl-7iAuTn1TsYvZVRFFmltr26dcCOtH4\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 348,\n",
      "    \"prompt_tokens\": 2868,\n",
      "    \"total_tokens\": 3216\n",
      "  }\n",
      "}\n",
      "{\n",
      "  \"0\": {\n",
      "    \"quote\": \"They analyzed the ECR from the bacterium Kitasatospora setae, currently the fastest known carboxylase.\",\n",
      "    \"index\": 1\n",
      "  },\n",
      "  \"1\": {\n",
      "    \"quote\": \"\\\"We were surprised to learn that only four amino acids are sufficient to provide control over the CO2 molecule”, explains Gabriele Stoffel, postdoctoral fellow in the Erb department and first author of the study. “Three amino acids – one asparagine, glutamate and a histidine – hold the CO2 in place from two sides. Another amino acid, a phenylalanine, shields the bound CO2 from water, which would inhibit the reaction”, says Stoffel.\",\n",
      "    \"index\": 2\n",
      "  },\n",
      "  \"2\": {\n",
      "    \"quote\": \"Both enzymes were already able to use CO2, but only with an efficiency of about five percent and with concentrated CO2. Computational models revealed that those enzymes only possessed some of the four amino acids required and those were also misaligned. Iria Bernhardsgrütter succeeded by exchanging amino acids to correct the \\\"miscasts\\\" in PCS. Immediately, the efficiency of CO2 increased to around 20 percent. Now the second aspect was targeted, namely shielding the binding site from water. Iria Bernhardsgrütter was also able to solve this problem: another amino acid replacement blocked the water's access to the binding site. The combination of both changes led to a carboxylation rate of almost 95%. Similar experiments with AER increased CO2-conversion efficiencies to almost 90%.\",\n",
      "    \"index\": 2\n",
      "  }\n",
      "}\n"
     ]
    }
   ],
   "source": [
    "res = chat_openai(seed_initial_question_decomposition_prompt, model=\"gpt-3.5-turbo\")\n",
    "print(res[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\n",
      "  \"0\": {\n",
      "    \"quote\": \"They analyzed the ECR from the bacterium Kitasatospora setae, currently the fastest known carboxylase.\",\n",
      "    \"index\": 1\n",
      "  },\n",
      "  \"1\": {\n",
      "    \"quote\": \"\\\"We were surprised to learn that only four amino acids are sufficient to provide control over the CO2 molecule”, explains Gabriele Stoffel, postdoctoral fellow in the Erb department and first author of the study. “Three amino acids – one asparagine, glutamate and a histidine – hold the CO2 in place from two sides. Another amino acid, a phenylalanine, shields the bound CO2 from water, which would inhibit the reaction”, says Stoffel.\",\n",
      "    \"index\": 2\n",
      "  },\n",
      "  \"2\": {\n",
      "    \"quote\": \"Both enzymes were already able to use CO2, but only with an efficiency of about five percent and with concentrated CO2. Computational models revealed that those enzymes only possessed some of the four amino acids required and those were also misaligned. Iria Bernhardsgrütter succeeded by exchanging amino acids to correct the \\\"miscasts\\\" in PCS. Immediately, the efficiency of CO2 increased to around 20 percent. Now the second aspect was targeted, namely shielding the binding site from water. Iria Bernhardsgrütter was also able to solve this problem: another amino acid replacement blocked the water's access to the binding site. The combination of both changes led to a carboxylation rate of almost 95%. Similar experiments with AER increased CO2-conversion efficiencies to almost 90%.\",\n",
      "    \"index\": 2\n",
      "  }\n",
      "} <class 'str'>\n"
     ]
    }
   ],
   "source": [
    "print(res[0], type(res[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [],
   "source": [
    "res_json = json.loads(res[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'str'>\n"
     ]
    }
   ],
   "source": [
    "print(type(res_json['2']['quote']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'0': {'quote': 'ECR enzymes are faster and more precise than RubisCo',\n",
       "  'index': 1},\n",
       " '1': {'quote': '\"We were surprised to learn that only four amino acids are sufficient to provide control over the CO2 molecule”, explains Gabriele Stoffel, postdoctoral fellow in the Erb department and first author of the study. “Three amino acids – one asparagine, glutamate and a histidine – hold the CO2 in place from two sides. Another amino acid, a phenylalanine, shields the bound CO2 from water, which would inhibit the reaction”, says Stoffel.',\n",
       "  'index': 2},\n",
       " '2': {'quote': 'Both enzymes were already able to use CO2, but only with an efficiency of about five percent and with concentrated CO2. Computational models revealed that those enzymes only possessed some of the four amino acids required and those were also misaligned. Iria Bernhardsgrütter succeeded by exchanging amino acids to correct the \"miscasts\" in PCS. Immediately, the efficiency of CO2 increased to around 20 percent. Now the second aspect was targeted, namely shielding the binding site from water. Iria Bernhardsgrütter was also able to solve this problem: another amino acid replacement blocked the water\\'s access to the binding site. The combination of both changes led to a carboxylation rate of almost 95%. Similar experiments with AER increased CO2-conversion efficiencies to almost 90%.',\n",
       "  'index': 2}}"
      ]
     },
     "execution_count": 156,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "res_json\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'res_json' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[5], line 3\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[39mimport\u001b[39;00m \u001b[39mos\u001b[39;00m\n\u001b[0;32m      2\u001b[0m \u001b[39m# iterate over the dictionary\u001b[39;00m\n\u001b[1;32m----> 3\u001b[0m \u001b[39mfor\u001b[39;00m key, value \u001b[39min\u001b[39;00m res_json\u001b[39m.\u001b[39mitems():\n\u001b[0;32m      4\u001b[0m     \u001b[39m# get the index and quote\u001b[39;00m\n\u001b[0;32m      5\u001b[0m     index \u001b[39m=\u001b[39m value[\u001b[39m'\u001b[39m\u001b[39mindex\u001b[39m\u001b[39m'\u001b[39m]\n\u001b[0;32m      6\u001b[0m     quote \u001b[39m=\u001b[39m value[\u001b[39m'\u001b[39m\u001b[39mquote\u001b[39m\u001b[39m'\u001b[39m]\n",
      "\u001b[1;31mNameError\u001b[0m: name 'res_json' is not defined"
     ]
    }
   ],
   "source": [
    "import os\n",
    "# iterate over the dictionary\n",
    "for key, value in res_json.items():\n",
    "    # get the index and quote\n",
    "    index = value['index']\n",
    "    quote = value['quote']\n",
    "\n",
    "    # create a file name with the format 'f_{index}.txt'\n",
    "    file_name = f'autoscious_logs/kd_1/h_1/f_{index}.txt'\n",
    "\n",
    "    # open the file in append mode ('a') to add the quote\n",
    "    # if you want to overwrite the existing file (if any), use 'w' mode\n",
    "    with open(file_name, 'a', encoding='utf-8') as f:\n",
    "        f.write(quote.replace('/\"', '\"') + os.linesep)  # write the quote to the file and add a new line"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get the list of key questions\n",
    "def get_key_questions_list_string(search_query_file_safe):\n",
    "    file_path = f'autoscious_logs/decompositions/{search_query_file_safe}.json'\n",
    "\n",
    "    with open(file_path, 'r') as f:\n",
    "            question_decomposition = json.load(f)\n",
    "    key_questions = question_decomposition['key_drivers']['1']['hypotheses']['1']['key_questions']\n",
    "    print(\"\\nkey_questions\\n\", key_questions)\n",
    "\n",
    "    numbered_key_questions_string = \"\"\n",
    "    for key, value in key_questions.items():\n",
    "        first_two_words = ' '.join(value.split())\n",
    "        numbered_key_questions_string += f'{int(key) - 1}. {first_two_words}\\n'\n",
    "\n",
    "    print(\"\\nnumbered_key_questions_string\\n\", numbered_key_questions_string)\n",
    "    return key_questions, numbered_key_questions_string"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "def extract_facts_from_website_text(key_questions_list_string, website_text, website_url):\n",
    "    seed_initial_question_decomposition_prompt = f'''\n",
    "Key questions (index : question): \n",
    "{key_questions_list_string}\n",
    "\n",
    "Task: \n",
    "Extract and output as many accurate direct quotes from the text that are relevant to answering the key questions and its most relevant key question index. Format as a JSON.\n",
    "```json\n",
    "{{\n",
    "  \"1\": {{\n",
    "    \"quote\": \"\",\n",
    "    \"index\": 0\n",
    "  }},\n",
    "  etc.\n",
    "}}\n",
    "```\n",
    "\n",
    "Text: {website_text}\n",
    "\n",
    "Respond only with the output, with no explanation or conversation.\n",
    "'''\n",
    "    # Ask GPT the prompt\n",
    "    print(\"seed_initial_question_decomposition_prompt\", seed_initial_question_decomposition_prompt)\n",
    "    res = chat_openai(seed_initial_question_decomposition_prompt, model=\"gpt-3.5-turbo\")\n",
    "    print(\"Extracted quotes: \", res[0])\n",
    "\n",
    "    # Save the quote to the corresponding key question index file\n",
    "    res_json = json.loads(res[0])\n",
    "    for key, value in res_json.items():\n",
    "        index = value['index']\n",
    "        quote = value['quote']\n",
    "\n",
    "        file_name = f'autoscious_logs/kd_1/h_1/f_{index}.txt'\n",
    "\n",
    "        with open(file_name, 'a', encoding='utf-8') as f:\n",
    "            f.write(quote.replace('/\"', '\"') + f\"[{website_url}]\" + os.linesep)\n",
    "\n",
    "    return"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def chunk_text(text: str, chunk_size: int, overlap: int) -> list[str]:\n",
    "    \"\"\"Chunk a text into smaller parts with some overlap.\n",
    "\n",
    "    Args:\n",
    "        text (str): The text to chunk.\n",
    "        chunk_size (int): The desired size of each chunk.\n",
    "        overlap (int): The desired overlap between chunks.\n",
    "\n",
    "    Returns:\n",
    "        list[str]: The text chunks.\n",
    "    \"\"\"\n",
    "    chunks = []\n",
    "    for i in range(0, len(text), chunk_size - overlap):\n",
    "        chunks.append(text[i : i + chunk_size])\n",
    "    return chunks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "key_questions {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity vary under different conditions?', '3': 'Are there any factors that enhance or inhibit ECR enzyme activity?'}\n",
      "\n",
      "numbered_key_questions_string 0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity vary under different conditions?\n",
      "2. Are there any factors that enhance or inhibit ECR enzyme activity?\n",
      "\n",
      "Search result:  Intersubunit Coupling Enables Fast CO2-Fixation by Reductive ... https://pubs.acs.org/doi/10.1021/acscentsci.2c00057\n",
      "Going through url:  https://pubs.acs.org/doi/10.1021/acscentsci.2c00057\n",
      "select chrome options!\n",
      "hard coding chrome\n",
      "setting up chrome driver\n",
      "Driver is getting url\n",
      "Driver got url\n",
      "Driver has found page source\n",
      "Handing off to Beautiful Soup!\n",
      "done extractin\n",
      "Chunk: 0 / 33\n",
      "seed_initial_question_decomposition_prompt \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity vary under different conditions?\n",
      "2. Are there any factors that enhance or inhibit ECR enzyme activity?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract and output as many accurate direct quotes from the text that are relevant to answering the key questions and its most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"1\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Text: ACS\n",
      "ACS Publications\n",
      "C&EN\n",
      "CAS\n",
      "Find my institution\n",
      "Log In\n",
      "Intersubunit Coupling Enables Fast CO2-Fixation by Reductive Carboxylases\n",
      "Quick View\n",
      "ShareShare onFacebookTwitterWeChatLinked InRedditEmail\n",
      "ACS Cent. Sci.All Publications/WebsiteOR SEARCH CITATIONSACS Central ScienceAccounts of Chemical ResearchAccounts of Materials ResearchACS Agricultural Science & TechnologyACS Applied Bio MaterialsACS Applied Electronic MaterialsACS Applied Energy MaterialsACS Applied Engineering MaterialsACS Applied Materials & InterfacesACS Applied Nano MaterialsACS Applied Optical MaterialsACS Applied Polymer MaterialsACS Bio & Med Chem AuACS Biomaterials Science & EngineeringACS CatalysisACS Chemical BiologyACS Chemical Health & SafetyACS Chemical NeuroscienceACS Combinatorial ScienceACS Earth and Space ChemistryACS Energy LettersACS Engineering AuACS Environmental AuACS ES&T AirACS ES&T EngineeringACS ES&T WaterACS Food Science & TechnologyACS Infectious DiseasesACS Macro LettersACS Materials AuACS Materials LettersACS Measurement Science AuACS Medicinal Chemistry LettersACS NanoACS Nanoscience AuACS OmegaACS Organic & Inorganic AuACS Pharmacology & Translational ScienceACS PhotonicsACS Physical Chemistry AuACS Polymers AuACS SensorsACS Sustainable Chemistry & EngineeringACS Sustainable Resource ManagementACS Synthetic BiologyAnalytical ChemistryBiochemistryBioconjugate ChemistryBiomacromoleculesBiotechnology ProgressC&EN Global EnterpriseChem & Bio EngineeringChemical & Biomedical ImagingChemical & Engineering News ArchiveChemical Health & SafetyChemical Health & SafetyChemical Research in ToxicologyChemical ReviewsChemistry of MaterialsCrystal Growth & DesignEnergy & FuelsEnvironment & HealthEnvironmental Science & TechnologyEnvironmental Science & Technology LettersI&EC Product Research and DevelopmentIndustrial & Engineering ChemistryIndustrial & Engineering Chemistry Analytical EditionIndustrial & Engineering Chemistry Chemical & Engineering Data SeriesIndustrial & Engineering Chemistry FundamentalsIndustrial & Engineering Chemistry Process Design and DevelopmentIndustrial & Engineering Chemistry Product Research and DevelopmentIndustrial & Engineering Chemistry ResearchIndustrial and Engineering Chemistry, News EditionInorganic ChemistryJACS AuJournal of the American Chemical SocietyJournal of Agricultural and Food ChemistryJournal of Chemical & Engineering DataJournal of Chemical DocumentationJournal of Chemical EducationJournal of Chemical Health & SafetyJournal of Chemical Information and Computer SciencesJournal of Chemical Information and ModelingJournal of Chemical Theory and ComputationJournal of Combinatorial ChemistryJournal of Industrial & Engineering ChemistryJournal of Medicinal and Pharmaceutical ChemistryJournal of Medicinal ChemistryJournal of Natural ProductsThe Journal of Organic ChemistryThe Journal of Physical ChemistryThe Journal of Physical ChemistryThe Journal of Physical Chemistry AThe Journal of Physical Chemistry BThe Journal of Physical Chemistry CThe Journal of Physical Chemistry LettersJournal of Proteome ResearchJournal of the American Society for Mass SpectrometryJournal of the American Society for Mass SpectrometryJournal of the American Society for Mass SpectrometryLangmuirMacromoleculesMolecular PharmaceuticsNano LettersNews Edition, American Chemical SocietyOrganic LettersOrganic Process Research & DevelopmentOrganometallicsPrecision ChemistryProduct R&DSciMeetingsThe Journal of Physical and Colloid Chemistry\n",
      "My Activity\n",
      "Recently ViewedYou have not visited any articles yet, Please visit some articles to see contents here.\n",
      "Publications\n",
      "publications\n",
      "my Activity\n",
      "Recently Viewed\n",
      "user resources\n",
      "Access Options\n",
      "Authors &\n",
      "Reviewers\n",
      "ACS Members\n",
      "Virtual Issues\n",
      "eAlerts\n",
      "RSS & Mobile\n",
      "for organizations\n",
      "Products & Services\n",
      "Get Access\n",
      "Manage My Account\n",
      "support\n",
      "Website Demos & Tutorials\n",
      "Support FAQs\n",
      "Live Chat with Agent\n",
      "For Advertisers\n",
      "For\n",
      "Librarians & Account Managers\n",
      "pairing\n",
      "Pair a device\n",
      "My Profile\n",
      "Login\n",
      "Logout\n",
      "Pair a device\n",
      "about us\n",
      "Overview\n",
      "ACS & Open Access\n",
      "Partners\n",
      "Blog\n",
      "Events\n",
      "Recently ViewedYou have not visited any articles yet, Please visit some articles to see contents here.\n",
      "PublicationsCONTENT TYPESAll TypesSUBJECTSPublications: All Types\n",
      "Download Hi-Res ImageDownload to MS-PowerPointCite This:ACS Cent. Sci. 2022, 8, 8, 1091-1101\n",
      "ADVERTISEMENT\n",
      "RETURN TO ISSUEPREVResearch ArticleNEXTGet e-AlertscloseIntersubunit Coupling Enables Fast CO2-Fixation by Reductive CarboxylasesHasan DeMirci*Hasan DeMirciBiosciences Division, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesPULSE Institute, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesDepartment of Molecular Biology and Genetics, Koc University, 34450 Sariyer/Istanbul, Turkey*Email for H.D.: hdemirci@ku.edu.trMore by Hasan DeMirci, Yashas RaoYashas RaoBiosciences Division, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesDepartamento de Físico Química, Fa\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Prompt:  \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity vary under different conditions?\n",
      "2. Are there any factors that enhance or inhibit ECR enzyme activity?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract and output as many accurate direct quotes from the text that are relevant to answering the key questions and its most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"1\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Text: ACS\n",
      "ACS Publications\n",
      "C&EN\n",
      "CAS\n",
      "Find my institution\n",
      "Log In\n",
      "Intersubunit Coupling Enables Fast CO2-Fixation by Reductive Carboxylases\n",
      "Quick View\n",
      "ShareShare onFacebookTwitterWeChatLinked InRedditEmail\n",
      "ACS Cent. Sci.All Publications/WebsiteOR SEARCH CITATIONSACS Central ScienceAccounts of Chemical ResearchAccounts of Materials ResearchACS Agricultural Science & TechnologyACS Applied Bio MaterialsACS Applied Electronic MaterialsACS Applied Energy MaterialsACS Applied Engineering MaterialsACS Applied Materials & InterfacesACS Applied Nano MaterialsACS Applied Optical MaterialsACS Applied Polymer MaterialsACS Bio & Med Chem AuACS Biomaterials Science & EngineeringACS CatalysisACS Chemical BiologyACS Chemical Health & SafetyACS Chemical NeuroscienceACS Combinatorial ScienceACS Earth and Space ChemistryACS Energy LettersACS Engineering AuACS Environmental AuACS ES&T AirACS ES&T EngineeringACS ES&T WaterACS Food Science & TechnologyACS Infectious DiseasesACS Macro LettersACS Materials AuACS Materials LettersACS Measurement Science AuACS Medicinal Chemistry LettersACS NanoACS Nanoscience AuACS OmegaACS Organic & Inorganic AuACS Pharmacology & Translational ScienceACS PhotonicsACS Physical Chemistry AuACS Polymers AuACS SensorsACS Sustainable Chemistry & EngineeringACS Sustainable Resource ManagementACS Synthetic BiologyAnalytical ChemistryBiochemistryBioconjugate ChemistryBiomacromoleculesBiotechnology ProgressC&EN Global EnterpriseChem & Bio EngineeringChemical & Biomedical ImagingChemical & Engineering News ArchiveChemical Health & SafetyChemical Health & SafetyChemical Research in ToxicologyChemical ReviewsChemistry of MaterialsCrystal Growth & DesignEnergy & FuelsEnvironment & HealthEnvironmental Science & TechnologyEnvironmental Science & Technology LettersI&EC Product Research and DevelopmentIndustrial & Engineering ChemistryIndustrial & Engineering Chemistry Analytical EditionIndustrial & Engineering Chemistry Chemical & Engineering Data SeriesIndustrial & Engineering Chemistry FundamentalsIndustrial & Engineering Chemistry Process Design and DevelopmentIndustrial & Engineering Chemistry Product Research and DevelopmentIndustrial & Engineering Chemistry ResearchIndustrial and Engineering Chemistry, News EditionInorganic ChemistryJACS AuJournal of the American Chemical SocietyJournal of Agricultural and Food ChemistryJournal of Chemical & Engineering DataJournal of Chemical DocumentationJournal of Chemical EducationJournal of Chemical Health & SafetyJournal of Chemical Information and Computer SciencesJournal of Chemical Information and ModelingJournal of Chemical Theory and ComputationJournal of Combinatorial ChemistryJournal of Industrial & Engineering ChemistryJournal of Medicinal and Pharmaceutical ChemistryJournal of Medicinal ChemistryJournal of Natural ProductsThe Journal of Organic ChemistryThe Journal of Physical ChemistryThe Journal of Physical ChemistryThe Journal of Physical Chemistry AThe Journal of Physical Chemistry BThe Journal of Physical Chemistry CThe Journal of Physical Chemistry LettersJournal of Proteome ResearchJournal of the American Society for Mass SpectrometryJournal of the American Society for Mass SpectrometryJournal of the American Society for Mass SpectrometryLangmuirMacromoleculesMolecular PharmaceuticsNano LettersNews Edition, American Chemical SocietyOrganic LettersOrganic Process Research & DevelopmentOrganometallicsPrecision ChemistryProduct R&DSciMeetingsThe Journal of Physical and Colloid Chemistry\n",
      "My Activity\n",
      "Recently ViewedYou have not visited any articles yet, Please visit some articles to see contents here.\n",
      "Publications\n",
      "publications\n",
      "my Activity\n",
      "Recently Viewed\n",
      "user resources\n",
      "Access Options\n",
      "Authors &\n",
      "Reviewers\n",
      "ACS Members\n",
      "Virtual Issues\n",
      "eAlerts\n",
      "RSS & Mobile\n",
      "for organizations\n",
      "Products & Services\n",
      "Get Access\n",
      "Manage My Account\n",
      "support\n",
      "Website Demos & Tutorials\n",
      "Support FAQs\n",
      "Live Chat with Agent\n",
      "For Advertisers\n",
      "For\n",
      "Librarians & Account Managers\n",
      "pairing\n",
      "Pair a device\n",
      "My Profile\n",
      "Login\n",
      "Logout\n",
      "Pair a device\n",
      "about us\n",
      "Overview\n",
      "ACS & Open Access\n",
      "Partners\n",
      "Blog\n",
      "Events\n",
      "Recently ViewedYou have not visited any articles yet, Please visit some articles to see contents here.\n",
      "PublicationsCONTENT TYPESAll TypesSUBJECTSPublications: All Types\n",
      "Download Hi-Res ImageDownload to MS-PowerPointCite This:ACS Cent. Sci. 2022, 8, 8, 1091-1101\n",
      "ADVERTISEMENT\n",
      "RETURN TO ISSUEPREVResearch ArticleNEXTGet e-AlertscloseIntersubunit Coupling Enables Fast CO2-Fixation by Reductive CarboxylasesHasan DeMirci*Hasan DeMirciBiosciences Division, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesPULSE Institute, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesDepartment of Molecular Biology and Genetics, Koc University, 34450 Sariyer/Istanbul, Turkey*Email for H.D.: hdemirci@ku.edu.trMore by Hasan DeMirci, Yashas RaoYashas RaoBiosciences Division, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesDepartamento de Físico Química, Fa\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"0\\\": {\\n    \\\"quote\\\": \\\"Intersubunit Coupling Enables Fast CO2-Fixation by Reductive Carboxylases\\\",\\n    \\\"index\\\": 0\\n  }\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1690769596,\n",
      "  \"id\": \"chatcmpl-7iChIGSrktnnwfxglqIxQGYr4Pegh\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 42,\n",
      "    \"prompt_tokens\": 1172,\n",
      "    \"total_tokens\": 1214\n",
      "  }\n",
      "}\n",
      "Extracted quotes:  {\n",
      "  \"0\": {\n",
      "    \"quote\": \"Intersubunit Coupling Enables Fast CO2-Fixation by Reductive Carboxylases\",\n",
      "    \"index\": 0\n",
      "  }\n",
      "}\n",
      "Chunk: 1 / 33\n",
      "seed_initial_question_decomposition_prompt \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity vary under different conditions?\n",
      "2. Are there any factors that enhance or inhibit ECR enzyme activity?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract and output as many accurate direct quotes from the text that are relevant to answering the key questions and its most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"1\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Text: nto de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, ChileMore by Yashas Rao, Gabriele M. StoffelGabriele M. StoffelDepartment of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, GermanyMore by Gabriele M. Stoffel, Bastian VögeliBastian VögeliDepartment of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, GermanyMore by Bastian Vögeli, Kristina SchellKristina SchellDepartment of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, GermanyMore by Kristina Schell, Aharon GomezAharon GomezDepartamento de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, ChileMore by Aharon Gomezhttps://orcid.org/0000-0003-4820-4697, Alexander BatyukAlexander BatyukLinac Coherent Light Source, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesMore by Alexander Batyukhttps://orcid.org/0000-0002-9393-2880, Cornelius GatiCornelius GatiBiosciences Division, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesStructural Biology Department, Stanford University Stanford, California 94305, United StatesMore by Cornelius Gati, Raymond G. SierraRaymond G. SierraLinac Coherent Light Source, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesMore by Raymond G. Sierra, Mark S. HunterMark S. HunterLinac Coherent Light Source, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesMore by Mark S. Hunter, E. Han DaoE. Han DaoBiosciences Division, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesPULSE Institute, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesMore by E. Han Dao, Halil I. CiftciHalil I. CiftciPULSE Institute, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesMore by Halil I. Ciftci, Brandon HayesBrandon HayesLinac Coherent Light Source, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesMore by Brandon Hayes, Fredric PoitevinFredric PoitevinLinac Coherent Light Source, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesMore by Fredric Poitevin, Po-Nan LiPo-Nan LiBiosciences Division, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesElectrical Engineering Department, Stanford University Stanford, California 94305, United StatesMore by Po-Nan Lihttps://orcid.org/0000-0002-7917-7444, Manat KaurManat KaurStructural Biology Department, Stanford University Stanford, California 94305, United StatesMore by Manat Kaur, Kensuke TonoKensuke TonoRIKEN SPring-8 Center, Sayo, Hyogo 679-5148, JapanJapan Synchrotron Radiation Research Institute, Sayo, Hyogo 679-5198, JapanMore by Kensuke Tono, David Adrian SaezDavid Adrian SaezDepartamento de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, ChileDepartamento de Farmacia, Facultad de Farmacia, Universidad de Concepción, Concepción 00000, ChileMore by David Adrian Saez, Samuel DeutschSamuel DeutschU.S. Department of Energy Joint Genome Institute, Lawrence Berkeley National Laboratory, Walnut Creek, California 94720, United StatesMore by Samuel Deutsch, Yasuo YoshikuniYasuo YoshikuniU.S. Department of Energy Joint Genome Institute, Lawrence Berkeley National Laboratory, Walnut Creek, California 94720, United StatesMore by Yasuo Yoshikunihttps://orcid.org/0000-0002-8372-640X, Helmut GrubmüllerHelmut GrubmüllerDepartment of Theoretical and Computational Biophysics, Max Planck Institute for Multidisciplinary Sciences, Am Fassberg 11, 37077 Göttingen, GermanyMore by Helmut Grubmüllerhttps://orcid.org/0000-0002-3270-3144, Tobias J. Erb*Tobias J. ErbDepartment of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, GermanyLOEWE Center for Synthetic Microbiology (SYNMIKRO), 35032 Marburg, Germany*Email for T.J.E.: toerb@mpi-marburg.mpg.deMore by Tobias J. Erbhttps://orcid.org/0000-0003-3685-0894, Esteban Vöhringer-Martinez*Esteban Vöhringer-MartinezDepartamento de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, Chile*Email for E.V.-M.: evohringer@udec.clMore by Esteban Vöhringer-Martinezhttps://orcid.org/0000-0003-1785-4558, and Soichi Wakatsuki*Soichi WakatsukiBiosciences Division, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesStructural Biology Department, Stanford University Stanford, California 94305, United States*Email for S.W.: soichi.wakatsuki@stanford.eduMore by Soichi Wakatsukihttps://orcid.org/0000-0001-5896-7968Cite this: ACS Cent. Sci. 2022, 8, 8, 1091–1101Publicati\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Prompt:  \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity vary under different conditions?\n",
      "2. Are there any factors that enhance or inhibit ECR enzyme activity?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract and output as many accurate direct quotes from the text that are relevant to answering the key questions and its most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"1\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Text: nto de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, ChileMore by Yashas Rao, Gabriele M. StoffelGabriele M. StoffelDepartment of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, GermanyMore by Gabriele M. Stoffel, Bastian VögeliBastian VögeliDepartment of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, GermanyMore by Bastian Vögeli, Kristina SchellKristina SchellDepartment of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, GermanyMore by Kristina Schell, Aharon GomezAharon GomezDepartamento de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, ChileMore by Aharon Gomezhttps://orcid.org/0000-0003-4820-4697, Alexander BatyukAlexander BatyukLinac Coherent Light Source, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesMore by Alexander Batyukhttps://orcid.org/0000-0002-9393-2880, Cornelius GatiCornelius GatiBiosciences Division, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesStructural Biology Department, Stanford University Stanford, California 94305, United StatesMore by Cornelius Gati, Raymond G. SierraRaymond G. SierraLinac Coherent Light Source, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesMore by Raymond G. Sierra, Mark S. HunterMark S. HunterLinac Coherent Light Source, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesMore by Mark S. Hunter, E. Han DaoE. Han DaoBiosciences Division, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesPULSE Institute, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesMore by E. Han Dao, Halil I. CiftciHalil I. CiftciPULSE Institute, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesMore by Halil I. Ciftci, Brandon HayesBrandon HayesLinac Coherent Light Source, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesMore by Brandon Hayes, Fredric PoitevinFredric PoitevinLinac Coherent Light Source, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesMore by Fredric Poitevin, Po-Nan LiPo-Nan LiBiosciences Division, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesElectrical Engineering Department, Stanford University Stanford, California 94305, United StatesMore by Po-Nan Lihttps://orcid.org/0000-0002-7917-7444, Manat KaurManat KaurStructural Biology Department, Stanford University Stanford, California 94305, United StatesMore by Manat Kaur, Kensuke TonoKensuke TonoRIKEN SPring-8 Center, Sayo, Hyogo 679-5148, JapanJapan Synchrotron Radiation Research Institute, Sayo, Hyogo 679-5198, JapanMore by Kensuke Tono, David Adrian SaezDavid Adrian SaezDepartamento de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, ChileDepartamento de Farmacia, Facultad de Farmacia, Universidad de Concepción, Concepción 00000, ChileMore by David Adrian Saez, Samuel DeutschSamuel DeutschU.S. Department of Energy Joint Genome Institute, Lawrence Berkeley National Laboratory, Walnut Creek, California 94720, United StatesMore by Samuel Deutsch, Yasuo YoshikuniYasuo YoshikuniU.S. Department of Energy Joint Genome Institute, Lawrence Berkeley National Laboratory, Walnut Creek, California 94720, United StatesMore by Yasuo Yoshikunihttps://orcid.org/0000-0002-8372-640X, Helmut GrubmüllerHelmut GrubmüllerDepartment of Theoretical and Computational Biophysics, Max Planck Institute for Multidisciplinary Sciences, Am Fassberg 11, 37077 Göttingen, GermanyMore by Helmut Grubmüllerhttps://orcid.org/0000-0002-3270-3144, Tobias J. Erb*Tobias J. ErbDepartment of Biochemistry and Synthetic Metabolism, Max Planck Institute for Terrestrial Microbiology, Karl-von-Frisch-Straße 10, D-35043 Marburg, GermanyLOEWE Center for Synthetic Microbiology (SYNMIKRO), 35032 Marburg, Germany*Email for T.J.E.: toerb@mpi-marburg.mpg.deMore by Tobias J. Erbhttps://orcid.org/0000-0003-3685-0894, Esteban Vöhringer-Martinez*Esteban Vöhringer-MartinezDepartamento de Físico Química, Facultad de Ciencias Químicas, Universidad de Concepción, Concepción 4030000, Chile*Email for E.V.-M.: evohringer@udec.clMore by Esteban Vöhringer-Martinezhttps://orcid.org/0000-0003-1785-4558, and Soichi Wakatsuki*Soichi WakatsukiBiosciences Division, SLAC National Accelerator Laboratory Menlo Park, California 94025, United StatesStructural Biology Department, Stanford University Stanford, California 94305, United States*Email for S.W.: soichi.wakatsuki@stanford.eduMore by Soichi Wakatsukihttps://orcid.org/0000-0001-5896-7968Cite this: ACS Cent. Sci. 2022, 8, 8, 1091–1101Publicati\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"0\\\": {\\n    \\\"quote\\\": \\\"Kitsatospor setae bacteria\\\",\\n    \\\"index\\\": 0\\n  },\\n  \\\"1\\\": {\\n    \\\"quote\\\": \\\"ECR enzyme activity\\\",\\n    \\\"index\\\": 0\\n  }\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1690769598,\n",
      "  \"id\": \"chatcmpl-7iChKMNq9zp6i8sgTvPhkis5ha0Fk\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 53,\n",
      "    \"prompt_tokens\": 1600,\n",
      "    \"total_tokens\": 1653\n",
      "  }\n",
      "}\n",
      "Extracted quotes:  {\n",
      "  \"0\": {\n",
      "    \"quote\": \"Kitsatospor setae bacteria\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  \"1\": {\n",
      "    \"quote\": \"ECR enzyme activity\",\n",
      "    \"index\": 0\n",
      "  }\n",
      "}\n",
      "Chunk: 2 / 33\n",
      "seed_initial_question_decomposition_prompt \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity vary under different conditions?\n",
      "2. Are there any factors that enhance or inhibit ECR enzyme activity?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract and output as many accurate direct quotes from the text that are relevant to answering the key questions and its most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"1\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Text:  8, 8, 1091–1101Publication Date (Web):April 25, 2022Publication History\n",
      "Published online25 April 2022Published inissue 24 August 2022https://doi.org/10.1021/acscentsci.2c00057Copyright © 2022 The Authors. Published by American Chemical SocietyRIGHTS & PERMISSIONSACS AuthorChoiceCC: Creative CommonsBY: Credit must be given to the creatorNC: Only noncommercial uses of the work are permittedND: No derivatives or adaptations of the work are permittedArticle Views6505Altmetric188Citations3LEARN ABOUT THESE METRICSArticle Views are the COUNTER-compliant sum of full text article downloads since November 2008 (both PDF and HTML) across all institutions and individuals. These metrics are regularly updated to reflect usage leading up to the last few days.Citations are the number of other articles citing this article, calculated by Crossref and updated daily. Find more information about Crossref citation counts.The Altmetric Attention Score is a quantitative measure of the attention that a research article has received online. Clicking on the donut icon will load a page at altmetric.com with additional details about the score and the social media presence for the given article. Find more information on the Altmetric Attention Score and how the score is calculated.\n",
      "Share\n",
      "Add toView InAdd Full Text with ReferenceAdd Description\n",
      "ExportRISCitationCitation and abstractCitation and referencesMore Options\n",
      "Share onFacebookTwitterWechatLinked InReddit\n",
      "PDF (3 MB)\n",
      "Get e-AlertscloseSupporting Info (14)»Supporting Information\n",
      "Supporting Information\n",
      "SUBJECTS:Adenine,Catalysis,Oligomers,Show MoreGet e-Alertsclose\n",
      "ACS Central Science\n",
      "Get e-AlertsAbstractHigh Resolution ImageDownload MS PowerPoint SlideEnoyl-CoA carboxylases/reductases (ECRs) are some of the most efficient CO2-fixing enzymes described to date. However, the molecular mechanisms underlying the extraordinary catalytic activity of ECRs on the level of the protein assembly remain elusive. Here we used a combination of ambient-temperature X-ray free electron laser (XFEL) and cryogenic synchrotron experiments to study the structural organization of the ECR from Kitasatospora setae. The K. setae ECR is a homotetramer that differentiates into a pair of dimers of open- and closed-form subunits in the catalytically active state. Using molecular dynamics simulations and structure-based mutagenesis, we show that catalysis is synchronized in the K. setae ECR across the pair of dimers. This conformational coupling of catalytic domains is conferred by individual amino acids to achieve high CO2-fixation rates. Our results provide unprecedented insights into the dynamic organization and synchronized inter- and intrasubunit communications of this remarkably efficient CO2-fixing enzyme during catalysis.SynopsisIntersubunit coupling drives fast carboxylation by enoyl-CoA carboxylases/reductases, providing the basis for engineering highly efficient CO2-fixing enzymes for bioenergy and bioproduct applications.IntroductionARTICLE SECTIONSJump ToAbstractIntroductionResults and DiscussionConclusionsMethodsSupporting InformationAuthor InformationAcknowledgmentsNote Added After ASAP PublicationReferencesThe capture and conversion of atmospheric CO2 remains a challenging task for chemistry, resulting in an ever-increasing interest to understand and exploit CO2 fixation mechanisms offered by biology. Nature has evolved different enzymes capable of capturing and/or converting CO2, including biotin-dependent CoA carboxylases, which incorporate CO2 into different alkyl-CoA esters, formate dehydrogenases, which directly reduce CO2 into the single-carbon compound formate, and RubisCO, the central enzyme in the Calvin–Benson–Bassham cycle of photosynthesis that produces two molecules of 3-phosphoglycerate from ribulose-1,5-bisphosphate and CO2. (1−4) The recently discovered family of enoyl-CoA carboxylases/reductases (ECRs) encompasses some of the most efficient CO2-fixing enzymes found in nature to date. (1) With respect to the catalytic rate, for instance, ECR family members from primary metabolism outcompete RubisCO by more than 1 order of magnitude, which makes them exquisite models to study the molecular basis of efficient CO2 catalysis. (2,5)ECRs catalyze the reduction of various α,β-unsaturated enoyl-CoAs into the corresponding alkylmalonyl-CoA esters using the reduced form of the cofactor nicotinamide adenine dinucleotide phosphate (NADPH). Hydride transfer from NADPH to the substrate generates a reactive enolate species, which acts as a nucleophile that attacks a bound CO2 molecule (Figure 1a). (2,5−7) However, the structural details of the carboxylation reaction have remained elusive, in part due to the lack of high-resolution structures of ECRs containing catalytic intermediates and carboxylated products.Figure 1Figure 1. Reaction scheme and structural organization of the K. setae ECR complex. (a) Carboxylation reaction scheme of ECR. (b) Anisotropic B-factors of the tetramer of the differen\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Prompt:  \n",
      "Key questions (index : question): \n",
      "0. What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\n",
      "1. How does ECR enzyme activity vary under different conditions?\n",
      "2. Are there any factors that enhance or inhibit ECR enzyme activity?\n",
      "\n",
      "\n",
      "Task: \n",
      "Extract and output as many accurate direct quotes from the text that are relevant to answering the key questions and its most relevant key question index. Format as a JSON.\n",
      "```json\n",
      "{\n",
      "  \"1\": {\n",
      "    \"quote\": \"\",\n",
      "    \"index\": 0\n",
      "  },\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Text:  8, 8, 1091–1101Publication Date (Web):April 25, 2022Publication History\n",
      "Published online25 April 2022Published inissue 24 August 2022https://doi.org/10.1021/acscentsci.2c00057Copyright © 2022 The Authors. Published by American Chemical SocietyRIGHTS & PERMISSIONSACS AuthorChoiceCC: Creative CommonsBY: Credit must be given to the creatorNC: Only noncommercial uses of the work are permittedND: No derivatives or adaptations of the work are permittedArticle Views6505Altmetric188Citations3LEARN ABOUT THESE METRICSArticle Views are the COUNTER-compliant sum of full text article downloads since November 2008 (both PDF and HTML) across all institutions and individuals. These metrics are regularly updated to reflect usage leading up to the last few days.Citations are the number of other articles citing this article, calculated by Crossref and updated daily. Find more information about Crossref citation counts.The Altmetric Attention Score is a quantitative measure of the attention that a research article has received online. Clicking on the donut icon will load a page at altmetric.com with additional details about the score and the social media presence for the given article. Find more information on the Altmetric Attention Score and how the score is calculated.\n",
      "Share\n",
      "Add toView InAdd Full Text with ReferenceAdd Description\n",
      "ExportRISCitationCitation and abstractCitation and referencesMore Options\n",
      "Share onFacebookTwitterWechatLinked InReddit\n",
      "PDF (3 MB)\n",
      "Get e-AlertscloseSupporting Info (14)»Supporting Information\n",
      "Supporting Information\n",
      "SUBJECTS:Adenine,Catalysis,Oligomers,Show MoreGet e-Alertsclose\n",
      "ACS Central Science\n",
      "Get e-AlertsAbstractHigh Resolution ImageDownload MS PowerPoint SlideEnoyl-CoA carboxylases/reductases (ECRs) are some of the most efficient CO2-fixing enzymes described to date. However, the molecular mechanisms underlying the extraordinary catalytic activity of ECRs on the level of the protein assembly remain elusive. Here we used a combination of ambient-temperature X-ray free electron laser (XFEL) and cryogenic synchrotron experiments to study the structural organization of the ECR from Kitasatospora setae. The K. setae ECR is a homotetramer that differentiates into a pair of dimers of open- and closed-form subunits in the catalytically active state. Using molecular dynamics simulations and structure-based mutagenesis, we show that catalysis is synchronized in the K. setae ECR across the pair of dimers. This conformational coupling of catalytic domains is conferred by individual amino acids to achieve high CO2-fixation rates. Our results provide unprecedented insights into the dynamic organization and synchronized inter- and intrasubunit communications of this remarkably efficient CO2-fixing enzyme during catalysis.SynopsisIntersubunit coupling drives fast carboxylation by enoyl-CoA carboxylases/reductases, providing the basis for engineering highly efficient CO2-fixing enzymes for bioenergy and bioproduct applications.IntroductionARTICLE SECTIONSJump ToAbstractIntroductionResults and DiscussionConclusionsMethodsSupporting InformationAuthor InformationAcknowledgmentsNote Added After ASAP PublicationReferencesThe capture and conversion of atmospheric CO2 remains a challenging task for chemistry, resulting in an ever-increasing interest to understand and exploit CO2 fixation mechanisms offered by biology. Nature has evolved different enzymes capable of capturing and/or converting CO2, including biotin-dependent CoA carboxylases, which incorporate CO2 into different alkyl-CoA esters, formate dehydrogenases, which directly reduce CO2 into the single-carbon compound formate, and RubisCO, the central enzyme in the Calvin–Benson–Bassham cycle of photosynthesis that produces two molecules of 3-phosphoglycerate from ribulose-1,5-bisphosphate and CO2. (1−4) The recently discovered family of enoyl-CoA carboxylases/reductases (ECRs) encompasses some of the most efficient CO2-fixing enzymes found in nature to date. (1) With respect to the catalytic rate, for instance, ECR family members from primary metabolism outcompete RubisCO by more than 1 order of magnitude, which makes them exquisite models to study the molecular basis of efficient CO2 catalysis. (2,5)ECRs catalyze the reduction of various α,β-unsaturated enoyl-CoAs into the corresponding alkylmalonyl-CoA esters using the reduced form of the cofactor nicotinamide adenine dinucleotide phosphate (NADPH). Hydride transfer from NADPH to the substrate generates a reactive enolate species, which acts as a nucleophile that attacks a bound CO2 molecule (Figure 1a). (2,5−7) However, the structural details of the carboxylation reaction have remained elusive, in part due to the lack of high-resolution structures of ECRs containing catalytic intermediates and carboxylated products.Figure 1Figure 1. Reaction scheme and structural organization of the K. setae ECR complex. (a) Carboxylation reaction scheme of ECR. (b) Anisotropic B-factors of the tetramer of the differen\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from autogpt.commands.web_selenium import scrape_text_with_selenium_no_agent\n",
    "import json\n",
    "import os\n",
    "import re\n",
    "\n",
    "chunk_size = 5000\n",
    "overlap = 25\n",
    "MAX_TOKENS = 10000 # Roughly 3K tokens, $0.10 per MAX_TOKENs\n",
    "curr_tokens = 0\n",
    "\n",
    "def sanitize_filename(filename):\n",
    "    return re.sub(r'[^a-zA-Z0-9]', '_', filename)\n",
    "\n",
    "# 1) Get key questions\n",
    "key_questions_dict, key_questions_list_string = get_key_questions_list_string(search_query_file_safe)\n",
    "\n",
    "# 2) Open the web search results for key question 1\n",
    "key_question = \"What is the level of ECR enzyme activity in Kitsatospor setae bacteria?\"\n",
    "key_question_file_safe = sanitize_filename(key_question)\n",
    "with open(f'autoscious_logs/web_search/{key_question_file_safe}.json', 'r') as f:\n",
    "    web_search_res = json.loads(json.loads(f.read()))\n",
    "\n",
    "driver = None\n",
    "for web_source in web_search_res:\n",
    "    print(\"Current tokens: \", curr_tokens)\n",
    "    if curr_tokens > MAX_TOKENS:\n",
    "        print(\"Max tokens reached!\")\n",
    "        break\n",
    "\n",
    "    url = web_source['href']\n",
    "    print(\"Search result: \", web_source['title'], url)\n",
    "    file_title = sanitize_filename(web_source[\"title\"])\n",
    "    file_path = f'autoscious_logs/web_search/{key_question_file_safe}/{file_title}_complete_text.txt'\n",
    "\n",
    "    # If we haven't already scraped the text\n",
    "    if not os.path.exists(file_path):\n",
    "        # Scrape url\n",
    "        driver, text = scrape_text_with_selenium_no_agent(url, driver)\n",
    "\n",
    "        # Extract relevant facts to key questions from web search results\n",
    "        # TODO: Chunk text if text is too long\n",
    "        chunks = chunk_text(text, chunk_size, overlap)\n",
    "        for i, chunk in enumerate(chunks):\n",
    "            print(f\"Chunk: {i} / {len(chunks)}\")\n",
    "            extract_facts_from_website_text(key_questions_list_string, chunk, url)\n",
    "\n",
    "            curr_tokens += len(chunks)\n",
    "            if curr_tokens > MAX_TOKENS:\n",
    "                print(\"Max tokens reached!\")\n",
    "                break\n",
    "\n",
    "        # Save complete text to mark already analyzed\n",
    "        with open(file_path, 'w', encoding='utf-8') as f:\n",
    "            f.write(text)\n",
    "    else:\n",
    "        print(\"Already scraped!\")\n",
    "\n",
    "driver.quit()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
