{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "from autogpt.commands.web_selenium import browse_website, scrape_text_with_selenium_no_agent\n",
    "import json\n",
    "from util import sanitize_filename\n",
    "import os\n",
    "from prompts import get_predicted_usefulness_of_text_prompt\n",
    "from collections import defaultdict\n",
    "from llm import chat_openai\n",
    "from autogpt.commands.web_search import web_search_ddg\n",
    "from datetime import datetime"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "search_query = \"How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "search_query_file_safe = sanitize_filename(search_query)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load in decomposition\n",
    "with open(f'autoscious_logs/{search_query_file_safe}/decompositions/improved_decomposition.json', 'r') as f:\n",
    "    decomposition = json.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity compare to other bacteria?', '3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?', '4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}\n"
     ]
    }
   ],
   "source": [
    "# Get key questions\n",
    "key_questions = decomposition['key_drivers']['1']['hypotheses']['1']['key_questions']\n",
    "print(key_questions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create sources folder\n",
    "if not os.path.exists(f'autoscious_logs/{search_query_file_safe}/sources'):\n",
    "    os.makedirs(f'autoscious_logs/{search_query_file_safe}/sources') \n",
    "if not os.path.exists(f'autoscious_logs/{search_query_file_safe}/sources/full_text'):\n",
    "    os.makedirs(f'autoscious_logs/{search_query_file_safe}/source/full_text') \n",
    "\n",
    "for key in key_questions.keys():\n",
    "    if not os.path.exists(f'autoscious_logs/{search_query_file_safe}/sources/kq{int(key)-1}'):\n",
    "        os.makedirs(f'autoscious_logs/{search_query_file_safe}/source/kq{int(key)-1}') "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Read Type 1: Rating search results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Key questions decomposition list:  [{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}, {'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'2': 'How does ECR enzyme activity compare to other bacteria?'}}}}}}, {'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}, {'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}]\n"
     ]
    }
   ],
   "source": [
    "# Create a decomposition for each key question only\n",
    "key_question_decomposition_list = []\n",
    "for driver_key, driver_value in decomposition['key_drivers'].items():\n",
    "    for hypothesis_key, hypothesis_value in driver_value['hypotheses'].items():\n",
    "        for question_key, question_value in hypothesis_value['key_questions'].items():\n",
    "            new_decomposition = decomposition.copy()\n",
    "            new_decomposition['key_drivers'] = {\n",
    "                driver_key: {\n",
    "                    'driver': driver_value['driver'],\n",
    "                    'hypotheses': {\n",
    "                        hypothesis_key: {\n",
    "                            'hypothesis': hypothesis_value['hypothesis'],\n",
    "                            'key_questions': {\n",
    "                                question_key: question_value\n",
    "                            }\n",
    "                        }\n",
    "                    }\n",
    "                }\n",
    "            }\n",
    "            key_question_decomposition_list.append(new_decomposition)\n",
    "print(\"Key questions decomposition list: \", key_question_decomposition_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Coming up with many good search queries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_initial_search_queries_prompt(context, key_question_decomposition):\n",
    "    return f'''\n",
    "Context:\n",
    "{context}\n",
    "\n",
    "Research question decomposition\n",
    "{key_question_decomposition}\n",
    "\n",
    "Task:\n",
    "For the key question, write a clear and comprehensive but short (around 3 queries) list of search keyword queries optimized for best search engine results, so that you can confidently and quickly surface the most relevant information to determine the best answer to the question. \n",
    "\n",
    "The output should be in JSON format: \n",
    "```json\n",
    "{{\n",
    "  1: \"<insert query>\",\n",
    "  etc.\n",
    "}}\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Prompt:  \n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}\n",
      "\n",
      "Task:\n",
      "For the key question, write a clear and comprehensive but short (around 3 queries) list of search keyword queries optimized for best search engine results, so that you can confidently and quickly surface the most relevant information to determine the best answer to the question. \n",
      "\n",
      "The output should be in JSON format: \n",
      "```json\n",
      "{\n",
      "  1: \"<insert query>\",\n",
      "  etc.\n",
      "}\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"1\\\": \\\"ECR enzyme activity Kitsatospor setae bacteria\\\",\\n  \\\"2\\\": \\\"Factors affecting ECR enzyme activity\\\",\\n  \\\"3\\\": \\\"Regulation of ECR enzyme activity in bacteria\\\"\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691111992,\n",
      "  \"id\": \"chatcmpl-7jdlo4Xrn6SD71aCLByDHDJJLcUQn\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 45,\n",
      "    \"prompt_tokens\": 264,\n",
      "    \"total_tokens\": 309\n",
      "  }\n",
      "}\n",
      "Prompt:  \n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'2': 'How does ECR enzyme activity compare to other bacteria?'}}}}}}\n",
      "\n",
      "Task:\n",
      "For the key question, write a clear and comprehensive but short (around 3 queries) list of search keyword queries optimized for best search engine results, so that you can confidently and quickly surface the most relevant information to determine the best answer to the question. \n",
      "\n",
      "The output should be in JSON format: \n",
      "```json\n",
      "{\n",
      "  1: \"<insert query>\",\n",
      "  etc.\n",
      "}\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"1\\\": \\\"ECR enzyme activity Kitsatospor setae bacteria\\\",\\n  \\\"2\\\": \\\"Comparison of ECR enzyme activity in Kitsatospor setae bacteria and other bacteria\\\",\\n  \\\"3\\\": \\\"Factors affecting ECR enzyme activity in Kitsatospor setae bacteria\\\"\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691111993,\n",
      "  \"id\": \"chatcmpl-7jdlpoqIBecpIonOclpbetl4SY7Vm\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 59,\n",
      "    \"prompt_tokens\": 258,\n",
      "    \"total_tokens\": 317\n",
      "  }\n",
      "}\n",
      "Prompt:  \n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}\n",
      "\n",
      "Task:\n",
      "For the key question, write a clear and comprehensive but short (around 3 queries) list of search keyword queries optimized for best search engine results, so that you can confidently and quickly surface the most relevant information to determine the best answer to the question. \n",
      "\n",
      "The output should be in JSON format: \n",
      "```json\n",
      "{\n",
      "  1: \"<insert query>\",\n",
      "  etc.\n",
      "}\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"1\\\": \\\"Factors influencing ECR enzyme activity in Kitsatospor setae bacteria\\\",\\n  \\\"2\\\": \\\"ECR enzyme activity regulation in Kitsatospor setae bacteria\\\",\\n  \\\"3\\\": \\\"Mechanisms of ECR enzyme activity in Kitsatospor setae bacteria\\\"\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691111998,\n",
      "  \"id\": \"chatcmpl-7jdlumJRI86BO6rp9cQVJUEjjP2EP\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 60,\n",
      "    \"prompt_tokens\": 262,\n",
      "    \"total_tokens\": 322\n",
      "  }\n",
      "}\n",
      "Prompt:  \n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}\n",
      "\n",
      "Task:\n",
      "For the key question, write a clear and comprehensive but short (around 3 queries) list of search keyword queries optimized for best search engine results, so that you can confidently and quickly surface the most relevant information to determine the best answer to the question. \n",
      "\n",
      "The output should be in JSON format: \n",
      "```json\n",
      "{\n",
      "  1: \"<insert query>\",\n",
      "  etc.\n",
      "}\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"1\\\": \\\"ECR enzyme activity Kitsatospor setae bacteria\\\",\\n  \\\"2\\\": \\\"Efficiency of ECR enzymes in Kitsatospor setae bacteria\\\",\\n  \\\"3\\\": \\\"Applications of high ECR enzyme activity in Kitsatospor setae bacteria\\\"\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691112000,\n",
      "  \"id\": \"chatcmpl-7jdlwBWFhkLN1eBx9IfyMnO0ZfdzS\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 57,\n",
      "    \"prompt_tokens\": 268,\n",
      "    \"total_tokens\": 325\n",
      "  }\n",
      "}\n"
     ]
    }
   ],
   "source": [
    "# TODO: Iterate through all key questions, or just ask to generate an initial set of search queries considering all key questions. Then when we iterate, we can see what facts have been added to what papers, and what unanswered questions we should focus our queries on.\n",
    "\n",
    "context = \"Enoyl-CoA carboxylase/reductase enzymes (ECRs)\"\n",
    "\n",
    "for decomposition_idx, key_question_decomposition in enumerate(key_question_decomposition_list):\n",
    "    key_question_initial_search_queries = json.loads(chat_openai(get_initial_search_queries_prompt(context, key_question_decomposition), model=\"gpt-3.5-turbo\")[0])\n",
    "\n",
    "    with open(f'autoscious_logs/{search_query_file_safe}/sources/kq{decomposition_idx}/initial_search_queries.json', 'w') as f:\n",
    "        json.dump(key_question_initial_search_queries, f, indent=2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Web search given search keywords"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "from dotenv import load_dotenv\n",
    "from googleapiclient.discovery import build\n",
    "load_dotenv()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import time\n",
    "def google_search_raw(search_term, cse_id, **kwargs):\n",
    "    service = build(\"customsearch\", \"v1\", developerKey=os.getenv('DEV_KEY'))\n",
    "    res = service.cse().list(q=search_term, cx=cse_id, **kwargs).execute()\n",
    "\n",
    "    search_results = res.get(\"items\", [])\n",
    "    time.sleep(1)\n",
    "\n",
    "    # Create a list of only the URLs from the search results\n",
    "    search_results_links = [item[\"link\"] for item in search_results]\n",
    "    return search_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "def search_google(search_query):\n",
    "    num_google_searches = 8\n",
    "    results = google_search_raw(search_query, os.getenv('MY_CSE_ID'), num=num_google_searches, lr=\"lang_en\", cr=\"countryUS\")\n",
    "    return results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "query:  ECR enzyme activity Kitsatospor setae bacteria\n",
      "trying google search!\n",
      "query:  Factors affecting ECR enzyme activity\n",
      "trying google search!\n",
      "query:  Regulation of ECR enzyme activity in bacteria\n",
      "trying google search!\n",
      "query:  ECR enzyme activity Kitsatospor setae bacteria\n",
      "trying google search!\n",
      "query:  Comparison of ECR enzyme activity in Kitsatospor setae bacteria and other bacteria\n",
      "trying google search!\n",
      "query:  Factors affecting ECR enzyme activity in Kitsatospor setae bacteria\n",
      "trying google search!\n",
      "query:  Factors influencing ECR enzyme activity in Kitsatospor setae bacteria\n",
      "trying google search!\n",
      "query:  ECR enzyme activity regulation in Kitsatospor setae bacteria\n",
      "trying google search!\n",
      "query:  Mechanisms of ECR enzyme activity in Kitsatospor setae bacteria\n",
      "trying google search!\n",
      "query:  ECR enzyme activity Kitsatospor setae bacteria\n",
      "trying google search!\n",
      "query:  Efficiency of ECR enzymes in Kitsatospor setae bacteria\n",
      "trying google search!\n",
      "query:  Applications of high ECR enzyme activity in Kitsatospor setae bacteria\n",
      "trying google search!\n"
     ]
    }
   ],
   "source": [
    "MAX_RETRIES = 3\n",
    "\n",
    "for decomposition_idx, key_question_decomposition in enumerate(key_question_decomposition_list):\n",
    "    with open(f'autoscious_logs/{search_query_file_safe}/sources/kq{decomposition_idx}/initial_search_queries.json', 'r') as f:\n",
    "        key_question_initial_search_queries = json.load(f)\n",
    "\n",
    "    for idx, query in key_question_initial_search_queries.items():\n",
    "        print(\"query: \", query)\n",
    "\n",
    "        # DDG\n",
    "        web_search_res = json.loads(web_search_ddg(query))\n",
    "        if len(web_search_res) == 0:\n",
    "            print(\"trying google search!\")\n",
    "            # Google\n",
    "            web_search_res_raw = search_google(query) # google uses 'link' instead of 'href'\n",
    "            web_search_res = [{\n",
    "                'title': web_search_res_raw[i]['title'], \n",
    "                'href': web_search_res_raw[i]['link'], \n",
    "                'body': web_search_res_raw[i]['snippet']\n",
    "                } for i in range(len(web_search_res_raw))\n",
    "            ]\n",
    "        \n",
    "        # save web search results\n",
    "        with open(f'autoscious_logs/{search_query_file_safe}/sources/kq{decomposition_idx}/initial_search_results_query_{idx}.json', 'w') as f:\n",
    "            json.dump(web_search_res, f, indent=2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reading type 1: filtering unlikely relevant sources based on title and body"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_filtering_web_results_ratings(context, decomposition, web_search_res):\n",
    "    return f'''\n",
    "Context:\n",
    "{context}\n",
    "\n",
    "Research question decomposition:\n",
    "{decomposition}\n",
    "\n",
    "Search results:\n",
    "{web_search_res}\n",
    "\n",
    "Task:\n",
    "Based on the key questions and each search result's title and body content, identify which key question indexes the search result content is likely most useful for or -1 if it isn't useful for any key question, and for each key question index identified, assign a relevance score of the search result content using a 5-point Likert scale, with 1 being very irrelevant to 5 being very relevant to the key question.\n",
    "\n",
    "The output should be in JSON format: \n",
    "```json\n",
    "{{\n",
    "  <'insert href'>: [['insert key question index', 'relevance score'], etc.],\n",
    "  etc.\n",
    "}}\n",
    "```\n",
    "\n",
    "Respond only with the output, with no explanation or conversation.\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Prompt:  \n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition:\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity compare to other bacteria?', '3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?', '4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}\n",
      "\n",
      "Search results:\n",
      "[{'title': 'Blue light treatment of Pseudomonas aeruginosa: Strong ...', 'href': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5626244/', 'body': 'The pseudolysin activity of elastase B was analyzed by performing and ECR assay. ... activity of virulence factors including (i) proteolytic enzymes (i.e.,\\xa0...'}, {'title': 'Sublethal Effects of Triflumezopyrim on Biological Traits ... - Frontiers', 'href': 'https://www.frontiersin.org/articles/10.3389/fphys.2020.00261/full', 'body': 'Mar 6, 2020 ... Our objective is to analyze the population dynamics, detoxification enzyme activities, Vg and VgR contents, and relative expression of the EcR,\\xa0...'}, {'title': 'Biosynthetic Enzyme GMP Synthetase Cooperates with Ubiquitin ...', 'href': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2812229/', 'body': 'Dec 7, 2009 ... Ecdysone receptor (EcR) interacts biochemically and genetically with ... Next we wondered whether the enzymatic activities of USP7 or GMPS\\xa0...'}, {'title': 'Aurintricarboxylic acid structure modifications lead to reduction of ...', 'href': 'https://link.springer.com/article/10.1007/s11274-016-2123-3', 'body': 'Aug 25, 2016 ... The free binding energy calculations show that ECR has a stronger binding ... Furthermore, we have assessed the enzymatic activity of YopH\\xa0...'}, {'title': 'A Broad Set of Chromatin Factors Influences Splicing | PLOS Genetics', 'href': 'https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1006318', 'body': 'Here, we show that a subset of chromatin proteins is physically in interaction with the enzyme responsible for RNA splicing. In addition, several chromatin\\xa0...'}, {'title': 'Chitin metabolism in insects: structure, function and regulation of ...', 'href': 'https://journals.biologists.com/jeb/article/206/24/4393/34291/Chitin-metabolism-in-insects-structure-function', 'body': 'Dec 15, 2003 ... Common features of most chitin synthases are that enzyme activity is ... of the EcR and the Drosophila retinoid X receptor homologue USP,\\xa0...'}, {'title': '20-Hydroxyecdysone (20E) signaling as a promising target for the ...', 'href': 'https://parasitesandvectors.biomedcentral.com/articles/10.1186/s13071-020-04558-5', 'body': 'Jan 29, 2021 ... Orthologues of these enzymes have been characterized in mosquitoes. b Once 20E binds to its heterodimer EcR/USP receptor, the latter is\\xa0...'}, {'title': 'Multiple functions of CREB-binding protein during postembryonic ...', 'href': 'https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-017-4373-3', 'body': 'Dec 29, 2017 ... Hence, understanding the molecular mechanisms of their action is of ... of 1306 genes coding for transcription factors and other proteins\\xa0...'}]\n",
      "\n",
      "Task:\n",
      "Based on the key questions and each search result's title and body content, identify which key question indexes the search result content is likely most useful for or -1 if it isn't useful for any key question, and for each key question index identified, assign a relevance score of the search result content using a 5-point Likert scale, with 1 being very irrelevant to 5 being very relevant to the key question.\n",
      "\n",
      "The output should be in JSON format: \n",
      "```json\n",
      "{\n",
      "  <'insert href'>: [['insert key question index', 'relevance score'], etc.],\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5626244/\\\": [[\\\"1\\\", 4]],\\n  \\\"https://www.frontiersin.org/articles/10.3389/fphys.2020.00261/full\\\": [[\\\"-1\\\"]],\\n  \\\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2812229/\\\": [[\\\"-1\\\"]],\\n  \\\"https://link.springer.com/article/10.1007/s11274-016-2123-3\\\": [[\\\"-1\\\"]],\\n  \\\"https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1006318\\\": [[\\\"-1\\\"]],\\n  \\\"https://journals.biologists.com/jeb/article/206/24/4393/34291/Chitin-metabolism-in-insects-structure-function\\\": [[\\\"-1\\\"]],\\n  \\\"https://parasitesandvectors.biomedcentral.com/articles/10.1186/s13071-020-04558-5\\\": [[\\\"-1\\\"]],\\n  \\\"https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-017-4373-3\\\": [[\\\"-1\\\"]]\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691114997,\n",
      "  \"id\": \"chatcmpl-7jeYHtgpdxRg82sydap3b4BWW0uZW\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 250,\n",
      "    \"prompt_tokens\": 1145,\n",
      "    \"total_tokens\": 1395\n",
      "  }\n",
      "}\n",
      "Prompt:  \n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition:\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity compare to other bacteria?', '3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?', '4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}\n",
      "\n",
      "Search results:\n",
      "[{'title': 'Isoform-Specific Regulation of a Steroid Hormone Nuclear Receptor ...', 'href': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3213379/', 'body': '... of EcR and usp genes. Thus, the activity of this enzyme results in the regulation of dimerizing receptors at the protein and gene transcription levels.'}, {'title': 'Developmental regulation of ecdysone receptor (EcR ... - Frontiers', 'href': 'https://www.frontiersin.org/articles/10.3389/fgene.2014.00445/full', 'body': 'Dec 4, 2014 ... Assessing Gene Transcription Patterns Associated to EcR Function ... regulate ecdysteroid titer and function since the enzymes encoded by\\xa0...'}, {'title': 'Antibiotic Treatment Decrease the Fitness of Honeybee (Apis ...', 'href': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8066305/', 'body': 'Mar 30, 2021 ... To determine the biologic function of gut bacteria with no host specificity ... expression of developmental regulation genes (ecr and usp),\\xa0...'}, {'title': 'Genome-wide regulation of innate immunity by juvenile hormone ...', 'href': 'https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-11-549', 'body': 'Oct 9, 2010 ... Molting and metamorphosis in insects are coordinately regulated by ... Antibacterial activities against both bacteria were high during the\\xa0...'}, {'title': 'Role of DUOX in gut inflammation: lessons from ... - Frontiers', 'href': 'https://www.frontiersin.org/articles/10.3389/fcimb.2013.00116/full', 'body': 'Dec 23, 2013 ... Several lines of evidence showed that the bacterial-modulated DUOX ... by the enzymatic action of lactoperoxidase in the mucosal fluids.'}, {'title': 'Chitin metabolism in insects: structure, function and regulation of ...', 'href': 'https://journals.biologists.com/jeb/article/206/24/4393/34291/Chitin-metabolism-in-insects-structure-function', 'body': 'Dec 15, 2003 ... Common features of most chitin synthases are that enzyme activity is ... Similar sequences have been found in bacterial and vertebrate\\xa0...'}, {'title': '20-Hydroxyecdysone (20E) signaling as a promising target for the ...', 'href': 'https://parasitesandvectors.biomedcentral.com/articles/10.1186/s13071-020-04558-5', 'body': 'Jan 29, 2021 ... Besides vitellogenesis, the timely regulation of metamorphosis also requires the presence cofactors to regulate EcR/USP. For example, in Ae.'}, {'title': '2106-Patent Subject Matter Eligibility', 'href': 'https://www.uspto.gov/web/offices/pac/mpep/s2106.html', 'body': 'Because abstract ideas, laws of nature, and natural phenomenon \"are the basic tools of ... Similarly, a genetically modified bacterium satisfies both the\\xa0...'}]\n",
      "\n",
      "Task:\n",
      "Based on the key questions and each search result's title and body content, identify which key question indexes the search result content is likely most useful for or -1 if it isn't useful for any key question, and for each key question index identified, assign a relevance score of the search result content using a 5-point Likert scale, with 1 being very irrelevant to 5 being very relevant to the key question.\n",
      "\n",
      "The output should be in JSON format: \n",
      "```json\n",
      "{\n",
      "  <'insert href'>: [['insert key question index', 'relevance score'], etc.],\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3213379/\\\": [[\\\"1\\\", 4]],\\n  \\\"https://www.frontiersin.org/articles/10.3389/fgene.2014.00445/full\\\": [[\\\"1\\\", 3]],\\n  \\\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8066305/\\\": [[\\\"-1\\\"]],\\n  \\\"https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-11-549\\\": [[\\\"-1\\\"]],\\n  \\\"https://www.frontiersin.org/articles/10.3389/fcimb.2013.00116/full\\\": [[\\\"-1\\\"]],\\n  \\\"https://journals.biologists.com/jeb/article/206/24/4393/34291/Chitin-metabolism-in-insects-structure-function\\\": [[\\\"-1\\\"]],\\n  \\\"https://parasitesandvectors.biomedcentral.com/articles/10.1186/s13071-020-04558-5\\\": [[\\\"-1\\\"]],\\n  \\\"https://www.uspto.gov/web/offices/pac/mpep/s2106.html\\\": [[\\\"-1\\\"]]\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691115005,\n",
      "  \"id\": \"chatcmpl-7jeYPxE8xS2IEmRsjDOu5hal249H4\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 245,\n",
      "    \"prompt_tokens\": 1088,\n",
      "    \"total_tokens\": 1333\n",
      "  }\n",
      "}\n"
     ]
    }
   ],
   "source": [
    "from collections import defaultdict\n",
    "\n",
    "context = \"Enoyl-CoA carboxylase/reductase enzymes (ECRs)\"\n",
    "\n",
    "for decomposition_idx, key_question_decomposition in enumerate(key_question_decomposition_list):\n",
    "    with open(f'autoscious_logs/{search_query_file_safe}/sources/kq{decomposition_idx}/initial_search_queries.json', 'r') as f:\n",
    "        key_question_initial_search_queries = json.load(f)\n",
    "\n",
    "    for query_idx, query in key_question_initial_search_queries.items():\n",
    "        # load web search results\n",
    "        with open(f'autoscious_logs/{search_query_file_safe}/sources/kq{decomposition_idx}/initial_search_results_query_{query_idx}.json', 'r') as f:\n",
    "            web_search_res = json.loads(f.read())\n",
    "        \n",
    "        filtered_web_results = {}\n",
    "        if web_search_res != []:\n",
    "            # filter web results based on title and body\n",
    "            filtered_web_results = json.loads(chat_openai(get_filtering_web_results_ratings(context, decomposition, web_search_res), model=\"gpt-3.5-turbo\")[0])\n",
    "\n",
    "        ratings_url_dict = defaultdict(list)\n",
    "        for url, ratings in filtered_web_results.items():\n",
    "            if '-1' not in str(ratings):  # only process if the ratings is not -1\n",
    "                for rating in ratings:\n",
    "                    ratings_url_dict[str(rating[1])].append(url)  # append the URL to the correct category\n",
    "            else:\n",
    "                ratings_url_dict['-1'].append(url) # Misc & unrelated\n",
    "\n",
    "        # save filtered search results\n",
    "        with open(f'autoscious_logs/{search_query_file_safe}/sources/kq{decomposition_idx}/rated_web_results_query_{int(query_idx)}.json', 'w') as f:\n",
    "            json.dump(ratings_url_dict, f, indent=2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reading type 2 & 3: filtering based on skimming and sampling from each source, and only saving most relevant sources for fact extraction and quotes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "# COMPLETE code for predicting usefulness of very relevant (5) and relevant (4) results.\n",
    "from autogpt.commands.web_selenium import scrape_text_with_selenium_no_agent\n",
    "\n",
    "CHUNK_SIZE = 1000\n",
    "SAMPLING_FACTOR = 0.1 # Also cap it so it falls under the max token limit\n",
    "MAX_TOKENS = 2500 * 4 # 1 token = 4 chars, 2500 + 500 (prompt) tokens is high for GPT3.5\n",
    "MAX_CHUNKS = int(MAX_TOKENS / CHUNK_SIZE)\n",
    "context = \"Enoyl-CoA carboxylase/reductase enzymes (ECRs)\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_sample_chunks(text, CHUNK_SIZE, num_chunk_samples):\n",
    "    step_size = len(text) // num_chunk_samples\n",
    "\n",
    "    chunks = []\n",
    "    for i in range(0, len(text), step_size):\n",
    "        chunk = text[i:i+CHUNK_SIZE]\n",
    "        chunks.append(chunk)\n",
    "\n",
    "        # Break after getting the required number of chunks\n",
    "        if len(chunks) >= num_chunk_samples:\n",
    "            break\n",
    "\n",
    "    return chunks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Need to determine how useful the text is likely to be for answering the key questions\n",
    "def get_predicted_usefulness_of_text_prompt(context, decomposition, sample_text_chunks):\n",
    "    return f'''\n",
    "Task: \n",
    "Based on the research question decomposition key questions and the sample text chunks of the source text, the goal is to identify how useful reading the full source text would be to extract direct quoted facts and determine the best answer to any of the key questions. \n",
    "\n",
    "Context:\n",
    "{context}\n",
    "\n",
    "Research question decomposition:\n",
    "{decomposition}\n",
    "\n",
    "Sample text chunks from the source text:\n",
    "{sample_text_chunks}\n",
    "\n",
    "Deliverables:\n",
    "For each key question, assign a predicted usefulness score of the full source text using a 5-point Likert scale, with 1 being very unlikely to be usefulness to 5 being very likely useful and containing facts that answer the key question.\n",
    "\n",
    "The output should be of the following JSON format\n",
    "{{\n",
    "    <insert key question index>: <insert predicted usefulness>,\n",
    "   etc.\n",
    "}}\n",
    "\n",
    "\n",
    "Respond only with the output, with no explanation or conversation.\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_sample_chunks(text, CHUNK_SIZE, num_chunk_samples):\n",
    "    step_size = len(text) // num_chunk_samples\n",
    "\n",
    "    chunks = []\n",
    "    for i in range(0, len(text), step_size):\n",
    "        chunk = text[i:i+CHUNK_SIZE]\n",
    "        chunks.append(chunk)\n",
    "\n",
    "        # Break after getting the required number of chunks\n",
    "        if len(chunks) >= num_chunk_samples:\n",
    "            break\n",
    "\n",
    "    return chunks\n",
    "\n",
    "def find_title(url, web_search_inf):\n",
    "    for item in web_search_inf:\n",
    "        if item[\"href\"] == url:\n",
    "            return item[\"title\"]\n",
    "    return None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rating_source_idx 0 Skimming url: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9413435/\n",
      "Going through url:  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9413435/\n",
      "select firefox options!\n",
      "hard coding chrome\n",
      "setting up chrome driver\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Driver is getting url\n",
      "set timeout!\n",
      "Page loaded within 15 seconds\n",
      "Driver got url\n",
      "Driver has found page source\n",
      "Handing off to Beautiful Soup!\n",
      "done extractin\n",
      "Text:  Back to Top\n",
      "Skip to main content\n",
      "The .gov means it’s official.\n",
      "Federal government websites often end in .gov or .mil. Before\n",
      "sharing sensitive information, make sure you’re on a federal\n",
      "government site.\n",
      "The site is secure.\n",
      "The https:// ensures that you are connecting to the\n",
      "official website and that any information you provide is encrypted\n",
      "and transmitted securely.\n",
      "Access keys\n",
      "NCBI Homepage\n",
      "MyNCBI Homepage\n",
      "Main Content\n",
      "Main Navigation\n",
      "Journal List\n",
      "ACS Cent Sci\n",
      "v.8(8); 2022 Aug 24\n",
      "PMC9413435\n",
      "As a\n",
      "len(sample_chunks) 6\n",
      "Prompt:  \n",
      "Task: \n",
      "Based on the research question decomposition key questions and the sample text chunks of the source text, the goal is to identify how useful reading the full source text would be to extract direct quoted facts and determine the best answer to any of the key questions. \n",
      "\n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition:\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity compare to other bacteria?', '3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?', '4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}\n",
      "\n",
      "Sample text chunks from the source text:\n",
      "['Back to Top\\nSkip to main content\\nThe .gov means it’s official.\\nFederal government websites often end in .gov or .mil. Before\\nsharing sensitive information, make sure you’re on a federal\\ngovernment site.\\nThe site is secure.\\nThe https:// ensures that you are connecting to the\\nofficial website and that any information you provide is encrypted\\nand transmitted securely.\\nAccess keys\\nNCBI Homepage\\nMyNCBI Homepage\\nMain Content\\nMain Navigation\\nJournal List\\nACS Cent Sci\\nv.8(8); 2022 Aug 24\\nPMC9413435\\nAs a library, NLM provides access to scientific literature. Inclusion in an NLM database does not imply endorsement of, or agreement with,\\nthe contents by NLM or the National Institutes of Health.\\nLearn more about our disclaimer.\\nACS Cent Sci. 2022 Aug 24; 8(8): 1091–1101. Published online 2022 Apr 25.\\ndoi:\\xa010.1021/acscentsci.2c00057PMCID: PMC9413435PMID: 36032767Intersubunit Coupling Enables Fast CO2-Fixation\\nby Reductive CarboxylasesHasan DeMirci,*†‡§▼ Yashas Rao,†⊥▼ Gabriele M. Stoffel,∥▼ Bastian', 's some of the\\nmost efficient CO2-fixing enzymes found in nature to date.1 With respect to the catalytic rate, for instance,\\nECR family members from primary metabolism outcompete RubisCO by more\\nthan 1 order of magnitude, which makes them exquisite models to study\\nthe molecular basis of efficient CO2 catalysis.2,5ECRs catalyze the reduction of various α,β-unsaturated\\nenoyl-CoAs into the corresponding alkylmalonyl-CoA esters using the\\nreduced form of the cofactor nicotinamide adenine dinucleotide phosphate\\n(NADPH). Hydride transfer from NADPH to the substrate generates a\\nreactive enolate species, which acts as a nucleophile that attacks\\na bound CO2 molecule (Figure \\u200bFigure11a).2,5−7 However, the structural details of the carboxylation reaction have\\nremained elusive, in part due to the lack of high-resolution structures\\nof ECRs containing catalytic intermediates and carboxylated products.Open in a separate windowFigure 1Reaction\\nscheme and structural organization of the K. setae ECR compl', 'bilizing the adenine\\nring of butyryl-CoA that stretches into the adjacent closed-form subunit.\\nRight inset: the adenine binding pocket of the closed-form subunit\\nholding the adenine ring of butyryl-CoA. Note that only the adenine\\nring of butyryl-CoA has visible electron density, while the rest of\\nthe molecule is disordered, and hence is indicated by a transparent\\nstick model. In both cases, three residues of the adjacent subunits,\\nK296, R303, and Y328, together hold the adenine ring.The binding of the cofactor and substrates in the closed-form\\nECR\\nsubunit is achieved by loops 37–44, 88–94, 338–350\\nand helices 6, 7, and 21 of the catalytic domains, which creates multiple\\ninteractions of the protein with the NADPH and butyryl-CoA (Figure \\u200bFigure33d–f). Notably,\\nthe CoA-thioester extends from the closed-form active site out to\\nthe oligomerization domain of the neighboring open-form subunit in\\nthe same dimer pair (from A to C in the A/C dimer, or from B to D\\nin the B/D dimer), where R352 a', 's represented as sequence\\nlogos. The numbering of residues, above the first row, is according\\nto their position in the K. setae ECR.\\n(d) Communication between the closed (green) and open (orange) subunits\\nacross the two dimers of the K. setae ECR. In the closed conformation the contacts between NADPH-H365-E165\\nand K332 of the adjacent open subunit allow for the correct intradimer\\ncommunication. In the open conformation the communication network\\nis compromised, as indicated by the increased distances between the\\namino acid side chains that cause the incorrect positioning of the\\nnicotinamide ring of NADPH.What are the molecular determinants that synchronize catalysis\\nacross the pair of open/closed form dimers? The intercatalytic domain\\ninterface is predominantly hydrophobic with some electrostatic interactions\\n(Figure \\u200bFigure55b). Most notable\\nare N218, which forms a hydrogen bond to N157 of the adjacent subunit\\nof the other dimer, and E151 interacting with the backbone nitrogen\\natom of ', 'nly electron density for the\\nadenine groups of the CoA thioester is observed (Figure \\u200bFigure33g). The “half-site reactivity”\\nis also consistent with a negative, rather than positive, cooperativity,\\nas almost all our kinetics data indicate.To understand if the\\nconventional half-site reactivity theory can\\nbe applied to the ECR tetramer case, we created a model for the enzyme\\nkinetics assuming positive or negative cooperativity between the pairs\\nof dimers. Instead of treating the four subunits as independent entities,\\nwe hypothesized that the two pairs of ECR dimers would cycle through\\nthe CO2-fixing reactions in synchrony: i.e., when one dimer\\nswitches from the open–close to the closed–open states,\\nthe other dimer would do the opposite, closed–open to open–closed\\nstates. Using this model, we simulated the case with just one dimer\\nwith two subunits working in negative or positive cooperativity using\\nthe method described by Hill and Levitski21 (Figure S12). For a thorough investigation\\nof ', '. U. S. A.\\n2009, 106, 8871–8876. 10.1073/pnas.0903939106. [PMC free article] [PubMed] [CrossRef] [Google Scholar]Erb T. J.; Zarzycki J.\\nBiochemical and synthetic biology approaches to improve\\nphotosynthetic CO2 -fixation. Curr.\\nOpin. Chem. Biol.\\n2016, 34, 72–79. 10.1016/j.cbpa.2016.06.026. [PMC free article] [PubMed] [CrossRef] [Google Scholar]Rosenthal R. G.; et al. Direct evidence for a covalent ene adduct intermediate\\nin NAD(P)H-dependent\\nenzymes. Nat. Chem. Biol.\\n2014, 10, 50–55. 10.1038/nchembio.1385. [PubMed] [CrossRef] [Google Scholar]Quade N.; Huo L.; Rachid S.; Heinz D. W.; Müller R.\\nUnusual carbon\\nfixation gives rise to diverse polyketide extender units. Nat. Chem. Biol.\\n2012, 8, 117–124. 10.1038/nchembio.734. [PubMed] [CrossRef] [Google Scholar]Zhang L.; et al. Rational Control of Polyketide Extender Units by Structure-Based\\nEngineering of a Crotonyl-CoA Carboxylase/Reductase in Antimycin Biosynthesis. Angew. Chemie Int. Ed.\\n2015, 54, 13462–13465. 10.1002/anie.201506899. [Pu']\n",
      "\n",
      "Deliverables:\n",
      "For each key question, assign a predicted usefulness score of the full source text using a 5-point Likert scale, with 1 being very unlikely to be usefulness to 5 being very likely useful and containing facts that answer the key question.\n",
      "\n",
      "The output should be of the following JSON format\n",
      "{\n",
      "    <insert key question index>: <insert predicted usefulness>,\n",
      "   etc.\n",
      "}\n",
      "\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n    \\\"1\\\": 4,\\n    \\\"2\\\": 3,\\n    \\\"3\\\": 2,\\n    \\\"4\\\": 1\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691116661,\n",
      "  \"id\": \"chatcmpl-7jez72soJTaVncN2nQOb2HI0HCZNH\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 30,\n",
      "    \"prompt_tokens\": 2070,\n",
      "    \"total_tokens\": 2100\n",
      "  }\n",
      "}\n",
      "rating_source_idx 0 Skimming url: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5626244/\n",
      "Going through url:  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5626244/\n",
      "select firefox options!\n",
      "hard coding chrome\n",
      "setting up chrome driver\n",
      "Driver is getting url\n",
      "set timeout!\n",
      "Page loaded within 15 seconds\n",
      "Driver got url\n",
      "Driver has found page source\n",
      "Handing off to Beautiful Soup!\n",
      "done extractin\n",
      "Text:  Back to Top\n",
      "Skip to main content\n",
      "The .gov means it’s official.\n",
      "Federal government websites often end in .gov or .mil. Before\n",
      "sharing sensitive information, make sure you’re on a federal\n",
      "government site.\n",
      "The site is secure.\n",
      "The https:// ensures that you are connecting to the\n",
      "official website and that any information you provide is encrypted\n",
      "and transmitted securely.\n",
      "Access keys\n",
      "NCBI Homepage\n",
      "MyNCBI Homepage\n",
      "Main Content\n",
      "Main Navigation\n",
      "Journal List\n",
      "Virulence\n",
      "v.8(6); 2017\n",
      "PMC5626244\n",
      "As a library, \n",
      "len(sample_chunks) 10\n",
      "Prompt:  \n",
      "Task: \n",
      "Based on the research question decomposition key questions and the sample text chunks of the source text, the goal is to identify how useful reading the full source text would be to extract direct quoted facts and determine the best answer to any of the key questions. \n",
      "\n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition:\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity compare to other bacteria?', '3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?', '4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}\n",
      "\n",
      "Sample text chunks from the source text:\n",
      "['Back to Top\\nSkip to main content\\nThe .gov means it’s official.\\nFederal government websites often end in .gov or .mil. Before\\nsharing sensitive information, make sure you’re on a federal\\ngovernment site.\\nThe site is secure.\\nThe https:// ensures that you are connecting to the\\nofficial website and that any information you provide is encrypted\\nand transmitted securely.\\nAccess keys\\nNCBI Homepage\\nMyNCBI Homepage\\nMain Content\\nMain Navigation\\nJournal List\\nVirulence\\nv.8(6); 2017\\nPMC5626244\\nAs a library, NLM provides access to scientific literature. Inclusion in an NLM database does not imply endorsement of, or agreement with,\\nthe contents by NLM or the National Institutes of Health.\\nLearn more about our disclaimer.\\nVirulence. 2017; 8(6): 938–958. Published online 2016 Oct 20.\\ndoi:\\xa010.1080/21505594.2016.1250995PMCID: PMC5626244PMID: 27763824Blue light treatment of Pseudomonas aeruginosa: Strong bactericidal activity, synergism with antibiotics and inactivation of virulence factorsGrzegorz Fila,a', 'ness for both MDR and wildtype (drug-sensitive) isolates.22 However, the aPDT approach combining a photosensitizer (PS) and light presents several challenges, such as the introducing of a photosensitizer in close proximity to bacterial cells and/or the attachment of a PS to the cell membrane and the development of strategies leading to increased photosensitizer selectivity for microbial cells instead of the host tissue. The use of UVC irradiation also has several limitations, primarily attributable to its harmful effects on eukaryotic cells and the possibility for host tissue damage, including carcinogenesis.23 Currently, a novel light-based approach, blue light therapy (BLT), is considered an attractive alternative. BLT exerts bactericidal (> 3 log10 units reduction in viable counts, up to bacterial eradication) effects without requiring the addition of exogenous photosensitizer.24,25 Moreover, BLT is considered safer for host tissue because it has fewer detrimental effects on mammali', 'termined using a reference heat-inactivated S. aureus Newman strain treated with P. aeruginosa extracellular fractions (“dark control:” kept in the dark; “illumination:” BLT; fluence 50 J/cm2; 15.7\\xa0mW/cm2; λ 405\\xa0nm; duration of irradiation: 3184 sec), and OD600 values were measured every 5\\xa0min during incubation (37°C). The figure represents the time at which the dark control supernatants reached 80% of maximum elastolytic activity. The values are the means of 3 separate experiments (the assay was performed for 3 independent experiments with one repetition per sample), and the bars represent the SE. The maximum value of absorbance was set as 1, and all other measurements were normalized as a ratio of 1. (B) Elastase (B)activity during BLT. The pseudolysin activity of elastase B was analyzed by performing and ECR assay. The reaction buffer containing ECR was treated with P. aeruginosa extracellular fractions (“dark control:” kept in the dark; “illumination:” BLT; fluence 50 J/cm2; 15.7\\xa0m', 'em (Fig.\\xa08). Sequential blue light irradiation and antibiotic administration controlled bacterial growth at minimum inhibitory concentration (MIC) values that were 2-, 8-, and even 6four-fold lower. The most pronounced effects occurred for ceftazidime (Fig.\\xa08, panels C and D) preceded by irradiation with a blue light fluence of 12.5 J/cm2 resulting in a 6four-fold decrease in the MIC value required for P. aeruginosa. In addition, the minimal bactericidal concentrations (MBCs) of antimicrobials acting synergistically with BLT were lower for blue light pre-treated P. aeruginosa strains (Table\\xa03). Moreover, no bacterial re-growth was observed at 24 hrs after the combination treatment.\\nOpen in a separate windowFigure 8.BLT renders P. aeruginosa susceptible to gentamycin, ceftazidime and meropenem. A checkerboard titration method was employed to analyze the effects of sequential BLT combined with the application of antimicrobial agents. We combined various concentrations of antimicrobial ag', ' microbial virulence factors, it has an advantage over conventional antibiotic therapy.Notably, all experiments were performed with stationary-phase P. aeruginosa cultures in which all relevant virulence factors were already expressed. Thus, the rationale for using cell-free supernatants to analyze the BLT-mediated inactivation of virulence factors may be called into question. During photodynamic processes, bacterial inactivation occurs via the loss of cell-membrane integrity and cell lysis, and all cytoplasmic components leak into the supernatant. If we would irradiate the cell culture and collect the supernatants after the treatment one could speculate that some other than photodynamic processes could occur and result in reduction of virulence factors activity. Moreover, photoinactivation utilizes the extracellular fractions of endogenous porphyrins in cell-free supernatants. Our results therefore suggest the realization of similar virulence factor photoinactivation, even in cell-fre', 'or 3 independent experiments with 3 repetitions per sample. The detection limit was 2.5\\xa0μM.71Hemolytic activityTo analyze the hemolytic activity of the bacterial supernatant, fresh sheep RBC (Innovative Research, USA) were employed using a previously described technique.15 The cells were washed 3\\xa0times in PBS (centrifuged for 5\\xa0min at 6000\\xa0rpm) and resuspended in RPMI-1640 medium (Roswell Park Memorial Institute, RPMI) (Sigma-Aldrich, Germany) to a final concentration of 2% (cell volume/medium volume). Next, 700\\xa0µl of the prepared RBC solution was mixed with 300\\xa0µl of the supernatant in an Eppendorf tube. The tubes were incubated horizontally at 37°C and 150\\xa0rpm for 25\\xa0min. After incubation, the suspension was centrifuged (10,000 rcf, 10\\xa0min, 4°C), and 240\\xa0µl of the sample was transferred into a 96 well plate. The released hemoglobin was determined by measuring the absorbance at λ 540\\xa0nm. The percentage (%) of lysed cells was calculated as follows: % = [(X-B)/(T-B)] × 100 (where B (bas', \"hodsStatistical analyses were performed using the statistical suite STATISTICA (data analysis software system) version 10.0 (StatSoft. Inc. (2011), www.statsoft.com) and Excel. Quantitative variables were characterized by the arithmetic mean of standard deviation. To test the homogeneity of the variances, the Leven (Brown-Forsythe) test was utilized. The statistical significance of differences between 2 groups was determined with Student's t-test. The significance level was P < 0.05.Go to:Supplementary MaterialKVIR_S_1250995.zip:Click here to view.(2.2M, zip)Go to:Disclosure of potential conflicts of interestNo potential conflicts of interest were disclosed.Go to:AcknowledgmentsThe isogenic pairs of isolates with affected phenazines production were kindly provided by Lars Dietrich from Department of Biological Sciences Columbia University New York. P. aeruginosa strains with disrupted movement systems were provided by Joanne N. Engel from Department of Medicine, University of Californi\", 'p://dx.doi.org/ 10.1007/s00784-013-1151-8 [PubMed] [CrossRef] [Google Scholar][33] Soukos NS, Som S, Abernethy AD, Ruggiero K, Dunham J, Lee C, Doukas AG, Goodson JM.\\nPhototargeting oral black-pigmented bacteria. Antimicrob Agents Chemother\\n2005; 49:1391-6; PMID:15793117; http://dx.doi.org/ 10.1128/AAC.49.4.1391-1396.2005 [PMC free article] [PubMed] [CrossRef] [Google Scholar][34] Maclean M, McKenzie K, Anderson JG, Gettinby G, MacGregor SJ.\\n405\\xa0nm light technology for the inactivation of pathogens and its potential role for environmental disinfection and infection control. J Hosp Infect\\n2014; 88:1-11; PMID:25066049; http://dx.doi.org/ 10.1016/j.jhin.2014.06.004 [PubMed] [CrossRef] [Google Scholar][35] Kleinpenning MM, Smits T, Frunt MH, van Erp PE, van de Kerkhof PC, Gerritsen RM.\\nClinical and histological effects of blue light on normal skin. Photodermatol Photoimmunol Photomed\\n2010; 26:16-21; PMID:20070834; http://dx.doi.org/ 10.1111/j.1600-0781.2009.00474.x [PubMed] [CrossRef] [Goo', '6; 16; PMID:27007376; http://dx.doi.org/ 10.3390/s16030408 [PMC free article] [PubMed] [CrossRef] [Google Scholar][72] Kessler E, Safrin M.\\nElastinolytic and proteolytic enzymes. Filloux A, and Ramos J. L.\\n135-69. 2014.\\nHumana Press. Pseudomonas. Methods and Protocols [Google Scholar][73] Ohman DE, Cryz SJ, Iglewski BH.\\nIsolation and characterization of Pseudomonas aeruginosa PAO mutant that produces altered elastase. J Bacteriol\\n1980; 142:836-42; PMID:6769912 [PMC free article] [PubMed] [Google Scholar][74] Kessler E, Safrin M, Abrams WR, Rosenbloom J, Ohman DE.\\nInhibitors and specificity of Pseudomonas aeruginosa LasA. J Biol Chem\\n1997; 272:9884-9; PMID:9092525; http://dx.doi.org/ 10.1074/jbc.272.15.9884 [PubMed] [CrossRef] [Google Scholar][75] Hamood AN, Griswold J, Colmer J.\\nCharacterization of elastase-deficient clinical isolates of Pseudomonas aeruginosa. Infect Immun\\n1996; 64:3154-60; PMID:8757847 [PMC free article] [PubMed] [Google Scholar][76] Kashid SG, Ghosh JS.\\nProduction, ', '81] Cumming G, Fidler F, Vaux DL.\\nError bars in experimental biology. J Cell Biol\\n2007; 177:7-11; PMID:17420288; http://dx.doi.org/ 10.1083/jcb.200611141 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][38] ASM\\nAntimicrobial Agents and Chemotherapy, Instructions to Authors. http://aac.asm.org/site/misc/2016AprilAACITA.pdf. 2016 [Google Scholar] [Ref list][39] Lau GW, Ran H, Kong F, Hassett DJ, Mavrodi D.\\nPseudomonas aeruginosa pyocyanin is critical for lung infection in mice. Infect Immun\\n2004; 72:4275-8; PMID:15213173; http://dx.doi.org/ 10.1128/IAI.72.7.4275-4278.2004 [PMC free article] [PubMed] [CrossRef] [Google Scholar] [Ref list][40] Reszka KJ, Bilski PJ, Britigan BE.\\nQuenching of singlet oxygen by pyocyanin and related phenazines. Photochem Photobiol\\n2010; 86:742-6; PMID:20408986; http://dx.doi.org/ 10.1111/j.1751-1097.2010.00728.x [PubMed] [CrossRef] [Google Scholar] [Ref list][41] Hawdon NA, Aval PS, Barnes RJ, Gravelle SK, Rosengren J, Khan S, Ciofu O, Johan']\n",
      "\n",
      "Deliverables:\n",
      "For each key question, assign a predicted usefulness score of the full source text using a 5-point Likert scale, with 1 being very unlikely to be usefulness to 5 being very likely useful and containing facts that answer the key question.\n",
      "\n",
      "The output should be of the following JSON format\n",
      "{\n",
      "    <insert key question index>: <insert predicted usefulness>,\n",
      "   etc.\n",
      "}\n",
      "\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n    \\\"1\\\": 4,\\n    \\\"2\\\": 2,\\n    \\\"3\\\": 3,\\n    \\\"4\\\": 1\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691116698,\n",
      "  \"id\": \"chatcmpl-7jezirf9HWNxyHKktiFxWYRVXOYiI\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 30,\n",
      "    \"prompt_tokens\": 3127,\n",
      "    \"total_tokens\": 3157\n",
      "  }\n",
      "}\n",
      "rating_source_idx 0 Skimming url: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3213379/\n",
      "Going through url:  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3213379/\n",
      "select firefox options!\n",
      "hard coding chrome\n",
      "setting up chrome driver\n",
      "Driver is getting url\n",
      "set timeout!\n",
      "Page loaded within 15 seconds\n",
      "Driver got url\n",
      "Driver has found page source\n",
      "Handing off to Beautiful Soup!\n",
      "done extractin\n",
      "Text:  Back to Top\n",
      "Skip to main content\n",
      "The .gov means it’s official.\n",
      "Federal government websites often end in .gov or .mil. Before\n",
      "sharing sensitive information, make sure you’re on a federal\n",
      "government site.\n",
      "The site is secure.\n",
      "The https:// ensures that you are connecting to the\n",
      "official website and that any information you provide is encrypted\n",
      "and transmitted securely.\n",
      "Access keys\n",
      "NCBI Homepage\n",
      "MyNCBI Homepage\n",
      "Main Content\n",
      "Main Navigation\n",
      "Journal List\n",
      "Genetics\n",
      "v.189(3); 2011 Nov\n",
      "PMC3213379\n",
      "As a libr\n",
      "len(sample_chunks) 8\n",
      "Prompt:  \n",
      "Task: \n",
      "Based on the research question decomposition key questions and the sample text chunks of the source text, the goal is to identify how useful reading the full source text would be to extract direct quoted facts and determine the best answer to any of the key questions. \n",
      "\n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition:\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity compare to other bacteria?', '3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?', '4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}\n",
      "\n",
      "Sample text chunks from the source text:\n",
      "['Back to Top\\nSkip to main content\\nThe .gov means it’s official.\\nFederal government websites often end in .gov or .mil. Before\\nsharing sensitive information, make sure you’re on a federal\\ngovernment site.\\nThe site is secure.\\nThe https:// ensures that you are connecting to the\\nofficial website and that any information you provide is encrypted\\nand transmitted securely.\\nAccess keys\\nNCBI Homepage\\nMyNCBI Homepage\\nMain Content\\nMain Navigation\\nJournal List\\nGenetics\\nv.189(3); 2011 Nov\\nPMC3213379\\nAs a library, NLM provides access to scientific literature. Inclusion in an NLM database does not imply endorsement of, or agreement with,\\nthe contents by NLM or the National Institutes of Health.\\nLearn more about our disclaimer.\\nGenetics. 2011 Nov; 189(3): 871–883.\\ndoi:\\xa010.1534/genetics.111.132191PMCID: PMC3213379PMID: 21900267Isoform-Specific Regulation of a Steroid Hormone Nuclear Receptor by an E3 Ubiquitin Ligase in Drosophila melanogasterAna-Citlali Gradilla,*†,1 Alicia Mansilla,*,1 and\\nAlberto Fer', 'at the DNA-binding domain); bon21B/TM6 (Bonus defective); P(ry+)ftz-f103649/TM3 (βFtz-f1 defective); Df(3R)Spf,P(ry+)l(3)ArBEip78C/MKRS (Eip78C defective); P(y+)Eip55EKG02526/CyO (Eip55E defective); P(y+)Eip75BKG03025 (Eip75B defective); P(y+)CG15523KG07288 (Eif2b defective); Df(2)Drlrv7(Spn1 defective); P(w+)UAS-EcRB13b; P(w+)UAS-EcRB23a; P(w+)UAS-EcRA3a; P(w+)UAS-CD8GFP; Beadex-Gal4-MS1096; how-Gal4-24B/TM2; tubulin-Gal4-LL7/TM3; Dpp-Gal4-PS6A/CyO; Distal-lesGal4-MD23/CyO; Gal4-C649; EcR.GET-BD-Gal4 (expressed in an EcRA domain); ari-1a-Gal4-NP1063 (expressed in an ari-1a domain); and phm-Gal4UAS-CD8-GFP (expressed in the phantom domain, the prothoracic cells of the ring gland). The expression domains of Gal4 drivers were tested using the reporters UAS-LacZ or UAS-GFPS65T. For gene interaction studies, ari-1a2/FM7i females were crossed with spn1 (serine protease inhibitor 1); eIF2b (translation initiation factor); Bonus (nuclear receptor cofactor); and EcRA, EcRA-B1-B2, EcRB1-B2 and ', 'ss-of-function mutant phenotypes (Aguilera et al. 2000), we addressed here the overexpression counterpart to explore possible trait similarities. Phenotypes caused by the excess of ARI-1a function were investigated using the UAS/Gal4 system (Brand and Perrimon 1993). Generalized overexpression by the tubulin-Gal4-LL7 driver causes lethality mostly at the white prepupae stage (Figure 1A). These mutants degenerate in their pupal cases and never emerge as adults. Using Gal4 drivers with more restricted expression domains (how-Gal4-24B) (see Supporting Information, Table S1), individuals develop further but arrest before head eversion and leg elongation, resulting in a cryptocephalic-like phenotype (Figure 1, B and C). In general, these traits are quite similar to those of genes involved in the ecdysone-signaling pathway (Bender et al. 1997; D’Avino and Thummel 2000; Fortier et al. 2003; Schubiger et al. 2003; Beckstead et al. 2005; Davis et al. 2005; King-Jones and Thummel 2005). Interest', 'lethal ari-1a2 adult males in heterozygous condition with ecdysone pathway mutants. Viability is calculated as a percentage with respect to the most viable sibling genotype. The allele ari-1a2 is a fully penetrant lethal. Thus, all rescue values are significant. Note that the mutant in EcRA is the most effective rescuer at 30%. The following ecdysone pathway mutant alleles were used: Df(2)Drlrv7(Spn1 defective); P(y+)CG15523KG07288 (Eif2b defective); bon21B (Bonus defective); Df(2)EcRA112 (EcRA defective) EcR225 (EcRB1, B2 defective) (Carney et al. 2004); EcRM554fs (EcRA, B1, B2 defective); P(ry+)ftz-f103649 (βFtz-f1 defective); Df(3R)Spf,P(ry+)l(3)ArBEip78C (Eip78C defective); P(y+)Eip75BKG03025 (Eip75B defective); and P(y+)Eip55EKG02526 (Eip55E defective). Data are from crosses in which a minimum of 100 individuals of the most viable genotype were screened. (B) Developmental phase reached by genotypes in which ari-1a and/or usp expression is altered. The various genotypes exhibit a p', 'n vitroSince ARI-1a is an E3 enzyme, we addressed the effects on the putative substrates. We measured protein levels of EcRA and USP under lack- and excess-of-function conditions for ARI-1a. Similar experiments were carried out for EcRB1 to define the specificity of ARI-1a activity. Prepupae of ari-1a2 and tub-Gal4-LL7 > UAS-ari-1a lethal mutants were used for protein extracts and Western blotting with EcRA, EcRB1, and USP-specific antibodies. Quantitative differences for EcRA, and to a lesser extent for USP, were clearly detected (Figure 6, A and B). While EcRB1 showed no apparent change, EcRA levels decreased fourfold when ARI-1a was overexpressed and increased sixfold in ari-1a lack-of-function mutants. By contrast, USP levels changed in the opposite direction to those of EcRA: they doubled in the first case and decreased twofold in the second.Open in a separate windowFigure 6\\u2002ARI-1a regulates the protein levels of EcRA and Utraspiracle (USP) but not those of EcRB1. (A, left column)', ' as well, given the conservation of the ARI type of E3 enzymes (Marin and Ferrus 2002; Mladek et al. 2003) and the EcR type of nuclear receptors (King-Jones and Thummel 2005). Thus, it will be worth investigating if the human liver X receptor, an ortholog of EcR, undergoes a similar regulatory process by the human ARI-1a. Considered in a wider context, the specificity of hormone receptor signaling seems to depend not only on the particular activated isoform, but also on the isoform-specific mechanism of modulation. The data reported here illustrate one such mechanism that yields specific phenotypes and has gene transcriptional effects. In addition, the described process represents a mechanism to coregulate other functionally linked nuclear receptors—in this case, USP.Go to:AcknowledgmentsWe are indebted to the Iowa Hybridoma Bank for antibodies, to Bloomington Stock Center for most fly strains, and to M. Bender (University of Georgia) for EcR mutants. The S2 cell line deficient for EcR', 'ogenesis. Annu. Rev. Pharmacol. Toxicol. 46: 451–480 [PubMed] [Google Scholar]Mladek C., Guger K., Hauser M. T., 2003. \\u2003Identification and characterization of the ARIADNE gene family in Arabidopsis: a group of putative E3 ligases. Plant Physiol. 131: 27–40 [PMC free article] [PubMed] [Google Scholar]Patel, N. V., and B. M. Forman, 2004\\u2003Linking lipids, Alzheimer’s and LXRs? Nucl. Recept. Signal. 2: e001. [PMC free article] [PubMed]Qian S. B., Waldron L., Choudhary N., Klevit R. E., Chazin W. J., et al., 2009. \\u2003Engineering a ubiquitin ligase reveals conformational flexibility required for ubiquitin transfer. J. Biol. Chem. 284: 26797–26802 [PMC free article] [PubMed] [Google Scholar]Riddiford L. M., Cherbas P., Truman J. W., 2000. \\u2003Ecdysone receptors and their biological actions. Vitam. Horm. 60: 1–73 [PubMed] [Google Scholar]Robertson K. M., Norgard M., Windahl S. H., Hultenby K., Ohlsson C., et al., 2006. \\u2003Cholesterol-sensing receptors, liver X receptor alpha and beta, have novel and d', '37 [PMC free article] [PubMed] [Google Scholar] [Ref list]Spyridon M., Moraes L. A., Jones C. I., Sage T., Sasikumar P., et al., 2011. \\u2003LXR as a novel antithrombotic target. Blood 117: 5751–5761 [PMC free article] [PubMed] [Google Scholar] [Ref list]Talbot W. S., Swyryd E. A., Hogness D. S., 1993. \\u2003Drosophila tissues with different metamorphic responses to ecdysone express different ecdysone receptor isoforms. Cell 73: 1323–1337 [PubMed] [Google Scholar] [Ref list]Kim S. J., Park J. G., Lee C. C., 1999. \\u2003Transcript titers of ecdysteroid receptor components vary between tissues and stages during Drosophila development. Mol. Cells 9: 61–66 [PubMed] [Google Scholar] [Ref list]Sung C., Robinow S., 2000. \\u2003Characterization of the regulatory elements controlling neuronal expression of the A-isoform of the ecdysone receptor gene of Drosophila melanogaster. Mech. Dev. 91: 237–248 [PubMed] [Google Scholar] [Ref list]Davis M. B., Carney G. E., Robertson A. E., Bender M., 2005. \\u2003Phenotypic analysi']\n",
      "\n",
      "Deliverables:\n",
      "For each key question, assign a predicted usefulness score of the full source text using a 5-point Likert scale, with 1 being very unlikely to be usefulness to 5 being very likely useful and containing facts that answer the key question.\n",
      "\n",
      "The output should be of the following JSON format\n",
      "{\n",
      "    <insert key question index>: <insert predicted usefulness>,\n",
      "   etc.\n",
      "}\n",
      "\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n    \\\"1\\\": 3,\\n    \\\"2\\\": 2,\\n    \\\"3\\\": 1,\\n    \\\"4\\\": 1\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691116723,\n",
      "  \"id\": \"chatcmpl-7jf07idAz0aZuSrGMdLNwbcu9PZ1H\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 30,\n",
      "    \"prompt_tokens\": 2834,\n",
      "    \"total_tokens\": 2864\n",
      "  }\n",
      "}\n"
     ]
    }
   ],
   "source": [
    "# Skimming through each highly relevant paper from skimming\n",
    "for decomposition_idx, key_question_decomposition in enumerate(key_question_decomposition_list):\n",
    "    with open(f'autoscious_logs/{search_query_file_safe}/sources/kq{decomposition_idx}/initial_search_queries.json', 'r') as f:\n",
    "        key_question_initial_search_queries = json.load(f)\n",
    "\n",
    "    for query_idx, query in key_question_initial_search_queries.items():\n",
    "        # open filtered search results\n",
    "        with open(f'autoscious_logs/{search_query_file_safe}/sources/kq{decomposition_idx}/rated_web_results_query_{int(query_idx)}.json', 'r') as f:\n",
    "            ratings_url_dict = json.loads(f.read())\n",
    "\n",
    "        # open web search info to extract metadata\n",
    "        with open(f'autoscious_logs/{search_query_file_safe}/sources/kq{decomposition_idx}/initial_search_results_query_{int(query_idx)}.json', 'r') as f:\n",
    "            web_search_info = json.load(f)\n",
    "        \n",
    "        for rating, urls in ratings_url_dict.items():\n",
    "            if rating == '5' or rating == '4': # TODO: figure out a good way to iterate and go to 4 after\n",
    "                # Start with iterating through 4s and 5s of ratings_url_dict\n",
    "                folder_path = f'autoscious_logs/{search_query_file_safe}/sources/kq{decomposition_idx}/predicted_usefulness_{rating}'\n",
    "                if not os.path.exists(folder_path):\n",
    "                    os.makedirs(folder_path)\n",
    "\n",
    "                for rating_source_idx, url in enumerate(urls):\n",
    "                    print(\"rating_source_idx\", rating_source_idx, \"Skimming url:\", url)\n",
    "                    text = scrape_text_with_selenium_no_agent(url, None)\n",
    "\n",
    "                    # Only evaluate websites you're able to scrape\n",
    "                    if text != \"No information found\":\n",
    "                        total_chunks = len(text) / CHUNK_SIZE\n",
    "                        num_chunk_samples = min(int(total_chunks * SAMPLING_FACTOR), MAX_CHUNKS)\n",
    "                        sample_chunks = get_sample_chunks(text, CHUNK_SIZE, num_chunk_samples)\n",
    "                        print(\"len(sample_chunks)\", len(sample_chunks))\n",
    "\n",
    "                        # Get predicted usefulness based on sample chunks\n",
    "                        predicted_usefulness_results = json.loads(chat_openai(get_predicted_usefulness_of_text_prompt(context, decomposition, sample_chunks), model=\"gpt-3.5-turbo\")[0])\n",
    "\n",
    "                        # save filtered search results\n",
    "                        with open(f'{folder_path}/query_{query_idx}_url_index_{rating_source_idx}.json', 'w') as f:\n",
    "                            json.dump(predicted_usefulness_results, f, indent=2)\n",
    "                        \n",
    "                        # Check if any scores were (4 or) 5, because then we should save the full text\n",
    "                        pred_usefulness = predicted_usefulness_results.values()\n",
    "                        if 5 in pred_usefulness or '5' in pred_usefulness or 4 in pred_usefulness or '4' in pred_usefulness:\n",
    "                            title = find_title(url, web_search_info)\n",
    "                            with open(f'autoscious_logs/{sanitize_filename(search_query)}/sources/full_text/{sanitize_filename(title)}.txt', 'w', encoding='utf-8') as f:\n",
    "                                f.write(title + '\\n')\n",
    "                                f.write(url + '\\n')\n",
    "                                f.write(text)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
