{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from autogpt.commands.web_selenium import browse_website, scrape_text_with_selenium_no_agent\n",
    "import json\n",
    "from util import sanitize_filename\n",
    "import os\n",
    "from prompts import get_predicted_usefulness_of_text_prompt\n",
    "from collections import defaultdict\n",
    "from llm import chat_openai\n",
    "from autogpt.commands.web_search import web_search_ddg\n",
    "from datetime import datetime"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "search_query = \"How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "search_query_file_safe = sanitize_filename(search_query)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load in decomposition\n",
    "with open(f'autoscious_logs/{search_query_file_safe}/decompositions/improved_decomposition.json', 'r') as f:\n",
    "    decomposition = json.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity compare to other bacteria?', '3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?', '4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}\n"
     ]
    }
   ],
   "source": [
    "# Get key questions\n",
    "key_questions = decomposition['key_drivers']['1']['hypotheses']['1']['key_questions']\n",
    "print(key_questions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create sources folder\n",
    "if not os.path.exists(f'autoscious_logs/{search_query_file_safe}/sources'):\n",
    "    os.makedirs(f'autoscious_logs/{search_query_file_safe}/sources') \n",
    "if not os.path.exists(f'autoscious_logs/{search_query_file_safe}/sources/full_text'):\n",
    "    os.makedirs(f'autoscious_logs/{search_query_file_safe}/source/full_text') \n",
    "\n",
    "for key in key_questions.keys():\n",
    "    if not os.path.exists(f'autoscious_logs/{search_query_file_safe}/sources/kq{int(key)-1}'):\n",
    "        os.makedirs(f'autoscious_logs/{search_query_file_safe}/source/kq{int(key)-1}') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "search_engine = \"General\"\n",
    "# search_engine = \"General\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Getting key questions and the decomposition and context"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Key questions decomposition list:  [{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}, {'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'2': 'How does ECR enzyme activity compare to other bacteria?'}}}}}}, {'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}, {'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}]\n"
     ]
    }
   ],
   "source": [
    "# Create a decomposition for each key question only\n",
    "context = \"Enoyl-CoA carboxylase/reductase enzymes (ECRs)\"\n",
    "key_question_decomposition_list = []\n",
    "for driver_key, driver_value in decomposition['key_drivers'].items():\n",
    "    for hypothesis_key, hypothesis_value in driver_value['hypotheses'].items():\n",
    "        for question_key, question_value in hypothesis_value['key_questions'].items():\n",
    "            new_decomposition = decomposition.copy()\n",
    "            new_decomposition['key_drivers'] = {\n",
    "                driver_key: {\n",
    "                    'driver': driver_value['driver'],\n",
    "                    'hypotheses': {\n",
    "                        hypothesis_key: {\n",
    "                            'hypothesis': hypothesis_value['hypothesis'],\n",
    "                            'key_questions': {\n",
    "                                question_key: question_value\n",
    "                            }\n",
    "                        }\n",
    "                    }\n",
    "                }\n",
    "            }\n",
    "            key_question_decomposition_list.append(new_decomposition)\n",
    "print(\"Key questions decomposition list: \", key_question_decomposition_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Coming up with many good search queries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_initial_search_queries_prompt(context, key_question_decomposition, search_engine):\n",
    "  if search_engine == \"General\":\n",
    "    return f'''\n",
    "Context:\n",
    "{context}\n",
    "\n",
    "Research question decomposition\n",
    "{key_question_decomposition}\n",
    "\n",
    "Task:\n",
    "For the key question, write a clear and comprehensive but short (around 3 queries) list of search keyword queries optimized for best search engine results, so that you can confidently and quickly surface the most relevant information to determine the best answer to the question. \n",
    "\n",
    "The output should be in JSON format: \n",
    "```json\n",
    "{{\n",
    "  1: \"<insert query>\",\n",
    "  etc.\n",
    "}}\n",
    "\n",
    "Respond only with the output, with no explanation or conversation.\n",
    "'''\n",
    "  elif search_engine == \"OpenAlex\":\n",
    "    return f'''\n",
    "Context:\n",
    "{context}\n",
    "\n",
    "Research question decomposition\n",
    "{key_question_decomposition}\n",
    "\n",
    "Task:\n",
    "For the key question, write a clear and comprehensive but short (around 3 queries) list of search keyword queries optimized for best results from a research paper database that searches using an index of word sequences called n-grams, so that you can confidently and quickly surface the most relevant information to determine the best answer to the question. \n",
    "\n",
    "The output should be in JSON format: \n",
    "```json\n",
    "{{\n",
    "  1: \"<insert query>\",\n",
    "  etc.\n",
    "}}\n",
    "\n",
    "Respond only with the output, with no explanation or conversation.\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}\n",
      "\n",
      "Task:\n",
      "For the key question, write a clear and comprehensive but short (around 3 queries) list of search keyword queries optimized for best search engine results, so that you can confidently and quickly surface the most relevant information to determine the best answer to the question. \n",
      "\n",
      "The output should be in JSON format: \n",
      "```json\n",
      "{\n",
      "  1: \"<insert query>\",\n",
      "  etc.\n",
      "}\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(get_initial_search_queries_prompt(context, key_question_decomposition_list[0], search_engine))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Prompt:  \n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}\n",
      "\n",
      "Task:\n",
      "For the key question, write a clear and comprehensive but short (around 3 queries) list of search keyword queries optimized for best search engine results, so that you can confidently and quickly surface the most relevant information to determine the best answer to the question. \n",
      "\n",
      "The output should be in JSON format: \n",
      "```json\n",
      "{\n",
      "  1: \"<insert query>\",\n",
      "  etc.\n",
      "}\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"1\\\": \\\"ECR enzyme activity Kitsatospor setae bacteria\\\",\\n  \\\"2\\\": \\\"Factors affecting ECR enzyme activity\\\",\\n  \\\"3\\\": \\\"Regulation of ECR enzyme activity in bacteria\\\"\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691161687,\n",
      "  \"id\": \"chatcmpl-7jqhLghbedyUReTc9IosfSPoPyySk\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 45,\n",
      "    \"prompt_tokens\": 276,\n",
      "    \"total_tokens\": 321\n",
      "  }\n",
      "}\n",
      "Prompt:  \n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'2': 'How does ECR enzyme activity compare to other bacteria?'}}}}}}\n",
      "\n",
      "Task:\n",
      "For the key question, write a clear and comprehensive but short (around 3 queries) list of search keyword queries optimized for best search engine results, so that you can confidently and quickly surface the most relevant information to determine the best answer to the question. \n",
      "\n",
      "The output should be in JSON format: \n",
      "```json\n",
      "{\n",
      "  1: \"<insert query>\",\n",
      "  etc.\n",
      "}\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"1\\\": \\\"ECR enzyme activity in Kitsatospor setae bacteria\\\",\\n  \\\"2\\\": \\\"Comparison of ECR enzyme activity in Kitsatospor setae bacteria with other bacteria\\\",\\n  \\\"3\\\": \\\"Factors affecting ECR enzyme activity in Kitsatospor setae bacteria\\\"\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691161690,\n",
      "  \"id\": \"chatcmpl-7jqhOuXC2xITeLBg6n0V8OOYIhNkl\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 60,\n",
      "    \"prompt_tokens\": 270,\n",
      "    \"total_tokens\": 330\n",
      "  }\n",
      "}\n",
      "Prompt:  \n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}\n",
      "\n",
      "Task:\n",
      "For the key question, write a clear and comprehensive but short (around 3 queries) list of search keyword queries optimized for best search engine results, so that you can confidently and quickly surface the most relevant information to determine the best answer to the question. \n",
      "\n",
      "The output should be in JSON format: \n",
      "```json\n",
      "{\n",
      "  1: \"<insert query>\",\n",
      "  etc.\n",
      "}\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"1\\\": \\\"Factors influencing ECR enzyme activity in Kitsatospor setae bacteria\\\",\\n  \\\"2\\\": \\\"ECR enzyme regulation in Kitsatospor setae bacteria\\\",\\n  \\\"3\\\": \\\"ECR enzyme kinetics in Kitsatospor setae bacteria\\\"\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691161693,\n",
      "  \"id\": \"chatcmpl-7jqhRzuOcqUkfCetO4zHnxBMZc39H\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 55,\n",
      "    \"prompt_tokens\": 274,\n",
      "    \"total_tokens\": 329\n",
      "  }\n",
      "}\n",
      "Prompt:  \n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}\n",
      "\n",
      "Task:\n",
      "For the key question, write a clear and comprehensive but short (around 3 queries) list of search keyword queries optimized for best search engine results, so that you can confidently and quickly surface the most relevant information to determine the best answer to the question. \n",
      "\n",
      "The output should be in JSON format: \n",
      "```json\n",
      "{\n",
      "  1: \"<insert query>\",\n",
      "  etc.\n",
      "}\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"1\\\": \\\"ECR enzyme activity Kitsatospor setae bacteria\\\",\\n  \\\"2\\\": \\\"Efficiency of ECR enzymes in Kitsatospor setae bacteria\\\",\\n  \\\"3\\\": \\\"Applications of high ECR enzyme activity in Kitsatospor setae bacteria\\\"\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691161695,\n",
      "  \"id\": \"chatcmpl-7jqhTxlFlKMcQGK9oB8ZgfvMVSUWo\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 57,\n",
      "    \"prompt_tokens\": 280,\n",
      "    \"total_tokens\": 337\n",
      "  }\n",
      "}\n"
     ]
    }
   ],
   "source": [
    "# TODO: Iterate through all key questions, or just ask to generate an initial set of search queries considering all key questions. Then when we iterate, we can see what facts have been added to what papers, and what unanswered questions we should focus our queries on.\n",
    "\n",
    "context = \"Enoyl-CoA carboxylase/reductase enzymes (ECRs)\"\n",
    "\n",
    "for decomposition_idx, key_question_decomposition in enumerate(key_question_decomposition_list):\n",
    "    key_question_initial_search_queries = json.loads(chat_openai(get_initial_search_queries_prompt(context, key_question_decomposition, search_engine), model=\"gpt-3.5-turbo\")[0])\n",
    "\n",
    "    with open(f'autoscious_logs/{search_query_file_safe}/sources/kq{decomposition_idx}/initial_search_queries.json', 'w') as f:\n",
    "        json.dump(key_question_initial_search_queries, f, indent=2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Web search given search keywords"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "from dotenv import load_dotenv\n",
    "from googleapiclient.discovery import build\n",
    "load_dotenv()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "import time\n",
    "def google_search_raw(search_term, cse_id, **kwargs):\n",
    "    service = build(\"customsearch\", \"v1\", developerKey=os.getenv('DEV_KEY'))\n",
    "    res = service.cse().list(q=search_term, cx=cse_id, **kwargs).execute()\n",
    "\n",
    "    search_results = res.get(\"items\", [])\n",
    "    time.sleep(1)\n",
    "\n",
    "    # Create a list of only the URLs from the search results\n",
    "    search_results_links = [item[\"link\"] for item in search_results]\n",
    "    return search_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "def search_google(search_query):\n",
    "    num_google_searches = 8\n",
    "    results = google_search_raw(search_query, os.getenv('MY_CSE_ID'), num=num_google_searches, lr=\"lang_en\", cr=\"countryUS\")\n",
    "    return results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "query:  ECR enzyme activity Kitsatospor setae bacteria\n",
      "query:  Factors affecting ECR enzyme activity\n",
      "trying google search!\n",
      "query:  Regulation of ECR enzyme activity in bacteria\n",
      "query:  ECR enzyme activity in Kitsatospor setae bacteria\n",
      "query:  Comparison of ECR enzyme activity in Kitsatospor setae bacteria with other bacteria\n",
      "trying google search!\n",
      "query:  Factors affecting ECR enzyme activity in Kitsatospor setae bacteria\n",
      "query:  Factors influencing ECR enzyme activity in Kitsatospor setae bacteria\n",
      "query:  ECR enzyme regulation in Kitsatospor setae bacteria\n",
      "trying google search!\n",
      "query:  ECR enzyme kinetics in Kitsatospor setae bacteria\n",
      "query:  ECR enzyme activity Kitsatospor setae bacteria\n",
      "query:  Efficiency of ECR enzymes in Kitsatospor setae bacteria\n",
      "trying google search!\n",
      "query:  Applications of high ECR enzyme activity in Kitsatospor setae bacteria\n"
     ]
    }
   ],
   "source": [
    "MAX_RETRIES = 3\n",
    "\n",
    "for decomposition_idx, key_question_decomposition in enumerate(key_question_decomposition_list):\n",
    "    with open(f'autoscious_logs/{search_query_file_safe}/sources/kq{decomposition_idx}/initial_search_queries.json', 'r') as f:\n",
    "        key_question_initial_search_queries = json.load(f)\n",
    "\n",
    "    for idx, query in key_question_initial_search_queries.items():\n",
    "        print(\"query: \", query)\n",
    "\n",
    "        # DDG\n",
    "        web_search_res = json.loads(web_search_ddg(query))\n",
    "        if len(web_search_res) == 0:\n",
    "            print(\"trying google search!\")\n",
    "            # Google\n",
    "            web_search_res_raw = search_google(query) # google uses 'link' instead of 'href'\n",
    "            web_search_res = [{\n",
    "                'title': web_search_res_raw[i]['title'], \n",
    "                'href': web_search_res_raw[i]['link'], \n",
    "                'body': web_search_res_raw[i]['snippet']\n",
    "                } for i in range(len(web_search_res_raw))\n",
    "            ]\n",
    "        \n",
    "        # save web search results\n",
    "        with open(f'autoscious_logs/{search_query_file_safe}/sources/kq{decomposition_idx}/initial_search_results_query_{idx}.json', 'w') as f:\n",
    "            json.dump(web_search_res, f, indent=2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reading type 1: filtering unlikely relevant sources based on title and body"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_filtering_web_results_ratings(context, decomposition, web_search_res):\n",
    "    return f'''\n",
    "Context:\n",
    "{context}\n",
    "\n",
    "Research question decomposition:\n",
    "{decomposition}\n",
    "\n",
    "Search results:\n",
    "{web_search_res}\n",
    "\n",
    "Task:\n",
    "Based on the key questions and each search result's title and body content, identify which key question indexes the search result content is likely most useful for or -1 if it isn't useful for any key question, and for each key question index identified, assign a relevance score of the search result content using a 5-point Likert scale, with 1 being very irrelevant to 5 being very relevant to the key question.\n",
    "\n",
    "The output should be in JSON format: \n",
    "```json\n",
    "{{\n",
    "  <'insert href'>: [['insert key question index', 'relevance score'], etc.],\n",
    "  etc.\n",
    "}}\n",
    "```\n",
    "\n",
    "Respond only with the output, with no explanation or conversation.\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Prompt:  \n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition:\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity compare to other bacteria?', '3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?', '4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}\n",
      "\n",
      "Search results:\n",
      "[{'title': 'Awakening the Sleeping Carboxylase Function of Enzymes: Engineering the ...', 'href': 'https://pubs.acs.org/doi/10.1021/jacs.9b03431', 'body': 'However, for enoyl-CoA carboxylase/reductase from Kitasatospora setae (ECR Ks), four conserved amino acids that form a CO 2-binding pocket at the active site were described recently (Figure 1a). These four amino acids anchor and position the CO 2 molecule during catalysis, in which a reactive enolate is formed that attacks the CO 2. Figure 1'}, {'title': 'How a soil microbe could rev up artificial photosynthesis', 'href': 'https://www6.slac.stanford.edu/news/2022-04-29-how-soil-microbe-could-rev-artificial-photosynthesis', 'body': 'This \\'photosynthesis 2.0\\' could take place in living or synthetic systems such as artificial chloroplasts - droplets of water suspended in oil.\". A close-up look at Kitasatospora setae, a bacterium isolated from soil in Japan. These bacteria fix carbon - turn carbon dioxide from their environment into biomolecules they need to survive ...'}, {'title': 'Four amino acids define the CO2 binding pocket of enoyl-CoA ...', 'href': 'https://pubmed.ncbi.nlm.nih.gov/31243147/', 'body': 'Here we investigated the interactions of CO 2 within the active site of Ecr from Kitasatospora setae Combining experimental biochemistry, protein crystallography, and advanced computer simulations we show that 4 amino acids, N81, F170, E171, and H365, are required to create a highly efficient CO 2 -fixing enzyme.'}, {'title': 'Bacterial Enzyme Converts CO2 Into Carbon Compounds 20x ... - SciTechDaily', 'href': 'https://scitechdaily.com/bacterial-enzyme-converts-co2-into-carbon-compounds-20x-faster-than-photosynthesis/', 'body': 'Improving on nature The enzyme the team studied is part of a family called enoyl-CoA carboxylases/reductases, or ECRs. It comes from soil bacteria called Kitasatospora setae, which in addition to their carbon-fixing skills can also produce antibiotics.'}, {'title': 'Reaction scheme and structural organization of the K. setae ECR ...', 'href': 'https://www.researchgate.net/figure/Reaction-scheme-and-structural-organization-of-the-K-setae-ECR-complex-a_fig1_360182999', 'body': 'We first determined four highresolution K. setae ECR structures in different conformational states: the apo form and three holo forms in a binary complex with the reduced cofactor NADPH, in a...'}, {'title': 'How a soil microbe could rev up artificial photosynthesis - Phys.org', 'href': 'https://phys.org/news/2022-04-soil-microbe-rev-artificial-photosynthesis.html', 'body': 'This depiction of ECR, an enzyme found in soil bacteria, shows each of its four identical molecules in a different color. These molecules work together in pairs—blue with white and green with ...'}, {'title': 'Protease Activity Analysis: A Toolkit for Analyzing Enzyme Activity ...', 'href': 'https://pubs.acs.org/doi/10.1021/acsomega.2c01559', 'body': 'Analyzing the activity of proteases and their substrates is critical to defining the biological functions of these enzymes and to designing new diagnostics and therapeutics that target protease dysregulation in disease.'}, {'title': 'Screening for extracellular enzyme activities by bacteria isolated from ...', 'href': 'https://annalsmicrobiology.biomedcentral.com/articles/10.1007/BF03175056', 'body': 'Ninety-five bacterial isolates, obtained in pure cultures from samples collected in various sites of the East sector of the Tyrrhenian Sea, were plate-screened for their ability to produce twelve extracellular enzyme activities in order to find new strains for possible applications.'}]\n",
      "\n",
      "Task:\n",
      "Based on the key questions and each search result's title and body content, identify which key question indexes the search result content is likely most useful for or -1 if it isn't useful for any key question, and for each key question index identified, assign a relevance score of the search result content using a 5-point Likert scale, with 1 being very irrelevant to 5 being very relevant to the key question.\n",
      "\n",
      "The output should be in JSON format: \n",
      "```json\n",
      "{\n",
      "  <'insert href'>: [['insert key question index', 'relevance score'], etc.],\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"https://pubs.acs.org/doi/10.1021/jacs.9b03431\\\": [[\\\"1\\\", 5]],\\n  \\\"https://www6.slac.stanford.edu/news/2022-04-29-how-soil-microbe-could-rev-artificial-photosynthesis\\\": [[\\\"1\\\", 4]],\\n  \\\"https://pubmed.ncbi.nlm.nih.gov/31243147/\\\": [[\\\"1\\\", 5]],\\n  \\\"https://scitechdaily.com/bacterial-enzyme-converts-co2-into-carbon-compounds-20x-faster-than-photosynthesis/\\\": [[\\\"1\\\", 5]],\\n  \\\"https://www.researchgate.net/figure/Reaction-scheme-and-structural-organization-of-the-K-setae-ECR-complex-a_fig1_360182999\\\": [[\\\"1\\\", 5]],\\n  \\\"https://phys.org/news/2022-04-soil-microbe-rev-artificial-photosynthesis.html\\\": [[\\\"1\\\", 4]],\\n  \\\"https://pubs.acs.org/doi/10.1021/acsomega.2c01559\\\": -1,\\n  \\\"https://annalsmicrobiology.biomedcentral.com/articles/10.1007/BF03175056\\\": -1\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691161740,\n",
      "  \"id\": \"chatcmpl-7jqiCscDo5VDJJPIhhVrIkBmr2zrI\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 266,\n",
      "    \"prompt_tokens\": 1295,\n",
      "    \"total_tokens\": 1561\n",
      "  }\n",
      "}\n",
      "Prompt:  \n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition:\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity compare to other bacteria?', '3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?', '4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}\n",
      "\n",
      "Search results:\n",
      "[{'title': 'Blue light treatment of Pseudomonas aeruginosa: Strong ...', 'href': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5626244/', 'body': 'The pseudolysin activity of elastase B was analyzed by performing and ECR assay. ... activity of virulence factors including (i) proteolytic enzymes (i.e.,\\xa0...'}, {'title': 'Sublethal Effects of Triflumezopyrim on Biological Traits ... - Frontiers', 'href': 'https://www.frontiersin.org/articles/10.3389/fphys.2020.00261/full', 'body': 'Mar 6, 2020 ... Our objective is to analyze the population dynamics, detoxification enzyme activities, Vg and VgR contents, and relative expression of the EcR,\\xa0...'}, {'title': 'Biosynthetic Enzyme GMP Synthetase Cooperates with Ubiquitin ...', 'href': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2812229/', 'body': 'Dec 7, 2009 ... Ecdysone receptor (EcR) interacts biochemically and genetically with ... Next we wondered whether the enzymatic activities of USP7 or GMPS\\xa0...'}, {'title': 'Aurintricarboxylic acid structure modifications lead to reduction of ...', 'href': 'https://link.springer.com/article/10.1007/s11274-016-2123-3', 'body': 'Aug 25, 2016 ... The free binding energy calculations show that ECR has a stronger binding ... Furthermore, we have assessed the enzymatic activity of YopH\\xa0...'}, {'title': 'A Broad Set of Chromatin Factors Influences Splicing | PLOS Genetics', 'href': 'https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1006318', 'body': 'Here, we show that a subset of chromatin proteins is physically in interaction with the enzyme responsible for RNA splicing. In addition, several chromatin\\xa0...'}, {'title': 'Chitin metabolism in insects: structure, function and regulation of ...', 'href': 'https://journals.biologists.com/jeb/article/206/24/4393/34291/Chitin-metabolism-in-insects-structure-function', 'body': 'Dec 15, 2003 ... Common features of most chitin synthases are that enzyme activity is ... of the EcR and the Drosophila retinoid X receptor homologue USP,\\xa0...'}, {'title': '20-Hydroxyecdysone (20E) signaling as a promising target for the ...', 'href': 'https://parasitesandvectors.biomedcentral.com/articles/10.1186/s13071-020-04558-5', 'body': 'Jan 29, 2021 ... Orthologues of these enzymes have been characterized in mosquitoes. b Once 20E binds to its heterodimer EcR/USP receptor, the latter is\\xa0...'}, {'title': 'Opportunistic binding of EcR to open chromatin drives tissue ...', 'href': 'https://www.pnas.org/doi/full/10.1073/pnas.2208935119', 'body': 'Surprisingly, the only ecdysone-induced transcription factor expressed in wild-type wings in late larvae, br, was still activated despite EcR loss of function.'}]\n",
      "\n",
      "Task:\n",
      "Based on the key questions and each search result's title and body content, identify which key question indexes the search result content is likely most useful for or -1 if it isn't useful for any key question, and for each key question index identified, assign a relevance score of the search result content using a 5-point Likert scale, with 1 being very irrelevant to 5 being very relevant to the key question.\n",
      "\n",
      "The output should be in JSON format: \n",
      "```json\n",
      "{\n",
      "  <'insert href'>: [['insert key question index', 'relevance score'], etc.],\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5626244/\\\": [[\\\"1\\\", 4]],\\n  \\\"https://www.frontiersin.org/articles/10.3389/fphys.2020.00261/full\\\": [[\\\"-1\\\"]],\\n  \\\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2812229/\\\": [[\\\"-1\\\"]],\\n  \\\"https://link.springer.com/article/10.1007/s11274-016-2123-3\\\": [[\\\"-1\\\"]],\\n  \\\"https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1006318\\\": [[\\\"-1\\\"]],\\n  \\\"https://journals.biologists.com/jeb/article/206/24/4393/34291/Chitin-metabolism-in-insects-structure-function\\\": [[\\\"-1\\\"]],\\n  \\\"https://parasitesandvectors.biomedcentral.com/articles/10.1186/s13071-020-04558-5\\\": [[\\\"-1\\\"]],\\n  \\\"https://www.pnas.org/doi/full/10.1073/pnas.2208935119\\\": [[\\\"-1\\\"]]\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691161752,\n",
      "  \"id\": \"chatcmpl-7jqiOIrz1WiU5a97LNc9B64FhEQSO\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 245,\n",
      "    \"prompt_tokens\": 1139,\n",
      "    \"total_tokens\": 1384\n",
      "  }\n",
      "}\n",
      "Prompt:  \n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition:\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity compare to other bacteria?', '3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?', '4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}\n",
      "\n",
      "Search results:\n",
      "[{'title': 'Extensive regulation of enzyme activity by phosphorylation in ... - Nature', 'href': 'https://www.nature.com/articles/s41467-021-25988-4', 'body': 'In vitro enzyme assays. In vitro activity of purified enzymes was determined by monitoring substrate decrease or product formation using an Infinite ® M Nano spectrophotometer (Tecan) and Tecan i ...'}, {'title': 'Regulation of Bacterial Metabolism - Textbook of Bacteriology', 'href': 'http://www.textbookofbacteriology.net/regulation.html', 'body': 'Regulation of Enzyme Reactions Not all enzymatic reactions occur in a cell to the same extent. Some substances are needed in large amounts and the reactions involved in their synthesis must therefore occur in large amounts.'}, {'title': '19.8: Enzyme Regulation - Biology LibreTexts', 'href': 'https://bio.libretexts.org/Bookshelves/Microbiology/Microbiology_(Kaiser)/Unit_7%3A_Microbial_Genetics_and_Microbial_Metabolism/19%3A_Review_of_Molecular_Genetics/19.8%3A_Enzyme_Regulation', 'body': 'Step 1: The regulator gene codes for an active repressor protein. Step 2: The repressor protein then binds to the operator region of the operon. Figure 19.8. 8: An Inducible Operon in the Presence of an Inducer (The Lactose Operon)Step 5: Since the inactive repressor protein is unable to bind to the operator region, RNA polymerase (the enzyme ...'}, {'title': 'Uncovering the activities, biological roles, and regulation of ...', 'href': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7062177/', 'body': 'Peptidoglycan hydrolases and cell wall-tailoring enzymes that regulate glycan strand length, the degree of cross-linking, and the addition of other modifications to peptidoglycan are central in determining the final architecture of the bacterial cell wall.'}, {'title': 'Methods for the Measurement of a Bacterial Enzyme Activity in Cell ...', 'href': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC140121/', 'body': 'The kinetic characteristics and regulation of aspartate carbamoyltransferase activity were studied in lysates and cell extracts of Helicobacter pylori by three diffirent methods. Nuclear magnetic resonance spectroscopy, radioactive tracer analysis, and spectrophotometry were employed in conjunction to identify the properties of the enzyme activity and to validate the results obtained with each ...'}, {'title': 'Regulation of RNA processing and degradation in bacteria', 'href': 'https://www.sciencedirect.com/science/article/pii/S1874939919302652', 'body': '1. Introduction Bacteria need to modulate their gene expression to adapt rapidly to ever changing environmental conditions. Post-transcriptional regulation allows adaptation of gene expression in a timely manner by altering mRNA stability and/or translation.'}, {'title': 'Regulation of enzyme activity in the cell: effect of enzyme concentration', 'href': 'https://faseb.onlinelibrary.wiley.com/doi/epdf/10.1096/fasebj.5.14.1752361', 'body': 'losteric enzyme whose activity is thought to be fundamental to the regulation of glycolysis (42). Several studies in the 1970s suggested a possible concentration-dependent activity for the mammalian enzyme, although with uncertain ex-trapolation to its concentration in vivo (4, 28, 43). When the glycolytic enzymes were examined in situ in ...'}, {'title': 'Regulation of Enzyme Activity | SpringerLink', 'href': 'https://link.springer.com/chapter/10.1007/978-981-13-0785-0_37', 'body': 'Regulation of Enzyme Activity N. S. Punekar Chapter First Online: 12 November 2018 132k Accesses 1 Citations Abstract Most organisms, however evolutionarily distant, contain a set of common metabolites. But their intracellular concentrations are unique to each individual species.'}]\n",
      "\n",
      "Task:\n",
      "Based on the key questions and each search result's title and body content, identify which key question indexes the search result content is likely most useful for or -1 if it isn't useful for any key question, and for each key question index identified, assign a relevance score of the search result content using a 5-point Likert scale, with 1 being very irrelevant to 5 being very relevant to the key question.\n",
      "\n",
      "The output should be in JSON format: \n",
      "```json\n",
      "{\n",
      "  <'insert href'>: [['insert key question index', 'relevance score'], etc.],\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"https://www.nature.com/articles/s41467-021-25988-4\\\": [[\\\"1\\\", 4]],\\n  \\\"http://www.textbookofbacteriology.net/regulation.html\\\": [[\\\"-1\\\", -1]],\\n  \\\"https://bio.libretexts.org/Bookshelves/Microbiology/Microbiology_(Kaiser)/Unit_7%3A_Microbial_Genetics_and_Microbial_Metabolism/19%3A_Review_of_Molecular_Genetics/19.8%3A_Enzyme_Regulation\\\": [[\\\"-1\\\", -1]],\\n  \\\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7062177/\\\": [[\\\"-1\\\", -1]],\\n  \\\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC140121/\\\": [[\\\"-1\\\", -1]],\\n  \\\"https://www.sciencedirect.com/science/article/pii/S1874939919302652\\\": [[\\\"-1\\\", -1]],\\n  \\\"https://faseb.onlinelibrary.wiley.com/doi/epdf/10.1096/fasebj.5.14.1752361\\\": [[\\\"-1\\\", -1]],\\n  \\\"https://link.springer.com/chapter/10.1007/978-981-13-0785-0_37\\\": [[\\\"-1\\\", -1]]\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691161761,\n",
      "  \"id\": \"chatcmpl-7jqiXVv6ArNv4ukDlDfz0vx7Vts09\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 286,\n",
      "    \"prompt_tokens\": 1294,\n",
      "    \"total_tokens\": 1580\n",
      "  }\n",
      "}\n",
      "Prompt:  \n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition:\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity compare to other bacteria?', '3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?', '4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}\n",
      "\n",
      "Search results:\n",
      "[{'title': 'Awakening the Sleeping Carboxylase Function of Enzymes: Engineering the ...', 'href': 'https://pubs.acs.org/doi/10.1021/jacs.9b03431', 'body': 'However, for enoyl-CoA carboxylase/reductase from Kitasatospora setae (ECR Ks), four conserved amino acids that form a CO 2-binding pocket at the active site were described recently (Figure 1a). These four amino acids anchor and position the CO 2 molecule during catalysis, in which a reactive enolate is formed that attacks the CO 2. Figure 1'}, {'title': 'Bacterial Enzyme Converts CO2 Into Carbon Compounds 20x ... - SciTechDaily', 'href': 'https://scitechdaily.com/bacterial-enzyme-converts-co2-into-carbon-compounds-20x-faster-than-photosynthesis/', 'body': 'Improving on nature The enzyme the team studied is part of a family called enoyl-CoA carboxylases/reductases, or ECRs. It comes from soil bacteria called Kitasatospora setae, which in addition to their carbon-fixing skills can also produce antibiotics.'}, {'title': 'Four amino acids define the CO2 binding pocket of enoyl-CoA ...', 'href': 'https://pubmed.ncbi.nlm.nih.gov/31243147/', 'body': 'Here we investigated the interactions of CO 2 within the active site of Ecr from Kitasatospora setae Combining experimental biochemistry, protein crystallography, and advanced computer simulations we show that 4 amino acids, N81, F170, E171, and H365, are required to create a highly efficient CO 2 -fixing enzyme.'}, {'title': 'Reaction scheme and structural organization of the K. setae ECR ...', 'href': 'https://www.researchgate.net/figure/Reaction-scheme-and-structural-organization-of-the-K-setae-ECR-complex-a_fig1_360182999', 'body': 'We first determined four highresolution K. setae ECR structures in different conformational states: the apo form and three holo forms in a binary complex with the reduced cofactor NADPH, in a...'}, {'title': 'Protease Activity Analysis: A Toolkit for Analyzing Enzyme Activity ...', 'href': 'https://pubs.acs.org/doi/10.1021/acsomega.2c01559', 'body': 'To facilitate protease research, here we present Protease Activity Analysis (PAA), a toolkit for the preprocessing, visualization, machine learning analysis, and querying of protease activity data sets.'}, {'title': 'Frontiers | Extracellular Enzyme Activities and Carbon/Nitrogen ...', 'href': 'https://www.frontiersin.org/articles/10.3389/fmicb.2021.787820/full', 'body': 'To fill this gap, we compared the activities of five extracellular enzymes [cellulase, xylanase, lignin peroxidase, laccase, and superoxide dismutase (SOD)] between fungi isolated from epiphytic and terrestrial orchids. In total, 24 fungal strains belonging to Tulasnellaceae were investigated.'}, {'title': 'Modified Enzyme Substrates for the Detection of Bacteria: A Review', 'href': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7465704/', 'body': \"This review focuses on the method for identification of bacteria based on the use of synthetic enzyme substrates, which allow for the detection of a species' specific enzymatic activity [1,2,3,4]. Herein, a synthetic substrate, specific for a marker enzyme in a specific bacterial species, is added to a bacterial culture in liquid or solid ...\"}, {'title': 'Screening for extracellular enzyme activities by bacteria isolated from ...', 'href': 'https://annalsmicrobiology.biomedcentral.com/articles/10.1007/BF03175056', 'body': 'Ninety-five bacterial isolates, obtained in pure cultures from samples collected in various sites of the East sector of the Tyrrhenian Sea, were plate-screened for their ability to produce twelve extracellular enzyme activities in order to find new strains for possible applications.'}]\n",
      "\n",
      "Task:\n",
      "Based on the key questions and each search result's title and body content, identify which key question indexes the search result content is likely most useful for or -1 if it isn't useful for any key question, and for each key question index identified, assign a relevance score of the search result content using a 5-point Likert scale, with 1 being very irrelevant to 5 being very relevant to the key question.\n",
      "\n",
      "The output should be in JSON format: \n",
      "```json\n",
      "{\n",
      "  <'insert href'>: [['insert key question index', 'relevance score'], etc.],\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"https://pubs.acs.org/doi/10.1021/jacs.9b03431\\\": [[\\\"1\\\", \\\"5\\\"]],\\n  \\\"https://scitechdaily.com/bacterial-enzyme-converts-co2-into-carbon-compounds-20x-faster-than-photosynthesis/\\\": [[\\\"1\\\", \\\"5\\\"]],\\n  \\\"https://pubmed.ncbi.nlm.nih.gov/31243147/\\\": [[\\\"1\\\", \\\"5\\\"]],\\n  \\\"https://www.researchgate.net/figure/Reaction-scheme-and-structural-organization-of-the-K-setae-ECR-complex-a_fig1_360182999\\\": [[\\\"1\\\", \\\"5\\\"]],\\n  \\\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7465704/\\\": [[\\\"-1\\\"]],\\n  \\\"https://www.frontiersin.org/articles/10.3389/fmicb.2021.787820/full\\\": [[\\\"-1\\\"]],\\n  \\\"https://annalsmicrobiology.biomedcentral.com/articles/10.1007/BF03175056\\\": [[\\\"-1\\\"]],\\n  \\\"https://pubs.acs.org/doi/10.1021/acsomega.2c01559\\\": [[\\\"-1\\\"]]\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691161774,\n",
      "  \"id\": \"chatcmpl-7jqikeWLxM8DcfklcYdH5eB4Y4W3R\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 248,\n",
      "    \"prompt_tokens\": 1315,\n",
      "    \"total_tokens\": 1563\n",
      "  }\n",
      "}\n",
      "Prompt:  \n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition:\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity compare to other bacteria?', '3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?', '4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}\n",
      "\n",
      "Search results:\n",
      "[{'title': 'Awakening the Sleeping Carboxylase Function of Enzymes: Engineering the ...', 'href': 'https://pubs.acs.org/doi/10.1021/jacs.9b03431', 'body': 'However, for enoyl-CoA carboxylase/reductase from Kitasatospora setae (ECR Ks), four conserved amino acids that form a CO 2-binding pocket at the active site were described recently (Figure 1a). These four amino acids anchor and position the CO 2 molecule during catalysis, in which a reactive enolate is formed that attacks the CO 2. Figure 1'}, {'title': 'Four amino acids define the CO2 binding pocket of enoyl-CoA ...', 'href': 'https://pubmed.ncbi.nlm.nih.gov/31243147/', 'body': 'Here we investigated the interactions of CO 2 within the active site of Ecr from Kitasatospora setae Combining experimental biochemistry, protein crystallography, and advanced computer simulations we show that 4 amino acids, N81, F170, E171, and H365, are required to create a highly efficient CO 2 -fixing enzyme.'}, {'title': 'Four amino acids define the CO 2 binding pocket of enoyl-CoA ... - PNAS', 'href': 'https://www.pnas.org/doi/10.1073/pnas.1901471116', 'body': 'Here we investigated the interactions of CO 2 within the active site of Ecr from Kitasatospora setae. Combining experimental biochemistry, protein crystallography, and advanced computer simulations we show that 4 amino acids, N81, F170, E171, and H365, are required to create a highly efficient CO 2 -fixing enzyme.'}, {'title': 'Genus Kitasatospora, taxonomic features and diversity of secondary ...', 'href': 'https://www.nature.com/articles/ja20178', 'body': 'Kitasatospora setae KM-6054 T is the type species and ... hence the possibility of bacterial contamination was ... biological activity and mechanism of action. J. Antibiot. 37, 829-835 (1984 ...'}, {'title': '7.1.1: Factors Affecting Enzyme Activity - Biology LibreTexts', 'href': 'https://bio.libretexts.org/Courses/Prince_Georges_Community_College/PGCC_Microbiology/07%3A_Microbial_Metabolism/7.01%3A_Introduction_to_Metabolism_and_Enzymes/7.1.01%3A_Factors_Affecting_Enzyme_Activity', 'body': 'Gary Kaiser Community College of Baltimore Country (Cantonsville) Learning Objectives Define or describe the following: metabolism catabolic reaction anabolic reaction enzyme substrate apoenzyme haloenzyme cofactor (coenzyme) State how enzymes are able to speed up the rate of chemical reactions.'}, {'title': '(PDF) Factors Affecting Enzyme Activity - ResearchGate', 'href': 'https://www.researchgate.net/publication/347439618_Factors_Affecting_Enzyme_Activity', 'body': 'Enzyme activity affected by a variety of factors, such as temperature, pH, and concentration. Substrate concentration: Increasing substrate concentration also increases the rate of reaction to a...'}, {'title': '5.5: Factos Affecting Enzyme Activity - Chemistry LibreTexts', 'href': 'https://chem.libretexts.org/Courses/Brevard_College/CHE_301_Biochemistry/05%3A_Enzymes/5.05%3A_Factos_Affecting_Enzyme_Activity', 'body': 'Factors that disrupt protein structure include temperature and pH; factors that affect catalysts in general include reactant or substrate concentration and catalyst or enzyme concentration. The activity of an enzyme can be measured by monitoring either the rate at which a substrate disappears or the rate at which a product forms.'}, {'title': 'Factors Affecting Enzyme Activity: Meaning, Mechanism - Embibe', 'href': 'https://www.embibe.com/exams/factors-affecting-enzyme-activity/', 'body': '1. Most enzymes are proteins except for a small group of catalytic RNA molecules. The catalytic activity of enzymes depends on the integrity of their protein structure. If an enzyme gets denatured or separated, then its catalytic activity is lost. 2. Enzymes are highly specific. Their actions are specific to the substrates. 3.'}]\n",
      "\n",
      "Task:\n",
      "Based on the key questions and each search result's title and body content, identify which key question indexes the search result content is likely most useful for or -1 if it isn't useful for any key question, and for each key question index identified, assign a relevance score of the search result content using a 5-point Likert scale, with 1 being very irrelevant to 5 being very relevant to the key question.\n",
      "\n",
      "The output should be in JSON format: \n",
      "```json\n",
      "{\n",
      "  <'insert href'>: [['insert key question index', 'relevance score'], etc.],\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"https://pubs.acs.org/doi/10.1021/jacs.9b03431\\\": [[\\\"1\\\", \\\"5\\\"]],\\n  \\\"https://pubmed.ncbi.nlm.nih.gov/31243147/\\\": [[\\\"1\\\", \\\"5\\\"]],\\n  \\\"https://www.pnas.org/doi/10.1073/pnas.1901471116\\\": [[\\\"1\\\", \\\"5\\\"]],\\n  \\\"https://www.nature.com/articles/ja20178\\\": -1,\\n  \\\"https://bio.libretexts.org/Courses/Prince_Georges_Community_College/PGCC_Microbiology/07%3A_Microbial_Metabolism/7.01%3A_Introduction_to_Metabolism_and_Enzymes/7.1.01%3A_Factors_Affecting_Enzyme_Activity\\\": [[\\\"1\\\", \\\"3\\\"]],\\n  \\\"https://www.researchgate.net/publication/347439618_Factors_Affecting_Enzyme_Activity\\\": [[\\\"1\\\", \\\"3\\\"]],\\n  \\\"https://chem.libretexts.org/Courses/Brevard_College/CHE_301_Biochemistry/05%3A_Enzymes/5.05%3A_Factos_Affecting_Enzyme_Activity\\\": [[\\\"1\\\", \\\"3\\\"]],\\n  \\\"https://www.embibe.com/exams/factors-affecting-enzyme-activity/\\\": [[\\\"1\\\", \\\"3\\\"]]\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691161787,\n",
      "  \"id\": \"chatcmpl-7jqixsOiD4xbZ7TBT9tDAAAU67mY0\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 304,\n",
      "    \"prompt_tokens\": 1382,\n",
      "    \"total_tokens\": 1686\n",
      "  }\n",
      "}\n",
      "Prompt:  \n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition:\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity compare to other bacteria?', '3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?', '4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}\n",
      "\n",
      "Search results:\n",
      "[{'title': 'Awakening the Sleeping Carboxylase Function of Enzymes: Engineering the ...', 'href': 'https://pubs.acs.org/doi/10.1021/jacs.9b03431', 'body': 'However, for enoyl-CoA carboxylase/reductase from Kitasatospora setae (ECR Ks), four conserved amino acids that form a CO 2-binding pocket at the active site were described recently (Figure 1a). These four amino acids anchor and position the CO 2 molecule during catalysis, in which a reactive enolate is formed that attacks the CO 2. Figure 1'}, {'title': 'Four amino acids define the CO2 binding pocket of enoyl-CoA ...', 'href': 'https://pubmed.ncbi.nlm.nih.gov/31243147/', 'body': 'Here we investigated the interactions of CO 2 within the active site of Ecr from Kitasatospora setae Combining experimental biochemistry, protein crystallography, and advanced computer simulations we show that 4 amino acids, N81, F170, E171, and H365, are required to create a highly efficient CO 2 -fixing enzyme.'}, {'title': 'Bacterial Enzyme Converts CO2 Into Carbon Compounds 20x Faster Than ...', 'href': 'https://scitechdaily.com/bacterial-enzyme-converts-co2-into-carbon-compounds-20x-faster-than-photosynthesis/', 'body': 'Researchers discover that a spot of molecular glue and a timely twist help a bacterial enzyme convert carbon dioxide into carbon compounds 20 times faster than plant enzymes do during photosynthesis. The results stand to accelerate progress toward converting carbon dioxide into a variety of products. Carbon fixation, or the conversion of carbon ...'}, {'title': 'PDF Genus Kitasatospora, taxonomic features and diversity of secondary ...', 'href': 'https://www.nature.com/articles/ja20178.pdf', 'body': 'show that the Kitasatospora is a legitimate genus, distinguishable from Streptomyces, on the basis of phenotypic and genetic differences. Five species, K. setae KM-6054 T , 2,11 K. griseola AM ...'}, {'title': 'Genus Kitasatospora, taxonomic features and diversity of secondary ...', 'href': 'https://www.nature.com/articles/ja20178', 'body': 'Kitasatospora setae KM-6054 T is the type species and ... hence the possibility of bacterial contamination was ... biological activity and mechanism of action. J. Antibiot. 37, 829-835 (1984 ...'}, {'title': 'PDF Characterization of bafilomycin biosynthesis in Kitasatospora setae KM ...', 'href': 'https://www.nature.com/articles/ja201733.pdf', 'body': 'setae was used to inoculate a 50 ml test tube containing 10 ml of vegetative medium 18 and the culture was allowed to grow with reciprocal shaking at 28 °C for 2 days.'}, {'title': '(PDF) Factors Affecting Enzyme Activity', 'href': 'https://www.researchgate.net/publication/347439618_Factors_Affecting_Enzyme_Activity', 'body': 'Abstract Enzyme activity affected by a variety of factors, such as temperature, pH, and concentration. Substrate concentration: Increasing substrate concentration also increases the rate of...'}, {'title': '7.1.1: Factors Affecting Enzyme Activity', 'href': 'https://bio.libretexts.org/Courses/Prince_Georges_Community_College/PGCC_Microbiology/07%3A_Microbial_Metabolism/7.01%3A_Introduction_to_Metabolism_and_Enzymes/7.1.01%3A_Factors_Affecting_Enzyme_Activity', 'body': 'pH salt concentration State how chemicals such as chlorine, iodine, iodophores, mercurials, and ethylene oxide inhibit or kill bacteria. State how high temperature and low temperature exert their effect on bacteria. To live, grow, and reproduce, microorganisms undergo a variety of chemical changes.'}]\n",
      "\n",
      "Task:\n",
      "Based on the key questions and each search result's title and body content, identify which key question indexes the search result content is likely most useful for or -1 if it isn't useful for any key question, and for each key question index identified, assign a relevance score of the search result content using a 5-point Likert scale, with 1 being very irrelevant to 5 being very relevant to the key question.\n",
      "\n",
      "The output should be in JSON format: \n",
      "```json\n",
      "{\n",
      "  <'insert href'>: [['insert key question index', 'relevance score'], etc.],\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"https://pubs.acs.org/doi/10.1021/jacs.9b03431\\\": [[\\\"1\\\", \\\"5\\\"]],\\n  \\\"https://pubmed.ncbi.nlm.nih.gov/31243147/\\\": [[\\\"1\\\", \\\"5\\\"]],\\n  \\\"https://scitechdaily.com/bacterial-enzyme-converts-co2-into-carbon-compounds-20x-faster-than-photosynthesis/\\\": [[\\\"1\\\", \\\"4\\\"]],\\n  \\\"https://www.nature.com/articles/ja20178.pdf\\\": [[\\\"-1\\\"]],\\n  \\\"https://www.nature.com/articles/ja20178\\\": [[\\\"-1\\\"]],\\n  \\\"https://www.nature.com/articles/ja201733.pdf\\\": [[\\\"-1\\\"]],\\n  \\\"https://www.researchgate.net/publication/347439618_Factors_Affecting_Enzyme_Activity\\\": [[\\\"1\\\", \\\"3\\\"]],\\n  \\\"https://bio.libretexts.org/Courses/Prince_Georges_Community_College/PGCC_Microbiology/07%3A_Microbial_Metabolism/7.01%3A_Introduction_to_Metabolism_and_Enzymes/7.1.01%3A_Factors_Affecting_Enzyme_Activity\\\": [[\\\"-1\\\"]]\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691161799,\n",
      "  \"id\": \"chatcmpl-7jqj9Bo4wu1i9OJgDJ678DCRrNJ55\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 265,\n",
      "    \"prompt_tokens\": 1293,\n",
      "    \"total_tokens\": 1558\n",
      "  }\n",
      "}\n",
      "Prompt:  \n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition:\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity compare to other bacteria?', '3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?', '4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}\n",
      "\n",
      "Search results:\n",
      "[{'title': 'A radical change in enzyme catalysis | Nature', 'href': 'https://www.nature.com/articles/540345a', 'body': 'Emmanuel et al. now report a striking new concept for generating enzymes that catalyse unnatural reactions: the authors use light to excite a cofactor (NAD (P)H) bound in the active site of a ...'}, {'title': 'Creatine kinase. Modification of the working enzyme. - PMC', 'href': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1163814/', 'body': 'This suggests that the degree of protection by the working enzyme represents the proportion of enzyme molecules that have a conformation complementary to a creatine-PO3-MgADP intermediate. Full text Get a printable copy (PDF file) of the complete article (1.3M), or click on a page image below to browse page by page.'}, {'title': 'Enzyme expression kinetics by Escherichia coli during ... - Nature', 'href': 'https://www.nature.com/articles/s41564-022-01310-w', 'body': 'As the enzymes in the same pathway had similar kinetics, attributed in part to operon structure (indicated in Extended Data Fig. 5), we summed up the abundances of enzymes in the same pathway to ...'}, {'title': 'Detailed Enzyme Kinetics in Terms of Biochemical Species: Study ... - PLOS', 'href': 'https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0001825', 'body': 'The compulsory-ordered ternary catalytic mechanism for two-substrate two-product enzymes is analyzed to account for binding of inhibitors to each of the four enzyme states and to maintain the relationship between the kinetic constants and the reaction equilibrium constant.'}, {'title': 'Enzymes: principles and biotechnological applications - PMC', 'href': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4692135/', 'body': 'Enzymes are biological catalysts (also known as biocatalysts) that speed up biochemical reactions in living organisms, and which can be extracted from cells and then used to catalyse a wide range of commercially important processes.'}, {'title': 'Bacterial Enzyme Converts CO2 Into Carbon Compounds 20x ... - SciTechDaily', 'href': 'https://scitechdaily.com/bacterial-enzyme-converts-co2-into-carbon-compounds-20x-faster-than-photosynthesis/', 'body': 'It comes from soil bacteria called Kitasatospora setae, which in addition to their carbon-fixing skills can also produce antibiotics. Wakatsuki heard about this enzyme family half a dozen years ago from Tobias Erb of the Max Planck Institute for Terrestrial Microbiology in Germany and Yasuo Yoshikuni of JGI.'}, {'title': 'Enzyme Kinetic Assays - How Does It Work? - Eppendorf', 'href': 'https://handling-solutions.eppendorf.com/sample-handling/photometry/applications/detailview-applications/news/enzyme-kinetic-assays-how-does-it-work/', 'body': 'Enzyme kinetics describes the speed at which an enzyme-catalyzed chemical reaction proceeds.First and foremost, the speed of the reaction depends on the amount of the enzyme used as well as on the amount of substrate. Whereas speed increases proportionally to the concentration of the enzyme, should substrate be added, speed will at first increase in a linear fashion until it eventually ...'}, {'title': 'Reaction scheme and structural organization of the K. setae ECR ...', 'href': 'https://www.researchgate.net/figure/Reaction-scheme-and-structural-organization-of-the-K-setae-ECR-complex-a_fig1_360182999', 'body': 'Enoyl-CoA carboxylases/reductases (ECRs) are some of the most efficient CO2-fixing enzymes described to date. However, the molecular mechanisms underlying the extraordinary catalytic activity of...'}]\n",
      "\n",
      "Task:\n",
      "Based on the key questions and each search result's title and body content, identify which key question indexes the search result content is likely most useful for or -1 if it isn't useful for any key question, and for each key question index identified, assign a relevance score of the search result content using a 5-point Likert scale, with 1 being very irrelevant to 5 being very relevant to the key question.\n",
      "\n",
      "The output should be in JSON format: \n",
      "```json\n",
      "{\n",
      "  <'insert href'>: [['insert key question index', 'relevance score'], etc.],\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"https://www.nature.com/articles/540345a\\\": [[\\\"1\\\", 3]],\\n  \\\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1163814/\\\": [[\\\"-1\\\", -1]],\\n  \\\"https://www.nature.com/articles/s41564-022-01310-w\\\": [[\\\"-1\\\", -1]],\\n  \\\"https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0001825\\\": [[\\\"-1\\\", -1]],\\n  \\\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4692135/\\\": [[\\\"-1\\\", -1]],\\n  \\\"https://scitechdaily.com/bacterial-enzyme-converts-co2-into-carbon-compounds-20x-faster-than-photosynthesis/\\\": [[\\\"1\\\", 5]],\\n  \\\"https://handling-solutions.eppendorf.com/sample-handling/photometry/applications/detailview-applications/news/enzyme-kinetic-assays-how-does-it-work/\\\": [[\\\"-1\\\", -1]],\\n  \\\"https://www.researchgate.net/figure/Reaction-scheme-and-structural-organization-of-the-K-setae-ECR-complex-a_fig1_360182999\\\": [[\\\"1\\\", 5]]\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691161810,\n",
      "  \"id\": \"chatcmpl-7jqjKqOB5PBFENw9i8IUxEGfpcdbJ\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 266,\n",
      "    \"prompt_tokens\": 1246,\n",
      "    \"total_tokens\": 1512\n",
      "  }\n",
      "}\n",
      "Prompt:  \n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition:\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity compare to other bacteria?', '3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?', '4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}\n",
      "\n",
      "Search results:\n",
      "[{'title': 'Creatine kinase. Modification of the working enzyme. - PMC', 'href': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1163814/', 'body': 'This suggests that the degree of protection by the working enzyme represents the proportion of enzyme molecules that have a conformation complementary to a creatine-PO3-MgADP intermediate. Full text Get a printable copy (PDF file) of the complete article (1.3M), or click on a page image below to browse page by page.'}, {'title': 'Enzyme Markers: Purpose, Procedure, and Results - Healthline', 'href': 'https://www.healthline.com/health/enzyme-markers', 'body': 'Enzyme markers are blood tests that analyze specific enzyme activity in the body. Some inherited diseases or conditions can cause these enzymes to stop working or be less efficient. Monitoring the ...'}, {'title': 'Protease Activity Analysis: A Toolkit for Analyzing Enzyme Activity ...', 'href': 'https://pubs.acs.org/doi/10.1021/acsomega.2c01559', 'body': 'Analyzing the activity of proteases and their substrates is critical to defining the biological functions of these enzymes and to designing new diagnostics and therapeutics that target protease dysregulation in disease.'}, {'title': 'Awakening the Sleeping Carboxylase Function of Enzymes: Engineering the ...', 'href': 'https://pubs.acs.org/doi/10.1021/jacs.9b03431', 'body': 'However, for enoyl-CoA carboxylase/reductase from Kitasatospora setae (ECR Ks), four conserved amino acids that form a CO 2-binding pocket at the active site were described recently (Figure 1a). These four amino acids anchor and position the CO 2 molecule during catalysis, in which a reactive enolate is formed that attacks the CO 2. Figure 1'}, {'title': 'Bacterial Enzyme Converts CO2 Into Carbon Compounds 20x ... - SciTechDaily', 'href': 'https://scitechdaily.com/bacterial-enzyme-converts-co2-into-carbon-compounds-20x-faster-than-photosynthesis/', 'body': 'Improving on nature The enzyme the team studied is part of a family called enoyl-CoA carboxylases/reductases, or ECRs. It comes from soil bacteria called Kitasatospora setae, which in addition to their carbon-fixing skills can also produce antibiotics.'}, {'title': 'Modified Enzyme Substrates for the Detection of Bacteria: A Review', 'href': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7465704/', 'body': \"This review focuses on the method for identification of bacteria based on the use of synthetic enzyme substrates, which allow for the detection of a species' specific enzymatic activity [1,2,3,4]. Herein, a synthetic substrate, specific for a marker enzyme in a specific bacterial species, is added to a bacterial culture in liquid or solid ...\"}, {'title': 'Methods for the Measurement of a Bacterial Enzyme Activity in Cell ...', 'href': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC140121/', 'body': 'Studies of bacterial enzyme activities provide fundamental information relevant to microbial physiology, and to a more complete understanding of cell metabolism, bacterial evolution, and to the interactions that occurs between bacterium and host.'}, {'title': 'Enzyme characterization and biological activities of a resuscitation ...', 'href': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6534110/', 'body': 'Introduction. Bacteria can adopt a dormant state when they are in harmful environments by reducing metabolic activity and ceasing to divide. Dormancy can take different forms, and some low G+C Gram-positive bacteria such as Clostridium and Bacillus have adopted sporulation as a means of surviving adverse environmental conditions (Rittershaus, Baek & Sassetti, 2013; Dworkin & Shah, 2010).'}]\n",
      "\n",
      "Task:\n",
      "Based on the key questions and each search result's title and body content, identify which key question indexes the search result content is likely most useful for or -1 if it isn't useful for any key question, and for each key question index identified, assign a relevance score of the search result content using a 5-point Likert scale, with 1 being very irrelevant to 5 being very relevant to the key question.\n",
      "\n",
      "The output should be in JSON format: \n",
      "```json\n",
      "{\n",
      "  <'insert href'>: [['insert key question index', 'relevance score'], etc.],\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1163814/\\\": [[\\\"1\\\", 4]],\\n  \\\"https://www.healthline.com/health/enzyme-markers\\\": [[\\\"1\\\", 3]],\\n  \\\"https://pubs.acs.org/doi/10.1021/acsomega.2c01559\\\": [[\\\"1\\\", 5]],\\n  \\\"https://pubs.acs.org/doi/10.1021/jacs.9b03431\\\": [[\\\"1\\\", 5]],\\n  \\\"https://scitechdaily.com/bacterial-enzyme-converts-co2-into-carbon-compounds-20x-faster-than-photosynthesis/\\\": [[\\\"1\\\", 5]],\\n  \\\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7465704/\\\": [[\\\"1\\\", 4]],\\n  \\\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC140121/\\\": [[\\\"1\\\", 5]],\\n  \\\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6534110/\\\": [[\\\"1\\\", 1]]\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691161821,\n",
      "  \"id\": \"chatcmpl-7jqjVwlqqOHDzEAoUfdCDJpgEGrxS\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 224,\n",
      "    \"prompt_tokens\": 1264,\n",
      "    \"total_tokens\": 1488\n",
      "  }\n",
      "}\n",
      "Prompt:  \n",
      "Context:\n",
      "Enoyl-CoA carboxylase/reductase enzymes (ECRs)\n",
      "\n",
      "Research question decomposition:\n",
      "{'project_question': 'How efficiently do the ECR enzymes work, especially in Kitsatospor setae bacteria?', 'project_objective': 'To determine the efficiency of ECR enzymes in Kitsatospor setae bacteria', 'key_drivers': {'1': {'driver': 'ECR enzyme activity', 'hypotheses': {'1': {'hypothesis': 'ECR enzyme activity is high in Kitsatospor setae bacteria', 'key_questions': {'1': 'What is the level of ECR enzyme activity in Kitsatospor setae bacteria?', '2': 'How does ECR enzyme activity compare to other bacteria?', '3': 'What factors influence ECR enzyme activity in Kitsatospor setae bacteria?', '4': 'What are the potential applications or implications of high ECR enzyme activity in Kitsatospor setae bacteria?'}}}}}}\n",
      "\n",
      "Search results:\n",
      "[{'title': 'Enhancing the compost maturation of swine manure and rice straw by ...', 'href': 'https://www.nature.com/articles/s41598-021-85615-6', 'body': 'The results indicated that the isolated Kitasatospora phosalacinea strain C1, which is a cellulose-degraded microorganism, presented the highest enzyme activity at 31 ℃ and pH 5.5, while the C/N ...'}, {'title': 'Four amino acids define the CO2 binding pocket of enoyl-CoA ...', 'href': 'https://pubmed.ncbi.nlm.nih.gov/31243147/', 'body': 'Here we investigated the interactions of CO 2 within the active site of Ecr from Kitasatospora setae Combining experimental biochemistry, protein crystallography, and advanced computer simulations we show that 4 amino acids, N81, F170, E171, and H365, are required to create a highly efficient CO 2 -fixing enzyme.'}, {'title': 'Purification and biochemical characterization of a novel ... - PubMed', 'href': 'https://pubmed.ncbi.nlm.nih.gov/36905451/', 'body': 'Purification and biochemical characterization of a novel chitosanase cloned from the gene of Kitasatospora setae KM-6054 and its application in the production of chitooligosaccharides Chitosanase could degrade chitosan efficiently under mild conditions to prepare chitosan oligosaccharides (COSs).'}, {'title': 'Bacterial Enzyme Converts CO2 Into Carbon Compounds 20x ... - SciTechDaily', 'href': 'https://scitechdaily.com/bacterial-enzyme-converts-co2-into-carbon-compounds-20x-faster-than-photosynthesis/', 'body': 'Improving on nature The enzyme the team studied is part of a family called enoyl-CoA carboxylases/reductases, or ECRs. It comes from soil bacteria called Kitasatospora setae, which in addition to their carbon-fixing skills can also produce antibiotics.'}, {'title': 'Structural basis for non-radical catalysis by TsrM, a radical SAM ...', 'href': 'https://www.nature.com/articles/s41589-020-00717-y', 'body': \"setae (KsTsrM), which shed light on these two questions and facilitate a deeper understanding of the enzyme's catalytic mechanism. In particular, we show that a conserved arginyl residue ...\"}, {'title': 'PDF Genus Kitasatospora, taxonomic features and diversity of secondary ...', 'href': 'https://www.nature.com/articles/ja20178.pdf', 'body': 'show that the Kitasatospora is a legitimate genus, distinguishable from Streptomyces, on the basis of phenotypic and genetic differences. Five species, K. setae KM-6054 T , 2,11 K. griseola AM ...'}, {'title': 'Genus Kitasatospora, taxonomic features and diversity of secondary ...', 'href': 'https://www.nature.com/articles/ja20178', 'body': 'Although Kitasatospora strains resemble Streptomyces strains in morphology, they are clearly different in cell-wall composition, as they contain both LL- and meso -diaminopimelic acid. Aerial and...'}, {'title': 'Purification and biochemical characterization of a novel ... - ResearchGate', 'href': 'https://www.researchgate.net/publication/369114393_Purification_and_biochemical_characterization_of_a_novel_chitosanase_cloned_from_the_gene_of_Kitasatospora_setae_KM-6054_and_its_application_in_the_production_of_chitooligosaccharides', 'body': 'After purification, CsnS is a cold-adapted enzyme with the highest activity at 60°C, showing 37.5% of the maximal activity at 0°C and 42.6% of the maximal activity at 10°C.'}]\n",
      "\n",
      "Task:\n",
      "Based on the key questions and each search result's title and body content, identify which key question indexes the search result content is likely most useful for or -1 if it isn't useful for any key question, and for each key question index identified, assign a relevance score of the search result content using a 5-point Likert scale, with 1 being very irrelevant to 5 being very relevant to the key question.\n",
      "\n",
      "The output should be in JSON format: \n",
      "```json\n",
      "{\n",
      "  <'insert href'>: [['insert key question index', 'relevance score'], etc.],\n",
      "  etc.\n",
      "}\n",
      "```\n",
      "\n",
      "Respond only with the output, with no explanation or conversation.\n",
      "\n",
      "Completion info:  {\n",
      "  \"choices\": [\n",
      "    {\n",
      "      \"finish_reason\": \"stop\",\n",
      "      \"index\": 0,\n",
      "      \"message\": {\n",
      "        \"content\": \"{\\n  \\\"https://www.nature.com/articles/s41598-021-85615-6\\\": [[\\\"1\\\", 4]],\\n  \\\"https://pubmed.ncbi.nlm.nih.gov/31243147/\\\": [[\\\"1\\\", 5]],\\n  \\\"https://pubmed.ncbi.nlm.nih.gov/36905451/\\\": [[\\\"1\\\", 3]],\\n  \\\"https://scitechdaily.com/bacterial-enzyme-converts-co2-into-carbon-compounds-20x-faster-than-photosynthesis/\\\": [[\\\"1\\\", 5]],\\n  \\\"https://www.nature.com/articles/s41589-020-00717-y\\\": [[\\\"-1\\\", -1]],\\n  \\\"https://www.nature.com/articles/ja20178.pdf\\\": [[\\\"-1\\\", -1]],\\n  \\\"https://www.nature.com/articles/ja20178\\\": [[\\\"-1\\\", -1]],\\n  \\\"https://www.researchgate.net/publication/369114393_Purification_and_biochemical_characterization_of_a_novel_chitosanase_cloned_from_the_gene_of_Kitasatospora_setae_KM-6054_and_its_application_in_the_production_of_chitooligosaccharides\\\": [[\\\"-1\\\", -1]]\\n}\",\n",
      "        \"role\": \"assistant\"\n",
      "      }\n",
      "    }\n",
      "  ],\n",
      "  \"created\": 1691161830,\n",
      "  \"id\": \"chatcmpl-7jqjeInMrwsV9pgIw9eOjjXs2yRx5\",\n",
      "  \"model\": \"gpt-3.5-turbo-0613\",\n",
      "  \"object\": \"chat.completion\",\n",
      "  \"usage\": {\n",
      "    \"completion_tokens\": 254,\n",
      "    \"prompt_tokens\": 1258,\n",
      "    \"total_tokens\": 1512\n",
      "  }\n",
      "}\n"
     ]
    }
   ],
   "source": [
    "from collections import defaultdict\n",
    "\n",
    "for decomposition_idx, key_question_decomposition in enumerate(key_question_decomposition_list):\n",
    "    with open(f'autoscious_logs/{search_query_file_safe}/sources/kq{decomposition_idx}/initial_search_queries.json', 'r') as f:\n",
    "        key_question_initial_search_queries = json.load(f)\n",
    "\n",
    "    for query_idx, query in key_question_initial_search_queries.items():\n",
    "        # load web search results\n",
    "        with open(f'autoscious_logs/{search_query_file_safe}/sources/kq{decomposition_idx}/initial_search_results_query_{query_idx}.json', 'r') as f:\n",
    "            web_search_res = json.loads(f.read())\n",
    "        \n",
    "        filtered_web_results = {}\n",
    "        if web_search_res != []:\n",
    "            # filter web results based on title and body\n",
    "            filtered_web_results = json.loads(chat_openai(get_filtering_web_results_ratings(context, decomposition, web_search_res), model=\"gpt-3.5-turbo\")[0])\n",
    "\n",
    "        ratings_url_dict = defaultdict(list)\n",
    "        for url, ratings in filtered_web_results.items():\n",
    "            if '-1' not in str(ratings):  # only process if the ratings is not -1\n",
    "                for rating in ratings:\n",
    "                    ratings_url_dict[str(rating[1])].append(url)  # append the URL to the correct category\n",
    "            else:\n",
    "                ratings_url_dict['-1'].append(url) # Misc & unrelated\n",
    "\n",
    "        # save filtered search results\n",
    "        with open(f'autoscious_logs/{search_query_file_safe}/sources/kq{decomposition_idx}/rated_web_results_query_{int(query_idx)}.json', 'w') as f:\n",
    "            json.dump(ratings_url_dict, f, indent=2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reading type 2 & 3: filtering based on skimming and sampling from each source, and only saving most relevant sources for fact extraction and quotes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# COMPLETE code for predicting usefulness of very relevant (5) and relevant (4) results.\n",
    "from autogpt.commands.web_selenium import scrape_text_with_selenium_no_agent\n",
    "\n",
    "CHUNK_SIZE = 1000\n",
    "SAMPLING_FACTOR = 0.1 # Also cap it so it falls under the max token limit\n",
    "MAX_TOKENS = 2500 * 4 # 1 token = 4 chars, 2500 + 500 (prompt) tokens is high for GPT3.5\n",
    "MAX_CHUNKS = int(MAX_TOKENS / CHUNK_SIZE)\n",
    "context = \"Enoyl-CoA carboxylase/reductase enzymes (ECRs)\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_sample_chunks(text, CHUNK_SIZE, num_chunk_samples):\n",
    "    step_size = len(text) // num_chunk_samples\n",
    "\n",
    "    chunks = []\n",
    "    for i in range(0, len(text), step_size):\n",
    "        chunk = text[i:i+CHUNK_SIZE]\n",
    "        chunks.append(chunk)\n",
    "\n",
    "        # Break after getting the required number of chunks\n",
    "        if len(chunks) >= num_chunk_samples:\n",
    "            break\n",
    "\n",
    "    return chunks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Need to determine how useful the text is likely to be for answering the key questions\n",
    "def get_predicted_usefulness_of_text_prompt(context, decomposition, sample_text_chunks):\n",
    "    return f'''\n",
    "Task: \n",
    "Based on the research question decomposition key questions and the sample text chunks of the source text, the goal is to identify how useful reading the full source text would be to extract direct quoted facts and determine the best answer to any of the key questions. \n",
    "\n",
    "Context:\n",
    "{context}\n",
    "\n",
    "Research question decomposition:\n",
    "{decomposition}\n",
    "\n",
    "Sample text chunks from the source text:\n",
    "{sample_text_chunks}\n",
    "\n",
    "Deliverables:\n",
    "For each key question, assign a predicted usefulness score of the full source text using a 5-point Likert scale, with 1 being very unlikely to be usefulness to 5 being very likely useful and containing facts that answer the key question.\n",
    "\n",
    "The output should be of the following JSON format\n",
    "{{\n",
    "    <insert key question index>: <insert predicted usefulness>,\n",
    "   etc.\n",
    "}}\n",
    "\n",
    "\n",
    "Respond only with the output, with no explanation or conversation.\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_sample_chunks(text, CHUNK_SIZE, num_chunk_samples):\n",
    "    step_size = len(text) // num_chunk_samples\n",
    "\n",
    "    chunks = []\n",
    "    for i in range(0, len(text), step_size):\n",
    "        chunk = text[i:i+CHUNK_SIZE]\n",
    "        chunks.append(chunk)\n",
    "\n",
    "        # Break after getting the required number of chunks\n",
    "        if len(chunks) >= num_chunk_samples:\n",
    "            break\n",
    "\n",
    "    return chunks\n",
    "\n",
    "def find_title(url, web_search_inf):\n",
    "    for item in web_search_inf:\n",
    "        if item[\"href\"] == url:\n",
    "            return item[\"title\"]\n",
    "    return None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "key question 0 query  1 rating_source_idx 0 Skimming url: https://pubs.acs.org/doi/10.1021/jacs.9b03431\n",
      "URL or text already visited!\n",
      "key question 0 query  1 rating_source_idx 1 Skimming url: https://pubmed.ncbi.nlm.nih.gov/31243147/\n",
      "URL or text already visited!\n",
      "key question 0 query  1 rating_source_idx 2 Skimming url: https://www.researchgate.net/figure/Reaction-scheme-and-structural-organization-of-the-K-setae-ECR-complex-a_fig1_360182999\n",
      "URL or text already visited!\n",
      "key question 1 query  1 rating_source_idx 0 Skimming url: https://pubs.acs.org/doi/10.1021/jacs.9b03431\n",
      "URL or text already visited!\n",
      "key question 1 query  1 rating_source_idx 1 Skimming url: https://pubmed.ncbi.nlm.nih.gov/31243147/\n",
      "URL or text already visited!\n",
      "key question 1 query  1 rating_source_idx 2 Skimming url: https://www.researchgate.net/figure/Reaction-scheme-and-structural-organization-of-the-K-setae-ECR-complex-a_fig1_360182999\n",
      "URL or text already visited!\n",
      "key question 1 query  3 rating_source_idx 0 Skimming url: https://pubs.acs.org/doi/10.1021/jacs.9b03431\n",
      "URL or text already visited!\n",
      "key question 1 query  3 rating_source_idx 1 Skimming url: https://pubmed.ncbi.nlm.nih.gov/31243147/\n",
      "URL or text already visited!\n",
      "key question 1 query  3 rating_source_idx 2 Skimming url: https://www.pnas.org/doi/10.1073/pnas.1901471116\n",
      "URL or text already visited!\n",
      "key question 2 query  1 rating_source_idx 0 Skimming url: https://pubs.acs.org/doi/10.1021/jacs.9b03431\n",
      "URL or text already visited!\n",
      "key question 2 query  1 rating_source_idx 1 Skimming url: https://pubmed.ncbi.nlm.nih.gov/31243147/\n",
      "URL or text already visited!\n",
      "key question 2 query  3 rating_source_idx 0 Skimming url: https://www.researchgate.net/figure/Reaction-scheme-and-structural-organization-of-the-K-setae-ECR-complex-a_fig1_360182999\n",
      "URL or text already visited!\n",
      "key question 3 query  1 rating_source_idx 0 Skimming url: https://pubs.acs.org/doi/10.1021/acsomega.2c01559\n",
      "URL or text already visited!\n",
      "key question 3 query  1 rating_source_idx 1 Skimming url: https://pubs.acs.org/doi/10.1021/jacs.9b03431\n",
      "URL or text already visited!\n",
      "key question 3 query  1 rating_source_idx 2 Skimming url: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC140121/\n",
      "URL or text already visited!\n",
      "key question 3 query  3 rating_source_idx 0 Skimming url: https://pubmed.ncbi.nlm.nih.gov/31243147/\n",
      "URL or text already visited!\n"
     ]
    }
   ],
   "source": [
    "# Skimming through each highly relevant paper from skimming\n",
    "for decomposition_idx, key_question_decomposition in enumerate(key_question_decomposition_list):\n",
    "    with open(f'autoscious_logs/{search_query_file_safe}/sources/kq{decomposition_idx}/initial_search_queries.json', 'r') as f:\n",
    "        key_question_initial_search_queries = json.load(f)\n",
    "\n",
    "    for query_idx, query in key_question_initial_search_queries.items():\n",
    "        # open filtered search results\n",
    "        with open(f'autoscious_logs/{search_query_file_safe}/sources/kq{decomposition_idx}/rated_web_results_query_{int(query_idx)}.json', 'r') as f:\n",
    "            ratings_url_dict = json.loads(f.read())\n",
    "\n",
    "        # open web search info to extract metadata\n",
    "        with open(f'autoscious_logs/{search_query_file_safe}/sources/kq{decomposition_idx}/initial_search_results_query_{int(query_idx)}.json', 'r') as f:\n",
    "            web_search_info = json.load(f)\n",
    "        \n",
    "        for rating, urls in ratings_url_dict.items():\n",
    "            if rating == '5': # TODO: figure out a good way to iterate and go to 4 after\n",
    "                # Start with iterating through 4s and 5s of ratings_url_dict\n",
    "                folder_path = f'autoscious_logs/{search_query_file_safe}/sources/kq{decomposition_idx}/predicted_usefulness_{rating}'\n",
    "                if not os.path.exists(folder_path):\n",
    "                    os.makedirs(folder_path)\n",
    "\n",
    "                for rating_source_idx, url in enumerate(urls):\n",
    "                    print(\"key question\", decomposition_idx, \"query \", query_idx, \"rating_source_idx\", rating_source_idx, \"Skimming url:\", url)\n",
    "\n",
    "                    # Ensure the url hasn't already been visited\n",
    "                    title = find_title(url, web_search_info)\n",
    "                    if not os.path.exists(f'autoscious_logs/{sanitize_filename(search_query)}/sources/full_text/{sanitize_filename(title)}.txt') and not os.path.exists(f'{folder_path}/query_{query_idx}_url_index_{rating_source_idx}.json'):\n",
    "                        text = scrape_text_with_selenium_no_agent(url, None, search_engine='chrome')\n",
    "\n",
    "                        # Only evaluate websites you're able to scrape\n",
    "                        if text != \"No information found\":\n",
    "                            total_chunks = len(text) / CHUNK_SIZE\n",
    "                            num_chunk_samples = min(int(total_chunks * SAMPLING_FACTOR), MAX_CHUNKS)\n",
    "                            sample_chunks = get_sample_chunks(text, CHUNK_SIZE, num_chunk_samples)\n",
    "                            print(\"len(sample_chunks)\", len(sample_chunks))\n",
    "\n",
    "                            # Get predicted usefulness based on sample chunks\n",
    "                            predicted_usefulness_results = json.loads(chat_openai(get_predicted_usefulness_of_text_prompt(context, decomposition, sample_chunks), model=\"gpt-3.5-turbo\")[0])\n",
    "\n",
    "                            # save filtered search results\n",
    "                            with open(f'{folder_path}/query_{query_idx}_url_index_{rating_source_idx}.json', 'w') as f:\n",
    "                                json.dump(predicted_usefulness_results, f, indent=2)\n",
    "                            \n",
    "                            # Check if any scores were (4 or) 5, because then we should save the full text\n",
    "                            pred_usefulness = predicted_usefulness_results.values()\n",
    "                            if 5 in pred_usefulness or '5' in pred_usefulness:\n",
    "                                with open(f'autoscious_logs/{sanitize_filename(search_query)}/sources/full_text/{sanitize_filename(title)}.txt', 'w', encoding='utf-8') as f:\n",
    "                                    f.write(title + '\\n')\n",
    "                                    f.write(url + '\\n')\n",
    "                                    f.write(text)\n",
    "                    else:\n",
    "                        print(\"URL or text already visited!\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
